<!DOCTYPE html>
<html class="no-js" lang="en"> <head><meta charset="utf-8"/><meta content="width=device-width,initial-scale=1" name="viewport"/><link href="https://docs.milaboratories.com/mixcr/reference/mixcr-postanalysis/" rel="canonical"/><link href="../../../assets/favicon.png" rel="icon"/><meta content="mkdocs-1.3.1, mkdocs-material-8.3.9" name="generator"/><title>postanalysis - Docs @ MiLaboratories</title><link href="../../../assets/stylesheets/main.1d29e8d0.min.css" rel="stylesheet"/><link href="../../../assets/stylesheets/palette.cbb835fc.min.css" rel="stylesheet"/><link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/><link href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link href="../../../css/timeago.css" rel="stylesheet"/><link href="../../../css/extra.css" rel="stylesheet"/><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-1KLG43HZP1"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-1KLG43HZP1",{page_path:e.pathname})})})</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-1KLG43HZP1"></script><link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>html.glightbox-open { overflow: initial; height: 100%; }</style><script src="../../../assets/javascripts/glightbox.min.js"></script></head> <body data-md-color-accent="" data-md-color-primary="" data-md-color-scheme="default" dir="ltr"> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/> <input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/> <label class="md-overlay" for="__drawer"></label> <div data-md-component="skip"> <a class="md-skip" href="#mixcr-postanalysis"> Skip to content </a> </div> <div data-md-component="announce"> </div> <header class="md-header" data-md-component="header"> <nav aria-label="Header" class="md-header__inner md-grid"> <a aria-label="Docs @ MiLaboratories" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> <label class="md-header__button md-icon" for="__drawer"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg> </label> <div class="md-header__title" data-md-component="header-title"> <div class="md-header__ellipsis"> <div class="md-header__topic"> <span class="md-ellipsis"> Docs @ MiLaboratories </span> </div> <div class="md-header__topic" data-md-component="header-topic"> <span class="md-ellipsis"> postanalysis </span> </div> </div> </div> <form class="md-header__option" data-md-component="palette"> <input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to dark mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z" fill="white" fill-rule="evenodd"></path> </svg> </label> <input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to light mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z" fill="white" fill-rule="evenodd"></path> </svg> </label> </form> <label class="md-header__button md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> </label> <div class="md-search" data-md-component="search" role="dialog"> <label class="md-search__overlay" for="__search"></label> <div class="md-search__inner" role="search"> <form class="md-search__form" name="search"> <input aria-label="Search by name or category" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search by name or category" required="" spellcheck="false" type="text"/> <label class="md-search__icon md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg> </label> <nav aria-label="Search" class="md-search__options"> <a aria-label="Share" class="md-search__icon md-icon" data-clipboard="" data-clipboard-text="" data-md-component="search-share" href="javascript:void(0)" tabindex="-1"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"></path></svg> </a> <button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg> </button> </nav> <div class="md-search__suggest" data-md-component="search-suggest"></div> </form> <div class="md-search__output"> <div class="md-search__scrollwrap" data-md-scrollfix=""> <div class="md-search-result" data-md-component="search-result"> <div class="md-search-result__meta"> Initializing search </div> <ol class="md-search-result__list"></ol> </div> </div> </div> </div> </div> <div class="md-header__source"> <a class="md-source" data-md-component="source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> </nav> </header> <div class="md-container" data-md-component="container"> <nav aria-label="Tabs" class="md-tabs" data-md-component="tabs"> <div class="md-tabs__inner md-grid"> <ul class="md-tabs__list"> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../.."> Home </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../getting-started/installation/"> Getting Started </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../guides/milaboratories-human-tcr-rna-multi/"> Guides </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link md-tabs__link--active" href="../mixcr-analyze/"> Reference </a> </li> </ul> </div> </nav> <main class="md-main" data-md-component="main"> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted has-section" data-md-level="0" data-test="1"> <!-- Site title --> <label class="md-nav__title" for="__drawer"> <a aria-label="Docs @ MiLaboratories" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class="md-nav__source"> <a class="md-source" data-md-component="source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../.."> Home </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_2"> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Getting Started" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_2"> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/installation/"> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/docker/"> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/milm/"> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/overview/"> MiXCR Overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/license/"> License </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_3"> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Guides" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_3"> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" id="__nav_3_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_1"> Kits &amp; Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Kits &amp; Protocols" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_1"> <span class="md-nav__icon md-icon"></span> Kits &amp; Protocols </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/milaboratories-human-tcr-rna-multi/"> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/biomed2-bcr/"> BioMed2 </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/abhelix-bcr/"> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/takara-mmu-tcr/"> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/takara-hsa-bcr/"> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/nebnext-bcr/"> NEBNext Immune Sequencing </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/qiaseq-tcr/"> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/rnaseq/"> RNA-Seq data </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" id="__nav_3_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_2"> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Analysis cases" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_2"> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/generic-umi-tcr/"> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/generic-multiplex-bcr/"> Targeted BCR Multiplex libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_3" id="__nav_3_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_3"> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Special guides" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_3"> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/create-custom-library/"> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../guides/external-libraries/"> Using external libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" id="__nav_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_4"> Reference <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reference" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_4"> <span class="md-nav__icon md-icon"></span> Reference </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_1" id="__nav_4_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_1"> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiXCR" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_1"> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-analyze/"> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-align/"> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-correctAndSortTags/"> correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-assemblePartial/"> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-extend/"> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-assemble/"> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-assembleContigs/"> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-export/"> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportPretty/"> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportAirr/"> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportReadsForClones/"> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportReports/"> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportQc/"> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for="__toc"> postanalysis <span class="md-nav__icon md-icon"></span> </label> <a class="md-nav__link md-nav__link--active" href="./"> postanalysis </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav aria-label="Table of contents" class="md-nav md-nav--secondary hello"> <label class="md-nav__title" for="__toc"> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix=""> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#common"> Common </a> <!-- Table of contents list --> <nav aria-label="Common" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#downsampling"> Downsampling </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#weight-functions"> Weight functions </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#metadata"> Metadata </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#isolation-groups"> Isolation groups </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#individual-postanalysis"> Individual postanalysis </a> <!-- Table of contents list --> <nav aria-label="Individual postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#command-line-options"> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#cdr3-metrics"> CDR3 metrics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#diversity-measures"> Diversity measures </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#segment-usage-metrics"> Segment usage metrics </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#overlap-postanalysis"> Overlap postanalysis </a> <!-- Table of contents list --> <nav aria-label="Overlap postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#command-line-options_1"> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#pairwise-distance-metrics"> Pairwise distance metrics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#preprocessing-summary-tables"> Preprocessing summary tables </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!-- Main navigation item --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportPlots/"> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportTables/"> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportPreprocTables/"> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-exportClonesOverlap/"> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mixcr-overlapScatterPlot/"> overlapScatterPlot </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" id="__nav_4_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_2"> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reports" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_2"> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-align/"> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-correctAndSortTags/"> mixcr correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-assemblePartial/"> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-extend/"> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-assemble/"> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../report-assembleContigs/"> mixcr assembleContigs </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" id="__nav_4_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_3"> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="RepSeq.io" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_3"> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-fasta/"> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-compile/"> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-debug/"> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-format/"> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-fromFasta/"> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-fromPaddedFasta/"> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-inferPoints/"> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-merge/"> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-stat/"> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-tsv/"> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_4" id="__nav_4_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_4"> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiTool" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_4"> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-parse/"> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-tag-stat/"> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-export-fastq/"> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-correct/"> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-consensus/"> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-process-fastq/"> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-format-fastq/"> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../mitool-compress-sequence-set/"> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_5" id="__nav_4_5" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_5"> Specs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Specs" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_5"> <span class="md-nav__icon md-icon"></span> Specs </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../tag-pattern/"> Tag Pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../geneFeatures/"> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../repseqio-json-format/"> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav aria-label="Table of contents" class="md-nav md-nav--secondary hello"> <label class="md-nav__title" for="__toc"> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix=""> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#common"> Common </a> <!-- Table of contents list --> <nav aria-label="Common" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#downsampling"> Downsampling </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#weight-functions"> Weight functions </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#metadata"> Metadata </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#isolation-groups"> Isolation groups </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#individual-postanalysis"> Individual postanalysis </a> <!-- Table of contents list --> <nav aria-label="Individual postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#command-line-options"> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#cdr3-metrics"> CDR3 metrics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#diversity-measures"> Diversity measures </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#segment-usage-metrics"> Segment usage metrics </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#overlap-postanalysis"> Overlap postanalysis </a> <!-- Table of contents list --> <nav aria-label="Overlap postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#command-line-options_1"> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#pairwise-distance-metrics"> Pairwise distance metrics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#preprocessing-summary-tables"> Preprocessing summary tables </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class="md-content" data-md-component="content"> <article class="md-content__inner md-typeset"> <a class="md-content__button md-icon" href="https://github.com/milaboratory/docs/edit/main/docs/mixcr/reference/mixcr-postanalysis.md" title="Edit this page"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg> </a> <h1 id="mixcr-postanalysis"><code>mixcr postanalysis</code></h1> <p>Runs basic postanalysis routines for a list of clonesets and writes the results to a compressed json file, which then may be used for tabular export with <a href="../mixcr-exportTables/"><code>exportTables</code></a> or graphical export with <a href="../mixcr-exportPlots/"><code>exportPlots</code></a>. There are two types of basic postanalysis:</p> <ul> <li><a href="./#individual-postanalysis">Individual</a> - computes different characteristics for each of the passed clonesets individually</li> <li><a href="./#overlap-postanalysis">Overlap</a> - builds clonesets overlap and computes various metrics on it</li> </ul> <p>In both cases schematically MiXCR works in the following way:</p> <ol> <li>Splits each of the passed clonesets into <code>TRAD</code>, <code>TRB</code>, <code>TRG</code>, <code>IGH</code>, <code>IGK</code> and <code>IGL</code> clonesets (of course if they present in the data)</li> <li>For each immunological chain it optionally filters productive clonotypes if the corresponding flag is specified</li> <li>Applies normalization of samples according to the specified downsampling</li> <li>Finally, it iterates over normalized datasets and computes either<ul> <li>(<code>individual</code>) all the characteristics for each dataset</li> <li>(<code>overlap</code>) overlap of the normalized clonesets and various pairwise metrics on the overlap</li> </ul> </li> </ol> <h2 id="common">Common</h2> <p>There are several important common aspects of individual and overlap postanalysis.</p> <h3 id="downsampling">Downsampling</h3> <p>Downsampling is a required procedure without which it is not possible to make a statistical comparisons between datasets.</p> <figure> <p><a class="glightbox" href="../pics/downsampling.svg"><img alt="downsampling" src="../pics/downsampling.svg"/></a></p> </figure> <p>There are three types of downsampling each of which may be applied on a one of three tag levels: <code>read</code>, <code>umi</code> or <code>cell</code>:</p> <dl> <dt><code>count-&lt;tag&gt;-(auto|min|fixed)[-&lt;number&gt;]</code></dt> <dd> <p>normalizes datasets to the same <em>tag number</em> (either reads, UMIs or cells). The exact downsampling value may be either</p> <ul> <li>computed automatically (<code>auto</code>)</li> <li>chosen as a value in the smallest sample (<code>min</code>)</li> <li>specified explicitly (<code>fixed</code>)</li> </ul> <p>For automatic downsampling MiXCR calculates 20-th quantile across all clonesets <em>Q20</em>, and takes the minimal sample which is above <em>0.5*Q20</em>.</p> <p>For memory efficiency, MiXCR uses random multivariate hypergeometric marginals algorithm for downsampling to the same number of reads.</p> <p>For downsampling to the same number of UMIs or cells, MiXCR uses random uniform sampling of the whole tag list. </p> </dd> <dt><code>top-&lt;tag&gt;-&lt;number&gt;</code></dt> <dd> <p>normalizes datasets to the same number of clonotypes. It takes top <code>&lt;number&gt;</code> of clonotypes by corresponding <code>tag</code> (i.e. top by number of reads, or by number of UMIs or by number of cells)</p> </dd> <dt><code>cumtop-&lt;tag&gt;-&lt;percent&gt;</code></dt> <dd> <p>normalizes datasets to the same cloneset abundance. It takes top clonotypes by corresponding <code>tag</code> (i.e. top by number of reads, or by number of UMIs or by number of cells) so that the top contains the specified <code>&lt;percent&gt;</code> of the cloneset abundance (again computed by corresponding <code>tag</code>) </p> </dd> <dt><code>none</code></dt> <dd> <p>no downsampling</p> </dd> </dl> <h3 id="weight-functions">Weight functions</h3> <p>MiXCR requires to select one of the four methods used to calculate weight of each clonotype:</p> <dl> <dt><code>read</code></dt> <dd> <p>weight by the number of reads</p> </dd> <dt><code>umi</code></dt> <dd> <p>weight by the number of UMI</p> </dd> <dt><code>cell</code></dt> <dd> <p>weight by the number of unique cell barcodes corresponding to the clone</p> </dd> <dt><code>none</code></dt> <dd> <p>for unweighted analysis</p> </dd> </dl> <h3 id="metadata">Metadata</h3> <p>Optionally, one can supply a metadata <code>.tsv</code> or <code>.csv</code> table which may be used further in <code>exportPlots</code> command to group samples and apply different statistical tests on a groups of samples. Metadata file must contain <code>sample</code> column. Values in this column should allow unambiguously match names of the input files using the longest common substring (LCS) algorithm. For example, suppose the input samples are:</p> <div class="highlight"><pre><span></span><code>&gt; ls results/*

results/sample_1_23_p.clns
results/sample_2_54_n.clns
results/sample_3_32_n.clns
results/sample_4_62_p.clns
...
</code></pre></div> <p>where <code>sample_i</code> part is unique for each sample. Then the sample metadata may look like:</p> <table> <thead> <tr> <th>sample</th> <th>age</th> <th>condition</th> </tr> </thead> <tbody> <tr> <td>sample_1</td> <td>23</td> <td>p</td> </tr> <tr> <td>sample_2</td> <td>54</td> <td>n</td> </tr> <tr> <td>sample_3</td> <td>32</td> <td>n</td> </tr> <tr> <td>sample_4</td> <td>62</td> <td>p</td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <p>Sometimes one project may contain samples corresponding to different immunological chains in separate files. In that case it is necessary to add <code>chain</code> column to the metadata file and MiXCR will use it. For example:</p> <div class="highlight"><pre><span></span><code>&gt; ls results/*

results/sample_1_23_p_alpha.clns
results/sample_2_54_n_beta.clns
results/sample_3_32_n_alpha.clns
results/sample_4_62_p_beta.clns
...
</code></pre></div> <p>and example metadata:</p> <table> <thead> <tr> <th>sample</th> <th>chain</th> <th>age</th> <th>condition</th> </tr> </thead> <tbody> <tr> <td>sample_1</td> <td>TRA</td> <td>23</td> <td>p</td> </tr> <tr> <td>sample_2</td> <td>TRB</td> <td>54</td> <td>n</td> </tr> <tr> <td>sample_3</td> <td>TRA</td> <td>32</td> <td>n</td> </tr> <tr> <td>sample_4</td> <td>TRB</td> <td>62</td> <td>p</td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <h3 id="isolation-groups">Isolation groups</h3> <p>Sometimes there is no need to apply normalization for <em>all</em> samples. For example, if there are two cell subsets in the data (e.g. CD4 and CD8) and you are interested in overlap postanalysis, then it is totally meaningless to compute overlap between CD4 and CD8 subsets. MiXCR allows to specify metadata column which will be used to separate samples into groups and apply downsampling and postanalysis separately for each group of samples. </p> <h2 id="individual-postanalysis">Individual postanalysis</h2> <div class="highlight"><pre><span></span><code>mixcr postanalysis individual [-f]
      --default-downsampling &lt;defaultDownsampling&gt;
      --default-weight-function &lt;defaultWeightFunction&gt; 
      [--drop-outliers]
      [--only-productive]
      [--chains &lt;chains&gt;]
      [--metadata &lt;file&gt;]
      [--preproc-tables &lt;path&gt;]
      [--tables &lt;path&gt;]
      [--group &lt;meta&gt;]...
      [-O&lt;String=String&gt;]...
      sample_1.clnx[&lt;sample_2.clnx&gt;...]
      output.json.gz
</code></pre></div> <p>Calculates <a href="./#cdr3-metrics">CDR3 metrics</a>, <a href="./#diversity-measures">Diversity measures</a> an <a href="./#segment-usage-metrics">Gene segment usage</a> for each of the specified samples.</p> <h3 id="command-line-options">Command line options</h3> <dl> <dt><code>--default-downsampling &lt;defaultDownsampling&gt;</code></dt> <dd> <p>default <a href="./#downsampling">downsampling</a> applied to normalize the clonesets</p> </dd> <dt><code>--drop-outliers</code></dt> <dd> <p>drop samples which are below downsampling value as computed according to specified default downsampling option</p> </dd> <dt><code>--default-weight-function &lt;defaultWeightFunction&gt;</code></dt> <dd> <p>default clonotype <a href="./#weight-functions">weight function</a></p> </dd> <dt><code>--only-productive</code></dt> <dd> <p>drop clonotypes with out-of-frame CDR3 sequence or CDR3 containing stop codon</p> </dd> <dt><code>--chains &lt;chains&gt;</code></dt> <dd> <p>possible values: <code>TRAD</code>, <code>TRG</code>, <code>TCR</code>, <code>IGH</code>, <code>IGK</code>, <code>IGL</code>, <code>IG</code>, <code>ALL</code>. Restrict the analysis to the specified immunological chains.</p> </dd> <dt><code>--metadata &lt;file&gt;</code></dt> <dd> <p><a href="./#metadata">metadata</a> file in a tab- (<code>.tsv</code>) or comma- (<code>.csv</code>) separated form. Must contain <code>sample</code> column which matches names of input files. Optionally may have <code>chains</code> column.</p> </dd> <dt><code>--preproc-tables &lt;path&gt;</code></dt> <dd> <p>output path for the <a href="./#preprocessing-summary-tables">preprocessing summary tables</a> (filtering and downsampling)</p> </dd> <dt><code>--tables &lt;path&gt;</code></dt> <dd> <p>output path for tabular postanalysis results </p> </dd> <dt><code>--group &lt;meta&gt;</code></dt> <dd> <p>metadata column used to group samples into <a href="./#isolation-groups">isolation groups</a>; postanalysis will be performed in each of the groups separately. It is possible to specify several isolation groups.</p> </dd> </dl> <p>MiXCR allows to override downsampling and filtering parameters for any postanalysis metric using <code>-Oparameter=value</code> syntax. Example:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr postanalysis individual <span class="se">\</span>
  -OvUsage.weightFunction<span class="o">=</span><span class="nb">read</span> <span class="se">\</span>
  ...
</code></pre></div> <p>The following options may be overridden:</p> <dl> <dt><code>-O&lt;metric&gt;.downsampling=&lt;downsampling&gt;</code></dt> <dd> <p>override default downsampling</p> </dd> <dt><code>-O&lt;metric&gt;.dropOutliers=(true|false)</code></dt> <dd> <p>override default setting for dropping outliers</p> </dd> <dt><code>-O&lt;metric&gt;.weightFunction=&lt;weightFunction&gt;</code></dt> <dd> <p>override default weight function</p> </dd> <dt><code>-O&lt;metric&gt;.onlyProductive=(true|false)</code></dt> <dd> <p>override default setting for filtering only productive clonotypes</p> </dd> </dl> <p>The following <code>&lt;metric&gt;</code> values are supported:</p> <ul> <li><a href="./#cdr3-metrics">CDR3 metrics</a>: <code>cdr3metrics.cdr3lenNT</code>, <code>cdr3metrics.cdr3lenAA</code>, <code>cdr3metrics.ndnLenNT</code>, <code>cdr3metrics.addedNNT</code>, <code>cdr3metrics.strength</code>, <code>cdr3metrics.hydrophobicity</code>, <code>cdr3metrics.surface</code>, <code>cdr3metrics.volume</code>, <code>cdr3metrics.charge</code>,</li> <li><a href="./#diversity-measures">Diversity measures</a>: <code>diversity.observed</code>, <code>diversity.shannonWiener</code>, <code>diversity.chao1</code>, <code>diversity.normalizedShannonWienerIndex</code>, <code>diversity.inverseSimpsonIndex</code>, <code>diversity.giniIndex</code>, <code>diversity.d50</code>, <code>diversity.efronThisted</code>,</li> <li><a href="./#segment-usage-metrics">Segment usage</a>: <code>vUsage</code>, <code>jUsage</code>, <code>isotypeUsage</code>, <code>vjUsage</code></li> </ul> <h3 id="cdr3-metrics">CDR3 metrics</h3> <p>CDR3 metrics may be separated into two groups. First includes "length" metrics:</p> <ul> <li>Length of CDR3, aa  length of amino acid CDR3 sequence</li> <li>Length of CDR3, nt  length of nucleotide CDR3 sequence</li> <li>Length of <code>VJJunction</code>, nt  nucleotide length of N-D-N region</li> <li>Added nucleotides  number of added <code>N</code> nucleotides</li> </ul> <p>The second group includes averaged physicochemical characteristics computed for the five amino acids in the middle of the CDR3:</p> <ul> <li>Strength of CDR3  estimated strength of the interaction between CDR3 and cognate peptide computed as a derivative of the interaction energy,</li> <li>Hydrophobicity of CDR3  estimated hydrophobicity (Kidera factor 4)</li> <li>Surface of CDR3  estimated surface of CDR3</li> <li>Volume of CDR3  estimated volume of CDR3</li> <li>Charge of CDR3  estimated charge of CDR3</li> </ul> <p>For details see the following publications and references therein:</p> <ul> <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6900484/"><em>De Simone G et al, J Immunol. 2019 Dec 15</em></a></li> <li><a href="https://pubmed.ncbi.nlm.nih.gov/32482872/"><em>Logunova et al, Proc Natl Acad Sci USA. 2020 Jun 16</em></a></li> </ul> <h3 id="diversity-measures">Diversity measures</h3> <p>Diversity measures reported by MiXCR include:</p> <ul> <li>Observed diversity  the number of clonotypes (<span class="arithmatex">\(D\)</span>)</li> <li> <p>Normalized Shannon-Wiener index  normalized (divided by log of the number of clonotypes) entropy of clonotype frequency distribution: $$ -\frac{1}{\log(D)}\sum f_i \log (f_i) $$</p> </li> <li> <p><a href="http://www.esajournals.org/doi/abs/10.2307/1934352">Shannon-Wiener diversity</a>  the exponent of clonotype frequency distribution entropy: $$ \exp \left(-\sum f_i \log (f_i)\right) $$</p> </li> <li> <p>Gini index: $$ 1 + \sum f_i^2 $$</p> </li> <li> <p><a href="http://www.esajournals.org/doi/abs/10.2307/1934352">Inverse Simpson index</a>: $$ 1 / \sum f_i^2 $$</p> </li> <li> <p><a href="https://academic.oup.com/jpe/article/5/1/3/1296712">Chao1 estimate</a>  Chao lower bound diversity estimate</p> </li> <li> <p><a href="http://www.jstor.org/stable/2335721">Efron-Thisted estimate</a>  lower bound total diversity estimate</p> </li> <li> <p><a href="http://www.google.com/patents/WO2012097374A1?cl=en">d50</a>  number of clones substituting a half of the sample abundance </p> </li> </ul> <h3 id="segment-usage-metrics">Segment usage metrics</h3> <p>Variable, Joining segment and Isotype usage vectors, i.e. the frequency of associated reads for each of V/J segments present in sample.</p> <h2 id="overlap-postanalysis">Overlap postanalysis</h2> <p><div class="highlight"><pre><span></span><code>mixcr postanalysis overlap [-f]
      --default-downsampling &lt;defaultDownsampling&gt;
      --default-weight-function &lt;defaultWeightFunction&gt;
      [--criteria &lt;overlapCriteria&gt;]
      [--factor-by &lt;meta&gt;[,&lt;meta&gt;...]]... 
      [--drop-outliers]
      [--only-productive]
      [--chains &lt;chains&gt;]
      [--metadata &lt;file&gt;]
      [--preproc-tables &lt;path&gt;]
      [--tables &lt;path&gt;]
      [--group &lt;meta&gt;]...
      [-O&lt;String=String&gt;]...
      sample_1.clnx[&lt;sample_2.clnx&gt;...]
      output.json.gz
</code></pre></div> Calculates pairwise <a href="./#overlap-metrics">Distance metrics</a>.</p> <h3 id="command-line-options_1">Command line options</h3> <dl> <dt><code>--default-downsampling &lt;defaultDownsampling&gt;</code></dt> <dd> <p>default <a href="./#downsampling">downsampling</a> applied to normalize the clonesets</p> </dd> <dt><code>--drop-outliers</code></dt> <dd> <p>drop samples which are below downsampling value as computed according to specified default downsampling option</p> </dd> <dt><code>--default-weight-function &lt;defaultWeightFunction&gt;</code></dt> <dd> <p>default clonotype <a href="./#weight-functions">weight function</a></p> </dd> <dt><code>--criteria &lt;overlapCriteria&gt;</code></dt> <dd> <p>overlap criteria. It allows to specify gene feature for overlap (nucleotide or amino acid), and optionally use V and J hits. Examples: <code>CDR3|AA|V|J</code> (overlap by a.a. CDR3 and V and J), <code>VDJRegion|AA</code> (overlap by a.a. <code>VDJRegion</code>), <code>CDR3|NT|V</code> (overlap by nt CDR3 and V).</p> </dd> <dt><code>--factor-by &lt;meta&gt;[,&lt;meta&gt;...]</code></dt> <dd> <p>pools samples with the same values of specified metadata columns and performs overlap between such pooled samples</p> </dd> <dt><code>--only-productive</code></dt> <dd> <p>drop clonotypes with out-of-frame CDR3 sequence or CDR3 containing stop codon</p> </dd> <dt><code>--chains &lt;chains&gt;</code></dt> <dd> <p>possible values: <code>TRAD</code>, <code>TRG</code>, <code>TCR</code>, <code>IGH</code>, <code>IGK</code>, <code>IGL</code>, <code>IG</code>, <code>ALL</code>. Restrict the analysis to the specified immunological chains.</p> </dd> <dt><code>--metadata &lt;file&gt;</code></dt> <dd> <p><a href="./#metadata">metadata</a> file in a tab- (<code>.tsv</code>) or comma- (<code>.csv</code>) separated form. Must contain <code>sample</code> column which matches names of input files. Optionally may have <code>chains</code> column.</p> </dd> <dt><code>--preproc-tables &lt;path&gt;</code></dt> <dd> <p>output path for the <a href="./#preprocessing-summary-tables">preprocessing summary tables</a> (filtering and downsampling)</p> </dd> <dt><code>--tables &lt;path&gt;</code></dt> <dd> <p>output path for tabular postanalysis results</p> </dd> <dt><code>--group &lt;meta&gt;</code></dt> <dd> <p>metadata column used to group samples into <a href="./#isolation-groups">isolation groups</a>; postanalysis will be performed in each of the groups separately. It is possible to specify several isolation groups.</p> </dd> </dl> <p>MiXCR allows to override downsampling and filtering parameters for any postanalysis metric using <code>-Oparameter=value</code> syntax. Example:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr postanalysis overlap <span class="se">\</span>
  -OvUsage.weightFunction<span class="o">=</span><span class="nb">read</span> <span class="se">\</span>
  ...
</code></pre></div> <p>The following options may be overridden:</p> <dl> <dt><code>-O&lt;metric&gt;.downsampling=&lt;downsampling&gt;</code></dt> <dd> <p>override default downsampling</p> </dd> <dt><code>-O&lt;metric&gt;.dropOutliers=(true|false)</code></dt> <dd> <p>override default setting for dropping outliers</p> </dd> <dt><code>-O&lt;metric&gt;.weightFunction=&lt;weightFunction&gt;</code></dt> <dd> <p>override default weight function</p> </dd> <dt><code>-O&lt;metric&gt;.onlyProductive=(true|false)</code></dt> <dd> <p>override default setting for filtering only productive clonotypes</p> </dd> </dl> <p>The following <code>&lt;metric&gt;</code> values are supported : <code>sharedClonotypes</code>, <code>f1Index</code>, <code>f2Index</code>, <code>jaccardIndex</code>, <code>pearson</code>, <code>pearsonAll</code>.</p> <h3 id="pairwise-distance-metrics">Pairwise distance metrics</h3> <p>Repertoire similarity measures include:</p> <ul> <li> <p>Shared clonotypes  number of shared clonotypes</p> </li> <li> <p>Relative diversity  overlap diversity, computed with the following normalization: $$ D_{ij} = \frac{d_{ij}}{d_{i}d_{j}} $$ where <span class="arithmatex">\(d_{ij}\)</span> is the number of clonotypes present in both samples and <span class="arithmatex">\(d_{i}\)</span> is the diversity of sample <span class="arithmatex">\(i\)</span>. See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3872297/">this paper</a> for the rationale behind normalization.</p> </li> <li> <p>Pearson  Pearson's correlation of clonotype frequencies, restricted only to the overlapping clonotypes: $$ R_{ij} = \frac{\sum^N_{k=1} \left(\phi_{ik} - \bar{\phi_{i}} \right) \left(\phi_{jk} - \bar{\phi_{j}} \right)}{\sqrt{\sum^N_{k=1} \left(\phi_{ik} - \bar{\phi_{i}} \right)^2 \sum^N_{k=1} \left(\phi_{jk} - \bar{\phi_{j}} \right)^2}} $$ where <span class="arithmatex">\(k=1 \to N\)</span> are the indices of overlapping clonotypes, <span class="arithmatex">\(\phi_{ik}\)</span> is the frequency of clonotype <span class="arithmatex">\(k\)</span> in sample <span class="arithmatex">\(i\)</span> and <span class="arithmatex">\(\bar{\phi_{i}}\)</span> is the average frequency of overlapping clonotypes in sample <span class="arithmatex">\(i\)</span>.</p> </li> <li> <p>Pearson (all)  Pearson's correlation of clonotype frequencies, restricted only to the overlapping clonotypes</p> </li> <li> <p>F1 index  geometric mean of relative overlap frequencies: $$ F_{ij} = \sqrt{f_{ij}f_{ji}} $$ where <span class="arithmatex">\(f_{ij}=\sum^N_{k=1}\phi_{ik}\)</span> is the total frequency of clonotypes that overlap between samples <span class="arithmatex">\(i\)</span> and <span class="arithmatex">\(j\)</span> in sample <span class="arithmatex">\(i\)</span>.</p> </li> <li> <p>F2 index  clonotype-wise sum of geometric mean frequencies: $$ F2_{ij} = \sum^N_{k=1}\sqrt{\phi_{ik}\phi_{jk}} $$ Note that this measure performs similar to <span class="arithmatex">\(F\)</span> and provides slightly more robust results in case cross-sample contamination is present.</p> </li> <li> <p><a href="http://en.wikipedia.org/wiki/Jaccard_index">Jaccard index</a>.</p> </li> </ul> <h3 id="preprocessing-summary-tables">Preprocessing summary tables</h3> <p>MiXCR provides detailed reports on preprocessing steps (filtering clonotypes and downsampling) applied for each of the computed postanalysis metrics. The typical report table looks like:</p> <table> <thead> <tr> <th>characteristic</th> <th>sample</th> <th>preprocessor</th> <th>nElementsBefore</th> <th>sumWeightBefore</th> <th>nElementsAfter</th> <th>sumWeightAfter</th> <th>preprocessor#1</th> <th>nElementsBefore#1</th> <th>sumWeightBefore#1</th> <th>nElementsAfter#1</th> <th>sumWeightAfter#1</th> <th>preprocessor#2</th> <th>nElementsBefore#2</th> <th>sumWeightBefore#2</th> <th>nElementsAfter#2</th> <th>sumWeightAfter#2</th> <th>preprocessor#3</th> <th>nElementsBefore#3</th> <th>sumWeightBefore#3</th> <th>nElementsAfter#3</th> <th>sumWeightAfter#3</th> </tr> </thead> <tbody> <tr> <td>VSegmentUsage</td> <td>P21-M1-PD1.clns</td> <td>Filter TRA,TRD chains &amp; Filter stops in CDR3, OOF in CDR3 &amp; Downsample by umi automatic</td> <td>309</td> <td>2141.0</td> <td>110</td> <td>796.0</td> <td>Filter TRA,TRD chains</td> <td>309</td> <td>2141.0</td> <td>162</td> <td>1186.0</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>162</td> <td>1186.0</td> <td>139</td> <td>1143.0</td> <td>Downsample by umi automatic</td> <td>139</td> <td>1143.0</td> <td>110</td> <td>796.0</td> </tr> <tr> <td>Added N, nt</td> <td>P14-T0-TIGIT.clns</td> <td>Filter TRA,TRD chains &amp; Filter stops in CDR3, OOF in CDR3 &amp; Downsample by umi automatic</td> <td>863</td> <td>8478.0</td> <td>189</td> <td>796.0</td> <td>Filter TRA,TRD chains</td> <td>863</td> <td>8478.0</td> <td>412</td> <td>3688.0</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>412</td> <td>3688.0</td> <td>311</td> <td>3071.0</td> <td>Downsample by umi automatic</td> <td>311</td> <td>3071.0</td> <td>189</td> <td>796.0</td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <p>The meaning of the columns is the following:</p> <dl> <dt><code>characteristic</code></dt> <dd> <p>name for a group of metrics for which the same downsampling was applied</p> </dd> <dt><code>preprocessor</code></dt> <dd> <p>the name of the <em>overall</em> preprocessing chain applied to the dataset</p> </dd> <dt><code>nElementsBefore</code></dt> <dd> <p>number of clonotypes before any preprocessing applied (that is in the initial dataset)</p> </dd> <dt><code>sumWeightBefore</code></dt> <dd> <p>total <a href="./#downsampling">weight</a> of all clonotypes before any preprocessing applied</p> </dd> <dt><code>nElementsAfter</code></dt> <dd> <p>number of clonotypes in the dataset after all preprocessing applied</p> </dd> <dt><code>sumWeightBefore</code></dt> <dd> <p>total weight of all clonotypes after all preprocessing applied.</p> </dd> <dt><code>preprocessor#i</code> and following columns</dt> <dd> <p>report columns for i-th step in the preprocessing chain</p> </dd> </dl> <hr/> <div class="md-source-file"> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class="timeago" datetime="2022-08-28T15:04:50+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2022-08-28</span> <br/> Created: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class="timeago" datetime="2022-08-28T15:04:50+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2022-08-28</span> </small> </div> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a class="md-top md-icon" data-md-component="top" hidden="" href="#"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg> Back to top </a> </main> <footer class="md-footer"> <nav aria-label="Footer" class="md-footer__inner md-grid"> <a aria-label="Previous: exportQc" class="md-footer__link md-footer__link--prev" href="../mixcr-exportQc/" rel="prev"> <div class="md-footer__button md-icon"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg> </div> <div class="md-footer__title"> <div class="md-ellipsis"> <span class="md-footer__direction"> Previous </span> exportQc </div> </div> </a> <a aria-label="Next: exportPlots" class="md-footer__link md-footer__link--next" href="../mixcr-exportPlots/" rel="next"> <div class="md-footer__title"> <div class="md-ellipsis"> <span class="md-footer__direction"> Next </span> exportPlots </div> </div> <div class="md-footer__button md-icon"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class="md-copyright"> Made with <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank"> Material for MkDocs </a> </div> </div> </div> </footer> </div> <div class="md-dialog" data-md-component="dialog"> <div class="md-dialog__inner md-typeset"></div> </div> <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src="../../../assets/javascripts/bundle.6c7ad80a.min.js"></script> <script src="../../../js/timeago.min.js"></script> <script src="../../../js/timeago_mkdocs_material.js"></script> <script src="../../../javascripts/svg.js"></script> <script src="../../../javascripts/onload.js"></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script> <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "width": "100%", "height": "auto", "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom"});})</script></body> </html>