<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="MiXCR documentation portal"><meta name=author content="MiLaboratories Inc"><link href=https://docs.milaboratories.com/mixcr/reference/mixcr-export/ rel=canonical><link rel=icon href=../../../assets/favicon.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>export - Docs @ MiLaboratories</title><link rel=stylesheet href=../../../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link rel=stylesheet href=../../../css/extra.css><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-2N4Z95ZG08"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-2N4Z95ZG08",{page_path:e.pathname})})})</script><script async src="https://www.googletagmanager.com/gtag/js?id=G-2N4Z95ZG08"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#export class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../.. title="Docs @ MiLaboratories" class="md-header__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Docs @ MiLaboratories </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> export </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z"/> </svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z"/> </svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label="Search by name or category" placeholder="Search by name or category" autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../.. class=md-tabs__link> Home </a> </li> <li class=md-tabs__item> <a href=../../getting-started/installation/ class=md-tabs__link> Getting Started </a> </li> <li class=md-tabs__item> <a href=../../guides/milaboratories-human-bcr-rna-multi/ class=md-tabs__link> Guides </a> </li> <li class=md-tabs__item> <a href=../overview-analysis-overview/ class="md-tabs__link md-tabs__link--active"> Docs </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav class="md-nav md-nav--primary md-nav--lifted has-section" aria-label=Navigation data-md-level=0 data-test=1> <!-- Site title --> <label class=md-nav__title for=__drawer> <a href=../../.. title="Docs @ MiLaboratories" class="md-nav__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class=md-nav__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class=md-nav__list data-md-scrollfix> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../.. class=md-nav__link> Home </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_2 type=checkbox id=__nav_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_2 data-level=1> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Getting Started" data-md-level=1> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/installation/ class=md-nav__link> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/docker/ class=md-nav__link> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/milm/ class=md-nav__link> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/first-run/ class=md-nav__link> First run </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/license/ class=md-nav__link> License </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/support/ class=md-nav__link> Support </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3 data-level=1> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Guides data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_1 data-level=2> Kits & Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Kits & Protocols" data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Kits & Protocols </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milaboratories-human-bcr-rna-multi/ class=md-nav__link> MiLaboratories Human BCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milaboratories-human-tcr-rna-multi/ class=md-nav__link> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milab-dna-multiplex-tcr/ class=md-nav__link> MiLaboratories Human TCR DNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/takara-hsa-bcr/ class=md-nav__link> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/takara-mmu-tcr/ class=md-nav__link> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/iRepertoire/ class=md-nav__link> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/nebnext-bcr/ class=md-nav__link> NEBNext Immune Sequencing </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/qiaseq-tcr/ class=md-nav__link> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/abhelix-bcr/ class=md-nav__link> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/biomed2-bcr/ class=md-nav__link> BioMed2 </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_2 data-level=2> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Analysis cases" data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-umi-tcr/ class=md-nav__link> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-multiplex-bcr/ class=md-nav__link> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-umi-bcr/ class=md-nav__link> Targeted BCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/rnaseq/ class=md-nav__link> RNA-Seq data </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/ig-trees-reconstruction/ class=md-nav__link> Immunoglobulin lineage trees reconstruction </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_3 type=checkbox id=__nav_3_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_3 data-level=2> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Special guides" data-md-level=2> <label class=md-nav__title for=__nav_3_3> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/create-custom-library/ class=md-nav__link> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/external-libraries/ class=md-nav__link> Using external libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4 data-level=1> Docs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Docs data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Docs </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_1 data-level=2> Overview <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Overview data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-analysis-overview/ class=md-nav__link> Analysis overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-presets/ class=md-nav__link> Presets </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-mixins-list/ class=md-nav__link> Mix-in options list </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-built-in-presets/ class=md-nav__link> Built-in presets </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_2 data-level=2> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiXCR data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-analyze/ class=md-nav__link> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-align/ class=md-nav__link> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-refineTagsAndSort/ class=md-nav__link> refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assemblePartial/ class=md-nav__link> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-extend/ class=md-nav__link> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assemble/ class=md-nav__link> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assembleContigs/ class=md-nav__link> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for=__toc> export <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> export </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#clonotype-tables class=md-nav__link> Clonotype tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alignments class=md-nav__link> Alignments </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-tables class=md-nav__link> SHM trees tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-with-nodes-tables class=md-nav__link> SHM trees with nodes tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#examples class=md-nav__link> Examples </a> <!-- Table of contents list --> <nav class=md-nav aria-label=Examples> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#default-export class=md-nav__link> Default export </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#umi-libraries class=md-nav__link> UMI libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#single-cell-libraries class=md-nav__link> Single cell libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-contigs-with-imputation class=md-nav__link> Export contigs with imputation </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-export-presets class=md-nav__link> SHM trees export presets </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-fields class=md-nav__link> Export fields </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Export fields"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#common-fields class=md-nav__link> Common fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alignment-specific-fields class=md-nav__link> Alignment-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#clonotype-specific-fields class=md-nav__link> Clonotype-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-tree-specific-fields class=md-nav__link> SHM tree-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-tree-node-specific-fields class=md-nav__link> SHM tree node-specific fields </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#default-anchor-point-positions class=md-nav__link> Default anchor point positions </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!-- Main navigation item --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPretty/ class=md-nav__link> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportAirr/ class=md-nav__link> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportReadsForClones/ class=md-nav__link> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportShmTreesNewick/ class=md-nav__link> exportShmTreesNewick </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportReports/ class=md-nav__link> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportQc/ class=md-nav__link> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-postanalysis/ class=md-nav__link> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-downsample/ class=md-nav__link> downsample </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPlots/ class=md-nav__link> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportTables/ class=md-nav__link> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPreprocTables/ class=md-nav__link> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportClonesOverlap/ class=md-nav__link> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-overlapScatterPlot/ class=md-nav__link> overlapScatterPlot </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-findAlleles/ class=md-nav__link> findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-findShmTrees/ class=md-nav__link> findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_3 data-level=2> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Reports data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-align/ class=md-nav__link> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-refineTagsAndSort/ class=md-nav__link> mixcr refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assemblePartial/ class=md-nav__link> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-extend/ class=md-nav__link> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assemble/ class=md-nav__link> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assembleContigs/ class=md-nav__link> mixcr assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-findAlleles/ class=md-nav__link> mixcr findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-findShmTrees/ class=md-nav__link> mixcr findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_4 data-level=2> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=RepSeq.io data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fasta/ class=md-nav__link> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-compile/ class=md-nav__link> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-debug/ class=md-nav__link> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-format/ class=md-nav__link> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fromFasta/ class=md-nav__link> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fromPaddedFasta/ class=md-nav__link> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-inferPoints/ class=md-nav__link> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-merge/ class=md-nav__link> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-stat/ class=md-nav__link> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-tsv/ class=md-nav__link> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_5 data-level=2> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiTool data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-parse/ class=md-nav__link> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-tag-stat/ class=md-nav__link> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-export-fastq/ class=md-nav__link> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-correct/ class=md-nav__link> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-consensus/ class=md-nav__link> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-process-fastq/ class=md-nav__link> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-format-fastq/ class=md-nav__link> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-compress-sequence-set/ class=md-nav__link> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_6 type=checkbox id=__nav_4_6> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_6 data-level=2> Specifications <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Specifications data-md-level=2> <label class=md-nav__title for=__nav_4_6> <span class="md-nav__icon md-icon"></span> Specifications </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-tag-pattern/ class=md-nav__link> Tag pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-input-file-name-expansion/ class=md-nav__link> Input file name expansion </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-gene-features/ class=md-nav__link> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-translation-rules/ class=md-nav__link> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-mutations-encoding/ class=md-nav__link> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-repseqio-json-format/ class=md-nav__link> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#clonotype-tables class=md-nav__link> Clonotype tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alignments class=md-nav__link> Alignments </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-tables class=md-nav__link> SHM trees tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-with-nodes-tables class=md-nav__link> SHM trees with nodes tables </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#examples class=md-nav__link> Examples </a> <!-- Table of contents list --> <nav class=md-nav aria-label=Examples> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#default-export class=md-nav__link> Default export </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#umi-libraries class=md-nav__link> UMI libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#single-cell-libraries class=md-nav__link> Single cell libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-contigs-with-imputation class=md-nav__link> Export contigs with imputation </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-trees-export-presets class=md-nav__link> SHM trees export presets </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-fields class=md-nav__link> Export fields </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Export fields"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#common-fields class=md-nav__link> Common fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alignment-specific-fields class=md-nav__link> Alignment-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#clonotype-specific-fields class=md-nav__link> Clonotype-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-tree-specific-fields class=md-nav__link> SHM tree-specific fields </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#shm-tree-node-specific-fields class=md-nav__link> SHM tree node-specific fields </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#default-anchor-point-positions class=md-nav__link> Default anchor point positions </a> <!-- Table of contents list --> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/milaboratory/docs/edit/main/docs/mixcr/reference/mixcr-export.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=export>Export</h1> <p>Export clonotypes or raw alignments in a tabular form.</p> <p><img alt src=../pics/export-light.svg#only-light> <img alt src=../pics/export-dark.svg#only-dark></p> <p>MiXCR uses four highly efficient binary formats that hold exhaustive information on the clonotypes, alignments, barcodes and original sequencing reads:</p> <ul> <li><code>.vdjca</code> produced by <a href=../mixcr-align/ ><code>align</code></a> and holds alignments</li> <li><code>.clns</code> produced by <a href=../mixcr-assemble/ ><code>assemble</code></a> and <a href=../mixcr-assembleContigs/ ><code>assembleContigs</code></a> holds clonotypes</li> <li><code>.clna</code> produced by <a href=../mixcr-assemble/ ><code>assemble</code></a> and holds both clonotypes and alignments</li> <li><code>.shmt</code> produced by <a href=../mixcr-findShmTrees/ ><code>findShmTrees</code></a> and holds clonotypes, alignments and SHM trees</li> </ul> <p>MiXCR provides functions for export <a href=#alignments>alignments</a>, <a href=#clonotype-tables>clonotype tables</a>, <a href=#shm-trees-tables>SHM trees tables</a> and <a href=#shm-trees-with-nodes-tables>SHM with nodes tables</a> in a tab-delimited way. For human-readable formatted output see <a href=../mixcr-exportPretty/ >pretty export</a>.</p> <h2 id=clonotype-tables>Clonotype tables</h2> <div class=highlight><pre><span></span><code>mixcr exportClones 
    [--chains &lt;chains&gt;] 
    [--filter-out-of-frames] 
    [--filter-stops] 
    [--split-by-tag &lt;tag&gt;] 
    [--split-files-by &lt;splitFilesBy&gt;]... 
    [--dont-split-files] 
    [--no-header] 
    [--drop-default-fields] 
    [--prepend-columns] 
    [&lt;exportField&gt;]...
    [--force-overwrite] 
    [--no-warnings] 
    [--verbose] 
    [--help] 
    data.(clns|clna) [table.tsv]
</code></pre></div> <p>Exports tab-delimited table of alignments from <code>.clns</code> and <code>.clna</code> files.</p> <p>Command line options:</p> <dl> <dt><code>data.(clns|clna)</code></dt> <dd>Path to input file with clones</dd> <dt><code>[table.tsv]</code></dt> <dd>Path where to write export table. Will write to output if omitted.</dd> <dt><code>-c, --chains &lt;chains&gt;</code></dt> <dd>Limit export to specific chain (e.g. TRA or IGH) (fractions will be recalculated). Default value determined by the preset.</dd> <dt><code>-o, --filter-out-of-frames</code></dt> <dd>Exclude clones with out-of-frame clone sequences (fractions will be recalculated). Default value determined by the preset.</dd> <dt><code>-t, --filter-stops</code></dt> <dd>Exclude sequences containing stop codons (fractions will be recalculated). Default value determined by the preset.</dd> <dt><code>--split-by-tag &lt;tag&gt;</code></dt> <dd>Split clones by tag values Default value determined by the preset.</dd> <dt><code>--split-files-by &lt;splitFilesBy&gt;</code></dt> <dd>Split files by (currently the only supported value is "geneLabel:reliableChain" etc... ). Default value determined by the preset.</dd> <dt><code>--dont-split-files</code></dt> <dd>Don't split files.</dd> <dt><code>--no-header</code></dt> <dd>Don't print first header line, print only data Default value determined by the preset.</dd> <dt><code>--drop-default-fields</code></dt> <dd>Don't export fields from preset.</dd> <dt><code>--prepend-columns</code></dt> <dd>Added columns will be inserted before default columns. By default columns will be added after default columns</dd> <dt><code>-f, --force-overwrite</code></dt> <dd>Force overwrite of output file(s).</dd> <dt><code>-nw, --no-warnings</code></dt> <dd>Suppress all warning messages.</dd> <dt><code>--verbose</code></dt> <dd>Verbose warning messages.</dd> <dt><code>-h, --help</code></dt> <dd>Show this help message and exit.</dd> <dt><code>&lt;exportField&gt;</code></dt> <dd>A list of <a href=#export-fields>export fields</a></dd> </dl> <h2 id=alignments>Alignments</h2> <div class=highlight><pre><span></span><code>mixcr exportAlignments [-f]
    [--chains &lt;chains&gt;] 
    [--no-header] 
    [--drop-default-fields] 
    [--prepend-columns] 
    [&lt;exportField&gt;]...
    [--force-overwrite] 
    [--no-warnings] 
    [--verbose] 
    [--help] 
    data.(vdjca|clns|clna) [table.tsv]
</code></pre></div> <p>Exports tab-delimited alignments from <code>.vdjca</code>, <code>.clns</code> and <code>.clna</code> files.</p> <p>Command line options:</p> <dl> <dt><code>data.[vdjca|clns|clna]</code></dt> <dd>Path to input file</dd> <dt><code>[table.tsv]</code></dt> <dd>Path where to write export table. Will write to output if omitted.</dd> <dt><code>-c, --chains &lt;chains&gt;</code></dt> <dd>Limit export to specific chain (e.g. TRA or IGH) (fractions will be recalculated) Default value determined by the preset.</dd> <dt><code>--no-header</code></dt> <dd>Don't print first header line, print only data Default value determined by the preset.</dd> <dt><code>--drop-default-fields</code></dt> <dd>Don't export fields from preset.</dd> <dt><code>--prepend-columns</code></dt> <dd>Added columns will be inserted before default columns. By default columns will be added after default columns</dd> <dt><code>-f, --force-overwrite</code></dt> <dd>Force overwrite of output file(s).</dd> <dt><code>-nw, --no-warnings</code></dt> <dd>Suppress all warning messages.</dd> <dt><code>--verbose</code></dt> <dd>Verbose warning messages.</dd> <dt><code>-h, --help</code></dt> <dd>Show this help message and exit.</dd> <dt><code>&lt;exportField&gt;</code></dt> <dd>A list of <a href=#export-fields>export fields</a></dd> </dl> <h2 id=shm-trees-tables>SHM trees tables</h2> <div class=highlight><pre><span></span><code>mixcr exportShmTrees [-f]
    [--ids &lt;id&gt;[,&lt;id&gt;...]]... 
    [--filter-min-nodes &lt;n&gt;] 
    [--filter-min-height &lt;n&gt;] 
    [--preset &lt;preset&gt;]
    [--preset-file &lt;file&gt;]
    [--no-header] 
    [&lt;exportField&gt;]...
    [--force-overwrite] 
    [--no-warnings] 
    [--verbose] 
    [--help] 
    [[--filter-in-feature &lt;gene_feature&gt;] [--pattern-max-errors &lt;n&gt;] (--filter-aa-pattern &lt;pattern&gt; | --filter-nt-pattern &lt;pattern&gt;)] 
    trees.shmt [trees.tsv]
</code></pre></div> <p>Exports tab-delimited alignments from <code>.shmt</code> file.</p> <p>Command line options:</p> <dl> <dt><code>trees.shmt</code></dt> <dd>Input file produced by 'findShmTrees' command.</dd> <dt><code>[trees.tsv]</code></dt> <dd>Path to output table. Print in stdout if omitted.</dd> <dt><code>--no-header</code></dt> <dd>Don't print column names</dd> <dt><code>--ids &lt;id&gt;[,&lt;id&gt;...]</code></dt> <dd>Filter specific trees by id</dd> <dt><code>--filter-min-nodes &lt;n&gt;</code></dt> <dd>Minimal number of nodes in tree</dd> <dt><code>--filter-min-height &lt;n&gt;</code></dt> <dd>Minimal height of the tree</dd> <dt><code>-p, --preset &lt;preset&gt;</code></dt> <dd>Specify preset of export fields (possible values: 'full', 'min'; 'full' by default)</dd> <dt><code>-pf, --preset-file &lt;presetFile&gt;</code></dt> <dd>Specify preset file of export fields</dd> <dt><code>-f, --force-overwrite</code></dt> <dd>Force overwrite of output file(s).</dd> <dt><code>-nw, --no-warnings</code></dt> <dd>Suppress all warning messages.</dd> <dt><code>--verbose</code></dt> <dd>Verbose warning messages.</dd> <dt><code>-h, --help</code></dt> <dd>Show this help message and exit.</dd> <dt><code>&lt;exportField&gt;</code></dt> <dd>A list of <a href=#export-fields>export fields</a></dd> </dl> <p>Filter by pattern options:</p> <dl> <dt><code>--filter-in-feature &lt;gene_feature&gt;</code></dt> <dd>Match pattern inside specified gene feature. Default: CDR3</dd> <dt><code>--pattern-max-errors &lt;n&gt;</code></dt> <dd>Max allowed subs &amp; indels. Default: 0</dd> <dt><code>--filter-aa-pattern &lt;pattern&gt;</code></dt> <dd>Filter specific trees by aa pattern.</dd> <dt><code>--filter-nt-pattern &lt;pattern&gt;</code></dt> <dd>Filter specific trees by nt pattern.</dd> </dl> <h2 id=shm-trees-with-nodes-tables>SHM trees with nodes tables</h2> <div class=highlight><pre><span></span><code>mixcr exportShmTreesWithNodes [-f]
    [--ids &lt;id&gt;[,&lt;id&gt;...]]... 
    [--filter-min-nodes &lt;n&gt;] 
    [--filter-min-height &lt;n&gt;] 
    [--onlyObserved] 
    [--preset &lt;preset&gt;] 
    [--preset-file &lt;presetFile&gt;] 
    [--no-header]
    [&lt;exportField&gt;]...
    [--force-overwrite] 
    [--no-warnings] 
    [--verbose]  
    [--help] 
    [[--filter-in-feature &lt;gene_feature&gt;] [--pattern-max-errors &lt;n&gt;] (--filter-aa-pattern &lt;pattern&gt; | --filter-nt-pattern &lt;pattern&gt;)] 
    trees.shmt [trees.tsv]
</code></pre></div> <p>Exports tab-delimited alignments from <code>.shmt</code> file.</p> <p>Command line options:</p> <dl> <dt><code>trees.shmt</code></dt> <dd>Input file produced by 'findShmTrees' command.</dd> <dt><code>[trees.tsv]</code></dt> <dd>Path where to write output export table. Print in stdout if omitted.</dd> <dt><code>--ids &lt;id&gt;[,&lt;id&gt;...]</code></dt> <dd>Filter specific trees by id</dd> <dt><code>--filter-min-nodes &lt;n&gt;</code></dt> <dd>Minimal number of nodes in tree</dd> <dt><code>--filter-min-height &lt;n&gt;</code></dt> <dd>Minimal height of the tree</dd> <dt><code>--no-header</code></dt> <dd>Don't print column names</dd> <dt><code>--onlyObserved</code></dt> <dd>Exclude nodes that was reconstructed by algorithm</dd> <dt><code>-p, --preset &lt;preset&gt;</code></dt> <dd>Specify preset of export fields (possible values: 'min', 'full'; 'full' by default)</dd> <dt><code>-pf, --preset-file &lt;presetFile&gt;</code></dt> <dd>Specify preset file of export fields</dd> <dt><code>-f, --force-overwrite</code></dt> <dd>Force overwrite of output file(s).</dd> <dt><code>-nw, --no-warnings</code></dt> <dd>Suppress all warning messages.</dd> <dt><code>--verbose</code></dt> <dd>Verbose warning messages.</dd> <dt><code>-h, --help</code></dt> <dd>Show this help message and exit.</dd> <dt><code>&lt;exportField&gt;</code></dt> <dd>A list of <a href=#export-fields>export fields</a></dd> </dl> <p>Filter by pattern options:</p> <dl> <dt><code>--filter-in-feature &lt;gene_feature&gt;</code></dt> <dd>Match pattern inside specified gene feature. Default: CDR3</dd> <dt><code>--pattern-max-errors &lt;n&gt;</code></dt> <dd>Max allowed subs &amp; indels. Default: 0</dd> <dt><code>--filter-aa-pattern &lt;pattern&gt;</code></dt> <dd>Filter specific trees by aa pattern.</dd> <dt><code>--filter-nt-pattern &lt;pattern&gt;</code></dt> <dd>Filter specific trees by nt pattern.</dd> </dl> <h2 id=examples>Examples</h2> <h3 id=default-export>Default export</h3> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones clones.clns clones.tsv
</code></pre></div> <p>The resulting tab-delimited text file will contain default set of columns (determined by the used <a href=../overview-built-in-presets/ >preset</a>) which includes clonotype abundances, nucleotide and amino acid clonotype sequences, Phred qualities, all or just best hit for V, D, J and C genes, corresponding alignments, nucleotide and amino acid sequences of gene regions present in sequence, etc. Example output (for BCR full-length data):</p> <table> <thead> <tr> <th>cloneId</th> <th>cloneCount</th> <th>cloneFraction</th> <th>allVHitsWithScore</th> <th>allDHitsWithScore</th> <th>allJHitsWithScore</th> <th>allCHitsWithScore</th> <th>allVAlignments</th> <th>allDAlignments</th> <th>allJAlignments</th> <th>allCAlignments</th> <th>nSeqFR1</th> <th>minQualFR1</th> <th>nSeqCDR1</th> <th>minQualCDR1</th> <th>nSeqFR2</th> <th>minQualFR2</th> <th>nSeqCDR2</th> <th>minQualCDR2</th> <th>nSeqFR3</th> <th>minQualFR3</th> <th>nSeqCDR3</th> <th>minQualCDR3</th> <th>nSeqFR4</th> <th>minQualFR4</th> <th>aaSeqFR1</th> <th>aaSeqCDR1</th> <th>aaSeqFR2</th> <th>aaSeqCDR2</th> <th>aaSeqFR3</th> <th>aaSeqCDR3</th> <th>aaSeqFR4</th> <th>refPoints</th> <th>targetSequences</th> <th>targetQualities</th> </tr> </thead> <tbody> <tr> <td>1</td> <td>157166.58695588264</td> <td>0.10205277935802338</td> <td>IGHV3-8*00(887.9)</td> <td></td> <td>IGHJ4*00(175.5)</td> <td>IGHG2B_hinge(26.6)</td> <td>0&#124;293&#124;316&#124;43&#124;336&#124;SG0CSG11CST18AST20GSG28CSA37GSA68CSA77CSA81CSA87GSG88CSG91ASC92TSA94TST95CSC97GSC98GSG101ASG103CSA115TSG117CST144GST149CSA154CSA156CSG157CSG160ASG169CST192ASG214ASC230TSC234GST237GSA238CSG253CSC254ASA259GSA260TSG266AST278ASA290G&#124;891.0</td> <td></td> <td>24&#124;68&#124;68&#124;342&#124;386&#124;ST31CSC39TSG45T&#124;178.0</td> <td></td> <td>CAGGTGCAGCTCGTGGAGACGGGGGGAGCCTTGGTGCGGCCTGGGGGGTCTCTGAGACTCTCCTGTGCCGCCTCT</td> <td>58</td> <td>GGCTTCCCCTTCGCTAATTTCGGG</td> <td>58</td> <td>ATAACCTGGGTCCGCCTGCCTCCAGGAAAGGGGCTCGAGTGGGTCGCAGAC</td> <td>58</td> <td>ATTACTCCTGATGGTGGTACCACA</td> <td>58</td> <td>TACTATGCAGACTCCGTGAAGGGCCGGTTCACCATCTCCAAAGACAACGCCAAGAATACGGTGGCTCTGCAAATGAACACACTGAGTCCTGAAGACACGGCCGTATATTAC</td> <td>58</td> <td>TGTGCGACCCCGATGTATGACCACTGG</td> <td>13</td> <td>GGTCAGGGTACCCAGGTCACCGTCTCCTCAG</td> <td>58</td> <td>QVQLVETGGALVRPGGSLRLSCAAS</td> <td>GFPFANFG</td> <td>ITWVRLPPGKGLEWVAD</td> <td>ITPDGGTT</td> <td>YYADSVKGRFTISKDNAKNTVALQMNTLSPEDTAVYY</td> <td>CATPMYDHW</td> <td>GQGTQVTVSS_</td> <td>::::43:118:142:193:217:328:-3:336:::::342:-4:355:386::</td> <td>CCTCGCGGCCCAGCCGGCCATGGCCGGCCCGGGAGCGGCCGCTCAGGTGCAGCTCGTGGAGACGGGGGGAGCCTTGGTGCGGCCTGGGGGGTCTCTGAGACTCTCCTGTGCCGCCTCTGGCTTCCCCTTCGCTAATTTCGGGATAACCTGGGTCCGCCTGCCTCCAGGAAAGGGGCTCGAGTGGGTCGCAGACATTACTCCTGATGGTGGTACCACATACTATGCAGACTCCGTGAAGGGCCGGTTCACCATCTCCAAAGACAACGCCAAGAATACGGTGGCTCTGCAAATGAACACACTGAGTCCTGAAGACACGGCCGTATATTACTGTGCGACCCCGATGTATGACCACTGGGGTCAGGGTACCCAGGTCACCGTCTCCTCAGCGCACCACAGCGAAGACCCCTCGGCGCGCCAGGCCTGCACTAGTGGTGCGCCGGT</td> <td>+;33335:4<a href="mailto:=678=8=9?98:=9?9@8C?9C@9:9:A9@9:9[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[...........................[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[9968amp#55;amp#54;amp#57;amp#52;amp#58;amp#56;amp#57;amp#49;amp#58;amp#58;amp#54;amp#57;amp#64;amp#57;amp#56;amp#56;amp#58;amp#58;amp#58;?:9:">&#61;&#54;&#55;&#56;&#61;&#56;&#61;&#57;&#63;&#57;&#56;&#58;&#61;&#57;&#63;&#57;&#64;&#56;&#67;&#63;&#57;&#67;&#64;&#57;&#58;&#57;&#58;&#65;&#57;&#64;&#57;&#58;&#57;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#46;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#91;&#57;&#57;&#54;&#56;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#53;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#52;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#55;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#50;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#54;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#55;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#52;&#57;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#52;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#55;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#54;&#52;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#55;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#54;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#54;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#2;&#97;&#109;&#112;&#3;&#35;&#53;&#56;&#59;&#63;&#58;&#57;&#58;</a>8::::99&lt;8:9:9::;997798&lt;79</td> </tr> <tr> <td>2</td> <td>113072.80475962501</td> <td>0.07342142002972953</td> <td>IGHV3-8*00(887.9)</td> <td></td> <td>IGHJ4*00(175.5)</td> <td>IGHG2B_hinge(26.6)</td> <td>0&#124;293&#124;316&#124;43&#124;336&#124;SG0CSG11CST18ASG28CSA37GSA68CSA77CSA81CSA87GSG88CSG91ASC92TSA94TST95CSC97GSC98GSG101ASG103CSA115TSG117CST144GST149CSA154CSA156CSG157CSG160ASG169CST192ASG214ASC230TSC234GST237GSA238CSG253CSC254ASA259GSA260TSG266AST278ASA290G&#124;905.0</td> <td></td> <td>24&#124;68&#124;68&#124;342&#124;386&#124;ST31CSC39TSG45T&#124;178.0</td> <td></td> <td>CAGGTGCAGCTCGTGGAGACTGGGGGAGCCTTGGTGCGGCCTGGGGGGTCTCTGAGACTCTCCTGTGCCGCCTCT</td> <td>58</td> <td>GGCTTCCCCTTCGCTAATTTCGGG</td> <td>58</td> <td>ATAACCTGGGTCCGCCTGCCTCCAGGAAAGGGGCTCGAGTGGGTCGCAGAC</td> <td>58</td> <td>ATTACTCCTGATGGTGGTACCACA</td> <td>58</td> <td>TACTATGCAGACTCCGTGAAGGGCCGGTTCACCATCTCCAAAGACAACGCCAAGAATACGGTGGCTCTGCAAATGAACACACTGAGTCCTGAAGACACGGCCGTATATTAC</td> <td>58</td> <td>TGTGCGACCCCGATGTATGACCACTGG</td> <td>13</td> <td>GGTCAGGGTACCCAGGTCACCGTCTCCTCAG</td> <td>58</td> <td>QVQLVETGGALVRPGGSLRLSCAAS</td> <td>GFPFANFG</td> <td>ITWVRLPPGKGLEWVAD</td> <td>ITPDGGTT</td> <td>YYADSVKGRFTISKDNAKNTVALQMNTLSPEDTAVYY</td> <td>CATPMYDHW</td> <td>GQGTQVTVSS_</td> <td>::::43:118:142:193:217:328:-3:336:::::342:-4:355:386::</td> <td>CCTCGCGGCCCAGCCGGCCATGGCCGGCCCGGGAGCGGCCGCTCAGGTGCAGCTCGTGGAGACTGGGGGAGCCTTGGTGCGGCCTGGGGGGTCTCTGAGACTCTCCTGTGCCGCCTCTGGCTTCCCCTTCGCTAATTTCGGGATAACCTGGGTCCGCCTGCCTCCAGGAAAGGGGCTCGAGTGGGTCGCAGACATTACTCCTGATGGTGGTACCACATACTATGCAGACTCCGTGAAGGGCCGGTTCACCATCTCCAAAGACAACGCCAAGAATACGGTGGCTCTGCAAATGAACACACTGAGTCCTGAAGACACGGCCGTATATTACTGTGCGACCCCGATGTATGACCACTGGGGTCAGGGTACCCAGGTCACCGTCTCCTCAGCGCACCACAGCGAAGACCCCTCGGCGCGCCAGGCCTGCACTAGTGGTGCGCCGGTNGT</td> <td>+&lt;33345:5==789&gt;9&gt;:@:::=;A;@9E@;EA:;:;A;@:;:[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[...........................[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[9968<a href=mailto:7694:891::69@988:::>&#55;&#54;&#57;&#52;&#58;&#56;&#57;&#49;&#58;&#58;&#54;&#57;&#64;&#57;&#56;&#56;&#58;&#58;&#58;</a>?:9:&gt;8::::99&lt;8:9:9::;997798&lt;79!.,</td> </tr> <tr> <td>3</td> <td>17367.706440640646</td> <td>0.011277350661302765</td> <td>IGHV3-3*00(1386.3)</td> <td>IGHD4(35),IGHD2(30)</td> <td>IGHJ4*00(224.4)</td> <td>IGHG2C_hinge(40.5)</td> <td>0&#124;295&#124;316&#124;141&#124;436&#124;SG11CSA14GST18ASG35TSG144TSA175T&#124;1391.0</td> <td>26&#124;33&#124;57&#124;453&#124;460&#124;&#124;35.0;54&#124;60&#124;102&#124;451&#124;457&#124;&#124;30.0</td> <td>22&#124;68&#124;68&#124;471&#124;517&#124;&#124;230.0</td> <td></td> <td>CAGGTGCAGCTCGTGGAGACTGGGGGAGGATTGGTTCAGGCTGGGGGCTCTCTGAGACTCTCCTGTGCAGCCTCT</td> <td>58</td> <td>GGACGCACCTTCAGTAGCTATGCC</td> <td>58</td> <td>ATGGGCTGGTTCCGCCAGGCTCCAGGGAAGGAGCGTGAGTTTGTATCAGCT</td> <td>58</td> <td>ATTAGCTGGAGTGGTGGTAGCACA</td> <td>58</td> <td>TTCTATGCAGACTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAACGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTAC</td> <td>58</td> <td>TGTGCAGCAGCAACGCACCGACACGATGGGTTGGCGCTAATCGGGGAGTATGACTACTGG</td> <td>9</td> <td>GGCCAGGGGACCCAGGTCACCGTCTCCTCAG</td> <td>58</td> <td>QVQLVETGGGLVQAGGSLRLSCAAS</td> <td>GRTFSSYA</td> <td>MGWFRQAPGKEREFVSA</td> <td>ISWSGGST</td> <td>FYADSVKGRFTISRDNAKNTVYLQMNSLKPEDTAVYY</td> <td>CAAATHRHDGLALIGEYDYW</td> <td>GQGTQVTVSS_</td> <td>::::141:216:240:291:315:426:-2:436:453:-7:-5:460:471:-2:486:517::</td> <td>CTCGCGGCCCAGCCGGCCATGGCCGGTTGGGCCGCGAGTAATAACAATCCAGCGGCTGCCGTAGGCAATAGGTATTTCATTTTAAATTCCTCCTGAANCCTCGCGGCCCAGCCGGCCATGGCCGGCCCGGGAGCGGCCGCTCAGGTGCAGCTCGTGGAGACTGGGGGAGGATTGGTTCAGGCTGGGGGCTCTCTGAGACTCTCCTGTGCAGCCTCTGGACGCACCTTCAGTAGCTATGCCATGGGCTGGTTCCGCCAGGCTCCAGGGAAGGAGCGTGAGTTTGTATCAGCTATTAGCTGGAGTGGTGGTAGCACATTCTATGCAGACTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAACGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTACTGTGCAGCAGCAACGCACCGACACGATGGGTTGGCGCTAATCGGGGAGTATGACTACTGGGGCCAGGGGACCCAGGTCACCGTCTCCTCAGAACCCAAGACACCAAAACCACAACCGGCGCGCCAGGCCTGCACTAGTGGTGCGCCGGTTGCGCCGGTCGGT</td> <td>CCCCCGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGCGGDDFGGGEGGGFGGGGGEEGGGGGGGGGGGGGGGGFFDFGGGGGGGGEGFA9EGFGGGG!);22335:5&gt;&lt;89;&gt;=&lt;&lt;@==&gt;&gt;?@B&gt;=G@@B@&gt;?@AC@I?F=[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[<strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[<span class=keys><kbd class=key-8>8</kbd><span>+</span><kbd class=key-7>7</kbd><span>+</span><kbd class=key-6>6</kbd><span>+</span><kbd class=key-4>4</kbd></span>++6+1+9&lt;+++++99+++?+++++<span class=keys><kbd class=key-9>9</kbd></span>+7+8</em>778++*85</strong>888[[30[26.1CCC8</td> </tr> <tr> <td>4</td> <td>14959.662854752114</td> <td>0.009713738792425266</td> <td>IGHV3-3*00(1386.3)</td> <td>IGHD4(35),IGHD2(30)</td> <td>IGHJ4*00(224.4)</td> <td>IGHG2C_hinge(40.5)</td> <td>0&#124;295&#124;316&#124;45&#124;340&#124;SG11CSA14GSG35TSG144T&#124;1419.0</td> <td>26&#124;33&#124;57&#124;357&#124;364&#124;&#124;35.0;54&#124;60&#124;102&#124;355&#124;361&#124;&#124;30.0</td> <td>22&#124;68&#124;68&#124;375&#124;421&#124;SA46T&#124;216.0</td> <td></td> <td>CAGGTGCAGCTCGTGGAGTCTGGGGGAGGATTGGTTCAGGCTGGGGGCTCTCTGAGACTCTCCTGTGCAGCCTCT</td> <td>58</td> <td>GGACGCACCTTCAGTAGCTATGCC</td> <td>58</td> <td>ATGGGCTGGTTCCGCCAGGCTCCAGGGAAGGAGCGTGAGTTTGTATCAGCT</td> <td>58</td> <td>ATTAGCTGGAGTGGTGGTAGCACA</td> <td>58</td> <td>TACTATGCAGACTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAACGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTAC</td> <td>58</td> <td>TGTGCAGCAGCAACGCACCGACACGATGGGTTGGCGCTAATCGGGGAGTATGACTACTGG</td> <td>9</td> <td>GGCCAGGGGTCCCAGGTCACCGTCTCCTCAG</td> <td>58</td> <td>QVQLVESGGGLVQAGGSLRLSCAAS</td> <td>GRTFSSYA</td> <td>MGWFRQAPGKEREFVSA</td> <td>ISWSGGST</td> <td>YYADSVKGRFTISRDNAKNTVYLQMNSLKPEDTAVYY</td> <td>CAAATHRHDGLALIGEYDYW</td> <td>GQGSQVTVSS_</td> <td>::::45:120:144:195:219:330:-2:340:357:-7:-5:364:375:-2:390:421::</td> <td>CTCCTCGCGGCCCAGCCGGCCATGGCCGGCCCGGGAGCGGCCGCTCAGGTGCAGCTCGTGGAGTCTGGGGGAGGATTGGTTCAGGCTGGGGGCTCTCTGAGACTCTCCTGTGCAGCCTCTGGACGCACCTTCAGTAGCTATGCCATGGGCTGGTTCCGCCAGGCTCCAGGGAAGGAGCGTGAGTTTGTATCAGCTATTAGCTGGAGTGGTGGTAGCACATACTATGCAGACTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAACGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTACTGTGCAGCAGCAACGCACCGACACGATGGGTTGGCGCTAATCGGGGAGTATGACTACTGGGGCCAGGGGTCCCAGGTCACCGTCTCCTCAGAACCCAAGACACCAAAACCACAACCGGCGCGCCAGGCCTGCACTAGTGGTGCGCCGGTTGCGCCGGTT</td> <td>AC+;33446&lt;6&gt;&gt;:;=A?A&gt;DA??&gt;BCAC&lt;DC@CG@C?BEBEAF=[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[<strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong>***<em>[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[<span class=keys><kbd class=key-8>8</kbd><span>+</span><kbd class=key-7>7</kbd><span>+</span><kbd class=key-6>6</kbd><span>+</span><kbd class=key-4>4</kbd></span>++6+1+9&lt;+++++99+++?+++++<span class=keys><kbd class=key-9>9</kbd></span>+7+8</em>778++48753;8:0----[-[-C</td> </tr> <tr> <td>5</td> <td>13653.505182577868</td> <td>0.008865613097855052</td> <td>IGHV3S61*00(1332.4)</td> <td>IGHD5(76)</td> <td>IGHJ7*00(210.9)</td> <td>IGHG2B_hinge(40)</td> <td>0&#124;297&#124;316&#124;63&#124;360&#124;SG0CSG39CSG103CST152ASA156CSG157AST174CSC185TSC221T&#124;1359.0,</td> <td>18&#124;36&#124;54&#124;371&#124;389&#124;SA20G&#124;76.0,</td> <td>26&#124;74&#124;74&#124;394&#124;442&#124;SG29CSA38T&#124;212.0,</td> <td>,</td> <td>CAGGTGCAGCTCGTGGAGTCTGGGGGAGGCTTGGTGCAGCCTGGGGGGTCTCTGAGACTCTCCTGTGCAGCCTCT</td> <td>58</td> <td>GGATTCACTTTGGATTATTATGCC</td> <td>58</td> <td>ATAGCCTGGTTCCGCCAGGCCCCAGGGAAGGAGCGCGAGGGGGTCTCATGT</td> <td>58</td> <td>ATAAGTCATAGTGATGGTAGCACA</td> <td>58</td> <td>CACTATGCAGATTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAATGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTAC</td> <td>58</td> <td>TGTGCGACAGATGCGCTTTCGCAGTGCGGTAGTAGCTGGTACCAAGACGCCATGGACTTCTGG</td> <td>9</td> <td>GGCAAAGGGACCCTGGTCACCGTCTCCTCAG</td> <td>58</td> <td>QVQLVESGGGLVQPGGSLRLSCAAS</td> <td>GFTLDYYA</td> <td>IAWFRQAPGKEREGVSC</td> <td>ISHSDGST</td> <td>HYADSVKGRFTISRDNAKNTVYLQMNSLKPEDTAVYY</td> <td>CATDALSQCGSSWYQDAMDFW</td> <td>GKGTLVTVSS_</td> <td>::::63:138:162:213:237:348:1:360:371:0:0:389:394:-6:411:442::,:::::::::::::::::::::</td> <td>CTCGCGGCCCNNNCGGCCNNCCTCGCGGCCCAGCCGGCCATGGCCGGCCCGGGAGCGGCCGCTCAGGTGCAGCTCGTGGAGTCTGGGGGAGGCTTGGTGCAGCCTGGGGGGTCTCTGAGACTCTCCTGTGCAGCCTCTGGATTCACTTTGGATTATTATGCCATAGCCTGGTTCCGCCAGGCCCCAGGGAAGGAGCGCGAGGGGGTCTCATGTATAAGTCATAGTGATGGTAGCACACACTATGCAGATTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAATGCCAAGAACACGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTTTATTACTGTGCGACAGATGCGCTTTCGCAGTGCGGTAGTAGCTGGTACCAAGACGCCATGGACTTCTGGGGCAAAGGGACCCTGGTCACCGTCTCCTCAGAACCCAAGACACCAAA,ACTAGTGGTGCGCCGGTTNGTCTTCGCTGTGGTGCGCCGGT</td> <td>CCCCCGGGG[!!![[[[[!!,:22235:5=&lt;78:&gt;;;;&gt;;:&lt;::@&lt;:8;@:&lt;:89;=&gt;=@9:9[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[<strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong><strong><em>*</em></strong>[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[''='&lt;':(7'''';'1,;(='&lt;9&gt;'''88(':7;'!GEEFCGFCGGFGGGGGGCCCCC</td> </tr> </tbody> </table> <p>One can customize the list of fields that will be exported. For example, in order to export just clone count, best hits for V and J genes with corresponding alignments and CDR3 amino acid sequence, one can do:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    --drop-default-fields <span class=se>\</span>
    -count <span class=se>\</span>
    -vHit -jHit <span class=se>\</span>
    -vAlignment -jAlignment <span class=se>\</span>
    -aaFeature CDR3 <span class=se>\</span>
    clones.clns <span class=se>\</span>
    clones.tsv
</code></pre></div> <p>The columns in the resulting file will be exported in exactly the same order as parameters on the command line.</p> <h3 id=umi-libraries>UMI libraries</h3> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    -uniqueTagCount umi <span class=se>\</span>
    clones.clns <span class=se>\</span>
    clones.tsv
</code></pre></div> <p>There are several options to export columns related to tagged analysis. In the above example we pass <code>-uniqueTagCount</code> option to add a column with UMI count. We also specify option to use full preset of other columns.</p> <p>It is also possible to export full list of UMIs with their read counts that were used to build a clonotype:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    -uniqueTagCount umi <span class=se>\</span>
    -tagCounts <span class=se>\</span>
    clones.clns <span class=se>\</span>
    clones.tsv
</code></pre></div> <table> <thead> <tr> <th>cloneId</th> <th>uniqueTagCountUMI</th> <th>tagCounts</th> <th>cloneCount</th> <th>...</th> </tr> </thead> <tbody> <tr> <td>1</td> <td>901</td> <td>{AGGATCTAGCTC=47.0,GATTCAGGCAAA=12.0,GTTTGTATATAG=119.0 ...</td> <td>157166.58695588264</td> <td>...</td> </tr> <tr> <td>2</td> <td>123</td> <td>{AGGGTACACCAG=12.0,GTTTAAAAATAA=42.0,ATTACAGCCTAA=19.0 ...</td> <td>113072.80475962501</td> <td>...</td> </tr> <tr> <td>3</td> <td>110</td> <td>{GCAAGCGCTGGC=40.0,TCGAAAAAAACA=42.0,AGCACAGGTGAT=113.0 ...</td> <td>17367.706440640646</td> <td>...</td> </tr> <tr> <td>4</td> <td>98</td> <td>{CGATCGAAGGAT=47.0,ACCCGCATCAGA=112.0,TCAGTTTGTAAA=1.0 ...</td> <td>14959.662854752114</td> <td>...</td> </tr> <tr> <td>5</td> <td>82</td> <td>{CTGTGGATAGTA=117.0,ATCCAGAAGCGT=12.0,ATCGGTGATCAC=93.0 ...</td> <td>13653.505182577868</td> <td>...</td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <h3 id=single-cell-libraries>Single cell libraries</h3> <p>Export paired TCR-alpha/beta or BCR-heavy/light clonotype pairs from single cell data:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    --drop-default-fields <span class=se>\</span>
    --split-by-tag cell <span class=se>\</span>
    -tag cell <span class=se>\</span>
    -cellGroup <span class=se>\</span>
    -uniqueTagCount UMI <span class=se>\</span>
    -count <span class=se>\</span>
    -vFamily -jFamily <span class=se>\</span>
    -aaFeature CDR3 <span class=se>\</span>
    -nFeatureImputed VDJRegion <span class=se>\</span>
    clones.clns <span class=se>\</span>
    clones.tsv
</code></pre></div> <p>Here we use <code>--split-by-tag</code> option to export cells that contain the same clonotype on separate rows, <code>-tag</code> to export cell barcode for each clonotype and <code>-cellGroup</code> which is cell identifier.</p> <table> <thead> <tr> <th>tagValueCELL</th> <th>cellGroup</th> <th>uniqueTagCountUMI</th> <th>cloneCount</th> <th>bestVFamily</th> <th>bestJFamily</th> <th>nSeqCDR3</th> <th>aaSeqVDJRegionImputed</th> </tr> </thead> <tbody> <tr> <td>CTAGGCTAGC</td> <td>11</td> <td>198</td> <td>10716.64</td> <td>IGLV1</td> <td>IGLJ3</td> <td>TGCGGAACATGGGATAGCAGCCTGAGTGCTTGGGTGTTC</td> <td>QVQLVESGGALVRPGGSLRLSCAASGFPFANFGITWVRLPPGKGLEWVADITPDGGTTYYADSVKGRFTISKDNAKNTVALQMNTLSPEDTAVYYCATPMYDHWGQGTQVTVSS_</td> </tr> <tr> <td>CTAGGCTAGC</td> <td>11</td> <td>93</td> <td>11372.5962501</td> <td>IGKV1</td> <td>IGKJ1</td> <td>TGTCAACAGTCTGAAAATCTCCCTCCGACGTTC</td> <td>QVQLVETGGGLVQAGGSLRLSCAASGRTFSSYAMGWFRQAPGKEREFVSAISWSGGSTFYADSVKGRFTISRDNAKNTVYLQMNSLKPEDTAVYYCAAATHRHDGLALIGEYDYWGQGTQVTVSS_</td> </tr> <tr> <td>CTAGGCTAGC</td> <td>11</td> <td>80</td> <td>17467.0640646</td> <td>IGHV2</td> <td>IGHJ5</td> <td>TGTGCACGGATACGGAGGTATAGCAGTGGCTGGTACTCAACGAACTGGTTCGACCCCTGG</td> <td>QVQLVETGGGLVQAGGSLRLSCAASGFTFDDYVIGWFRQAPGKEREGVSCINSSDGSTYYADSVKGRFTISSDNAKNTVYLQMNSLKPEDTAVYYCAAELIDRLIAIMGASCPLEYDYWGQGTQVTVSS_</td> </tr> <tr> <td>TGCTGAATCG</td> <td>187</td> <td>98</td> <td>12959.5475211</td> <td>IGKV3</td> <td>IGKJ2</td> <td>TGTCAACTCGATTGCATTGCACCTCCGACGTTC</td> <td>QVQLVETGGRLGAGWGVSETLLCLLWIQFP*I*YRVVPPGPREGA*GSWMY*FQRW*YIPSRLREGPIHHLPRQFEECGISAHEQLET*RHGRLLLCKRSGRMCCVYRGLLPRHGLLGQRDPGHRLL_</td> </tr> <tr> <td>TGCTGAATCG</td> <td>187</td> <td>82</td> <td>11653.182577868</td> <td>IGHV1</td> <td>IGHJ2</td> <td>TGTGCACTACGTAGCAAGGTATAGCAGCTAGGCTGCTGGTGCAACTAGGCTAGCTTCGACCCCTGG</td> <td>QLQLVESGGGLVQAGGSLRLSCAASGRTDSRYTMGWFRQAPGKEREIVAQISPFGGNQYYADSVKGRFTISRDNAKNTVYLQMNSLKAEDTAVYYCYAEGPGRWVAGTWTRDYWGQGTQVTISS_</td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <p>In the above example we specified particular columns to export. To export all columns one can use simply:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    --split-by-tag cell <span class=se>\</span>
    -tag cell <span class=se>\</span>
    -cellGroup <span class=se>\</span>
    -uniqueTagCount UMI <span class=se>\</span>
    clones.clns <span class=se>\</span>
    clones.tsv
</code></pre></div> <h3 id=export-contigs-with-imputation>Export contigs with imputation</h3> <p>When V-D-J contigs assembled with <a href=../mixcr-assembleContigs/ >assembleContigs</a> does not cover all gene features, it still might be useful to impute non-covered parts from germline (for example for the purposes of synthesis). Typically, there may be uncovered parts in <code>VDJRegion</code> for example due to long CDR3 region and non-overlapping R1 or R2, or in case of fragmented data (RNA-Seq or 10x single cell).</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    -aaFeatureImputed VDJRegion <span class=se>\</span>
    -nFeatureImputed VDJRegion <span class=se>\</span>
    clones.clns clones.tsv  
</code></pre></div> <p>MiXCR allows to export gene features with imputation using <code>-nFeatureImputed</code> and <code>-aaFeatureImputed</code> export fields. The resulting sequences will have imputed letters in lower case:</p> <table> <thead> <tr> <th>cloneId</th> <th>aaSeqImputedVDJRegion</th> <th>nSeqImputedVDJRegion</th> <th>cloneCount</th> <th>cloneFraction</th> <th>allVHitsWithScore</th> <th>...</th> </tr> </thead> <tbody> <tr> <td>1550</td> <td>evqlvesggglvqpggslrlscaasgftfssyamswvrqapgkgpEXXAAITSGGIXXXXXX<em>RAXHHLQRQCQR</em>E<em>GVSANEQPET</em>GHGRLLLRSGRI*ARLLGPGDPGHCLL_</td> <td>gaggtgcagctggtggagtctgggggaggcttggtgcagcctggggggtctctgagactctcctgtgcagcctctggattcaccttcagtagctatgccatgagctgggtccgccaggctccaggaaaggggcccgAGTNGGNCGCAGCTATTACTAGTGGTGGTATCNNAANATGNNANNCNNCNTGAAGGGCNNTTCACCATCTCCAGAGACAATGTCAACGCTAAGAATAGGGTGTATCTGCAAATGAACAGCCTGAAACCTGAGGACACGGCCGTCTATTACTGCGAAGCGGTAGGATATGAGCTCGACTACTGGGGCCAGGGGACCCAGGTCACTGTCTCCTCAG</td> <td>14959.662854752114</td> <td>0.009713738792425266</td> <td>IGHV3-3*00(1386.3)</td> <td>...</td> </tr> <tr> <td>2739</td> <td>QVQLVETGGALVHPGGSLRLSCVVSGFT<em>IIMP</em>PGSARSRGRSVRGSXVLvvvmvahtmqtp<em>radspspettprtrcick</em>ta*nlrtrpfiTAQQPTSGRAMKVVSTRNSMITGARGPRSPSLQ</td> <td>CAGGTGCAGCTCGTGGAGACGGGTGGAGCGTTGGTGCACCCTGGGGGGTCGCTGAGACTCTCCTGTGTCGTCTCTGGATTCACTTGAATTATTATGCCATAGCCTGGTTCCGCCAGGTCCCGGGGAAGGAGCGTGAGGGGATCTCANGTATTAGtagtagtgatggtagcacatactatgcagactccgtgaagggccgattcaccatctccagagacaacgccaagaacacggtgtatctgcaaatgaacagcctgaaacctgaggacacggccgtttattacTGCGCAGCAGCCCACTTCGGGGCGTGCTATGAAGGTAGTTTCGACGCGAAACTCTATGATCACTGGGGCCAGGGGACCCAGGTCACCGTCTCTTCAG</td> <td>17367.706440640646</td> <td>0.011277350661302765</td> <td>IGHV3-3*00(1386.3)</td> <td>...</td> </tr> <tr> <td>2940</td> <td>QVQLVEXGGGLXQXGGSLKLSCAASGLTFDDYAIGWFRQVPGKXREGIXCVGSRGXxyyadsvXXXXXXXIXXXXXTXSLDLNSLIPEDTATYQCAAVTSDLGCTHYMLQNDIEYDYWGRGTQVTVST_</td> <td>CAGGTGCAGCTCGTGGAGNCNGGGGGAGGNTTGGNGCAGNCTGGGGGGTCTCTGAAACTCTCCTGTGCAGCCTCTGGACTCACTTTCGACGATTATGCCATCGGCTGGTTCCGNCAGGTTCCAGGGAAGNAGCGCGAGGGGATCTGNTGTGTCGGTAGTCGAGGNNNCNCatactatgcagactccgtgaANNNNNGATTNNNCATNNCCATTGNCANCGNCAANNACACGNAGTCTCTGGATTTGAACAGCCTGATCCCTGAGGACACGGCCACATATCAATGTGCGGCAGTCACCTCGGACCTGGGATGTACGCACTATATGTTGCAGAATGATATCGAGTATGACTACTGGGGCCGGGGGACCCAGGTCACCGTCTCGACAG</td> <td>157166.58695588264</td> <td>0.10205277935802338</td> <td>IGHV3-8*00(887.9)</td> <td>...</td> </tr> <tr> <td>3283</td> <td>qvqlvesggglvqaggslrlscaasgrtfssyamgwfrqapgkerefvaaiswXGSTXXYADSAKDRFVISRDNGKNMAYLXLTSLKPDDTGIYLCAADIQCQTDPRHLPFGSWGWGQGTQVTVSS_</td> <td>caggtgcagctggtagagtctgggggaggattggtgcaggctgggggctctctgagactctcctgtgcagcctctggacgcaccttcagtagctatgccatgggctggttccgccaggctccagggaaggagcgtgagtttgtagcagctattagctggANAGGTTCAACCANGNACTATGCCGACTCCGCGAAGGACCGATTCGTCATTTCCAGAGACAACGGCAAGAACATGGCGTACTTGTANTTAACCAGCCTGAAGCCTGACGACACTGGCATTTATCTCTGTGCGGCGGACATCCAGTGTCAGACTGACCCCCGTCATCTCCCTTTTGGTTCCTGGGGTTGGGGCCAGGGGACGCAAGTCACCGTCTCCTCGG</td> <td>113072.80475962501</td> <td>0.07342142002972953</td> <td>IGHV3-8*00(887.9)</td> <td>...</td> </tr> </tbody> </table> <p>One can also use default presets with imputation (all gene features will use imputation option):</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportClones <span class=se>\</span>
    --drop-default-fields <span class=se>\</span>
    -aaFeatureImputed VDJRegion <span class=se>\</span>
    -nFeatureImputed VDJRegion <span class=se>\</span>
    clones.clns clones.tsv  
</code></pre></div> <h3 id=shm-trees-export-presets>SHM trees export presets</h3> <p>For convenience, MiXCR provides two predefined sets of fields for exporting SHM trees: <code>min</code> (will export minimal required information about tree or tree nodes) and <code>full</code> (used by default); one can use these sets by specifying the <code>--preset</code> option:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportShmTrees --preset min trees.shmt trees.tsv
</code></pre></div> <p>One can add additional columns to the preset in the following way:</p> <div class=highlight><pre><span></span><code>&gt; mixcr exportShmTreesWithNodes --preset min -qFeature CDR2 trees.shmt trees_with_nodes.tsv
</code></pre></div> <p>One can also put all specify export fields in a separate file</p> <div class=highlight><pre><span></span><code>&gt; cat myFields.txt
-vHits
-dHits
-nFeature CDR3

&gt; mixcr exportShmTreesWithNodes --preset-file myFields.txt trees.shmt trees_with_nodes.tsv
</code></pre></div> <h2 id=export-fields>Export fields</h2> <h3 id=common-fields>Common fields</h3> <p>These fields available for <code>exportAlignments</code>, <code>exportClones</code> and <code>exportShmTreesWithNodes</code> (for <code>exportShmTreesWithNodes</code> only for nodes with clones):</p> <dl> <dt><code>-targets</code></dt> <dd>Export number of targets</dd> <dt><code>-vHit</code></dt> <dd>Export best V hit</dd> <dt><code>-dHit</code></dt> <dd>Export best D hit</dd> <dt><code>-jHit</code></dt> <dd>Export best J hit</dd> <dt><code>-cHit</code></dt> <dd>Export best C hit</dd> <dt><code>-vGene</code></dt> <dd>Export best V hit gene name (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-dGene</code></dt> <dd>Export best D hit gene name (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-jGene</code></dt> <dd>Export best J hit gene name (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-cGene</code></dt> <dd>Export best C hit gene name (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-vFamily</code></dt> <dd>Export best V hit family name (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-dFamily</code></dt> <dd>Export best D hit family name (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-jFamily</code></dt> <dd>Export best J hit family name (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-cFamily</code></dt> <dd>Export best C hit family name (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-vHitScore</code></dt> <dd>Export score for best V hit</dd> <dt><code>-dHitScore</code></dt> <dd>Export score for best D hit</dd> <dt><code>-jHitScore</code></dt> <dd>Export score for best J hit</dd> <dt><code>-cHitScore</code></dt> <dd>Export score for best C hit</dd> <dt><code>-vHitsWithScore</code></dt> <dd>Export all V hits with score</dd> <dt><code>-dHitsWithScore</code></dt> <dd>Export all D hits with score</dd> <dt><code>-jHitsWithScore</code></dt> <dd>Export all J hits with score</dd> <dt><code>-cHitsWithScore</code></dt> <dd>Export all C hits with score</dd> <dt><code>-vHits</code></dt> <dd>Export all V hits</dd> <dt><code>-dHits</code></dt> <dd>Export all D hits</dd> <dt><code>-jHits</code></dt> <dd>Export all J hits</dd> <dt><code>-cHits</code></dt> <dd>Export all C hits</dd> <dt><code>-vGenes</code></dt> <dd>Export all V gene names (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-dGenes</code></dt> <dd>Export all D gene names (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-jGenes</code></dt> <dd>Export all J gene names (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-cGenes</code></dt> <dd>Export all C gene names (e.g. TRBV12-3 for TRBV12-3*00)</dd> <dt><code>-vFamilies</code></dt> <dd>Export all V gene family anmes (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-dFamilies</code></dt> <dd>Export all D gene family anmes (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-jFamilies</code></dt> <dd>Export all J gene family anmes (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-cFamilies</code></dt> <dd>Export all C gene family anmes (e.g. TRBV12 for TRBV12-3*00)</dd> <dt><code>-vAlignment</code></dt> <dd>Export best V alignment</dd> <dt><code>-dAlignment</code></dt> <dd>Export best D alignment</dd> <dt><code>-jAlignment</code></dt> <dd>Export best J alignment</dd> <dt><code>-cAlignment</code></dt> <dd>Export best C alignment</dd> <dt><code>-vAlignments</code></dt> <dd>Export all V alignments</dd> <dt><code>-dAlignments</code></dt> <dd>Export all D alignments</dd> <dt><code>-jAlignments</code></dt> <dd>Export all J alignments</dd> <dt><code>-cAlignments</code></dt> <dd>Export all C alignments</dd> <dt><code>-qFeature &lt;gene_feature&gt;</code></dt> <dd>Export quality string of specified gene feature</dd> <dt><code>-nFeatureImputed &lt;gene_feature&gt;</code></dt> <dd>Export nucleotide sequence of specified gene feature using letters from germline (marked lowercase) for uncovered regions</dd> <dt><code>-allNFeaturesImputed [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export nucleotide sequence using letters from germline (marked lowercase) for uncovered regions for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allNFeaturesImputed FR3Begin FR4End</code> will export <code>-nFeatureImputed FR3</code>, <code>-nFeatureImputed CDR3</code>, <code>-nFeatureImputed FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-aaFeatureImputed &lt;gene_feature&gt;</code></dt> <dd>Export amino acid sequence of specified gene feature using letters from germline (marked lowercase) for uncovered regions</dd> <dt><code>-allAaFeaturesImputed [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export amino acid sequence using letters from germline (marked lowercase) for uncovered regions for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allAaFeaturesImputed FR3Begin FR4End</code> will export <code>-aaFeatureImputed FR3</code>, <code>-aaFeatureImputed CDR3</code>, <code>-aaFeatureImputed FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-minFeatureQuality &lt;gene_feature&gt;</code></dt> <dd>Export minimal quality of specified gene feature</dd> <dt><code>-allMinFeaturesQuality [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export minimal quality for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allMinFeaturesQuality FR3Begin FR4End</code> will export <code>-minFeatureQuality FR3</code>, <code>-minFeatureQuality CDR3</code>, <code>-minFeatureQuality FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-allNFeaturesWithMinQuality [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export nucleotide sequences and minimal quality for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allNFeaturesWithMinQuality FR3Begin FR4End</code> will export <code>-nFeature FR3</code>, <code>-minFeatureQuality FR3</code>, <code>-nFeature CDR3</code>, <code>-minFeatureQuality CDR3</code>, <code>-nFeature FR4</code>, <code>-minFeatureQuality FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-allNFeaturesImputedWithMinQuality [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export nucleotide sequences and minimal quality for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allNFeaturesImputedWithMinQuality FR3Begin FR4End</code> will export <code>-nFeatureImputed FR3</code>, <code>-minFeatureQuality FR3</code>, <code>-nFeatureImputed CDR3</code>, <code>-minFeatureQuality CDR3</code>, <code>-nFeatureImputed FR4</code>, <code>-minFeatureQuality FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-avrgFeatureQuality &lt;gene_feature&gt;</code></dt> <dd>Export average quality of specified gene feature</dd> <dt><code>-allAvrgFeaturesQuality [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export average quality for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allAvrgFeaturesQuality FR3Begin FR4End</code> will export <code>-avrgFeatureQuality FR3</code>, <code>-avrgFeatureQuality CDR3</code>, <code>-avrgFeatureQuality FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-positionInReferenceOf &lt;reference_point&gt;</code></dt> <dd>Export position of specified reference point inside reference sequences (clonal sequence / read sequence).</dd> <dt><code>-allPositionsInReference [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export position inside reference sequences (clonal sequence / read sequence) for all reference points between specified (in separate columns).</dd> </dl> <p>For example, <code>-allPositionsInReference FR3Begin FR4End</code> will export <code>-positionInReferenceOf FR3Begin</code>, -positionInReferenceOf CDR3Begin<code>, -positionInReferenceOf CDR3End</code> and <code>-positionInReferenceOf FR4End</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-positionOf &lt;reference_point&gt;</code></dt> <dd>Export position of specified reference point inside target sequences (clonal sequence / read sequence).</dd> <dt><code>-allPositions [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export position inside target sequences (clonal sequence / read sequence) for all reference points between specified (in separate columns).</dd> </dl> <p>For example, <code>-allPositions FR3Begin FR4End</code> will export <code>-positionOf FR3Begin</code>, -positionOf CDR3Begin<code>, -positionOf CDR3End</code> and <code>-positionOf FR4End</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-defaultAnchorPoints</code></dt> <dd>Outputs a list of default reference points (like CDR2Begin, FR4End, etc. see documentation bellow for the full list and formatting)</dd> <dt><code>-targetSequences</code></dt> <dd>Export aligned sequences (targets), separated with comma</dd> <dt><code>-targetQualities</code></dt> <dd>Export aligned sequence (target) qualities, separated with comma</dd> <dt><code>-vIdentityPercents</code></dt> <dd>V alignment identity percents</dd> <dt><code>-dIdentityPercents</code></dt> <dd>D alignment identity percents</dd> <dt><code>-jIdentityPercents</code></dt> <dd>J alignment identity percents</dd> <dt><code>-cIdentityPercents</code></dt> <dd>C alignment identity percents</dd> <dt><code>-vBestIdentityPercent</code></dt> <dd>V best alignment identity percent</dd> <dt><code>-dBestIdentityPercent</code></dt> <dd>D best alignment identity percent</dd> <dt><code>-jBestIdentityPercent</code></dt> <dd>J best alignment identity percent</dd> <dt><code>-cBestIdentityPercent</code></dt> <dd>C best alignment identity percent</dd> <dt><code>-chains</code></dt> <dd>Chains</dd> <dt><code>-topChains</code></dt> <dd>Top chains</dd> <dt><code>-geneLabel &lt;label&gt;</code></dt> <dd>Export gene label (i.e. ReliableChain)</dd> <dt><code>-tagCounts</code></dt> <dd>All tags with counts</dd> <dt><code>-tag &lt;tag_name&gt;</code></dt> <dd>Tag value (i.e. CELL barcode or UMI sequence)</dd> <dt><code>-allTags &lt;(Sample|Cell|Molecule)&gt;</code></dt> <dd>Tag values (i.e. CELL barcode or UMI sequence) for all available tags in separate columns.</dd> </dl> <p>Tag type will be used for filtering tags for export.</p> <dl> <dt><code>-uniqueTagCount &lt;tag_name&gt;</code></dt> <dd>Unique tag count</dd> <dt><code>-allUniqueTagsCount &lt;(Sample|Cell|Molecule)&gt;</code></dt> <dd>Unique tag count for all available tags in separate columns.</dd> </dl> <p>Tag type will be used for filtering tags for export.</p> <p>These fields available for both <code>exportAlignments</code> and <code>exportClones</code>:</p> <dl> <dt><code>-lengthOf &lt;gene_feature&gt;</code></dt> <dd>Export length of specified gene feature.</dd> <dt><code>-allLengthOf [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export length for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allLengthOf FR3Begin FR4End</code> will export <code>-lengthOf FR3</code>, <code>-lengthOf CDR3</code>, <code>-lengthOf FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-nFeature &lt;gene_feature&gt;</code></dt> <dd>Export nucleotide sequence of specified gene feature</dd> <dt><code>-allNFeatures [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export nucleotide sequences for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allNFeatures FR3Begin FR4End</code> will export <code>-nFeature FR3</code>, <code>-nFeature CDR3</code>, <code>-nFeature FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-aaFeature &lt;gene_feature&gt;</code></dt> <dd>Export amino acid sequence of specified gene feature</dd> <dt><code>-allAaFeatures [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Export amino acid sequence for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allAaFeatures FR3Begin FR4End</code> will export <code>-aaFeature FR3</code>, <code>-aaFeature CDR3</code>, <code>-aaFeature FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-nMutations &lt;gene_feature&gt;</code></dt> <dd>Extract nucleotide mutations for specific gene feature; relative to germline sequence.</dd> <dt><code>-allNMutations [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Extract nucleotide mutations relative to germline sequence for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allNMutations FR3Begin FR4End</code> will export <code>-nMutations FR3</code>, <code>-nMutations CDR3</code>, <code>-nMutations FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-nMutationsRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt;</code></dt> <dd>Extract nucleotide mutations for specific gene feature relative to another feature.</dd> <dt><code>-aaMutations &lt;gene_feature&gt;</code></dt> <dd>Extract amino acid mutations for specific gene feature</dd> <dt><code>-allAaMutations [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Extract amino acid nucleotide mutations relative to germline sequence for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allAaMutations FR3Begin FR4End</code> will export <code>-aaMutations FR3</code>, <code>-aaMutations CDR3</code>, <code>-aaMutations FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-aaMutationsRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt;</code></dt> <dd>Extract amino acid mutations for specific gene feature relative to another feature.</dd> <dt><code>-mutationsDetailed &lt;gene_feature&gt;</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations. Format <code>&lt;nt_mutation&gt;:&lt;aa_mutation_individual&gt;:&lt;aa_mutation_cumulative&gt;</code>, where <code>&lt;aa_mutation_individual&gt;</code> is an expected amino acid mutation given no other mutations have occurred, and <code>&lt;aa_mutation_cumulative&gt;</code> amino acid mutation is the observed amino acid mutation combining effect from all others.</dd> <dt><code>-allMutationsDetailed [&lt;from_reference_point&gt; &lt;to_reference_point&gt;]</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations for all gene features between specified reference points (in separate columns).</dd> </dl> <p>For example, <code>-allMutationsDetailed FR3Begin FR4End</code> will export <code>-mutationsDetailed FR3</code>, <code>-mutationsDetailed CDR3</code>, <code>-mutationsDetailed FR4</code>.</p> <p>By default, boundaries will be got from analysis parameters if possible or <code>FR1Begin FR4End</code> otherwise.</p> <dl> <dt><code>-mutationsDetailedRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt;</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations written, positions relative to specified gene feature. Format <nt_mutation>:<aa_mutation_individual>:<aa_mutation_cumulative>, where <aa_mutation_individual> is an expected amino acid mutation given no other mutations have occurred, and <aa_mutation_cumulative> amino acid mutation is the observed amino acid mutation combining effect from all other. WARNING: format may change in following versions.</dd> </dl> <h3 id=alignment-specific-fields>Alignment-specific fields</h3> <p>The following fields are only available for <code>exportAlignments</code>:</p> <dl> <dt><code>-readIds</code></dt> <dd>Id(s) of read(s) corresponding to alignment</dd> <dt><code>-descrsR1</code></dt> <dd>Export description lines from initial <code>.fasta</code> or <code>.fastq</code> file for R1 reads (only available if <code>-OsaveOriginalReads= true</code> was used in <code>align</code> command)</dd> <dt><code>-descrsR2</code></dt> <dd>Export description lines from initial <code>.fastq</code> file for R2 reads (only available if <code>-OsaveOriginalReads=true</code> was used in <code>align</code>n command)</dd> <dt><code>-readHistory</code></dt> <dd>Read history</dd> <dt><code>-cloneId</code></dt> <dd>To which clone alignment was attached (make sure using <code>.clna</code> file as input for <code>exportAlignments</code>)</dd> <dt><code>-cloneIdWithMappingType</code></dt> <dd>To which clone alignment was attached with additional info on mapping type (make sure using <code>.clna</code> file as input for <code>exportAlignments</code>)</dd> </dl> <h3 id=clonotype-specific-fields>Clonotype-specific fields</h3> <p>The following fields are available for <code>exportClones</code> and <code>exportShmTreesWithNodes</code> (for <code>exportShmTreesWithNodes</code> only for nodes with clones):</p> <dl> <dt><code>-cloneId</code></dt> <dd>Unique clone identifier</dd> <dt><code>-readCount</code></dt> <dd>Number of reads assigned to the clonotype</dd> <dt><code>-readFraction</code></dt> <dd>Fraction of reads assigned to the clonotype</dd> <dt><code>-uniqueTagFraction &lt;tag_name&gt;</code></dt> <dd>Fraction of unique tags (UMI, CELL, etc.) the clone or alignment collected.</dd> <dt><code>-allUniqueTagFractions &lt;(Sample|Cell|Molecule)&gt;</code></dt> <dd>Fractions of unique tags (i.e. CELL barcode or UMI sequence) for all available tags in separate columns.</dd> </dl> <p>Tag type will be used for filtering tags for export.</p> <dl> <dt><code>-cellGroup</code></dt> <dd>Cell group (for single cell analysis)</dd> </dl> <h3 id=shm-tree-specific-fields>SHM tree-specific fields</h3> <p>The following fields are available for both <code>exportShmTrees</code> and <code>exportShmTreesWithNodes</code>:</p> <dl> <dt><code>-treeId</code></dt> <dd>SHM tree id</dd> <dt><code>-uniqClonesCount</code></dt> <dd>Number of uniq clones in the SHM tree</dd> <dt><code>-totalClonesCount</code></dt> <dd>Total sum of counts of clones in the SHM tree</dd> </dl> <p>The following fields are only available for <code>exportShmTrees</code>:</p> <dl> <dt><code>-nFeature &lt;gene_feature&gt; &lt;germline|mrca&gt;</code></dt> <dd>Export nucleotide sequence of specified gene feature of specified node type.</dd> <dt><code>-allNFeatures &lt;(germline|mrca)&gt;</code></dt> <dd>Export nucleotide sequences for all covered gene features.</dd> <dt><code>-aaFeature &lt;gene_feature&gt; &lt;germline|mrca&gt;</code></dt> <dd>Export amino acid sequence of specified gene feature of specified node type.</dd> <dt><code>-allAaFeatures &lt;(germline|mrca)&gt;</code></dt> <dd>Export nucleotide sequences for all covered gene features.</dd> </dl> <h3 id=shm-tree-node-specific-fields>SHM tree node-specific fields</h3> <p>The following fields are available only for <code>exportShmTreesWithNodes</code>:</p> <dl> <dt><code>-nodeId</code></dt> <dd>Node id in SHM tree</dd> <dt><code>-isObserved</code></dt> <dd>Is node have clones. All other nodes are reconstructed by algorithm</dd> <dt><code>-parentId</code></dt> <dd>Parent node id in SHM tree</dd> <dt><code>-distance &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Distance from another node</dd> <dt><code>-fileName</code></dt> <dd>Name of clns file with sample</dd> <dt><code>-nFeature &lt;gene_feature&gt; [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export nucleotide sequence of specified gene feature. If second arg is omitted, then feature will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-allNFeatures [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export nucleotide sequences for all covered gene features. If second arg is omitted, then feature will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-aaFeature &lt;gene_feature&gt; [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export amino acid sequence of specified gene feature. If second arg is omitted, then feature will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-allAaFeatures [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export amino acid sequences for all covered gene features. If second arg is omitted, then feature will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-lengthOf &lt;gene_feature&gt; [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export length of specified gene feature. If second arg is omitted, then feature length will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-allLengthOf [&lt;(germline|mrca|parent)&gt;]</code></dt> <dd>Export lengths for all covered gene features. If second arg is omitted, then feature will be printed for current node. Otherwise - for corresponding <code>parent</code>, <code>germline</code> or <code>mrca</code></dd> <dt><code>-nMutations &lt;gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract nucleotide mutations from specific node for specific gene feature.</dd> <dt><code>-allNMutations &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract nucleotide mutations from specific node for all covered gene features.</dd> <dt><code>-nMutationsRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract nucleotide mutations from specific node for specific gene feature relative to another feature.</dd> <dt><code>-aaMutations &lt;gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract amino acid mutations from specific node for specific gene feature.</dd> <dt><code>-allAaMutations &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract amino acid mutations from specific node for all covered gene features.</dd> <dt><code>-aaMutationsRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Extract amino acid mutations from specific node for specific gene feature relative to another feature.</dd> <dt><code>-mutationsDetailed &lt;gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations from specific node. Format <code>&lt;nt_mutation&gt;:&lt;aa_mutation_individual&gt;:&lt;aa_mutation_cumulative&gt;</code>, where <code>&lt;aa_mutation_individual&gt;</code> is an expected amino acid mutation given no other mutations have occurred, and <code>&lt;aa_mutation_cumulative&gt;</code> amino acid mutation is the observed amino acid mutation combining effect from all others.</dd> <dt><code>-allMutationsDetailed &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations from specific node for all covered gene features.</dd> <dt><code>-mutationsDetailedRelative &lt;gene_feature&gt; &lt;relative_to_gene_feature&gt; &lt;(germline|mrca|parent)&gt;</code></dt> <dd>Detailed list of nucleotide and corresponding amino acid mutations written, positions relative to specified gene feature. Format <nt_mutation>:<aa_mutation_individual>:<aa_mutation_cumulative>, where <aa_mutation_individual> is an expected amino acid mutation given no other mutations have occurred, and <aa_mutation_cumulative> amino acid mutation is the observed amino acid mutation combining effect from all other. WARNING: format may change in following versions.</dd> </dl> <h2 id=default-anchor-point-positions>Default anchor point positions</h2> <p>Positions of anchor points produced by the <code>-defaultAnchorPoints</code> option are outputted as a colon separated list. If an anchor point is not covered by the target sequence nothing is printed for it, but flanking colon symbols are preserved to maintain positions in array. See example:</p> <div class=highlight><pre><span></span><code>:::::::::108:117:125:152:186:213:243:244:
</code></pre></div> <p>If there are several target sequences (e.g. paired-end reads or multi-part clonal sequnce), an array is outputted for each target sequence. In this case arrays are separated by a comma:</p> <div class=highlight><pre><span></span><code><span class=m>2</span>:61:107:107:118:::::::::::::,:::::::::103:112:120:147:181:208:238:239:
</code></pre></div> <p>Even if there are no anchor points in one of the parts:</p> <div class=highlight><pre><span></span><code>:::::::::::::::::,:::::::::108:117:125:152:186:213:243:244:
</code></pre></div> <p>The following table shows the correspondence between anchor points and positions in the default anchor point array:</p> <table> <thead> <tr> <th>Anchors point</th> <th>Zero-based position</th> <th>One-based position</th> </tr> </thead> <tbody> <tr> <td>V5UTRBeginTrimmed</td> <td>0</td> <td>1</td> </tr> <tr> <td>V5UTREnd / L1Begin</td> <td>1</td> <td>2</td> </tr> <tr> <td>L1End / VIntronBegin</td> <td>2</td> <td>3</td> </tr> <tr> <td>VIntronEnd / L2Begin</td> <td>3</td> <td>4</td> </tr> <tr> <td>L2End / FR1Begin</td> <td>4</td> <td>5</td> </tr> <tr> <td>FR1End / CDR1Begin</td> <td>5</td> <td>6</td> </tr> <tr> <td>CDR1End / FR2Begin</td> <td>6</td> <td>7</td> </tr> <tr> <td>FR2End / CDR2Begin</td> <td>7</td> <td>8</td> </tr> <tr> <td>CDR2End / FR3Begin</td> <td>8</td> <td>9</td> </tr> <tr> <td>FR3End / CDR3Begin</td> <td>9</td> <td>10</td> </tr> <tr> <td>Number of 3 V deletions (negative value), or length of 3 V P-segment (positive value)</td> <td>10</td> <td>11</td> </tr> <tr> <td>VEndTrimmed, next position after last aligned nucleotide of V gene</td> <td>11</td> <td>12</td> </tr> <tr> <td>DBeginTrimmed, position of first aligned nucleotide of D gene</td> <td>12</td> <td>13</td> </tr> <tr> <td>Number of 5 D deletions (negative value), or length of 5 D P-segment (positive value)</td> <td>13</td> <td>14</td> </tr> <tr> <td>Number of 3 D deletions (negative value), or length of 3 D P-segment (positive value)</td> <td>14</td> <td>15</td> </tr> <tr> <td>DEndTrimmed, next position after last aligned nucleotide of D gene</td> <td>15</td> <td>16</td> </tr> <tr> <td>JBeginTrimmed, position of first aligned nucleotide of J gene</td> <td>16</td> <td>17</td> </tr> <tr> <td>Number of 3 J deletions (negative value), or length of 3 J P-segment (positive value)</td> <td>17</td> <td>18</td> </tr> <tr> <td>CDR3End / FR4Begin</td> <td>18</td> <td>19</td> </tr> <tr> <td>FR4End</td> <td>19</td> <td>20</td> </tr> <tr> <td>CBegin</td> <td>20</td> <td>21</td> </tr> <tr> <td>CExon1End</td> <td>21</td> <td>22</td> </tr> </tbody> </table> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../mixcr-assembleContigs/ class="md-footer__link md-footer__link--prev" aria-label="Previous: assembleContigs" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> assembleContigs </div> </div> </a> <a href=../mixcr-exportPretty/ class="md-footer__link md-footer__link--next" aria-label="Next: exportPretty" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> exportPretty </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../../../javascripts/svg.js></script> <script src=../../../javascripts/onload.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>