<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="MiXCR: ultimate software platform for analysis of Next-Generation Sequencing (NGS) data for immune profiling."><meta name=author content="MiLaboratories Inc"><link href=https://docs.milaboratories.com/mixcr/guides/nebnext-bcr/ rel=canonical><link rel=icon href=../../../assets/favicon.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>NEBNext Immune Sequencing - Docs @ MiLaboratories</title><link rel=stylesheet href=../../../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link rel=stylesheet href=../../../css/extra.css><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-2N4Z95ZG08"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-2N4Z95ZG08",{page_path:e.pathname})})})</script><script async src="https://www.googletagmanager.com/gtag/js?id=G-2N4Z95ZG08"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#nebnext-immune-sequencing class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../.. title="Docs @ MiLaboratories" class="md-header__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Docs @ MiLaboratories </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> NEBNext Immune Sequencing </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z"/> </svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z"/> </svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label="Search by name or category" placeholder="Search by name or category" autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../about/ class=md-tabs__link> About </a> </li> <li class=md-tabs__item> <a href=../../getting-started/installation/ class=md-tabs__link> Downloads </a> </li> <li class=md-tabs__item> <a href=../milaboratories-human-bcr-rna-multi/ class="md-tabs__link md-tabs__link--active"> Guides </a> </li> <li class=md-tabs__item> <a href=../../reference/overview-analysis-overview/ class=md-tabs__link> Docs </a> </li> <li class=md-tabs__item> <a href=https://vdj.online/ class=md-tabs__link> VDJ.online </a> </li> <li class=md-tabs__item> <a href=https://github.com/milaboratory/mixcr/discussions class=md-tabs__link> Discussions </a> </li> <li class=md-tabs__item> <a href=https://github.com/milaboratory/mixcr class=md-tabs__link> Github </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav class="md-nav md-nav--primary md-nav--lifted has-section" aria-label=Navigation data-md-level=0 data-test=1> <!-- Site title --> <label class=md-nav__title for=__drawer> <a href=../../.. title="Docs @ MiLaboratories" class="md-nav__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class=md-nav__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class=md-nav__list data-md-scrollfix> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../about/ class=md-nav__link> About </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_2 type=checkbox id=__nav_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_2 data-level=1> Downloads <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Downloads data-md-level=1> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Downloads </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/installation/ class=md-nav__link> Binary distributions </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/docker/ class=md-nav__link> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/milm/ class=md-nav__link> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/first-run/ class=md-nav__link> First run </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/license/ class=md-nav__link> License </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/support/ class=md-nav__link> Support </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3 data-level=1> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Guides data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_1 data-level=2> Kits & Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Kits & Protocols" data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Kits & Protocols </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-bcr-rna-multi/ class=md-nav__link> MiLaboratories Human IG RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-tcr-rna-multi/ class=md-nav__link> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milab-dna-multiplex-tcr/ class=md-nav__link> MiLaboratories Human TCR DNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-hsa-bcr/ class=md-nav__link> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-mmu-tcr/ class=md-nav__link> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../iRepertoire/ class=md-nav__link> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../qiaseq-tcr/ class=md-nav__link> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../cellecta-air-tcr-bcr/ class=md-nav__link> Cellecta AIR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../abhelix-bcr/ class=md-nav__link> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../biomed2-bcr/ class=md-nav__link> BioMed2 </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for=__toc> NEBNext Immune Sequencing <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> NEBNext Immune Sequencing </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-libraries class=md-nav__link> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#under-the-hood-pipeline class=md-nav__link> Under the hood pipeline: </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Under the hood pipeline:"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#align class=md-nav__link> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#refinetagsandsort class=md-nav__link> refineTagsAndSort </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#assemble class=md-nav__link> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export class=md-nav__link> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#full-length-clonotype-assembly class=md-nav__link> Full-length clonotype assembly </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#reports class=md-nav__link> Reports </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!-- Main navigation item --> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_2 data-level=2> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Analysis cases" data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-tcr/ class=md-nav__link> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-multiplex-bcr/ class=md-nav__link> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-bcr/ class=md-nav__link> Targeted BCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../rnaseq/ class=md-nav__link> RNA-Seq data </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ig-trees-reconstruction/ class=md-nav__link> Immunoglobulin lineage trees reconstruction </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_3 type=checkbox id=__nav_3_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_3 data-level=2> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Special guides" data-md-level=2> <label class=md-nav__title for=__nav_3_3> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../create-custom-library/ class=md-nav__link> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../external-libraries/ class=md-nav__link> Using external libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../b-cell-lineages-webinar/ class=md-nav__link> B cell lineage tracing </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4 data-level=1> Docs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Docs data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Docs </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_1 data-level=2> Overview <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Overview data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-analysis-overview/ class=md-nav__link> Analysis overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-presets/ class=md-nav__link> Presets </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-mixins-list/ class=md-nav__link> Mix-in options list </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-built-in-presets/ class=md-nav__link> Built-in presets </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_2 data-level=2> Data quality <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Data quality" data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> Data quality </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/qc-overview/ class=md-nav__link> When the analysis results look poor </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/qc-sequencing-quality/ class=md-nav__link> Sequencing quality </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/qc-repseq/ class=md-nav__link> Rep-Seq libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/qc-rnaseq/ class=md-nav__link> Non-targeted data </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_3 data-level=2> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiXCR data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-analyze/ class=md-nav__link> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-align/ class=md-nav__link> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-refineTagsAndSort/ class=md-nav__link> refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemblePartial/ class=md-nav__link> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-extend/ class=md-nav__link> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemble/ class=md-nav__link> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assembleContigs/ class=md-nav__link> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-export/ class=md-nav__link> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPretty/ class=md-nav__link> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportAirr/ class=md-nav__link> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReadsForClones/ class=md-nav__link> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportShmTrees/ class=md-nav__link> exportShmTrees </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReports/ class=md-nav__link> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReportsTable/ class=md-nav__link> exportReportsTable </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportQc/ class=md-nav__link> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-postanalysis/ class=md-nav__link> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-downsample/ class=md-nav__link> downsample </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPlots/ class=md-nav__link> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportTables/ class=md-nav__link> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPreprocTables/ class=md-nav__link> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportClonesOverlap/ class=md-nav__link> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-overlapScatterPlot/ class=md-nav__link> overlapScatterPlot </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findAlleles/ class=md-nav__link> findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findShmTrees/ class=md-nav__link> findShmTrees </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPreset/ class=md-nav__link> exportPreset </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_4 data-level=2> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Reports data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-align/ class=md-nav__link> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-refineTagsAndSort/ class=md-nav__link> mixcr refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemblePartial/ class=md-nav__link> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-extend/ class=md-nav__link> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemble/ class=md-nav__link> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assembleContigs/ class=md-nav__link> mixcr assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findAlleles/ class=md-nav__link> mixcr findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findShmTrees/ class=md-nav__link> mixcr findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_5 data-level=2> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=RepSeq.io data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fasta/ class=md-nav__link> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-compile/ class=md-nav__link> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-debug/ class=md-nav__link> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-format/ class=md-nav__link> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromFasta/ class=md-nav__link> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromPaddedFasta/ class=md-nav__link> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-inferPoints/ class=md-nav__link> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-merge/ class=md-nav__link> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-stat/ class=md-nav__link> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-tsv/ class=md-nav__link> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_6 type=checkbox id=__nav_4_6> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_6 data-level=2> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiTool data-md-level=2> <label class=md-nav__title for=__nav_4_6> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-parse/ class=md-nav__link> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-tag-stat/ class=md-nav__link> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-export-fastq/ class=md-nav__link> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-correct/ class=md-nav__link> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-consensus/ class=md-nav__link> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-process-fastq/ class=md-nav__link> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-format-fastq/ class=md-nav__link> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-compress-sequence-set/ class=md-nav__link> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_7 type=checkbox id=__nav_4_7> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_7 data-level=2> Specifications <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Specifications data-md-level=2> <label class=md-nav__title for=__nav_4_7> <span class="md-nav__icon md-icon"></span> Specifications </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-tag-pattern/ class=md-nav__link> Tag pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-input-file-name-expansion/ class=md-nav__link> Input file name expansion </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-samples-table/ class=md-nav__link> Samples table </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-gene-features/ class=md-nav__link> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-translation-rules/ class=md-nav__link> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-mutations-encoding/ class=md-nav__link> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-repseqio-json-format/ class=md-nav__link> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://vdj.online/ class=md-nav__link> VDJ.online </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://github.com/milaboratory/mixcr/discussions class=md-nav__link> Discussions </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://github.com/milaboratory/mixcr class=md-nav__link> Github </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-libraries class=md-nav__link> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#under-the-hood-pipeline class=md-nav__link> Under the hood pipeline: </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Under the hood pipeline:"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#align class=md-nav__link> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#refinetagsandsort class=md-nav__link> refineTagsAndSort </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#assemble class=md-nav__link> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export class=md-nav__link> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#full-length-clonotype-assembly class=md-nav__link> Full-length clonotype assembly </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#reports class=md-nav__link> Reports </a> <!-- Table of contents list --> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/milaboratory/docs/edit/main/docs/mixcr/guides/nebnext-bcr.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=nebnext-immune-sequencing>NEBNext Immune Sequencing</h1> <p>Here we will discuss how to process BCR cDNA libraries obtained with NEBNext Immune Sequencing Kit.</p> <h2 id=data-libraries>Data libraries</h2> <p>This tutorial uses the data from the following publication: <em>Germinal centre-driven maturation of B cell response to mRNA vaccination</em> Wooseob, et al, Nature volume 604, pages141145 (2022) <a href=https://doi.org/10.1038/s41586-022-04527-1>doi:10.1038/s41586-022-04527-1</a></p> <p>The data includes samples taken from 8 patients before and several time points after receiving SARS-CoV-2 mRNA vaccine. Samples include: germinal center B cells from lymph node, plasmablasts from lymph node, plasma cells from bone marrow and memory B cells from blood. Sorted GC B cells and LNPCs from FNA, enriched BMPCs from bone marrow or enriched MBCs from PBMCs from blood were used for library preparation for bulk BCR sequecning. Circulating MBCs were magnetically isolated by first staining with IgD-PE and MojoSort anti-PE Nanobeads (BioLegend), and then processing with the EasySep Human B Cell Isolation Kit (StemCell Technologies) to negatively enrich IgDlo B cells. RNA was prepared from each sample using the RNeasy Plus Micro kit (Qiagen). Libraries were prepared using the NEBNext Immune Sequencing Kit for Human (New England Biolabs) according to the manufacturers instructions without modifications. Only heavy chain primers were used. High-throughput paired-end sequencing was performed on the Illumina MiSeq platform, 325 cycles for read 1 and 275 cycles for read 2.</p> <p>On the scheme bellow you can see structure of cDNA library. UMI is located in the first 17 bp of R2, followed by 7 to 10 bp occurred in the process of template switching (TS).</p> <details> <summary>The set of primers used in NEBNext Immune Sequencing Kits. The list includes primers for human and mice bcr and tcr kits.</summary> <div class="tabbed-set tabbed-alternate" data-tabs=1:2><input checked=checked id=__tabbed_1_1 name=__tabbed_1 type=radio><input id=__tabbed_1_2 name=__tabbed_1 type=radio><div class=tabbed-labels><label for=__tabbed_1_1>Human</label><label for=__tabbed_1_2>Mouse</label></div> <div class=tabbed-content> <div class=tabbed-block> <div class=highlight><pre><span></span><code>&gt;Human-IGHM
GAATTCTCACAGGAGACGAGG
&gt;Human-IGHD
TGTCTGCACCCTGATATGATGG
&gt;Human-IGHA
GGGTGCTGYMGAGGCTCAG
&gt;Human-IGHE
TTGCAGCAGCGGGTCAAGG
&gt;Human-IGHG
CCAGGGGGAAGACSGATG
&gt;Human-IGK
GACAGATGGTGCAGCCACAG
&gt;Human-IGL
AGGGYGGGAACAGAGTGAC
&gt;Human-TRA
CACGGCAGGGTCAGGGTTC
&gt;Human-TRB
CGACCTCGGGTGGGAACAC
</code></pre></div> </div> <div class=tabbed-block> <div class=highlight><pre><span></span><code>&gt;Mus-p5-IgGb
CCAGGGACCAAGGGATAGAC
&gt;Mus-p5-IgGa
CCAGTGGATAGACHGATGGGG
&gt;Mus-p5-IgE
GCTTTAAGGGGTAGAGCTGAG
&gt;Mus-p5-IgD
CTCTGAGAGGAGGAACATGTCA
&gt;Mus-p5-IgM
GGGAAGACATTTGGGAAGGAC
&gt;Mus-p5-IgA
GAATCAGGCAGCCGATTATCAC
&gt;Mus-p5-IgK
AGATGGATGCAGTTGGTGCA
&gt;Mus-p5-IgL
TCCTCAGAGGAAGGTGGAAAC
&gt;Mus-p5-TRAC
ATCTTTTAACTGGTACACAGCAGG
&gt;Mus-p5-TRBC
CAAGGAGACCTTGGGTGGAG
&gt;Mus-p5-TRGC
AAGGAAGAAAAATAGTGGGCTTGG
&gt;Mus-p5-TRDC
CATGATGAAAACAGATGGTTTGGC<span class=w>        </span>
</code></pre></div> </div> </div> </div> </details> <p><img alt src=../../reference/pics/NEBNext-human-bcr-kit-light.svg#only-light> <img alt src=../../reference/pics/NEBNext-human-bcr-kit-dark.svg#only-dark></p> <p>All data may be downloaded directly from SRA (PRJNA777934) using e.g. <a href=https://sra-explorer.info>SRA Explorer</a>.</p> <details class=tip> <summary>Use <a href=https://aria2.github.io>aria2c</a> for efficient download of the full dataset with the proper filenames:</summary> <p><div class=highlight><span class=filename>download.sh</span><pre><span></span><code>mkdir<span class=w> </span>-p<span class=w> </span>raw
aria2c<span class=w> </span>-c<span class=w> </span>-s<span class=w> </span><span class=m>16</span><span class=w> </span>-x<span class=w> </span><span class=m>16</span><span class=w> </span>-k<span class=w> </span>1M<span class=w> </span>-j<span class=w> </span><span class=m>8</span><span class=w> </span>-i<span class=w> </span>download-list.txt
</code></pre></div> <div class=highlight><span class=filename>download-list.txt</span><pre><span></span><code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952573/ERR5952573_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl1_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952573/ERR5952573_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl1_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952574/ERR5952574_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl2_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952574/ERR5952574_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl2_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952575/ERR5952575_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl3_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952575/ERR5952575_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl3_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952576/ERR5952576_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl4_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952576/ERR5952576_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl4_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952577/ERR5952577_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl5_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952577/ERR5952577_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl5_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952578/ERR5952578_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl6_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952578/ERR5952578_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl6_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952579/ERR5952579_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl7_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952579/ERR5952579_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl7_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/000/ERR5952580/ERR5952580_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl8_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/000/ERR5952580/ERR5952580_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl8_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/001/ERR5952581/ERR5952581_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl9_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/001/ERR5952581/ERR5952581_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl9_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/002/ERR5952582/ERR5952582_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl10_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/002/ERR5952582/ERR5952582_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl10_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952583/ERR5952583_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl11_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952583/ERR5952583_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl11_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952584/ERR5952584_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl12_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952584/ERR5952584_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl12_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952585/ERR5952585_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl13_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952585/ERR5952585_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl13_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952586/ERR5952586_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl14_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952586/ERR5952586_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl14_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952587/ERR5952587_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl15_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952587/ERR5952587_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/FebControl15_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952588/ERR5952588_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC1_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952588/ERR5952588_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC1_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952589/ERR5952589_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC2_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952589/ERR5952589_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC2_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/000/ERR5952590/ERR5952590_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC3_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/000/ERR5952590/ERR5952590_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC3_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/001/ERR5952591/ERR5952591_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC4_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/001/ERR5952591/ERR5952591_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC4_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/002/ERR5952592/ERR5952592_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC5_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/002/ERR5952592/ERR5952592_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC5_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952593/ERR5952593_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC6_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/003/ERR5952593/ERR5952593_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC6_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952594/ERR5952594_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC7_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/004/ERR5952594/ERR5952594_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC7_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952595/ERR5952595_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC8_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/005/ERR5952595/ERR5952595_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC8_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952596/ERR5952596_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC9_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/006/ERR5952596/ERR5952596_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC9_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952597/ERR5952597_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC10_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/007/ERR5952597/ERR5952597_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC10_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952598/ERR5952598_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC11_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/008/ERR5952598/ERR5952598_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC11_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952599/ERR5952599_1.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC12_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR595/009/ERR5952599/ERR5952599_2.fastq.gz
<span class=w>  </span><span class=nv>out</span><span class=o>=</span>raw/MISC12_R2.fastq.gz
</code></pre></div></p> </details> <p>The project contains 48 paired FASTQ files. Each file name encodes the information about donor, time-point relative to vaccination, tissue of origin and cell population. For example for the first file from the above listing:</p> <p>13_d60_LN_germCenterB_R1.fastq.gz - 13 - donor id - d60 - 60 days after vaccination - LN - lymph node (BM-bone marrow) - germCenterB - germinal center B cell (cell population)</p> <h2 id=upstream-analysis>Upstream analysis</h2> <p>MiXCR has a dedicated preset for this protocol, thus analysing the data is as easy as:</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>analyze<span class=w> </span>nebnext-human-bcr-cdr3<span class=w> </span><span class=se>\</span>
<span class=w>    </span>raw/13_d60_LN_germCenterB_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>raw/13_d60_LN_germCenterB_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB
</code></pre></div> <p>Running the command above will generate the following files:</p> <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>ls<span class=w> </span>result/

<span class=c1># human-readable reports </span>
13_d60_LN_germCenterB.report
<span class=c1># raw alignments (highly compressed binary file)</span>
13_d60_LN_germCenterB.vdjca
<span class=c1># alignments with corrected UMI barcode sequences </span>
13_d60_LN_germCenterB.refined.vdjca
<span class=c1># IGH, IGK and IGL CDR3 clonotypes (highly compressed binary file)</span>
13_d60_LN_germCenterB.clns
<span class=c1># IGH, IGK and IGL CDR3 clonotypes exported in tab-delimited txt</span>
13_d60_LN_germCenterB.clones.IGH.tsv
13_d60_LN_germCenterB.clones.IGK.tsv
13_d60_LN_germCenterB.clones.IGL.tsv
</code></pre></div> <p>In order to run the analysis for all samples in the project on Linux we can use <a href=https://www.gnu.org/software/parallel/ >GNU Parallel</a> in the following way:</p> <div class=highlight><pre><span></span><code><span class=ch>#!/usr/bin/env bash</span>

mkdir<span class=w> </span>-p<span class=w> </span>results

ls<span class=w> </span>/raw/*R1*<span class=w> </span><span class=p>|</span>
<span class=w>    </span>parallel<span class=w> </span>-j<span class=w> </span><span class=m>2</span><span class=w> </span>--line-buffer<span class=w> </span><span class=se>\</span>
<span class=w>    </span><span class=s2>&quot;mixcr analyze nebnext-human-bcr-cdr3 \</span>
<span class=s2>    {} \</span>
<span class=s2>    {=s:R1:R2:=} \</span>
<span class=s2>    {=s:.*/:results/:;s:_R.*::=}&quot;</span>
</code></pre></div> <p>While <code>.clns</code> file holds all data and is used for downstream analysis using <a href=../../reference/mixcr-postanalysis/ ><code>mixcr postanalisis</code></a>, the output <code>.txt</code> clonotype table will contain exhaustive information about each clonotype as well:</p> <details class=tip> <summary>See first 100 records from FebControl1.clones.IGH.tsv clonotype table</summary> <table> <thead> <tr> <th align=right>cloneId</th> <th align=right>cloneCount</th> <th align=right>uniqueTagCountUMI</th> <th align=right>cloneFraction</th> <th align=left>targetSequences</th> <th align=left>targetQualities</th> <th align=left>allVHitsWithScore</th> <th align=left>allDHitsWithScore</th> <th align=left>allJHitsWithScore</th> <th align=left>allCHitsWithScore</th> <th align=left>allVAlignments</th> <th align=left>allDAlignments</th> <th align=left>allJAlignments</th> <th align=left>allCAlignments</th> <th align=right>nSeqFR1</th> <th align=right>minQualFR1</th> <th align=right>nSeqCDR1</th> <th align=right>minQualCDR1</th> <th align=right>nSeqFR2</th> <th align=right>minQualFR2</th> <th align=right>nSeqCDR2</th> <th align=right>minQualCDR2</th> <th align=right>nSeqFR3</th> <th align=right>minQualFR3</th> <th align=left>nSeqCDR3</th> <th align=right>minQualCDR3</th> <th align=right>nSeqFR4</th> <th align=right>minQualFR4</th> <th align=right>aaSeqFR1</th> <th align=right>aaSeqCDR1</th> <th align=right>aaSeqFR2</th> <th align=right>aaSeqCDR2</th> <th align=right>aaSeqFR3</th> <th align=left>aaSeqCDR3</th> <th align=right>aaSeqFR4</th> <th align=left>refPoints</th> </tr> </thead> <tbody> <tr> <td align=right>0</td> <td align=right>4763</td> <td align=right>969</td> <td align=right>0.108675</td> <td align=left>TGTGCGAGAGATTCCGTCTTCGACTTATTAATTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2146.9)</td> <td align=left>IGHD3-10<em>00(31),IGHD3-22</em>00(30),IGHD3-3*00(30)</td> <td align=left>IGHJ4*00(416.3)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>288</td> <td align=left>300</td> <td align=left>319</td> <td align=left>0</td> <td align=right>12</td> <td align=right>SC297G</td> <td align=right>91.0</td> <td align=right>49</td> <td align=right>58</td> <td align=right>93</td> <td align=right>21</td> <td align=right>30</td> <td align=right>SG51C</td> <td align=right>31.0;52</td> <td align=left>58</td> <td align=right>93</td> <td align=right>24</td> <td align=right>30</td> <td align=right></td> <td align=right>30.0;52</td> <td align=right>58</td> <td align=right>93</td> <td align=right>24</td> <td align=left>30</td> <td align=right></td> <td align=left>30.0</td> </tr> <tr> <td align=right>1</td> <td align=right>2392</td> <td align=right>486</td> <td align=right>0.054577</td> <td align=left>TGTGCGACTGCTTTGGTGGTGACTGCTGCATACTTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2317.7)</td> <td align=left>IGHD2-21*00(56)</td> <td align=left>IGHJ4*00(211.9)</td> <td align=left>IGHG1<em>00(75.1),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>37</td> <td align=right>51</td> <td align=right>84</td> <td align=right>13</td> <td align=right>27</td> <td align=right>ST46C</td> <td align=right>56.0</td> <td align=left>29</td> <td align=right>37</td> <td align=right>68</td> <td align=right>31</td> <td align=right>39</td> <td align=right>SA32T</td> <td align=right>51.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>2</td> <td align=right>2237</td> <td align=right>476</td> <td align=right>0.0510404</td> <td align=left>TGTGCGTCCCTACGGTGGCTACGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2270.8)</td> <td align=left>IGHD4-17*00(56)</td> <td align=left>IGHJ4*00(429.3)</td> <td align=left>IGHG1<em>00(75.2),IGHG2</em>00(75.2),IGHG4<em>00(75.2),IGHGP</em>00(75.2)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=right>23</td> <td align=right>SA27G</td> <td align=right>56.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>3</td> <td align=right>1342</td> <td align=right>284</td> <td align=right>0.0306197</td> <td align=left>TGTGCGAGAGGCGACCGGAATTATCATGACAGTGGGTGGTCCGGCTGGTTCTTCGGTCTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(1365.2)</td> <td align=left>IGHD6-19*00(42)</td> <td align=left>IGHJ2*00(293.3)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.3)</td> <td align=left>282</td> <td align=left>294</td> <td align=left>313</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>29</td> <td align=right>43</td> <td align=right>63</td> <td align=right>29</td> <td align=right>43</td> <td align=right>SC35GSA40C</td> <td align=right>42.0</td> <td align=left>23</td> <td align=right>42</td> <td align=right>73</td> <td align=right>44</td> <td align=right>63</td> <td align=right>SA28TSA34G</td> <td align=right>132.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>4</td> <td align=right>1135</td> <td align=right>255</td> <td align=right>0.0258967</td> <td align=left>TGTGCGAGAGGCAATATGATAGTAGTGGCCTTTAGTGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2388)</td> <td align=left>IGHD3-22*00(70)</td> <td align=left>IGHJ3*00(443.4)</td> <td align=left>IGHG1<em>00(75.4),IGHG2</em>00(75.4),IGHG4<em>00(75.4),IGHGP</em>00(75.4)</td> <td align=left>282</td> <td align=left>294</td> <td align=left>313</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>38</td> <td align=right>52</td> <td align=right>93</td> <td align=right>14</td> <td align=right>28</td> <td align=right></td> <td align=right>70.0</td> <td align=left>23</td> <td align=right>39</td> <td align=right>70</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>5</td> <td align=right>904</td> <td align=right>187</td> <td align=right>0.0206261</td> <td align=left>TGTGCGAGTGAGAACGTGGATACAGGTATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2148.7)</td> <td align=left>IGHD5-5*00(86)</td> <td align=left>IGHJ5*00(465.8)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>296</td> <td align=left>319</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30G</td> <td align=right>86.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>6</td> <td align=right>736</td> <td align=right>148</td> <td align=right>0.0167929</td> <td align=left>TGTGCGAGTGAGAACGTGGATACAGGAATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2268.5)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(465.4)</td> <td align=left>IGHG1<em>00(75.5),IGHG2</em>00(75.5),IGHG4<em>00(75.5),IGHGP</em>00(75.5)</td> <td align=left>288</td> <td align=left>296</td> <td align=left>319</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30GST31A</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>7</td> <td align=right>709</td> <td align=right>139</td> <td align=right>0.0161769</td> <td align=left>TGTGCACACAGACTCGGCTCGTGGTTAAAGGGACTTCAGACTCACCCTATGATAGACCCAAGCGGGGACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-5*00(2446.5)</td> <td align=left>IGHD3-22*00(45)</td> <td align=left>IGHJ5*00(478.5)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>288</td> <td align=left>301</td> <td align=left>321</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>37</td> <td align=right>46</td> <td align=right>93</td> <td align=right>46</td> <td align=right>55</td> <td align=right></td> <td align=right>45.0</td> <td align=left>23</td> <td align=right>40</td> <td align=right>71</td> <td align=right>67</td> <td align=right>84</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>8</td> <td align=right>704</td> <td align=right>143</td> <td align=right>0.0160628</td> <td align=left>TGTGCGAGACGGCCCCCAGTAGTGGTTATTCCGTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2164)</td> <td align=left>IGHD3-22*00(65)</td> <td align=left>IGHJ4*00(427.9)</td> <td align=left>IGHG1<em>00(75.8),IGHG2</em>00(75.8),IGHG4<em>00(75.8),IGHGP</em>00(75.8)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>44</td> <td align=right>57</td> <td align=right>93</td> <td align=right>17</td> <td align=right>30</td> <td align=right></td> <td align=right>65.0</td> <td align=left>25</td> <td align=right>37</td> <td align=right>68</td> <td align=right>33</td> <td align=right>45</td> <td align=right></td> <td align=right>120.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>9</td> <td align=right>693</td> <td align=right>160</td> <td align=right>0.0158118</td> <td align=left>TGTGCGAGGTGGGGGCCTACGGTGACTCCCTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2094.5)</td> <td align=left>IGHD4-17*00(55)</td> <td align=left>IGHJ4*00(465.4)</td> <td align=left>IGHA1<em>00(144.4),IGHA2</em>00(144.4)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>19</td> <td align=right>30</td> <td align=right>48</td> <td align=right>16</td> <td align=right>27</td> <td align=right></td> <td align=right>55.0</td> <td align=left>21</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>45</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>10</td> <td align=right>679</td> <td align=right>155</td> <td align=right>0.0154924</td> <td align=left>TGTGCGAGAGCCAACCATATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2105.7)</td> <td align=left>IGHD3-10<em>00(25),IGHD3-22</em>00(25),IGHD3-3*00(25)</td> <td align=left>IGHJ6*00(435.8)</td> <td align=left>IGHG1<em>00(75.3),IGHG2</em>00(75.3),IGHG4<em>00(75.3),IGHGP</em>00(75.3)</td> <td align=left>285</td> <td align=left>295</td> <td align=left>316</td> <td align=left>0</td> <td align=right>10</td> <td align=right>SA292G</td> <td align=right>71.0</td> <td align=right>17</td> <td align=right>22</td> <td align=right>93</td> <td align=right>12</td> <td align=right>17</td> <td align=right></td> <td align=right>25.0;8</td> <td align=left>13</td> <td align=right>93</td> <td align=right>12</td> <td align=right>17</td> <td align=right></td> <td align=right>25.0;8</td> <td align=right>13</td> <td align=right>93</td> <td align=right>12</td> <td align=left>17</td> <td align=right></td> <td align=left>25.0</td> </tr> <tr> <td align=right>11</td> <td align=right>627</td> <td align=right>128</td> <td align=right>0.0143059</td> <td align=left>TGTGCGGGCCAATTTGTAGTAGTTCCAGCTGCTAGTTACTACTACTACGCTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2084.7)</td> <td align=left>IGHD2-2*00(91)</td> <td align=left>IGHJ6*00(546.8)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>37</td> <td align=right>58</td> <td align=right>93</td> <td align=right>13</td> <td align=right>34</td> <td align=right>SA47T</td> <td align=right>91.0</td> <td align=left>25</td> <td align=right>52</td> <td align=right>83</td> <td align=right>36</td> <td align=right>63</td> <td align=right>SG38C</td> <td align=right>241.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>12</td> <td align=right>591</td> <td align=right>142</td> <td align=right>0.0134845</td> <td align=left>TGTGCGAAATCCTATAGTGGGAGCTACTCCGCCGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2218.3)</td> <td align=left>IGHD1-26*00(81)</td> <td align=left>IGHJ3*00(438.3)</td> <td align=left>IGHG1<em>00(74.7),IGHG2</em>00(74.3),IGHG4<em>00(74.3),IGHGP</em>00(74.3)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>22</td> <td align=right>41</td> <td align=right>60</td> <td align=right>12</td> <td align=right>31</td> <td align=right>SA38C</td> <td align=right>81.0</td> <td align=left>24</td> <td align=right>39</td> <td align=right>70</td> <td align=right>33</td> <td align=right>48</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>13</td> <td align=right>490</td> <td align=right>99</td> <td align=right>0.0111801</td> <td align=left>TGTGCGCGTGAGAACGTGGATACAGGAATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2257.6)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(466.4)</td> <td align=left>IGHG1<em>00(74.6),IGHG2</em>00(74.6),IGHG4<em>00(74.6),IGHGP</em>00(74.6)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30GST31A</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>14</td> <td align=right>432</td> <td align=right>84</td> <td align=right>0.00985671</td> <td align=left>TGTGCGTCCCTAAGGTGGCTACGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2280.2)</td> <td align=left>IGHD5-12<em>00(45),IGHD5-18</em>00(45),IGHD4-17*00(42)</td> <td align=left>IGHJ4*00(428.6)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>32</td> <td align=right>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;32</td> <td align=left>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=left>23</td> <td align=right>SC22ASA27G</td> <td align=left>42.0</td> </tr> <tr> <td align=right>15</td> <td align=right>419</td> <td align=right>102</td> <td align=right>0.0095601</td> <td align=left>TGTGCGAGAGATCAGGGATACAGCTATGGGCCATGGGGGGACTACTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-21*00(2077.3)</td> <td align=left>IGHD5-5*00(70)</td> <td align=left>IGHJ6*00(537.1)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>285</td> <td align=left>298</td> <td align=left>316</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>22</td> <td align=right>36</td> <td align=right>60</td> <td align=right>15</td> <td align=right>29</td> <td align=right></td> <td align=right>70.0</td> <td align=left>29</td> <td align=right>52</td> <td align=right>83</td> <td align=right>40</td> <td align=right>63</td> <td align=right></td> <td align=right>230.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>16</td> <td align=right>410</td> <td align=right>83</td> <td align=right>0.00935475</td> <td align=left>TGTGCGACTGAGAACGTGGATACAGGCATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2221.2)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(439.7)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30GST31C</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>17</td> <td align=right>387</td> <td align=right>88</td> <td align=right>0.00882997</td> <td align=left>TGTGCGAGGACCCTTTACGATTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2307.8)</td> <td align=left>IGHD3-3<em>00(35),IGHD3-9</em>00(35),IGHD5-12*00(30)</td> <td align=left>IGHJ4*00(458.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>34</td> <td align=right>41</td> <td align=right>93</td> <td align=right>14</td> <td align=right>21</td> <td align=right></td> <td align=right>35.0;34</td> <td align=left>41</td> <td align=right>93</td> <td align=right>14</td> <td align=right>21</td> <td align=right></td> <td align=right>35.0;37</td> <td align=right>43</td> <td align=right>69</td> <td align=right>15</td> <td align=left>21</td> <td align=right></td> <td align=left>30.0</td> </tr> <tr> <td align=right>18</td> <td align=right>324</td> <td align=right>70</td> <td align=right>0.00739253</td> <td align=left>TGTGCGTCCCTACGGTGGCTTCGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2207.9)</td> <td align=left>IGHD4-23<em>00(45),IGHD4-17</em>00(42)</td> <td align=left>IGHJ4*00(408.8)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>22</td> <td align=right>31</td> <td align=right>57</td> <td align=right>9</td> <td align=right>18</td> <td align=right></td> <td align=right>45.0;19</td> <td align=left>33</td> <td align=right>48</td> <td align=right>9</td> <td align=right>23</td> <td align=right>SA27GSA30T</td> <td align=right>42.0</td> <td align=right>24</td> <td align=right>37</td> <td align=right>68</td> <td align=left>29</td> <td align=right>42</td> <td align=left></td> </tr> <tr> <td align=right>19</td> <td align=right>319</td> <td align=right>73</td> <td align=right>0.00727845</td> <td align=left>TGTGCGAGAATCTCCGTCTATGATAGTAGTGGTTATTACTCTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2193.7)</td> <td align=left>IGHD3-22*00(80)</td> <td align=left>IGHJ4*00(426.6)</td> <td align=left>IGHG1<em>00(74.7),IGHGP</em>00(74.7),IGHG2<em>00(73.8),IGHG4</em>00(73.8)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>37</td> <td align=right>53</td> <td align=right>93</td> <td align=right>17</td> <td align=right>33</td> <td align=right></td> <td align=right>80.0</td> <td align=left>19</td> <td align=right>37</td> <td align=right>68</td> <td align=right>33</td> <td align=right>51</td> <td align=right>SC21TST26C</td> <td align=right>122.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>20</td> <td align=right>307</td> <td align=right>75</td> <td align=right>0.00700465</td> <td align=left>TGTGCGAGGCCGTATAGTGGGAGCTACTTTTCTTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2040.4)</td> <td align=left>IGHD1-26*00(85)</td> <td align=left>IGHJ4*00(457.8)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>285</td> <td align=left>292</td> <td align=left>316</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>21</td> <td align=right>38</td> <td align=right>60</td> <td align=right>11</td> <td align=right>28</td> <td align=right></td> <td align=right>85.0</td> <td align=left>22</td> <td align=right>37</td> <td align=right>68</td> <td align=right>33</td> <td align=right>48</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>21</td> <td align=right>301</td> <td align=right>69</td> <td align=right>0.00686776</td> <td align=left>TGTGCGAGAGATGACGGAAGCAGCGGCTGGTATGGACCGGGGAACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2140.9)</td> <td align=left>IGHD6-13<em>00(58),IGHD6-19</em>00(58),IGHD6-25*00(50)</td> <td align=left>IGHJ4*00(387.9)</td> <td align=left>IGHG1<em>00(75.6),IGHG2</em>00(75.6),IGHG4<em>00(75.6),IGHGP</em>00(75.6)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>27</td> <td align=right>47</td> <td align=right>63</td> <td align=right>18</td> <td align=right>38</td> <td align=right>SA33GSC41TST43G</td> <td align=right>58.0;27</td> <td align=left>47</td> <td align=right>63</td> <td align=right>18</td> <td align=right>38</td> <td align=right>ST32CSC41TST43G</td> <td align=right>58.0;24</td> <td align=right>34</td> <td align=right>54</td> <td align=right>18</td> <td align=left>28</td> <td align=right></td> <td align=left>50.0</td> </tr> <tr> <td align=right>22</td> <td align=right>296</td> <td align=right>62</td> <td align=right>0.00675367</td> <td align=left>TGTGCGAGTGAGAACGTGGACACAGGTATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2253.5)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(463.9)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>288</td> <td align=left>296</td> <td align=left>319</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>ST25CSC30G</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>23</td> <td align=right>284</td> <td align=right>62</td> <td align=right>0.00647988</td> <td align=left>TGTGCGACCGGGATACAGTTACAGCTATGGTCTCACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2432.8)</td> <td align=left>IGHD5-5*00(60)</td> <td align=left>IGHJ4*00(474.6)</td> <td align=left>IGHA1<em>00(144.2),IGHA2</em>00(144.2)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>25</td> <td align=right>37</td> <td align=right>60</td> <td align=right>19</td> <td align=right>31</td> <td align=right></td> <td align=right>60.0</td> <td align=left>20</td> <td align=right>37</td> <td align=right>68</td> <td align=right>34</td> <td align=right>51</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>24</td> <td align=right>264</td> <td align=right>59</td> <td align=right>0.00602355</td> <td align=left>TGTGCGACTGCTTTGGTGGTGACTGCTGCATACTTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2487.2)</td> <td align=left>IGHD2-21*00(56)</td> <td align=left>IGHJ1*00(139.5)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>37</td> <td align=right>51</td> <td align=right>84</td> <td align=right>13</td> <td align=right>27</td> <td align=right>ST46C</td> <td align=right>56.0</td> <td align=left>37</td> <td align=right>41</td> <td align=right>72</td> <td align=right>35</td> <td align=right>39</td> <td align=right></td> <td align=right>40.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>25</td> <td align=right>258</td> <td align=right>52</td> <td align=right>0.00588665</td> <td align=left>TGTGCGACTGAGAACGTGGATACAGGTATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2360.1)</td> <td align=left>IGHD5-5*00(86)</td> <td align=left>IGHJ5*00(467)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30G</td> <td align=right>86.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>26</td> <td align=right>250</td> <td align=right>53</td> <td align=right>0.00570412</td> <td align=left>TGTGCACACAGGCGGGGAAGGGCCACATGGATGGCTGGTTATTTCGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-5*00(2447.1)</td> <td align=left>IGHD6-19<em>00(40),IGHD5-12</em>00(36),IGHD5-5*00(36)</td> <td align=left>IGHJ4*00(400.5)</td> <td align=left>IGHG1<em>00(75.2),IGHG2</em>00(75.2),IGHG4<em>00(75.2),IGHGP</em>00(73.8)</td> <td align=left>288</td> <td align=left>302</td> <td align=left>321</td> <td align=left>0</td> <td align=right>14</td> <td align=right>SA299G</td> <td align=right>111.0</td> <td align=right>32</td> <td align=right>40</td> <td align=right>63</td> <td align=right>31</td> <td align=right>39</td> <td align=right></td> <td align=right>40.0;19</td> <td align=left>29</td> <td align=right>69</td> <td align=right>22</td> <td align=right>32</td> <td align=right>SG23A</td> <td align=right>36.0;16</td> <td align=right>26</td> <td align=right>60</td> <td align=right>22</td> <td align=left>32</td> <td align=right>SG20A</td> <td align=left>36.0</td> </tr> <tr> <td align=right>27</td> <td align=right>244</td> <td align=right>49</td> <td align=right>0.00556722</td> <td align=left>TGTGCGAGAGATGACGGAAGTACCGGCTGGTATGGACCGGGGAACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2165.1)</td> <td align=left>IGHD2-2*00(52)</td> <td align=left>IGHJ4*00(386)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>44</td> <td align=right>60</td> <td align=right>93</td> <td align=right>18</td> <td align=right>34</td> <td align=right>SA50GSC55G</td> <td align=right>52.0</td> <td align=left>29</td> <td align=right>37</td> <td align=right>68</td> <td align=right>43</td> <td align=right>51</td> <td align=right></td> <td align=right>80.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>28</td> <td align=right>244</td> <td align=right>44</td> <td align=right>0.00556722</td> <td align=left>TGTGCGAGAAGGGACTTCCAGAGGTCGTTCCTCTTTTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2190.5)</td> <td align=left>IGHD1-1<em>00(41),IGHD1-20</em>00(41)</td> <td align=left>IGHJ4<em>00(337.4),IGHJ5</em>00(337.4)</td> <td align=left>IGHG1<em>00(76.5),IGHG2</em>00(76.5),IGHG4<em>00(76.5),IGHGP</em>00(76.5)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>31</td> <td align=right>42</td> <td align=right>51</td> <td align=right>20</td> <td align=right>31</td> <td align=right>SC33G</td> <td align=right>41.0;31</td> <td align=left>42</td> <td align=right>51</td> <td align=right>20</td> <td align=right>31</td> <td align=right>SC33G</td> <td align=right>41.0</td> <td align=right>34</td> <td align=right>37</td> <td align=right>68</td> <td align=left>36</td> <td align=right>39</td> <td align=left></td> </tr> <tr> <td align=right>29</td> <td align=right>238</td> <td align=right>55</td> <td align=right>0.00543032</td> <td align=left>TGTGCGAGACGGCCGGCTACTATGATATTTCTGGCCCACACCAAGAAAGACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2147.8)</td> <td align=left>IGHD3-22*00(50)</td> <td align=left>IGHJ4*00(478.2)</td> <td align=left>IGHG1<em>00(74.7),IGHG2</em>00(74.7),IGHG4<em>00(74.7),IGHGP</em>00(74.7)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>35</td> <td align=right>45</td> <td align=right>93</td> <td align=right>17</td> <td align=right>27</td> <td align=right></td> <td align=right>50.0</td> <td align=left>20</td> <td align=right>37</td> <td align=right>68</td> <td align=right>49</td> <td align=right>66</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>30</td> <td align=right>237</td> <td align=right>49</td> <td align=right>0.0054075</td> <td align=left>TGTGCCAGACGGCCCCCAGTAGTGGTTATTCCGTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2228.4)</td> <td align=left>IGHD3-22*00(65)</td> <td align=left>IGHJ4*00(428.9)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right>SG287C</td> <td align=right>61.0</td> <td align=right>44</td> <td align=right>57</td> <td align=right>93</td> <td align=right>17</td> <td align=right>30</td> <td align=right></td> <td align=right>65.0</td> <td align=left>25</td> <td align=right>37</td> <td align=right>68</td> <td align=right>33</td> <td align=right>45</td> <td align=right></td> <td align=right>120.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>31</td> <td align=right>229</td> <td align=right>51</td> <td align=right>0.00522497</td> <td align=left>TGTGCGAGAGATGACGGAAGCAGCAGCTGGTATGGACCGGGGAACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2258)</td> <td align=left>IGHD6-13*00(72)</td> <td align=left>IGHJ4*00(389.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>27</td> <td align=right>47</td> <td align=right>63</td> <td align=right>18</td> <td align=right>38</td> <td align=right>SC41TST43G</td> <td align=right>72.0</td> <td align=left>29</td> <td align=right>37</td> <td align=right>68</td> <td align=right>43</td> <td align=right>51</td> <td align=right></td> <td align=right>80.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>32</td> <td align=right>226</td> <td align=right>52</td> <td align=right>0.00515652</td> <td align=left>TGTGCGAGCAGTGGCTGGCACCGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-66*00(2370.4)</td> <td align=left>IGHD6-19*00(51)</td> <td align=left>IGHJ4*00(436.5)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>29</td> <td align=right>42</td> <td align=right>63</td> <td align=right>8</td> <td align=right>21</td> <td align=right>ST39C</td> <td align=right>51.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>33</td> <td align=right>212</td> <td align=right>55</td> <td align=right>0.00483709</td> <td align=left>TGTGCGAGCGCTCCCCGCCCGTATTACTATGATAGTAGTGCTTATTACCCATTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-48*00(2081.7)</td> <td align=left>IGHD3-22*00(131)</td> <td align=left>IGHJ4*00(428.4)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>293</td> <td align=left>316</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>30</td> <td align=right>59</td> <td align=right>93</td> <td align=right>19</td> <td align=right>48</td> <td align=right>SG51C</td> <td align=right>131.0</td> <td align=left>25</td> <td align=right>37</td> <td align=right>68</td> <td align=right>51</td> <td align=right>63</td> <td align=right></td> <td align=right>120.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>34</td> <td align=right>196</td> <td align=right>37</td> <td align=right>0.00447203</td> <td align=left>TGTGCGAGAGGAGCTACGGGAGATGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2201.8)</td> <td align=left>IGHD1-26*00(40)</td> <td align=left>IGHJ3*00(438.3)</td> <td align=left>IGHG1<em>00(74.1),IGHG2</em>00(74.1),IGHG4<em>00(74.1),IGHGP</em>00(74.1)</td> <td align=left>288</td> <td align=left>297</td> <td align=left>319</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>29</td> <td align=right>37</td> <td align=right>60</td> <td align=right>9</td> <td align=right>17</td> <td align=right></td> <td align=right>40.0</td> <td align=left>21</td> <td align=right>39</td> <td align=right>70</td> <td align=right>21</td> <td align=right>39</td> <td align=right></td> <td align=right>180.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>35</td> <td align=right>195</td> <td align=right>43</td> <td align=right>0.00444921</td> <td align=left>TGTGCGATTGAGAACGTGGATACAGGTATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2276)</td> <td align=left>IGHD5-5*00(86)</td> <td align=left>IGHJ5*00(469.6)</td> <td align=left>IGHG1<em>00(75.5),IGHG2</em>00(75.5),IGHG4<em>00(75.5),IGHGP</em>00(75.5)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30G</td> <td align=right>86.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>36</td> <td align=right>182</td> <td align=right>44</td> <td align=right>0.0041526</td> <td align=left>TGTGCGAGAGAACGGGATATTGTAGTAGTACCAGCGTCTTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-46*00(2381.6)</td> <td align=left>IGHD2-2*00(105)</td> <td align=left>IGHJ4*00(456.1)</td> <td align=left>IGHG1<em>00(74.7),IGHG2</em>00(74.7),IGHG4<em>00(74.7),IGHGP</em>00(74.7)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>32</td> <td align=right>53</td> <td align=right>93</td> <td align=right>14</td> <td align=right>35</td> <td align=right></td> <td align=right>105.0</td> <td align=left>22</td> <td align=right>37</td> <td align=right>68</td> <td align=right>39</td> <td align=right>54</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>37</td> <td align=right>173</td> <td align=right>38</td> <td align=right>0.00394725</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGCATTGTAGTGGAGGTAGCTGCTACGACCAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(1962.1)</td> <td align=left>IGHD2-15*00(101)</td> <td align=left>IGHJ5*00(495)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right>ST46A</td> <td align=right>101.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right></td> <td align=right>190.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>38</td> <td align=right>170</td> <td align=right>37</td> <td align=right>0.0038788</td> <td align=left>TGTGCGAAAGATCGGGGCGATGACTACGCCAGCTACTACTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2161.3)</td> <td align=left>IGHD4-17<em>00(45),IGHD4-23</em>00(45),IGHD4-11*00(40)</td> <td align=left>IGHJ6*00(502.4)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>298</td> <td align=left>316</td> <td align=left>0</td> <td align=right>13</td> <td align=right>SG292A</td> <td align=right>101.0</td> <td align=right>15</td> <td align=right>24</td> <td align=right>48</td> <td align=right>19</td> <td align=right>28</td> <td align=right></td> <td align=right>45.0;18</td> <td align=left>27</td> <td align=right>57</td> <td align=right>19</td> <td align=right>28</td> <td align=right></td> <td align=right>45.0;15</td> <td align=right>23</td> <td align=right>48</td> <td align=right>19</td> <td align=left>27</td> <td align=right></td> <td align=left>40.0</td> </tr> <tr> <td align=right>39</td> <td align=right>167</td> <td align=right>45</td> <td align=right>0.00381035</td> <td align=left>TGTGCGAGAGTGAATGGCTACGGAAATGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-23*00(2161.1)</td> <td align=left>IGHD5-12<em>00(40),IGHD5-18</em>00(40),IGHD5-24*00(40)</td> <td align=left>IGHJ3*00(398.4)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>295</td> <td align=left>316</td> <td align=left>0</td> <td align=right>10</td> <td align=right>SA292G</td> <td align=right>71.0</td> <td align=right>33</td> <td align=right>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>22</td> <td align=right></td> <td align=right>40.0;33</td> <td align=left>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>22</td> <td align=right></td> <td align=right>40.0;26</td> <td align=right>34</td> <td align=right>60</td> <td align=right>13</td> <td align=left>21</td> <td align=right></td> <td align=left>40.0</td> </tr> <tr> <td align=right>40</td> <td align=right>163</td> <td align=right>37</td> <td align=right>0.00371908</td> <td align=left>TGTGCGAGACCAGGGTGGCTACGAGGTGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2345)</td> <td align=left>IGHD5-12<em>00(50),IGHD5-18</em>00(50)</td> <td align=left>IGHJ3*00(462.2)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>288</td> <td align=left>298</td> <td align=left>319</td> <td align=left>0</td> <td align=right>10</td> <td align=right></td> <td align=right>100.0</td> <td align=right>32</td> <td align=right>42</td> <td align=right>69</td> <td align=right>14</td> <td align=right>24</td> <td align=right></td> <td align=right>50.0;32</td> <td align=left>42</td> <td align=right>69</td> <td align=right>14</td> <td align=right>24</td> <td align=right></td> <td align=right>50.0</td> <td align=right>23</td> <td align=right>39</td> <td align=right>70</td> <td align=left>26</td> <td align=right>42</td> <td align=left></td> </tr> <tr> <td align=right>41</td> <td align=right>155</td> <td align=right>36</td> <td align=right>0.00353655</td> <td align=left>TGTGCGAGACATCCCTCACCTTATTACTATGGTTCGGGGAGATATGAGTACGGTGTGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV5-51*00(2241.9)</td> <td align=left>IGHD3-10*00(92)</td> <td align=left>IGHJ6*00(454.7)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>32</td> <td align=right>56</td> <td align=right>93</td> <td align=right>21</td> <td align=right>45</td> <td align=right>SA46GST52A</td> <td align=right>92.0</td> <td align=left>34</td> <td align=right>52</td> <td align=right>83</td> <td align=right>48</td> <td align=right>66</td> <td align=right>SA40G</td> <td align=right>151.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>42</td> <td align=right>155</td> <td align=right>34</td> <td align=right>0.00353655</td> <td align=left>TGTGCGAGAGATCAGGGATACAGCTATGGGCCATGGGGGGACTACTACGGTATGGACGACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-21*00(2242.4)</td> <td align=left>IGHD5-5*00(70)</td> <td align=left>IGHJ6*00(475.5)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(72.8)</td> <td align=left>285</td> <td align=left>298</td> <td align=left>316</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>22</td> <td align=right>36</td> <td align=right>60</td> <td align=right>15</td> <td align=right>29</td> <td align=right></td> <td align=right>70.0</td> <td align=left>29</td> <td align=right>52</td> <td align=right>83</td> <td align=right>40</td> <td align=right>63</td> <td align=right>ST47A</td> <td align=right>201.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>43</td> <td align=right>155</td> <td align=right>41</td> <td align=right>0.00353655</td> <td align=left>TGTGCAAAAGATATAGGGTATAGCAGTGGTGGGACCGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-9*00(2385.3)</td> <td align=left>IGHD6-19*00(74)</td> <td align=left>IGHJ4*00(396)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>300</td> <td align=left>318</td> <td align=left>0</td> <td align=right>15</td> <td align=right></td> <td align=right>150.0</td> <td align=right>21</td> <td align=right>42</td> <td align=right>63</td> <td align=right>15</td> <td align=right>35</td> <td align=right>DC35ST39G</td> <td align=right>74.0</td> <td align=left>28</td> <td align=right>37</td> <td align=right>68</td> <td align=right>36</td> <td align=right>45</td> <td align=right></td> <td align=right>90.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>44</td> <td align=right>154</td> <td align=right>35</td> <td align=right>0.00351374</td> <td align=left>TGTGCGGGCCAATTTGTAGTAGTACCAGCTGCTAGTTACTACTACTACGCTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2289.6)</td> <td align=left>IGHD2-2*00(105)</td> <td align=left>IGHJ6*00(542.6)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>37</td> <td align=right>58</td> <td align=right>93</td> <td align=right>13</td> <td align=right>34</td> <td align=right></td> <td align=right>105.0</td> <td align=left>25</td> <td align=right>52</td> <td align=right>83</td> <td align=right>36</td> <td align=right>63</td> <td align=right>SG38C</td> <td align=right>241.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>45</td> <td align=right>154</td> <td align=right>39</td> <td align=right>0.00351374</td> <td align=left>TGTGCGAGGTGTTTCGGTGGGAACTATTACGGGCCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2123.5)</td> <td align=left>IGHD1-26<em>00(47),IGHD4-23</em>00(41)</td> <td align=left>IGHJ4*00(428.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>292</td> <td align=left>316</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>26</td> <td align=right>41</td> <td align=right>60</td> <td align=right>16</td> <td align=right>31</td> <td align=right>SG32ASC36T</td> <td align=right>47.0;25</td> <td align=left>36</td> <td align=right>57</td> <td align=right>14</td> <td align=right>25</td> <td align=right>ST31G</td> <td align=right>41.0</td> <td align=right>24</td> <td align=right>37</td> <td align=right>68</td> <td align=left>35</td> <td align=right>48</td> <td align=left></td> </tr> <tr> <td align=right>46</td> <td align=right>151</td> <td align=right>33</td> <td align=right>0.00344529</td> <td align=left>TGTGCGGGCCAATCTGTAGTAGTACCAGCTGCTAGTTACTACTACTACGCTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2130.8)</td> <td align=left>IGHD2-2*00(100)</td> <td align=left>IGHJ6*00(549.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>38</td> <td align=right>58</td> <td align=right>93</td> <td align=right>14</td> <td align=right>34</td> <td align=right></td> <td align=right>100.0</td> <td align=left>25</td> <td align=right>52</td> <td align=right>83</td> <td align=right>36</td> <td align=right>63</td> <td align=right>SG38C</td> <td align=right>241.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>47</td> <td align=right>139</td> <td align=right>34</td> <td align=right>0.00317149</td> <td align=left>TGTGCGAAGTCCTATAGTGGGAGCTACTCCGCCGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2227.7)</td> <td align=left>IGHD1-26*00(81)</td> <td align=left>IGHJ3*00(372.4)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>293</td> <td align=left>316</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>22</td> <td align=right>41</td> <td align=right>60</td> <td align=right>12</td> <td align=right>31</td> <td align=right>SA38C</td> <td align=right>81.0</td> <td align=left>24</td> <td align=right>39</td> <td align=right>70</td> <td align=right>33</td> <td align=right>48</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>48</td> <td align=right>136</td> <td align=right>37</td> <td align=right>0.00310304</td> <td align=left>TGTGCGAGCAGTGGCTGGCACCGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2225.4)</td> <td align=left>IGHD6-19*00(51)</td> <td align=left>IGHJ4*00(438.2)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>29</td> <td align=right>42</td> <td align=right>63</td> <td align=right>8</td> <td align=right>21</td> <td align=right>ST39C</td> <td align=right>51.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>49</td> <td align=right>130</td> <td align=right>28</td> <td align=right>0.00296614</td> <td align=left>TGTGCGACCGGGATACAGTTACACCTATGGTCTCACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2349.5)</td> <td align=left>IGHD5-5*00(46)</td> <td align=left>IGHJ4*00(475.6)</td> <td align=left>IGHA1<em>00(144.5),IGHA2</em>00(144.5)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>25</td> <td align=right>37</td> <td align=right>60</td> <td align=right>19</td> <td align=right>31</td> <td align=right>SG29C</td> <td align=right>46.0</td> <td align=left>20</td> <td align=right>37</td> <td align=right>68</td> <td align=right>34</td> <td align=right>51</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>50</td> <td align=right>130</td> <td align=right>30</td> <td align=right>0.00296614</td> <td align=left>TGTGCGAGACTTACCAGCAGTTTTCCTCCGTACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-23*00(1988.5)</td> <td align=left>IGHD6-13<em>00(36),IGHD6-19</em>00(36),IGHD2-2*00(35)</td> <td align=left>IGHJ4*00(496.8)</td> <td align=left>IGHA1<em>00(143.4),IGHA2</em>00(143.4)</td> <td align=left>285</td> <td align=left>292</td> <td align=left>316</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>40</td> <td align=right>50</td> <td align=right>63</td> <td align=right>8</td> <td align=right>18</td> <td align=right>SG42T</td> <td align=right>36.0;40</td> <td align=left>50</td> <td align=right>63</td> <td align=right>8</td> <td align=right>18</td> <td align=right>SG42T</td> <td align=right>36.0;46</td> <td align=right>53</td> <td align=right>93</td> <td align=right>11</td> <td align=left>18</td> <td align=right></td> <td align=left>35.0</td> </tr> <tr> <td align=right>51</td> <td align=right>128</td> <td align=right>29</td> <td align=right>0.00292051</td> <td align=left>TGTGCGAGAGATCTGGGGGGCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(1972.5)</td> <td align=left>IGHD3-16*00(25)</td> <td align=left>IGHJ5*00(465.4)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>299</td> <td align=left>316</td> <td align=left>0</td> <td align=right>14</td> <td align=right>SA292G</td> <td align=right>111.0</td> <td align=right>54</td> <td align=right>59</td> <td align=right>111</td> <td align=right>14</td> <td align=right>19</td> <td align=right></td> <td align=right>25.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>20</td> <td align=right>36</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>52</td> <td align=right>124</td> <td align=right>22</td> <td align=right>0.00282924</td> <td align=left>TGTGCGAGAAGGCAGATGCCCTACACCCTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2353.8)</td> <td align=left>IGHD2-8<em>00(30),IGHD5-24</em>00(29),IGHD6-25*00(26)</td> <td align=left>IGHJ4*00(419.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>9</td> <td align=right>15</td> <td align=right>93</td> <td align=right>21</td> <td align=right>27</td> <td align=right></td> <td align=right>30.0;24</td> <td align=left>35</td> <td align=right>60</td> <td align=right>13</td> <td align=right>25</td> <td align=right>I29CSG29C</td> <td align=right>29.0;10</td> <td align=right>18</td> <td align=right>54</td> <td align=right>20</td> <td align=left>28</td> <td align=right>ST13C</td> <td align=left>26.0</td> </tr> <tr> <td align=right>53</td> <td align=right>123</td> <td align=right>30</td> <td align=right>0.00280643</td> <td align=left>TGTGCGAGAGATCAGGGATACAGCTATGGGCCATGGGGGGACTACTACGGTCTGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-21*00(2122.7)</td> <td align=left>IGHD5-5*00(70)</td> <td align=left>IGHJ6*00(507.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>298</td> <td align=left>316</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>22</td> <td align=right>36</td> <td align=right>60</td> <td align=right>15</td> <td align=right>29</td> <td align=right></td> <td align=right>70.0</td> <td align=left>29</td> <td align=right>52</td> <td align=right>83</td> <td align=right>40</td> <td align=right>63</td> <td align=right>SA40C</td> <td align=right>201.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>54</td> <td align=right>122</td> <td align=right>24</td> <td align=right>0.00278361</td> <td align=left>TGTGTGTCCCTACGGTGGCTACGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2304.3)</td> <td align=left>IGHD4-17*00(56)</td> <td align=left>IGHJ4*00(438.4)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>292</td> <td align=left>319</td> <td align=left>0</td> <td align=right>4</td> <td align=right></td> <td align=right>40.0</td> <td align=right>19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=right>23</td> <td align=right>SA27G</td> <td align=right>56.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>55</td> <td align=right>122</td> <td align=right>25</td> <td align=right>0.00278361</td> <td align=left>TGTGCGTCCCTAGGGTGGCTACGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2248.3)</td> <td align=left>IGHD5-12<em>00(45),IGHD5-18</em>00(45),IGHD4-17*00(42)</td> <td align=left>IGHJ4*00(427.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>32</td> <td align=right>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;32</td> <td align=left>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=left>23</td> <td align=right>SC22GSA27G</td> <td align=left>42.0</td> </tr> <tr> <td align=right>56</td> <td align=right>120</td> <td align=right>27</td> <td align=right>0.00273798</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGGATTGTAGTGGTGCTAGCTGCTACGACCAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(2068.5)</td> <td align=left>IGHD2-15*00(101)</td> <td align=left>IGHJ5*00(495.3)</td> <td align=left>IGHG1<em>00(74.1),IGHG2</em>00(74.1),IGHG4<em>00(74.1),IGHGP</em>00(74.1)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right>SG48C</td> <td align=right>101.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right></td> <td align=right>190.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>57</td> <td align=right>119</td> <td align=right>23</td> <td align=right>0.00271516</td> <td align=left>TGTGCGGCAGATGACGGAAGCAGCGGCTGGTTTGGACCGGGGAACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2168.4)</td> <td align=left>IGHD6-13<em>00(51),IGHD6-19</em>00(51),IGHD6-25*00(50)</td> <td align=left>IGHJ4*00(386.6)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right>SA291GSG292C</td> <td align=right>62.0</td> <td align=right>27</td> <td align=right>40</td> <td align=right>63</td> <td align=right>18</td> <td align=right>31</td> <td align=right>SA33G</td> <td align=right>51.0;27</td> <td align=left>40</td> <td align=right>63</td> <td align=right>18</td> <td align=right>31</td> <td align=right>ST32C</td> <td align=right>51.0;24</td> <td align=right>34</td> <td align=right>54</td> <td align=right>18</td> <td align=left>28</td> <td align=right></td> <td align=left>50.0</td> </tr> <tr> <td align=right>58</td> <td align=right>117</td> <td align=right>26</td> <td align=right>0.00266953</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGGATTGTAGTGATGGTAGCTGCTACGACCAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(2073)</td> <td align=left>IGHD2-15*00(101)</td> <td align=left>IGHJ5*00(497.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right>SG45A</td> <td align=right>101.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right></td> <td align=right>190.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>59</td> <td align=right>116</td> <td align=right>21</td> <td align=right>0.00264671</td> <td align=left>TGTGCGCGACTTTCCTACTACTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-59*00(2214.1)</td> <td align=left>nan</td> <td align=left>IGHJ6*00(541.6)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right>SA288C</td> <td align=right>61.0</td> <td align=right>nan</td> <td align=right>27</td> <td align=right>52</td> <td align=right>83</td> <td align=right>14</td> <td align=right>39</td> <td align=right></td> <td align=left>250.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>60</td> <td align=right>115</td> <td align=right>25</td> <td align=right>0.00262389</td> <td align=left>TGTGCGAGACAACCCTTACGATATTTTGACTGGAACTGGCACTTCGATCTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-59*00(2239.1)</td> <td align=left>IGHD3-9*00(90)</td> <td align=left>IGHJ2*00(473.6)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>34</td> <td align=right>52</td> <td align=right>93</td> <td align=right>15</td> <td align=right>33</td> <td align=right></td> <td align=right>90.0</td> <td align=left>22</td> <td align=right>42</td> <td align=right>73</td> <td align=right>34</td> <td align=right>54</td> <td align=right>ST27C</td> <td align=right>171.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>61</td> <td align=right>115</td> <td align=right>20</td> <td align=right>0.00262389</td> <td align=left>TGTGCGAGAAGGGACTTCCAGACGTCGTTCCTCTTTTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2367.2)</td> <td align=left>IGHD1-1<em>00(55),IGHD1-20</em>00(55)</td> <td align=left>IGHJ4<em>00(338.4),IGHJ5</em>00(338.4)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>31</td> <td align=right>42</td> <td align=right>51</td> <td align=right>20</td> <td align=right>31</td> <td align=right></td> <td align=right>55.0;31</td> <td align=left>42</td> <td align=right>51</td> <td align=right>20</td> <td align=right>31</td> <td align=right></td> <td align=right>55.0</td> <td align=right>34</td> <td align=right>37</td> <td align=right>68</td> <td align=left>36</td> <td align=right>39</td> <td align=left></td> </tr> <tr> <td align=right>62</td> <td align=right>113</td> <td align=right>30</td> <td align=right>0.00257826</td> <td align=left>TGTGCGAGAGATTACACCCCTCTCTCCTTCGGGGGGGCTCGCTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-21*00(2323.2)</td> <td align=left>IGHD2-8<em>00(30),IGHD3-16</em>00(26),IGHD5-24*00(26)</td> <td align=left>IGHJ6*00(486.2)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>297</td> <td align=left>316</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>9</td> <td align=right>15</td> <td align=right>93</td> <td align=right>12</td> <td align=right>18</td> <td align=right></td> <td align=right>30.0;51</td> <td align=left>59</td> <td align=right>111</td> <td align=right>27</td> <td align=right>35</td> <td align=right>ST53C</td> <td align=right>26.0;10</td> <td align=right>18</td> <td align=right>60</td> <td align=right>17</td> <td align=left>25</td> <td align=right>SA12C</td> <td align=left>26.0</td> </tr> <tr> <td align=right>63</td> <td align=right>112</td> <td align=right>19</td> <td align=right>0.00255544</td> <td align=left>TGTGCACAGAGCAGTGGCTGGTCGTACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-5*00(2452.4)</td> <td align=left>IGHD6-19*00(45)</td> <td align=left>IGHJ4*00(490.5)</td> <td align=left>IGHA1<em>00(143.9),IGHA2</em>00(143.9)</td> <td align=left>288</td> <td align=left>299</td> <td align=left>321</td> <td align=left>0</td> <td align=right>11</td> <td align=right>SC296G</td> <td align=right>81.0</td> <td align=right>29</td> <td align=right>38</td> <td align=right>63</td> <td align=right>11</td> <td align=right>20</td> <td align=right></td> <td align=right>45.0</td> <td align=left>15</td> <td align=right>37</td> <td align=right>68</td> <td align=right>20</td> <td align=right>42</td> <td align=right>SA17C</td> <td align=right>191.0</td> <td align=right>;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>64</td> <td align=right>110</td> <td align=right>24</td> <td align=right>0.00250981</td> <td align=left>TGTGCGAGGCTACTATGGTTCGGGGCATATGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2233.6)</td> <td align=left>IGHD3-10*00(61)</td> <td align=left>IGHJ3*00(475.7)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>296</td> <td align=left>319</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>35</td> <td align=right>50</td> <td align=right>93</td> <td align=right>10</td> <td align=right>25</td> <td align=right>SA46G</td> <td align=right>61.0</td> <td align=left>22</td> <td align=right>39</td> <td align=right>70</td> <td align=right>28</td> <td align=right>45</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>65</td> <td align=right>109</td> <td align=right>28</td> <td align=right>0.00248699</td> <td align=left>TGTGCGAGCACCGATTGGGGTACTGGGAGATACTACAGTGCCTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2269)</td> <td align=left>IGHD2-8<em>00(35),IGHD1-26</em>00(31),IGHD2-2*00(30)</td> <td align=left>IGHJ4*00(397.5)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>39</td> <td align=right>46</td> <td align=right>93</td> <td align=right>19</td> <td align=right>26</td> <td align=right></td> <td align=right>35.0;24</td> <td align=left>33</td> <td align=right>60</td> <td align=right>20</td> <td align=right>29</td> <td align=right>SG26C</td> <td align=right>31.0;12</td> <td align=right>18</td> <td align=right>93</td> <td align=right>18</td> <td align=left>24</td> <td align=right></td> <td align=left>30.0</td> </tr> <tr> <td align=right>66</td> <td align=right>106</td> <td align=right>25</td> <td align=right>0.00241855</td> <td align=left>TGTGCGTCTGTTTGGTGGCTGAGGGGGTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-48*00(2190.8)</td> <td align=left>IGHD6-19<em>00(35),IGHD2-21</em>00(33),IGHD2-15*00(31)</td> <td align=left>IGHJ4*00(459.1)</td> <td align=left>IGHG1<em>00(75.6),IGHG2</em>00(75.6),IGHG4<em>00(75.6),IGHGP</em>00(75.6)</td> <td align=left>285</td> <td align=left>291</td> <td align=left>316</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>31</td> <td align=right>38</td> <td align=right>63</td> <td align=right>14</td> <td align=right>21</td> <td align=right></td> <td align=right>35.0;37</td> <td align=left>46</td> <td align=right>84</td> <td align=right>12</td> <td align=right>22</td> <td align=right>I43C</td> <td align=right>33.0;46</td> <td align=right>55</td> <td align=right>93</td> <td align=right>12</td> <td align=left>21</td> <td align=right>SA50G</td> <td align=left>31.0</td> </tr> <tr> <td align=right>67</td> <td align=right>103</td> <td align=right>22</td> <td align=right>0.0023501</td> <td align=left>TGTGCGAGCCTCATGTATTTCTATGGTCCCCTTGACCACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2333.4)</td> <td align=left>IGHD3-10*00(51)</td> <td align=left>IGHJ4*00(388.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>293</td> <td align=left>316</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>31</td> <td align=right>44</td> <td align=right>93</td> <td align=right>14</td> <td align=right>27</td> <td align=right>SA36T</td> <td align=right>51.0</td> <td align=left>26</td> <td align=right>37</td> <td align=right>68</td> <td align=right>31</td> <td align=right>42</td> <td align=right>ST31C</td> <td align=right>81.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>68</td> <td align=right>103</td> <td align=right>26</td> <td align=right>0.0023501</td> <td align=left>TGTGCGAGAGGGTTACGATATTTTGACTGGTTATTTCACAGTCAAGTCAACGAACACAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2097.1)</td> <td align=left>IGHD3-9*00(115)</td> <td align=left>IGHJ5*00(505.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>297</td> <td align=left>319</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>34</td> <td align=right>57</td> <td align=right>93</td> <td align=right>12</td> <td align=right>35</td> <td align=right></td> <td align=right>115.0</td> <td align=left>20</td> <td align=right>40</td> <td align=right>71</td> <td align=right>55</td> <td align=right>75</td> <td align=right></td> <td align=right>200.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>69</td> <td align=right>102</td> <td align=right>22</td> <td align=right>0.00232728</td> <td align=left>TGTGCGACTGCTTTGGTGGTGACTGCTGCATACTTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2462.4)</td> <td align=left>IGHD2-21*00(56)</td> <td align=left>IGHJ6*00(135)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>37</td> <td align=right>51</td> <td align=right>84</td> <td align=right>13</td> <td align=right>27</td> <td align=right>ST46C</td> <td align=right>56.0</td> <td align=left>44</td> <td align=right>52</td> <td align=right>83</td> <td align=right>31</td> <td align=right>39</td> <td align=right>SG46T</td> <td align=right>51.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>70</td> <td align=right>101</td> <td align=right>18</td> <td align=right>0.00230446</td> <td align=left>TGTGCACACACAGTGGATACGATTTTTGGAGTGGTTATTTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-5*00(2407.5)</td> <td align=left>IGHD3-3*00(110)</td> <td align=left>IGHJ4*00(416)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>298</td> <td align=left>321</td> <td align=left>0</td> <td align=right>10</td> <td align=right></td> <td align=right>100.0</td> <td align=right>35</td> <td align=right>57</td> <td align=right>93</td> <td align=right>17</td> <td align=right>39</td> <td align=right></td> <td align=right>110.0</td> <td align=left>25</td> <td align=right>37</td> <td align=right>68</td> <td align=right>39</td> <td align=right>51</td> <td align=right></td> <td align=right>120.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>71</td> <td align=right>100</td> <td align=right>22</td> <td align=right>0.00228165</td> <td align=left>TGTGCTAGAGTGGGCGGCGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-46*00(2335.5)</td> <td align=left>IGHD6-25*00(25)</td> <td align=left>IGHJ3*00(453.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>295</td> <td align=left>316</td> <td align=left>0</td> <td align=right>10</td> <td align=right>SG290T</td> <td align=right>71.0</td> <td align=right>28</td> <td align=right>33</td> <td align=right>54</td> <td align=right>13</td> <td align=right>18</td> <td align=right></td> <td align=right>25.0</td> <td align=left>24</td> <td align=right>39</td> <td align=right>70</td> <td align=right>18</td> <td align=right>33</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>72</td> <td align=right>100</td> <td align=right>21</td> <td align=right>0.00228165</td> <td align=left>TGTGCGAGAGATCGGGGCGATGACTACGCCAGTTACTACTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-69D*00(2349.7)</td> <td align=left>IGHD4-17<em>00(45),IGHD4-23</em>00(45),IGHD4-11*00(40)</td> <td align=left>IGHJ6*00(544.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>298</td> <td align=left>316</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>15</td> <td align=right>24</td> <td align=right>48</td> <td align=right>19</td> <td align=right>28</td> <td align=right></td> <td align=right>45.0;18</td> <td align=left>27</td> <td align=right>57</td> <td align=right>19</td> <td align=right>28</td> <td align=right></td> <td align=right>45.0;15</td> <td align=right>23</td> <td align=right>48</td> <td align=right>19</td> <td align=left>27</td> <td align=right></td> <td align=left>40.0</td> </tr> <tr> <td align=right>73</td> <td align=right>100</td> <td align=right>27</td> <td align=right>0.00228165</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGGATTGTAGTGGTGATAGCTACTACGACCAACTGGTTTGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(2136.3)</td> <td align=left>IGHD2-15*00(87)</td> <td align=left>IGHJ5*00(467.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right>SG48ASG54A</td> <td align=right>87.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right>SC30T</td> <td align=right>161.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>74</td> <td align=right>99</td> <td align=right>22</td> <td align=right>0.00225883</td> <td align=left>TGTGCGTCCCTACGGTGGCTACGGGGGGCCTTTGAGTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2273.1)</td> <td align=left>IGHD4-17*00(56)</td> <td align=left>IGHJ4*00(397.6)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=right>23</td> <td align=right>SA27G</td> <td align=right>56.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right>SC30G</td> <td align=right>101.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>75</td> <td align=right>99</td> <td align=right>24</td> <td align=right>0.00225883</td> <td align=left>TGTGCGAAGGAAGGGGCGTTCGGGGAGTTAGGGCTCGGCCTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(1863.7)</td> <td align=left>IGHD3-10<em>00(51),IGHD3-16</em>00(46)</td> <td align=left>IGHJ4*00(416.5)</td> <td align=left>IGHA1<em>00(144.5),IGHA2</em>00(144.5)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right>SA293G</td> <td align=right>81.0</td> <td align=right>42</td> <td align=right>55</td> <td align=right>93</td> <td align=right>17</td> <td align=right>30</td> <td align=right>SA46G</td> <td align=right>51.0;52</td> <td align=left>64</td> <td align=right>111</td> <td align=right>18</td> <td align=right>30</td> <td align=right>SG54C</td> <td align=right>46.0</td> <td align=right>26</td> <td align=right>37</td> <td align=right>68</td> <td align=left>40</td> <td align=right>51</td> <td align=left></td> </tr> <tr> <td align=right>76</td> <td align=right>99</td> <td align=right>23</td> <td align=right>0.00225883</td> <td align=left>TGTACGAGACTGCCGGATTACTATGATAGTAGTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2195.4)</td> <td align=left>IGHD3-22*00(81)</td> <td align=left>IGHJ4*00(408.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>298</td> <td align=left>319</td> <td align=left>0</td> <td align=right>10</td> <td align=right>SG291A</td> <td align=right>71.0</td> <td align=right>30</td> <td align=right>49</td> <td align=right>93</td> <td align=right>13</td> <td align=right>32</td> <td align=right>ST32G</td> <td align=right>81.0</td> <td align=left>27</td> <td align=right>37</td> <td align=right>68</td> <td align=right>32</td> <td align=right>42</td> <td align=right></td> <td align=right>100.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>77</td> <td align=right>98</td> <td align=right>21</td> <td align=right>0.00223601</td> <td align=left>TGTGCGGGCCAATTTGTAGTAGTGCCAGCCGCTAGTTACTACTATTACGCTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2242)</td> <td align=left>IGHD2-2*00(77)</td> <td align=left>IGHJ6*00(519.2)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>37</td> <td align=right>58</td> <td align=right>93</td> <td align=right>13</td> <td align=right>34</td> <td align=right>SA47GST53C</td> <td align=right>77.0</td> <td align=left>25</td> <td align=right>52</td> <td align=right>83</td> <td align=right>36</td> <td align=right>63</td> <td align=right>SC33TSG38C</td> <td align=right>212.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>78</td> <td align=right>98</td> <td align=right>20</td> <td align=right>0.00223601</td> <td align=left>TGTGCGAGACTTTCCTACTACTACGGTATGGACGTCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-59*00(2199.1)</td> <td align=left>nan</td> <td align=left>IGHJ6*00(524.8)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>nan</td> <td align=right>27</td> <td align=right>52</td> <td align=right>83</td> <td align=right>14</td> <td align=right>39</td> <td align=right></td> <td align=left>250.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>79</td> <td align=right>98</td> <td align=right>24</td> <td align=right>0.00223601</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGGATTGTAGTGGTGTTAGCTGCTACGACCAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(2118)</td> <td align=left>IGHD2-15*00(101)</td> <td align=left>IGHJ5*00(497.8)</td> <td align=left>IGHG1*00(74.6)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right>SG48T</td> <td align=right>101.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right></td> <td align=right>190.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>80</td> <td align=right>97</td> <td align=right>21</td> <td align=right>0.0022132</td> <td align=left>TGTGCGTCCCTTCGGTGGCTACGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2375.9)</td> <td align=left>IGHD5-12<em>00(45),IGHD5-18</em>00(45),IGHD4-17*00(42)</td> <td align=left>IGHJ4*00(437.2)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>32</td> <td align=right>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;32</td> <td align=left>41</td> <td align=right>69</td> <td align=right>14</td> <td align=right>23</td> <td align=right></td> <td align=right>45.0;19</td> <td align=right>33</td> <td align=right>48</td> <td align=right>9</td> <td align=left>23</td> <td align=right>SA21TSA27G</td> <td align=left>42.0</td> </tr> <tr> <td align=right>81</td> <td align=right>94</td> <td align=right>21</td> <td align=right>0.00214475</td> <td align=left>TGTGCGAGACGTCCGGCCACTACGATAGTACTGGCCCACACCAAGAAAGACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2089.9)</td> <td align=left>IGHD3-22<em>00(52),IGHD4-11</em>00(46),IGHD4-4*00(46)</td> <td align=left>IGHJ4*00(475.8)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>36</td> <td align=right>52</td> <td align=right>93</td> <td align=right>18</td> <td align=right>34</td> <td align=right>ST40CSG48C</td> <td align=right>52.0;27</td> <td align=left>42</td> <td align=right>48</td> <td align=right>18</td> <td align=right>33</td> <td align=right>I33ADT36</td> <td align=right>46.0;27</td> <td align=right>42</td> <td align=right>48</td> <td align=right>18</td> <td align=left>33</td> <td align=right>I33ADT36</td> <td align=left>46.0</td> </tr> <tr> <td align=right>82</td> <td align=right>94</td> <td align=right>23</td> <td align=right>0.00214475</td> <td align=left>TGTGCGAAATCCTACAGTGGGAGCTACTCCGCCGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2374.9)</td> <td align=left>IGHD1-26*00(67)</td> <td align=left>IGHJ3*00(370.2)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>294</td> <td align=left>316</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>22</td> <td align=right>41</td> <td align=right>60</td> <td align=right>12</td> <td align=right>31</td> <td align=right>ST24CSA38C</td> <td align=right>67.0</td> <td align=left>24</td> <td align=right>39</td> <td align=right>70</td> <td align=right>33</td> <td align=right>48</td> <td align=right></td> <td align=right>150.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>83</td> <td align=right>91</td> <td align=right>21</td> <td align=right>0.0020763</td> <td align=left>TGTGCGAGAGAAAGTCCCGAGCGGATTGTAGTGGTGGTAGCTGCTACGACCAACTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV1-2*00(1923.5)</td> <td align=left>IGHD2-15*00(115)</td> <td align=left>IGHJ5*00(495.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>36</td> <td align=right>59</td> <td align=right>93</td> <td align=right>24</td> <td align=right>47</td> <td align=right></td> <td align=right>115.0</td> <td align=left>21</td> <td align=right>40</td> <td align=right>71</td> <td align=right>50</td> <td align=right>69</td> <td align=right></td> <td align=right>190.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>84</td> <td align=right>89</td> <td align=right>20</td> <td align=right>0.00203067</td> <td align=left>TGTGCGGGTGAGAACGTGGATACAGGAATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2194.4)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(450.8)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4<em>00(74.9),IGHGP</em>00(74.9)</td> <td align=left>288</td> <td align=left>294</td> <td align=left>319</td> <td align=left>0</td> <td align=right>6</td> <td align=right></td> <td align=right>60.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30GST31A</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>85</td> <td align=right>89</td> <td align=right>26</td> <td align=right>0.00203067</td> <td align=left>TGTGCGAGGTGTTTCGGTGGGAGCTACTACGGGCCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-30*00(2097.5)</td> <td align=left>IGHD1-26*00(75)</td> <td align=left>IGHJ4*00(438)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>292</td> <td align=left>316</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>26</td> <td align=right>41</td> <td align=right>60</td> <td align=right>16</td> <td align=right>31</td> <td align=right></td> <td align=right>75.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>35</td> <td align=right>48</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>86</td> <td align=right>86</td> <td align=right>18</td> <td align=right>0.00196222</td> <td align=left>TGTGCACACAGACGGGGAAGGGCCACATGGATGGCTGGTTATTTCGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-5*00(2398.7)</td> <td align=left>IGHD6-19<em>00(40),IGHD5-12</em>00(36),IGHD5-5*00(36)</td> <td align=left>IGHJ4*00(383.1)</td> <td align=left>IGHG1<em>00(74.7),IGHG2</em>00(74.7),IGHG4<em>00(74.7),IGHGP</em>00(74.7)</td> <td align=left>288</td> <td align=left>302</td> <td align=left>321</td> <td align=left>0</td> <td align=right>14</td> <td align=right></td> <td align=right>140.0</td> <td align=right>32</td> <td align=right>40</td> <td align=right>63</td> <td align=right>31</td> <td align=right>39</td> <td align=right></td> <td align=right>40.0;19</td> <td align=left>29</td> <td align=right>69</td> <td align=right>22</td> <td align=right>32</td> <td align=right>SG23A</td> <td align=right>36.0;16</td> <td align=right>26</td> <td align=right>60</td> <td align=right>22</td> <td align=left>32</td> <td align=right>SG20A</td> <td align=left>36.0</td> </tr> <tr> <td align=right>87</td> <td align=right>85</td> <td align=right>19</td> <td align=right>0.0019394</td> <td align=left>TGTGCGACCGGGATACAGTTACAGTTATGGTCTCACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2351.9)</td> <td align=left>IGHD1-7<em>00(46),IGHD5-5</em>00(46),IGHD1-20*00(41)</td> <td align=left>IGHJ4*00(477)</td> <td align=left>IGHA1<em>00(144.6),IGHA2</em>00(144.6)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>2</td> <td align=right>14</td> <td align=right>51</td> <td align=right>16</td> <td align=right>28</td> <td align=right>SC6A</td> <td align=right>46.0;25</td> <td align=left>37</td> <td align=right>60</td> <td align=right>19</td> <td align=right>31</td> <td align=right>SC30T</td> <td align=right>46.0;3</td> <td align=right>14</td> <td align=right>51</td> <td align=right>17</td> <td align=left>28</td> <td align=right>SC6A</td> <td align=left>41.0</td> </tr> <tr> <td align=right>88</td> <td align=right>85</td> <td align=right>22</td> <td align=right>0.0019394</td> <td align=left>TGTGCGAGGTGGGGGCCTACGGTGACTCCCTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2380.4)</td> <td align=left>IGHD4-17*00(55)</td> <td align=left>IGHJ4*00(466.9)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>290</td> <td align=left>313</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>19</td> <td align=right>30</td> <td align=right>48</td> <td align=right>16</td> <td align=right>27</td> <td align=right></td> <td align=right>55.0</td> <td align=left>21</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>45</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>89</td> <td align=right>83</td> <td align=right>17</td> <td align=right>0.00189377</td> <td align=left>TGTGCGAGACGGCCCCCACTAGTGGTTATTCCGTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2173.2)</td> <td align=left>IGHD3-22<em>00(55),IGHD3-3</em>00(50)</td> <td align=left>IGHJ4*00(427.2)</td> <td align=left>IGHG1<em>00(75.9),IGHG2</em>00(75.9),IGHG4<em>00(75.9),IGHGP</em>00(75.9)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>46</td> <td align=right>57</td> <td align=right>93</td> <td align=right>19</td> <td align=right>30</td> <td align=right></td> <td align=right>55.0;47</td> <td align=left>57</td> <td align=right>93</td> <td align=right>20</td> <td align=right>30</td> <td align=right></td> <td align=right>50.0</td> <td align=right>25</td> <td align=right>37</td> <td align=right>68</td> <td align=left>33</td> <td align=right>45</td> <td align=left></td> </tr> <tr> <td align=right>90</td> <td align=right>83</td> <td align=right>18</td> <td align=right>0.00189377</td> <td align=left>TGTGCGAGAGACGAAATTCGGGGAATTAGATCCAGAACCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2455.5)</td> <td align=left>IGHD3-10<em>00(32),IGHD3-16</em>00(32),IGHD1-7*00(31)</td> <td align=left>IGHJ4*00(436.1)</td> <td align=left>IGHG1<em>00(74.9),IGHG2</em>00(74.9),IGHG4*00(74.9)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>43</td> <td align=right>55</td> <td align=right>93</td> <td align=right>16</td> <td align=right>28</td> <td align=right>SA46GSG51A</td> <td align=right>32.0;52</td> <td align=left>64</td> <td align=right>111</td> <td align=right>16</td> <td align=right>28</td> <td align=right>SG54CSG60A</td> <td align=right>32.0;1</td> <td align=right>10</td> <td align=right>51</td> <td align=right>26</td> <td align=left>35</td> <td align=right>ST4A</td> <td align=left>31.0</td> </tr> <tr> <td align=right>91</td> <td align=right>82</td> <td align=right>20</td> <td align=right>0.00187095</td> <td align=left>TGTGCACGGATACCGGGGGATTTTTGGAGTGGTTATTCCTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV2-70*00(2387.2)</td> <td align=left>IGHD3-3*00(95)</td> <td align=left>IGHJ4*00(465.1)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>301</td> <td align=left>321</td> <td align=left>0</td> <td align=right>13</td> <td align=right></td> <td align=right>130.0</td> <td align=right>38</td> <td align=right>57</td> <td align=right>93</td> <td align=right>18</td> <td align=right>37</td> <td align=right></td> <td align=right>95.0</td> <td align=left>21</td> <td align=right>37</td> <td align=right>68</td> <td align=right>38</td> <td align=right>54</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>92</td> <td align=right>81</td> <td align=right>17</td> <td align=right>0.00184813</td> <td align=left>TGTGCGAGAGGCAATATGATAATAGTGGCCTTTAGTGCTTTTGATATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2438)</td> <td align=left>IGHD3-22*00(56)</td> <td align=left>IGHJ3*00(417.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>294</td> <td align=left>313</td> <td align=left>0</td> <td align=right>12</td> <td align=right></td> <td align=right>120.0</td> <td align=right>38</td> <td align=right>52</td> <td align=right>93</td> <td align=right>14</td> <td align=right>28</td> <td align=right>SG45A</td> <td align=right>56.0</td> <td align=left>23</td> <td align=right>39</td> <td align=right>70</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>93</td> <td align=right>81</td> <td align=right>20</td> <td align=right>0.00184813</td> <td align=left>TGTGCGAGAGGAGGTGGATACAGCTATGGGGCCCTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2432.1)</td> <td align=left>IGHD5-5*00(80)</td> <td align=left>IGHJ4*00(418.7)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>282</td> <td align=left>293</td> <td align=left>313</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>20</td> <td align=right>36</td> <td align=right>60</td> <td align=right>13</td> <td align=right>29</td> <td align=right></td> <td align=right>80.0</td> <td align=left>26</td> <td align=right>37</td> <td align=right>68</td> <td align=right>34</td> <td align=right>45</td> <td align=right></td> <td align=right>110.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>94</td> <td align=right>80</td> <td align=right>18</td> <td align=right>0.00182532</td> <td align=left>TGTGCGCGAGATCTCACTACGGTGTCGATCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-48*00(2152.3)</td> <td align=left>IGHD4-17<em>00(45),IGHD4-23</em>00(45)</td> <td align=left>IGHJ3*00(362.3)</td> <td align=left>IGHG1<em>00(74.7),IGHG2</em>00(74.7),IGHG4<em>00(74.7),IGHGP</em>00(74.7)</td> <td align=left>285</td> <td align=left>300</td> <td align=left>316</td> <td align=left>0</td> <td align=right>15</td> <td align=right>SA291C</td> <td align=right>121.0</td> <td align=right>18</td> <td align=right>27</td> <td align=right>48</td> <td align=right>15</td> <td align=right>24</td> <td align=right></td> <td align=right>45.0;21</td> <td align=left>30</td> <td align=right>57</td> <td align=right>15</td> <td align=right>24</td> <td align=right></td> <td align=right>45.0</td> <td align=right>33</td> <td align=right>39</td> <td align=right>70</td> <td align=left>27</td> <td align=right>33</td> <td align=left></td> </tr> <tr> <td align=right>95</td> <td align=right>80</td> <td align=right>22</td> <td align=right>0.00182532</td> <td align=left>TGTGCGAGCGCTCCCCGCCCGTATTACTATGATAGTAGTGGTTATTACCCATTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-48*00(2155.1)</td> <td align=left>IGHD3-22*00(145)</td> <td align=left>IGHJ4*00(426.6)</td> <td align=left>IGHG1<em>00(74.6),IGHG2</em>00(74.6),IGHG4<em>00(74.6),IGHGP</em>00(74.6)</td> <td align=left>285</td> <td align=left>293</td> <td align=left>316</td> <td align=left>0</td> <td align=right>8</td> <td align=right></td> <td align=right>80.0</td> <td align=right>30</td> <td align=right>59</td> <td align=right>93</td> <td align=right>19</td> <td align=right>48</td> <td align=right></td> <td align=right>145.0</td> <td align=left>25</td> <td align=right>37</td> <td align=right>68</td> <td align=right>51</td> <td align=right>63</td> <td align=right></td> <td align=right>120.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>96</td> <td align=right>78</td> <td align=right>15</td> <td align=right>0.00177968</td> <td align=left>TGTGCGACTGAGAACGTGGATACAGGAATGGTTCCCTGGTTCGACCCCTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-39*00(2367.5)</td> <td align=left>IGHD5-5*00(72)</td> <td align=left>IGHJ5*00(467.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>288</td> <td align=left>295</td> <td align=left>319</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>18</td> <td align=right>38</td> <td align=right>60</td> <td align=right>13</td> <td align=right>33</td> <td align=right>SC30GST31A</td> <td align=right>72.0</td> <td align=left>24</td> <td align=right>40</td> <td align=right>71</td> <td align=right>35</td> <td align=right>51</td> <td align=right></td> <td align=right>160.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>97</td> <td align=right>78</td> <td align=right>21</td> <td align=right>0.00177968</td> <td align=left>TGTGCGAACAGTGGCTGGCACCGGGGGGCCTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV3-53*00(2155.6)</td> <td align=left>IGHD6-19*00(51)</td> <td align=left>IGHJ4*00(435.5)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>282</td> <td align=left>289</td> <td align=left>313</td> <td align=left>0</td> <td align=right>7</td> <td align=right></td> <td align=right>70.0</td> <td align=right>29</td> <td align=right>42</td> <td align=right>63</td> <td align=right>8</td> <td align=right>21</td> <td align=right>ST39C</td> <td align=right>51.0</td> <td align=left>24</td> <td align=right>37</td> <td align=right>68</td> <td align=right>29</td> <td align=right>42</td> <td align=right></td> <td align=right>130.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>98</td> <td align=right>76</td> <td align=right>18</td> <td align=right>0.00173405</td> <td align=left>TGTGCGAGACGGCCGGCTACTCTGATAGTAATGGCCCACACCAAGAATGACTACTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-34*00(2074.4)</td> <td align=left>IGHD3-22*00(57)</td> <td align=left>IGHJ4*00(452.7)</td> <td align=left>IGHG1<em>00(74.8),IGHG2</em>00(74.8),IGHG4<em>00(74.8),IGHGP</em>00(74.8)</td> <td align=left>282</td> <td align=left>291</td> <td align=left>313</td> <td align=left>0</td> <td align=right>9</td> <td align=right></td> <td align=right>90.0</td> <td align=right>35</td> <td align=right>52</td> <td align=right>93</td> <td align=right>17</td> <td align=right>34</td> <td align=right>SA39CSG48A</td> <td align=right>57.0</td> <td align=left>20</td> <td align=right>37</td> <td align=right>68</td> <td align=right>49</td> <td align=right>66</td> <td align=right></td> <td align=right>170.0</td> <td align=right>;;;</td> <td align=right>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>99</td> <td align=right>74</td> <td align=right>15</td> <td align=right>0.00168842</td> <td align=left>TGTGCGAGAGACGAAATTCGGGGAGTTAGATCCAGAATGTTTGACTACTGG</td> <td align=left>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</td> <td align=left>IGHV4-4*00(2253.9)</td> <td align=left>IGHD3-10<em>00(46),IGHD3-16</em>00(46)</td> <td align=left>IGHJ4*00(396.7)</td> <td align=left>IGHG1<em>00(75),IGHG2</em>00(75),IGHG4<em>00(75),IGHGP</em>00(75)</td> <td align=left>285</td> <td align=left>296</td> <td align=left>316</td> <td align=left>0</td> <td align=right>11</td> <td align=right></td> <td align=right>110.0</td> <td align=right>43</td> <td align=right>55</td> <td align=right>93</td> <td align=right>16</td> <td align=right>28</td> <td align=right>SA46G</td> <td align=right>46.0;52</td> <td align=left>64</td> <td align=right>111</td> <td align=right>16</td> <td align=right>28</td> <td align=right>SG54C</td> <td align=right>46.0</td> <td align=right>25</td> <td align=right>37</td> <td align=right>68</td> <td align=left>39</td> <td align=right>51</td> <td align=left></td> </tr> </tbody> </table> </details> <h3 id=under-the-hood-pipeline>Under the hood pipeline:</h3> <p>Under the hood the command above actually executes the following pipeline:</p> <h4 id=align><code>align</code></h4> <p>Alignment of raw sequencing reads against reference database of V-, D-, J- and C- gene segments.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>align<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--species<span class=w> </span>hsa<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-p<span class=w> </span>kAligner2_4.0<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.geneFeatureToAlign<span class=o>=</span><span class=s2>&quot;VTranscriptWithout5UTRWithP&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.floatingLeftBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OjParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OcParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--tagPattern<span class=w> </span><span class=s2>&quot;^N{22}(R1:*) \ ^(UMI:N{17})(R2:*)&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.report.txt<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.report.json<span class=w> </span><span class=se>\</span>
<span class=w>     </span>raw/13_d60_LN_germCenterB_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>     </span>raw/13_d60_LN_germCenterB_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>     </span>results/13_d60_LN_germCenterB.vdjca
</code></pre></div> <p>Option <code>--report</code> is specified here explicitly.</p> <dl> <dt><code>--species hsa</code></dt> <dd>determines the organism species (hsa for <em>Homo Sapiens</em>).</dd> <dt><code>-p kaligner2_4.0</code></dt> <dd>a default preset of MiXCR parameters which includes a dedicated BCR aligner.</dd> <dt><code>-OvParameters.geneFeatureToAlign="VTranscriptWithout5UTRWithP"</code></dt> <dd>Sets a V gene feature to align. Check <a href=../../reference/ref-gene-features/ >gene features</a> for more info.</dd> <dt><code>-OvParameters.parameters.floatingLeftBound=false</code></dt> <dd>Results in a global alignment algorithm for V gene left bound. We use it because we don't have any primers covering V gene coding sequence.</dd> <dt><code>-OjParameters.parameters.floatingRightBound=false -OcParameters.parameters.floatingRightBound=false</code></dt> <dd>Results in a global alignment algorithm for J and C gene right bound, because primer sequences have been trimmed by <code>--tagPattern</code>.</dd> <dt><code>--tagPattern "^N{22}(R1:*) \ ^(UMI:N{17})(R2:*)"</code></dt> <dd>tag pattern specifies the location of the UMI barcode and also trims C primer sequences located in the beginning of R1 (22 b.p. is the length of the longest primer).</dd> </dl> <h4 id=refinetagsandsort><code>refineTagsAndSort</code></h4> <p><a href=../../reference/mixcr-refineTagsAndSort/ >Corrects</a> sequencing and PCR errors <em>inside</em> barcode sequences. This step does extremely important job by correcting artificial diversity caused by errors in barcodes. In the considered example project it corrects only sequences of UMIs.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>refineTagsAndSort<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.report.txt<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.report.json<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.vdjca<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.refined.vdjca
</code></pre></div> <h4 id=assemble><code>assemble</code></h4> <p>Assembles alignments into clonotypes and applies several layers of errors correction:</p> <ul> <li>quality-dependent correction for sequencing errors</li> <li>PCR-error correction by clustering </li> <li>UMI-based error correction)</li> </ul> <p>Check <a href=../../reference/mixcr-assemble/ ><code>mixcr assemble</code></a> for more information.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>assemble<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OassemblingFeatures<span class=o>=</span><span class=s2>&quot;CDR3&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OseparateByJ<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OseparateByV<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OseparateByC<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.report.txt<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.report.json<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.refined.vdjca<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clns
</code></pre></div> <p>Options <code>--report</code> and <code>--json-report</code> are specified here explicitly so that the report files will be appended with assembly report.</p> <dl> <dt><code>-OassemblingFeatures=CDR3</code></dt> <dd>By default <code>nebnext-mouse-bcr-cdr3</code> preset assembles clones by <code>CDR3</code> sequence.</dd> <dt><code>-separateByJ: true</code></dt> <dd>Separate clones with the same assembling feature, but different V-genes.</dd> <dt><code>-separateByV: true</code></dt> <dd>Separate clones with the same assembling feature, but different J-genes.</dd> <dt><code>-separateByC: true</code></dt> <dd>Separate clones with the same assembling feature, but different C-genes, which is essential for isotype identification.</dd> </dl> <h4 id=export><code>export</code></h4> <p>Exports clonotypes from .clns file into human-readable tables.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>exportClones<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-c<span class=w> </span>IGH<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-uniqueTagCount<span class=w> </span>UMI<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clns<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clones.IGH.txt
</code></pre></div> <dl> <dt><code>- IGH</code></dt> <dd>defines a specific chain to be exported.</dd> <dt><code>-uniqueTagCount UMI</code></dt> <dd>adds a column with the number of UMIs for each clone.</dd> </dl> <h2 id=quality-control>Quality control</h2> <p>Now when we have all files processed lets perform Quality Control. That can be easily done using <a href=../../reference/mixcr-exportQc/ ><code>mixcr exportQc</code></a> function.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>exportQc<span class=w> </span>align<span class=w> </span>results/*.clns<span class=w> </span>figs/alignQc.pdf
</code></pre></div> <p><img alt="align QC" src=figs/alignQc.svg></p> <p>Most of the samples have a high successful alignment score. But samples that come from blood memory B cells clearly have a lower percentage of aligned reads, and the major reason for that is the lack of immune receptor sequences. That might be due to some issues during sample preparation. It is recommended to realign one of these samples and save not aligned reads (see <a href=../../reference/mixcr-align/ ><code>mixcr align</code></a>) into separate file for manual inspection. That can be done with the following command:</p> <div class=highlight><pre><span></span><code>mkdir<span class=w> </span>-p<span class=w> </span>debug

mixcr<span class=w> </span>align<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--species<span class=w> </span>hsa<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-p<span class=w> </span>kAligner2_4.0<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.geneFeatureToAlign<span class=o>=</span><span class=s2>&quot;VTranscriptWithout5UTRWithP&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.floatingLeftBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OjParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OcParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--tag-pattern<span class=w> </span><span class=s2>&quot;^N{22}(R1:*) \ ^(UMI:N{17})(R2:*)&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--report<span class=w> </span>debug/13_d60_LN_germCenterB.report.txt<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json-report<span class=w> </span>debug/13_d60_LN_germCenterB.report.json<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--not-aligned-R1<span class=w> </span>debug/13_d60_LN_germCenterB_notAligned_R1.fastq<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--not-aligned-R2<span class=w> </span>debug/13_d60_LN_germCenterB_notAligned_R2.fastq<span class=w> </span><span class=se>\</span>
<span class=w>     </span>raw/13_d60_LN_germCenterB_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>     </span>raw/13_d60_LN_germCenterB_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>     </span>debug/13_d60_LN_germCenterB.vdjca
</code></pre></div> <p>Resulting <code>13_d60_LN_germCenterB_notAligned_R1.fastq</code> and <code>13_d60_LN_germCenterB_notAligned_R2.fastq</code> files can be manually inspected. A brief <a href=https://blast.ncbi.nlm.nih.gov/Blast.cgi>BLAST</a> search revealed most not aligned sequences come from DNA contamination (they align within sequence between segments) and with immunoglobulin like genes (e.g.IGLL5).</p> <p>Now Lets look at the chain distribution in every sample.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>exportQc<span class=w> </span>chainUsage<span class=w> </span>results/*.clns<span class=w> </span>figs/chainUsage.pdf
</code></pre></div> <p>If we look at the chain usage plot, we see, as expected, that the samples are almost entirely consist of IGH chains, since only heavy BCR chains we sequenced.</p> <p><img alt=chainUsage.svg src=figs/chainUsage.svg></p> <h2 id=full-length-clonotype-assembly>Full-length clonotype assembly</h2> <p>In the example above, we have assembled clones by a default <code>CDR3</code> feature. But NEBNext Immune Sequencing Kit covers the whole sequence of V and J genes, because it is a 5'RACE based protocol. And also C primers allow to capture enough sequence for isotyping. </p> <p>Taking into account what is mentioned above, the longest possible assembling feature for this protocol is <code>VDJRegion</code>.</p> <p>MiXCR has a specific preset to obtain full-length BCR clones with NEBNext Immune Sequencing Kit:</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>analyze<span class=w> </span>nebnext-human-bcr-full-length.yaml<span class=w> </span><span class=se>\</span>
<span class=w>    </span>raw/13_d60_LN_germCenterB_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>raw/13_d60_LN_germCenterB_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB
</code></pre></div> <p>The <code>mixcr assemble</code> step in this preset differs from the one above in the following manner:</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>assemble<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OassemblingFeatures<span class=o>=</span><span class=s2>&quot;VDJRegion&quot;</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OseparateByC<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.report.txt<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.report.json<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.refined.vdjca<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clns
</code></pre></div> <dl> <dt><code>-OassemblingFeatures="VDJRegion"</code></dt> <dd>sets the assembling feature to the region which starts from <code>FR1Begin</code> and ends at the end of <code>FR4</code>.</dd> </dl> <p>Notice we omit <code>-OseparateByV=true</code> and <code>-OseparateByJ=true</code> in this case because assembling feature already covers full V and J genes sequences, thus in case if clones have identical <code>CDR3</code> they will still be separated. We still use <code>-OseparateByC=true</code> option for isotype identification.</p> <h2 id=reports>Reports</h2> <p>Finally, MiXCR provides a very convenient way to look at the reports generated at ech step. Every <code>.vdjca</code>, <code>.clns</code> and <code>.clna</code> file holds all the reports for every MiXCR function that has been applied to this sample. E.g. in our case <code>.clns</code> file contains reports for <code>mixcr align</code> and <code>mixcr assemble</code>. To output this report use <a href=../../reference/mixcr-exportReports/ ><code>mixcr exportReports</code></a> as shown bellow. Note <code>--json</code> parameter will output a JSON-formatted report.</p> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>exportReports<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clns<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.report.txt
</code></pre></div> <div class=highlight><pre><span></span><code>mixcr<span class=w> </span>exportReports<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--json<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.clns<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/13_d60_LN_germCenterB.report.json
</code></pre></div> <details> <summary>Show report file</summary> <div class="tabbed-set tabbed-alternate" data-tabs=2:2><input checked=checked id=__tabbed_2_1 name=__tabbed_2 type=radio><input id=__tabbed_2_2 name=__tabbed_2 type=radio><div class=tabbed-labels><label for=__tabbed_2_1><code>.txt</code></label><label for=__tabbed_2_2><code>.json</code></label></div> <div class=tabbed-content> <div class=tabbed-block> <div class=highlight><pre><span></span><code><span class=o>==============</span><span class=w> </span>Align<span class=w> </span><span class=nv>Report</span><span class=w> </span><span class=o>==============</span>
Input<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R1.fastq.gz,/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R2.fastq.gz
Output<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>results/13_d60_LN_germCenterB.vdjca
Version:<span class=w> </span><span class=p>;</span><span class=w> </span><span class=nv>built</span><span class=o>=</span>Mon<span class=w> </span>Sep<span class=w> </span><span class=m>26</span><span class=w> </span><span class=m>10</span>:55:18<span class=w> </span>CEST<span class=w> </span><span class=m>2022</span><span class=p>;</span><span class=w> </span><span class=nv>rev</span><span class=o>=</span>8c998df1ab<span class=p>;</span><span class=w> </span><span class=nv>lib</span><span class=o>=</span>repseqio.v2.0
Command<span class=w> </span>line<span class=w> </span>arguments:<span class=w> </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.align.report.txt<span class=w> </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.align.report.json<span class=w> </span>--preset<span class=w> </span>local:nebnext-human-bcr-cdr3<span class=w> </span>+limitInput<span class=w> </span><span class=m>100000</span><span class=w> </span>/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R1.fastq.gz<span class=w> </span>/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R2.fastq.gz<span class=w> </span>results/13_d60_LN_germCenterB.vdjca
Analysis<span class=w> </span>time:<span class=w> </span>0ns
Total<span class=w> </span>sequencing<span class=w> </span>reads:<span class=w> </span><span class=m>100000</span>
Successfully<span class=w> </span>aligned<span class=w> </span>reads:<span class=w> </span><span class=m>88595</span><span class=w> </span><span class=o>(</span><span class=m>88</span>.59%<span class=o>)</span>
Paired-end<span class=w> </span>alignment<span class=w> </span>conflicts<span class=w> </span>eliminated:<span class=w> </span><span class=m>30693</span><span class=w> </span><span class=o>(</span><span class=m>30</span>.69%<span class=o>)</span>
Alignment<span class=w> </span>failed,<span class=w> </span>no<span class=w> </span>hits<span class=w> </span><span class=o>(</span>not<span class=w> </span>TCR/IG?<span class=o>)</span>:<span class=w> </span><span class=m>3851</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.85%<span class=o>)</span>
Alignment<span class=w> </span>failed<span class=w> </span>because<span class=w> </span>of<span class=w> </span>absence<span class=w> </span>of<span class=w> </span>V<span class=w> </span>hits:<span class=w> </span><span class=m>3260</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.26%<span class=o>)</span>
Alignment<span class=w> </span>failed<span class=w> </span>because<span class=w> </span>of<span class=w> </span>absence<span class=w> </span>of<span class=w> </span>J<span class=w> </span>hits:<span class=w> </span><span class=m>2521</span><span class=w> </span><span class=o>(</span><span class=m>2</span>.52%<span class=o>)</span>
No<span class=w> </span>target<span class=w> </span>with<span class=w> </span>both<span class=w> </span>V<span class=w> </span>and<span class=w> </span>J<span class=w> </span>alignments:<span class=w> </span><span class=m>1773</span><span class=w> </span><span class=o>(</span><span class=m>1</span>.77%<span class=o>)</span>
Overlapped:<span class=w> </span><span class=m>49087</span><span class=w> </span><span class=o>(</span><span class=m>49</span>.09%<span class=o>)</span>
Overlapped<span class=w> </span>and<span class=w> </span>aligned:<span class=w> </span><span class=m>44737</span><span class=w> </span><span class=o>(</span><span class=m>44</span>.74%<span class=o>)</span>
Alignment-aided<span class=w> </span>overlaps:<span class=w> </span><span class=m>2058</span><span class=w> </span><span class=o>(</span><span class=m>4</span>.6%<span class=o>)</span>
Overlapped<span class=w> </span>and<span class=w> </span>not<span class=w> </span>aligned:<span class=w> </span><span class=m>4350</span><span class=w> </span><span class=o>(</span><span class=m>4</span>.35%<span class=o>)</span>
No<span class=w> </span>CDR3<span class=w> </span>parts<span class=w> </span>alignments,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>successfully<span class=w> </span>aligned:<span class=w> </span><span class=m>44</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.05%<span class=o>)</span>
Partial<span class=w> </span>aligned<span class=w> </span>reads,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>successfully<span class=w> </span>aligned:<span class=w> </span><span class=m>1970</span><span class=w> </span><span class=o>(</span><span class=m>2</span>.22%<span class=o>)</span>
V<span class=w> </span>gene<span class=w> </span>chimeras:<span class=w> </span><span class=m>2068</span><span class=w> </span><span class=o>(</span><span class=m>2</span>.07%<span class=o>)</span>
J<span class=w> </span>gene<span class=w> </span>chimeras:<span class=w> </span><span class=m>5</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.01%<span class=o>)</span>
IGH<span class=w> </span>chains:<span class=w> </span><span class=m>88288</span><span class=w> </span><span class=o>(</span><span class=m>99</span>.65%<span class=o>)</span>
IGH<span class=w> </span>non-functional:<span class=w> </span><span class=m>2924</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.31%<span class=o>)</span>
IGK<span class=w> </span>chains:<span class=w> </span><span class=m>253</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.29%<span class=o>)</span>
IGK<span class=w> </span>non-functional:<span class=w> </span><span class=m>10</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.95%<span class=o>)</span>
IGL<span class=w> </span>chains:<span class=w> </span><span class=m>54</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.06%<span class=o>)</span>
IGL<span class=w> </span>non-functional:<span class=w> </span><span class=m>5</span><span class=w> </span><span class=o>(</span><span class=m>9</span>.26%<span class=o>)</span>
Realigned<span class=w> </span>with<span class=w> </span>forced<span class=w> </span>non-floating<span class=w> </span>bound:<span class=w> </span><span class=m>105942</span><span class=w> </span><span class=o>(</span><span class=m>105</span>.94%<span class=o>)</span>
Realigned<span class=w> </span>with<span class=w> </span>forced<span class=w> </span>non-floating<span class=w> </span>right<span class=w> </span>bound<span class=w> </span><span class=k>in</span><span class=w> </span>left<span class=w> </span>read:<span class=w> </span><span class=m>25837</span><span class=w> </span><span class=o>(</span><span class=m>25</span>.84%<span class=o>)</span>
Realigned<span class=w> </span>with<span class=w> </span>forced<span class=w> </span>non-floating<span class=w> </span>left<span class=w> </span>bound<span class=w> </span><span class=k>in</span><span class=w> </span>right<span class=w> </span>read:<span class=w> </span><span class=m>25837</span><span class=w> </span><span class=o>(</span><span class=m>25</span>.84%<span class=o>)</span>
<span class=o>==============</span><span class=w> </span>RefineTagsAndSort<span class=w> </span><span class=nv>Report</span><span class=w> </span><span class=o>==============</span>
Input<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>results/13_d60_LN_germCenterB.vdjca
Output<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>results/13_d60_LN_germCenterB.refined.vdjca
Version:<span class=w> </span><span class=p>;</span><span class=w> </span><span class=nv>built</span><span class=o>=</span>Mon<span class=w> </span>Sep<span class=w> </span><span class=m>26</span><span class=w> </span><span class=m>10</span>:55:18<span class=w> </span>CEST<span class=w> </span><span class=m>2022</span><span class=p>;</span><span class=w> </span><span class=nv>rev</span><span class=o>=</span>8c998df1ab<span class=p>;</span><span class=w> </span><span class=nv>lib</span><span class=o>=</span>repseqio.v2.0
Command<span class=w> </span>line<span class=w> </span>arguments:<span class=w> </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.refine.report.txt<span class=w> </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.refine.report.json<span class=w> </span>results/13_d60_LN_germCenterB.vdjca<span class=w> </span>results/13_d60_LN_germCenterB.refined.vdjca
Analysis<span class=w> </span>time:<span class=w> </span>0ns
Time<span class=w> </span>spent<span class=w> </span><span class=k>in</span><span class=w> </span>correction:<span class=w> </span>0ns
UMI<span class=w> </span>input<span class=w> </span>diversity:<span class=w> </span><span class=m>44135</span>
UMI<span class=w> </span>output<span class=w> </span>diversity:<span class=w> </span><span class=m>23581</span><span class=w> </span><span class=o>(</span><span class=m>53</span>.43%<span class=o>)</span>
UMI<span class=w> </span>input<span class=w> </span>reads:<span class=w> </span><span class=m>88595</span>
UMI<span class=w> </span>output<span class=w> </span>count:<span class=w> </span><span class=m>70348</span><span class=w> </span><span class=o>(</span><span class=m>79</span>.4%<span class=o>)</span>
UMI<span class=w> </span>mean<span class=w> </span>reads<span class=w> </span>per<span class=w> </span>tag:<span class=w> </span><span class=m>2</span>.01
UMI<span class=w> </span>input<span class=w> </span>core<span class=w> </span>diversity:<span class=w> </span><span class=m>23770</span><span class=w> </span><span class=o>(</span><span class=m>53</span>.86%<span class=o>)</span>
UMI<span class=w> </span>input<span class=w> </span>core<span class=w> </span>reads:<span class=w> </span><span class=m>64187</span><span class=w> </span><span class=o>(</span><span class=m>72</span>.45%<span class=o>)</span>
UMI<span class=w> </span>directly<span class=w> </span>corrected<span class=w> </span>diversity:<span class=w> </span><span class=m>5918</span><span class=w> </span><span class=o>(</span><span class=m>13</span>.41%<span class=o>)</span>
UMI<span class=w> </span>directly<span class=w> </span>corrected<span class=w> </span>reads:<span class=w> </span><span class=m>6351</span><span class=w> </span><span class=o>(</span><span class=m>7</span>.17%<span class=o>)</span>
UMI<span class=w> </span>diversity<span class=w> </span>filtered<span class=w> </span>by<span class=w> </span>tag<span class=w> </span>quality:<span class=w> </span><span class=m>14636</span><span class=w> </span><span class=o>(</span><span class=m>33</span>.16%<span class=o>)</span>
UMI<span class=w> </span>reads<span class=w> </span>filtered<span class=w> </span>by<span class=w> </span>tag<span class=w> </span>quality:<span class=w> </span><span class=m>18247</span><span class=w> </span><span class=o>(</span><span class=m>20</span>.6%<span class=o>)</span>
UMI<span class=w> </span>diversity<span class=w> </span>filtered<span class=w> </span>by<span class=w> </span>whitelist:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
UMI<span class=w> </span>recursively<span class=w> </span>corrected:<span class=w> </span><span class=m>6</span>
Number<span class=w> </span>of<span class=w> </span>output<span class=w> </span>records:<span class=w> </span><span class=m>52064</span><span class=w> </span><span class=o>(</span><span class=m>58</span>.77%<span class=o>)</span>
Filter<span class=w> </span>report:
<span class=w>  </span>Number<span class=w> </span>of<span class=w> </span>groups:<span class=w> </span><span class=m>23581</span>
<span class=w>  </span>Number<span class=w> </span>of<span class=w> </span>groups<span class=w> </span>accepted:<span class=w> </span><span class=m>11348</span><span class=w> </span><span class=o>(</span><span class=m>48</span>.12%<span class=o>)</span>
<span class=w>  </span>Total<span class=w> </span>records<span class=w> </span>weight:<span class=w> </span><span class=m>70348</span>.0
<span class=w>  </span>Records<span class=w> </span>weight<span class=w> </span>accepted:<span class=w> </span><span class=m>52064</span>.0<span class=w> </span><span class=o>(</span><span class=m>74</span>.01%<span class=o>)</span>
<span class=w>  </span>Operator<span class=w> </span><span class=c1>#0:</span>
<span class=w>    </span>Effective<span class=w> </span>threshold:<span class=w> </span><span class=m>3</span>.0
<span class=o>==============</span><span class=w> </span>Assemble<span class=w> </span><span class=nv>Report</span><span class=w> </span><span class=o>==============</span>
Input<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>results/13_d60_LN_germCenterB.refined.vdjca
Output<span class=w> </span>file<span class=o>(</span>s<span class=o>)</span>:<span class=w> </span>results/13_d60_LN_germCenterB.clns
Version:<span class=w> </span><span class=p>;</span><span class=w> </span><span class=nv>built</span><span class=o>=</span>Mon<span class=w> </span>Sep<span class=w> </span><span class=m>26</span><span class=w> </span><span class=m>10</span>:55:18<span class=w> </span>CEST<span class=w> </span><span class=m>2022</span><span class=p>;</span><span class=w> </span><span class=nv>rev</span><span class=o>=</span>8c998df1ab<span class=p>;</span><span class=w> </span><span class=nv>lib</span><span class=o>=</span>repseqio.v2.0
Command<span class=w> </span>line<span class=w> </span>arguments:<span class=w> </span>--report<span class=w> </span>results/13_d60_LN_germCenterB.assemble.report.txt<span class=w> </span>--json-report<span class=w> </span>results/13_d60_LN_germCenterB.assemble.report.json<span class=w> </span>results/13_d60_LN_germCenterB.refined.vdjca<span class=w> </span>results/13_d60_LN_germCenterB.clns
Analysis<span class=w> </span>time:<span class=w> </span>0ns
Number<span class=w> </span>of<span class=w> </span>input<span class=w> </span>groups:<span class=w> </span><span class=m>11348</span>
Number<span class=w> </span>of<span class=w> </span>input<span class=w> </span>alignments:<span class=w> </span><span class=m>52064</span>
Number<span class=w> </span>of<span class=w> </span>output<span class=w> </span>pre-clonotypes:<span class=w> </span><span class=m>10644</span>
Number<span class=w> </span>of<span class=w> </span>clonotypes<span class=w> </span>per<span class=w> </span>group:
<span class=w>  </span><span class=m>0</span>:<span class=w> </span>+<span class=w> </span><span class=m>674</span><span class=w> </span><span class=o>(</span><span class=m>5</span>.96%<span class=o>)</span><span class=w> </span><span class=o>=</span><span class=w> </span><span class=m>674</span><span class=w> </span><span class=o>(</span><span class=m>5</span>.96%<span class=o>)</span>
<span class=w>  </span><span class=m>1</span>:<span class=w> </span>+<span class=w> </span><span class=m>10644</span><span class=w> </span><span class=o>(</span><span class=m>94</span>.04%<span class=o>)</span><span class=w> </span><span class=o>=</span><span class=w> </span><span class=m>11318</span><span class=w> </span><span class=o>(</span><span class=m>100</span>%<span class=o>)</span>
Number<span class=w> </span>of<span class=w> </span>core<span class=w> </span>alignments:<span class=w> </span><span class=m>47804</span><span class=w> </span><span class=o>(</span><span class=m>91</span>.82%<span class=o>)</span>
Discarded<span class=w> </span>core<span class=w> </span>alignments:<span class=w> </span><span class=m>1661</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.47%<span class=o>)</span>
Empirically<span class=w> </span>assigned<span class=w> </span>alignments:<span class=w> </span><span class=m>1695</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.26%<span class=o>)</span>
Empirical<span class=w> </span>assignment<span class=w> </span>conflicts:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
UMI+VJ-gene<span class=w> </span>empirically<span class=w> </span>assigned<span class=w> </span>alignments:<span class=w> </span><span class=m>1695</span><span class=w> </span><span class=o>(</span><span class=m>3</span>.26%<span class=o>)</span>
VJ-gene<span class=w> </span>empirically<span class=w> </span>assigned<span class=w> </span>alignments:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
UMI<span class=w> </span>empirically<span class=w> </span>assigned<span class=w> </span>alignments:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Number<span class=w> </span>of<span class=w> </span>ambiguous<span class=w> </span>UMIs:<span class=w> </span><span class=m>0</span>
Number<span class=w> </span>of<span class=w> </span>ambiguous<span class=w> </span>UMI+V/J-gene<span class=w> </span>combinations:<span class=w> </span><span class=m>0</span>
Unassigned<span class=w> </span>alignments:<span class=w> </span><span class=m>2430</span><span class=w> </span><span class=o>(</span><span class=m>4</span>.67%<span class=o>)</span>
Final<span class=w> </span>clonotype<span class=w> </span>count:<span class=w> </span><span class=m>1180</span>
Average<span class=w> </span>number<span class=w> </span>of<span class=w> </span>reads<span class=w> </span>per<span class=w> </span>clonotype:<span class=w> </span><span class=m>39</span>.9
Reads<span class=w> </span>used<span class=w> </span><span class=k>in</span><span class=w> </span>clonotypes,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>47080</span><span class=w> </span><span class=o>(</span><span class=m>47</span>.08%<span class=o>)</span>
Reads<span class=w> </span>used<span class=w> </span><span class=k>in</span><span class=w> </span>clonotypes<span class=w> </span>before<span class=w> </span>clustering,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>49371</span><span class=w> </span><span class=o>(</span><span class=m>49</span>.37%<span class=o>)</span>
Number<span class=w> </span>of<span class=w> </span>reads<span class=w> </span>used<span class=w> </span>as<span class=w> </span>a<span class=w> </span>core,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>used:<span class=w> </span><span class=m>49353</span><span class=w> </span><span class=o>(</span><span class=m>99</span>.96%<span class=o>)</span>
Mapped<span class=w> </span>low<span class=w> </span>quality<span class=w> </span>reads,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>used:<span class=w> </span><span class=m>18</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.04%<span class=o>)</span>
Reads<span class=w> </span>clustered<span class=w> </span><span class=k>in</span><span class=w> </span>PCR<span class=w> </span>error<span class=w> </span>correction,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>used:<span class=w> </span><span class=m>2291</span><span class=w> </span><span class=o>(</span><span class=m>4</span>.64%<span class=o>)</span>
Reads<span class=w> </span>pre-clustered<span class=w> </span>due<span class=w> </span>to<span class=w> </span>the<span class=w> </span>similar<span class=w> </span>VJC-lists,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>used:<span class=w> </span><span class=m>108</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.23%<span class=o>)</span>
Reads<span class=w> </span>dropped<span class=w> </span>due<span class=w> </span>to<span class=w> </span>the<span class=w> </span>lack<span class=w> </span>of<span class=w> </span>a<span class=w> </span>clone<span class=w> </span>sequence,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>2599</span><span class=w> </span><span class=o>(</span><span class=m>2</span>.6%<span class=o>)</span>
Reads<span class=w> </span>dropped<span class=w> </span>due<span class=w> </span>to<span class=w> </span>a<span class=w> </span>too<span class=w> </span>short<span class=w> </span>clonal<span class=w> </span>sequence,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads<span class=w> </span>dropped<span class=w> </span>due<span class=w> </span>to<span class=w> </span>low<span class=w> </span>quality,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>9</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.01%<span class=o>)</span>
Reads<span class=w> </span>dropped<span class=w> </span>due<span class=w> </span>to<span class=w> </span>failed<span class=w> </span>mapping,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>71</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.07%<span class=o>)</span>
Reads<span class=w> </span>dropped<span class=w> </span>with<span class=w> </span>low<span class=w> </span>quality<span class=w> </span>clones,<span class=w> </span>percent<span class=w> </span>of<span class=w> </span>total:<span class=w> </span><span class=m>57</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.06%<span class=o>)</span>
Clonotypes<span class=w> </span>eliminated<span class=w> </span>by<span class=w> </span>PCR<span class=w> </span>error<span class=w> </span>correction:<span class=w> </span><span class=m>223</span>
Clonotypes<span class=w> </span>dropped<span class=w> </span>as<span class=w> </span>low<span class=w> </span>quality:<span class=w> </span><span class=m>9</span>
Clonotypes<span class=w> </span>pre-clustered<span class=w> </span>due<span class=w> </span>to<span class=w> </span>the<span class=w> </span>similar<span class=w> </span>VJC-lists:<span class=w> </span><span class=m>12</span>
IGH<span class=w> </span>chains:<span class=w> </span><span class=m>1173</span><span class=w> </span><span class=o>(</span><span class=m>99</span>.41%<span class=o>)</span>
IGH<span class=w> </span>non-functional:<span class=w> </span><span class=m>31</span><span class=w> </span><span class=o>(</span><span class=m>2</span>.64%<span class=o>)</span>
IGK<span class=w> </span>chains:<span class=w> </span><span class=m>5</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.42%<span class=o>)</span>
IGK<span class=w> </span>non-functional:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
IGL<span class=w> </span>chains:<span class=w> </span><span class=m>2</span><span class=w> </span><span class=o>(</span><span class=m>0</span>.17%<span class=o>)</span>
IGL<span class=w> </span>non-functional:<span class=w> </span><span class=m>0</span><span class=w> </span><span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
</code></pre></div> </div> <div class=tabbed-block> <div class=highlight><pre><span></span><code><span class=p>{</span>
<span class=w>  </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;alignerReport&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;commandLine&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;--report results/13_d60_LN_germCenterB.align.report.txt --json-report results/13_d60_LN_germCenterB.align.report.json --preset local:nebnext-human-bcr-cdr3 +limitInput 100000 /raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R1.fastq.gz /raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R2.fastq.gz results/13_d60_LN_germCenterB.vdjca&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;inputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R1.fastq.gz&quot;</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;/raw/PRJNA777934_NEB_Immune_sequencing/13_d60_LN_germCenterB_R2.fastq.gz&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;outputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;results/13_d60_LN_germCenterB.vdjca&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;version&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;; built=Mon Sep 26 10:55:18 CEST 2022; rev=8c998df1ab; lib=repseqio.v2.0&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;trimmingReport&quot;</span><span class=o>:</span><span class=w> </span><span class=kc>null</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;totalReadsProcessed&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>100000</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;aligned&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>88595</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;notAligned&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>11405</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;notAlignedReasons&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;NoJHits&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2521</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;VAndJOnDifferentTargets&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1773</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;NoCDR3Parts&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;NoBarcode&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;NoHits&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3851</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;LowTotalScore&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;NoVHits&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3260</span>
<span class=w>  </span><span class=p>},</span>
<span class=w>  </span><span class=s2>&quot;chimeras&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;overlapped&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>49087</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;alignmentAidedOverlaps&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2058</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;overlappedAligned&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>44737</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;overlappedNotAligned&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>4350</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;pairedEndAlignmentConflicts&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>30693</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;vChimeras&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2068</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;jChimeras&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;chainUsage&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;chainUsage&quot;</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;chimeras&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>88595</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;chains&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>      </span><span class=s2>&quot;IGH&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>88288</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2924</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>516</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2408</span>
<span class=w>      </span><span class=p>},</span>
<span class=w>      </span><span class=s2>&quot;IGK&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>253</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>10</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>7</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3</span>
<span class=w>      </span><span class=p>},</span>
<span class=w>      </span><span class=s2>&quot;IGL&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>54</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span>
<span class=w>      </span><span class=p>}</span>
<span class=w>    </span><span class=p>}</span>
<span class=w>  </span><span class=p>},</span>
<span class=w>  </span><span class=s2>&quot;realignedWithForcedNonFloatingBound&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>105942</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;realignedWithForcedNonFloatingRightBoundInLeftRead&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>25837</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;realignedWithForcedNonFloatingLeftBoundInRightRead&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>25837</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;noCDR3PartsAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>44</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;partialAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1970</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;tagReport&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;tagReport&quot;</span>
<span class=w>  </span><span class=p>}</span>
<span class=p>}</span>
<span class=p>{</span>
<span class=w>  </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;refineTagsAndSort&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;commandLine&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;--report results/13_d60_LN_germCenterB.refine.report.txt --json-report results/13_d60_LN_germCenterB.refine.report.json results/13_d60_LN_germCenterB.vdjca results/13_d60_LN_germCenterB.refined.vdjca&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;inputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;results/13_d60_LN_germCenterB.vdjca&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;outputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;results/13_d60_LN_germCenterB.refined.vdjca&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;version&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;; built=Mon Sep 26 10:55:18 CEST 2022; rev=8c998df1ab; lib=repseqio.v2.0&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;correctionReport&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;inputRecords&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>88595</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;outputRecords&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>52064</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;steps&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>      </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;tagName&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;UMI&quot;</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;inputGroups&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;inputDiversity&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>44135</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;inputCount&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>88595</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;coreDiversity&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>23770</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;coreCount&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>64187</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;directlyCorrectedDiversity&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5918</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;directlyCorrectedCount&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6351</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;filteredDiversity&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>14636</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;filteredCount&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>18247</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;recursivelyCorrected&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;diversityFilteredByWhitelist&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;outputDiversity&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>23581</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;outputCount&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>70348</span>
<span class=w>      </span><span class=p>}</span>
<span class=w>    </span><span class=p>],</span>
<span class=w>    </span><span class=s2>&quot;filterReport&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>      </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;filter_groups_report&quot;</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;groupingKeys&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>        </span><span class=s2>&quot;UMI&quot;</span>
<span class=w>      </span><span class=p>],</span>
<span class=w>      </span><span class=s2>&quot;numberOfGroups&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>23581</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;numberOfGroupsAccepted&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>11348</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;totalWeight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>70348</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;totalWeightAccepted&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>52064</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;operatorReports&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>        </span><span class=p>{</span>
<span class=w>          </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;generic_hist_report&quot;</span><span class=p>,</span>
<span class=w>          </span><span class=s2>&quot;threshold&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3</span>
<span class=w>        </span><span class=p>}</span>
<span class=w>      </span><span class=p>],</span>
<span class=w>      </span><span class=s2>&quot;metricHists&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>        </span><span class=p>{</span>
<span class=w>          </span><span class=s2>&quot;metric&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>            </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;group_metric_sum_weight&quot;</span><span class=p>,</span>
<span class=w>            </span><span class=s2>&quot;reportHist&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>              </span><span class=s2>&quot;log&quot;</span><span class=o>:</span><span class=w> </span><span class=kc>true</span><span class=p>,</span>
<span class=w>              </span><span class=s2>&quot;binNumber&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>              </span><span class=s2>&quot;minBinWidth&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0.2</span>
<span class=w>            </span><span class=p>}</span>
<span class=w>          </span><span class=p>},</span>
<span class=w>          </span><span class=s2>&quot;hist&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>            </span><span class=s2>&quot;bins&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1.5848931924611136</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6182</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1.5848931924611136</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2.51188643150958</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6051</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2.51188643150958</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3.981071705534973</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>4316</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>3.981071705534973</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6.309573444801933</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5455</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>6.309573444801933</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>10</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1223</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>10</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>15.848931924611142</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>333</span>
<span class=w>              </span><span class=p>},</span>
<span class=w>              </span><span class=p>{</span>
<span class=w>                </span><span class=s2>&quot;from&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>15.848931924611142</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;to&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>25.11886431509581</span><span class=p>,</span>
<span class=w>                </span><span class=s2>&quot;weight&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>21</span>
<span class=w>              </span><span class=p>}</span>
<span class=w>            </span><span class=p>],</span>
<span class=w>            </span><span class=s2>&quot;collectionSpec&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>              </span><span class=s2>&quot;log&quot;</span><span class=o>:</span><span class=w> </span><span class=kc>true</span><span class=p>,</span>
<span class=w>              </span><span class=s2>&quot;binNumber&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>              </span><span class=s2>&quot;minBinWidth&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0.2</span>
<span class=w>            </span><span class=p>}</span>
<span class=w>          </span><span class=p>}</span>
<span class=w>        </span><span class=p>}</span>
<span class=w>      </span><span class=p>]</span>
<span class=w>    </span><span class=p>}</span>
<span class=w>  </span><span class=p>}</span>
<span class=p>}</span>
<span class=p>{</span>
<span class=w>  </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;assemblerReport&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;commandLine&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;--report results/13_d60_LN_germCenterB.assemble.report.txt --json-report results/13_d60_LN_germCenterB.assemble.report.json results/13_d60_LN_germCenterB.refined.vdjca results/13_d60_LN_germCenterB.clns&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;inputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;results/13_d60_LN_germCenterB.refined.vdjca&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;outputFiles&quot;</span><span class=o>:</span><span class=w> </span><span class=p>[</span>
<span class=w>    </span><span class=s2>&quot;results/13_d60_LN_germCenterB.clns&quot;</span>
<span class=w>  </span><span class=p>],</span>
<span class=w>  </span><span class=s2>&quot;version&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;; built=Mon Sep 26 10:55:18 CEST 2022; rev=8c998df1ab; lib=repseqio.v2.0&quot;</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;preCloneAssemblerReport&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;preCloneAssemblerReport&quot;</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;inputGroups&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>11348</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;inputAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>52064</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;clonotypes&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>10644</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;clonotypesPerGroup&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>      </span><span class=s2>&quot;0&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>674</span><span class=p>,</span>
<span class=w>      </span><span class=s2>&quot;1&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>10644</span>
<span class=w>    </span><span class=p>},</span>
<span class=w>    </span><span class=s2>&quot;coreAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>47804</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;discardedCoreAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1661</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;empiricallyAssignedAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1695</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;vjEmpiricallyAssignedAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;umiEmpiricallyAssignedAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;gatEmpiricallyAssignedAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1695</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;empiricalAssignmentConflicts&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;unassignedAlignments&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2430</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;umiConflicts&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;gatConflicts&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;geneConflicts&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{},</span>
<span class=w>    </span><span class=s2>&quot;coreClonotypesDroppedByTagSuffix&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;coreAlignmentsDroppedByTagSuffix&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span>
<span class=w>  </span><span class=p>},</span>
<span class=w>  </span><span class=s2>&quot;totalReadsProcessed&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>100000</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;initialClonesCreated&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1424</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsDroppedNoTargetSequence&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2599</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsDroppedTooShortClonalSequence&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsDroppedLowQuality&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;coreReads&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>49353</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsDroppedFailedMapping&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>71</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;lowQualityRescued&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>18</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;clonesClustered&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>223</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsClustered&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2291</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;clones&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1180</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;clonesDroppedAsLowQuality&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>9</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;clonesPreClustered&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>12</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsPreClustered&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>108</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsInClones&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>47080</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsInClonesBeforeClustering&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>49371</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;readsDroppedWithLowQualityClones&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>57</span><span class=p>,</span>
<span class=w>  </span><span class=s2>&quot;clonalChainUsage&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>    </span><span class=s2>&quot;type&quot;</span><span class=o>:</span><span class=w> </span><span class=s2>&quot;chainUsage&quot;</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;chimeras&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1180</span><span class=p>,</span>
<span class=w>    </span><span class=s2>&quot;chains&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>      </span><span class=s2>&quot;IGH&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>1173</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>31</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>27</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>4</span>
<span class=w>      </span><span class=p>},</span>
<span class=w>      </span><span class=s2>&quot;IGK&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>5</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span>
<span class=w>      </span><span class=p>},</span>
<span class=w>      </span><span class=s2>&quot;IGL&quot;</span><span class=o>:</span><span class=w> </span><span class=p>{</span>
<span class=w>        </span><span class=s2>&quot;total&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>2</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;nonFunctional&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;isOOF&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span><span class=p>,</span>
<span class=w>        </span><span class=s2>&quot;hasStops&quot;</span><span class=o>:</span><span class=w> </span><span class=mf>0</span>
<span class=w>      </span><span class=p>}</span>
<span class=w>    </span><span class=p>}</span>
<span class=w>  </span><span class=p>}</span>
<span class=p>}</span>
</code></pre></div> </div> </div> </div> </details> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../biomed2-bcr/ class="md-footer__link md-footer__link--prev" aria-label="Previous: BioMed2" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> BioMed2 </div> </div> </a> <a href=../generic-umi-tcr/ class="md-footer__link md-footer__link--next" aria-label="Next: Targeted TCR UMI libraries" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Targeted TCR UMI libraries </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../../../javascripts/svg.js></script> <script src=../../../javascripts/onload.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>