<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="MiXCR: ultimate software platform for analysis of Next-Generation Sequencing (NGS) data for immune profiling."><meta name=author content="MiLaboratories Inc"><link href=https://docs.milaboratories.com/mixcr/reference/mixcr-align/ rel=canonical><link rel=icon href=../../../assets/favicon.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>align - Docs @ MiLaboratories</title><link rel=stylesheet href=../../../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link rel=stylesheet href=../../../css/extra.css><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-2N4Z95ZG08"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-2N4Z95ZG08",{page_path:e.pathname})})})</script><script async src="https://www.googletagmanager.com/gtag/js?id=G-2N4Z95ZG08"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#mixcr-align class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../.. title="Docs @ MiLaboratories" class="md-header__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Docs @ MiLaboratories </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> align </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z"/> </svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z"/> </svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label="Search by name or category" placeholder="Search by name or category" autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../about/ class=md-tabs__link> About </a> </li> <li class=md-tabs__item> <a href=../../getting-started/installation/ class=md-tabs__link> Getting Started </a> </li> <li class=md-tabs__item> <a href=../../guides/milaboratories-human-bcr-rna-multi/ class=md-tabs__link> Guides </a> </li> <li class=md-tabs__item> <a href=../overview-analysis-overview/ class="md-tabs__link md-tabs__link--active"> Docs </a> </li> <li class=md-tabs__item> <a href=https://vdj.online/ class=md-tabs__link> VDJ.online </a> </li> <li class=md-tabs__item> <a href=https://github.com/milaboratory/mixcr/discussions class=md-tabs__link> Discussions </a> </li> <li class=md-tabs__item> <a href=https://github.com/milaboratory/mixcr class=md-tabs__link> Github </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav class="md-nav md-nav--primary md-nav--lifted has-section" aria-label=Navigation data-md-level=0 data-test=1> <!-- Site title --> <label class=md-nav__title for=__drawer> <a href=../../.. title="Docs @ MiLaboratories" class="md-nav__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class=md-nav__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class=md-nav__list data-md-scrollfix> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../about/ class=md-nav__link> About </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_2 type=checkbox id=__nav_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_2 data-level=1> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Getting Started" data-md-level=1> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/installation/ class=md-nav__link> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/docker/ class=md-nav__link> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/milm/ class=md-nav__link> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/first-run/ class=md-nav__link> First run </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/license/ class=md-nav__link> License </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/support/ class=md-nav__link> Support </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3 data-level=1> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Guides data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_1 data-level=2> Kits & Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Kits & Protocols" data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Kits & Protocols </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milaboratories-human-bcr-rna-multi/ class=md-nav__link> MiLaboratories Human IG RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milaboratories-human-tcr-rna-multi/ class=md-nav__link> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/milab-dna-multiplex-tcr/ class=md-nav__link> MiLaboratories Human TCR DNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/takara-hsa-bcr/ class=md-nav__link> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/takara-mmu-tcr/ class=md-nav__link> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/iRepertoire/ class=md-nav__link> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/qiaseq-tcr/ class=md-nav__link> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/cellecta-air-tcr-bcr/ class=md-nav__link> Cellecta AIR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/abhelix-bcr/ class=md-nav__link> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/biomed2-bcr/ class=md-nav__link> BioMed2 </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/nebnext-bcr/ class=md-nav__link> NEBNext Immune Sequencing </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_2 data-level=2> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Analysis cases" data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-umi-tcr/ class=md-nav__link> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-multiplex-bcr/ class=md-nav__link> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/generic-umi-bcr/ class=md-nav__link> Targeted BCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/rnaseq/ class=md-nav__link> RNA-Seq data </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/ig-trees-reconstruction/ class=md-nav__link> Immunoglobulin lineage trees reconstruction </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_3 type=checkbox id=__nav_3_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_3 data-level=2> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Special guides" data-md-level=2> <label class=md-nav__title for=__nav_3_3> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/create-custom-library/ class=md-nav__link> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/external-libraries/ class=md-nav__link> Using external libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../guides/b-cell-lineages-webinar/ class=md-nav__link> B cell lineage tracing </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4 data-level=1> Docs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Docs data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Docs </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_1 data-level=2> Overview <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Overview data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-analysis-overview/ class=md-nav__link> Analysis overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-presets/ class=md-nav__link> Presets </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-mixins-list/ class=md-nav__link> Mix-in options list </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../overview-built-in-presets/ class=md-nav__link> Built-in presets </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_2 data-level=2> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiXCR data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-analyze/ class=md-nav__link> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for=__toc> align <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> align </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#command-line-options class=md-nav__link> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#concatenating-across-multiple-files class=md-nav__link> Concatenating across multiple files </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#aligner-parameters class=md-nav__link> Aligner parameters </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Aligner parameters"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#merging-algorithm-parameters class=md-nav__link> Merging algorithm parameters </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#gene-features-to-align class=md-nav__link> Gene features to align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#v-j-and-c-aligners-parameters class=md-nav__link> V, J and C aligners parameters </a> <!-- Table of contents list --> <nav class=md-nav aria-label="V, J and C aligners parameters"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#parameters-for-kaligner class=md-nav__link> Parameters for kAligner </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#parameters-for-kaligner2 class=md-nav__link> Parameters for kAligner2 </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#d-aligner-parameters class=md-nav__link> D aligner parameters </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#hardware-recommendations class=md-nav__link> Hardware recommendations </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!-- Main navigation item --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-refineTagsAndSort/ class=md-nav__link> refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assemblePartial/ class=md-nav__link> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-extend/ class=md-nav__link> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assemble/ class=md-nav__link> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-assembleContigs/ class=md-nav__link> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-export/ class=md-nav__link> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPretty/ class=md-nav__link> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportAirr/ class=md-nav__link> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportReadsForClones/ class=md-nav__link> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportShmTrees/ class=md-nav__link> exportShmTrees </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportReports/ class=md-nav__link> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportQc/ class=md-nav__link> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-postanalysis/ class=md-nav__link> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-downsample/ class=md-nav__link> downsample </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPlots/ class=md-nav__link> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportTables/ class=md-nav__link> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportPreprocTables/ class=md-nav__link> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-exportClonesOverlap/ class=md-nav__link> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-overlapScatterPlot/ class=md-nav__link> overlapScatterPlot </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-findAlleles/ class=md-nav__link> findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mixcr-findShmTrees/ class=md-nav__link> findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_3 data-level=2> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Reports data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-align/ class=md-nav__link> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-refineTagsAndSort/ class=md-nav__link> mixcr refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assemblePartial/ class=md-nav__link> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-extend/ class=md-nav__link> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assemble/ class=md-nav__link> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-assembleContigs/ class=md-nav__link> mixcr assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-findAlleles/ class=md-nav__link> mixcr findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../report-findShmTrees/ class=md-nav__link> mixcr findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_4 data-level=2> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=RepSeq.io data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fasta/ class=md-nav__link> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-compile/ class=md-nav__link> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-debug/ class=md-nav__link> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-format/ class=md-nav__link> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fromFasta/ class=md-nav__link> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-fromPaddedFasta/ class=md-nav__link> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-inferPoints/ class=md-nav__link> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-merge/ class=md-nav__link> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-stat/ class=md-nav__link> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../repseqio-tsv/ class=md-nav__link> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_5 data-level=2> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiTool data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-parse/ class=md-nav__link> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-tag-stat/ class=md-nav__link> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-export-fastq/ class=md-nav__link> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-correct/ class=md-nav__link> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-consensus/ class=md-nav__link> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-process-fastq/ class=md-nav__link> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-format-fastq/ class=md-nav__link> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../mitool-compress-sequence-set/ class=md-nav__link> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_6 type=checkbox id=__nav_4_6> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_6 data-level=2> Specifications <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Specifications data-md-level=2> <label class=md-nav__title for=__nav_4_6> <span class="md-nav__icon md-icon"></span> Specifications </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-tag-pattern/ class=md-nav__link> Tag pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-input-file-name-expansion/ class=md-nav__link> Input file name expansion </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-gene-features/ class=md-nav__link> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-translation-rules/ class=md-nav__link> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-mutations-encoding/ class=md-nav__link> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ref-repseqio-json-format/ class=md-nav__link> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://vdj.online/ class=md-nav__link> VDJ.online </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://github.com/milaboratory/mixcr/discussions class=md-nav__link> Discussions </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=https://github.com/milaboratory/mixcr class=md-nav__link> Github </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#command-line-options class=md-nav__link> Command line options </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#concatenating-across-multiple-files class=md-nav__link> Concatenating across multiple files </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#aligner-parameters class=md-nav__link> Aligner parameters </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Aligner parameters"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#merging-algorithm-parameters class=md-nav__link> Merging algorithm parameters </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#gene-features-to-align class=md-nav__link> Gene features to align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#v-j-and-c-aligners-parameters class=md-nav__link> V, J and C aligners parameters </a> <!-- Table of contents list --> <nav class=md-nav aria-label="V, J and C aligners parameters"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#parameters-for-kaligner class=md-nav__link> Parameters for kAligner </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#parameters-for-kaligner2 class=md-nav__link> Parameters for kAligner2 </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#d-aligner-parameters class=md-nav__link> D aligner parameters </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#hardware-recommendations class=md-nav__link> Hardware recommendations </a> <!-- Table of contents list --> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/milaboratory/docs/edit/main/docs/mixcr/reference/mixcr-align.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=mixcr-align><code>mixcr align</code></h1> <p>Aligns raw sequencing data against V-, D-, J- and C- gene segment references library database for specified species. If the input sequences have barcodes (UMIs, cell barcodes etc.), MiXCR allows to parse barcodes using powerful <a href=../ref-tag-pattern/ >pattern matching syntax</a> and assign them to every alignment. Additionally, read trimming may be applied if corresponding options are specified. MiXCR supports paired-end and single-end <a href=https://en.wikipedia.org/wiki/FASTQ_format><code>.fastq</code></a>, <a href=https://en.wikipedia.org/wiki/FASTA_format><code>.fasta</code></a>, <a href=https://en.wikipedia.org/wiki/Binary_Alignment_Map><code>.bam</code> and <code>.sam</code></a> formats.</p> <p><img alt src=../pics/align-light.svg#only-light> <img alt src=../pics/align-dark.svg#only-dark></p> <h2 id=command-line-options>Command line options</h2> <p><div class=highlight><pre><span></span><code>mixcr align --preset &lt;name&gt; 
    [--trimming-quality-threshold &lt;n&gt;] 
    [--trimming-window-size &lt;n&gt;] 
    [--write-all] 
    [--tag-pattern-file &lt;path&gt;] 
    [--tag-parse-unstranded] 
    [--tag-max-budget &lt;n&gt;] 
    [--read-id-as-cell-tag]
    [--read-buffer &lt;n&gt;] 
    [--high-compression] 
    [--not-aligned-R1 &lt;path.fastq[.gz]&gt;] 
    [--not-aligned-R2 &lt;path.fastq[.gz]&gt;] 
    [--not-parsed-R1 &lt;path.fastq[.gz]&gt;] 
    [--not-parsed-R2 &lt;path.fastq[.gz]&gt;] 
    [--species &lt;species&gt;] 
    [--library &lt;library&gt;] 
    [--dna] 
    [--rna] 
    [--floating-left-alignment-boundary [&lt;anchor_point&gt;]] 
    [--rigid-left-alignment-boundary [&lt;anchor_point&gt;]] 
    [--floating-right-alignment-boundary (&lt;gene_type&gt;|&lt;anchor_point&gt;)] 
    [--rigid-right-alignment-boundary [(&lt;gene_type&gt;|&lt;anchor_point&gt;)]] 
    [--tag-pattern &lt;pattern&gt;] 
    [--keep-non-CDR3-alignments] 
    [--drop-non-CDR3-alignments] 
    [--limit-input &lt;n&gt;] 
    [--assemble-clonotypes-by &lt;gene_features&gt;] 
    [--split-clones-by &lt;gene_type&gt;]... 
    [--dont-split-clones-by &lt;gene_type&gt;]... 
    [--assemble-contigs-by &lt;gene_features&gt;] 
    [--impute-germline-on-export] 
    [--dont-impute-germline-on-export] 
    [--prepend-export-clones-field &lt;field&gt; [&lt;param&gt;...]]... 
    [--append-export-clones-field &lt;field&gt; [&lt;param&gt;...]]...
    [--prepend-export-alignments-field &lt;field&gt; [&lt;param&gt;...]]... 
    [--append-export-alignments-field &lt;field&gt; [&lt;param&gt;...]]... 
    [-O &lt;key=value&gt;]... 
    [-M &lt;key=value&gt;]... 
    [--repor) &lt;path&gt;] 
    [--json-report &lt;path&gt;] 
    [--threads &lt;n&gt;] 
    [--force-overwrite] 
    [--no-warnings] 
    [--verbose] 
    [--help]
    (file_R1.fastq[.gz] file_R2.fastq[.gz]|file_RN.(fastq[.gz]|fasta|bam|sam)) alignments.vdjca
</code></pre></div> The command returns a highly-compressed, memory- and CPU-efficient binary <code>.vdjca</code> file that holds exhaustive information about alignments. Alignments can be further extracted in tabular form using <a href=../mixcr-export/#alignments><code>exportAlignments</code></a> or in human-readable form using <a href=../mixcr-exportPretty/#raw-alignments><code>exportAlignmentsPretty</code></a>. Additionally, MiXCR produces a comprehensive <a href=../report-align/ >report</a> which provides a detailed overview of the alignment performance and quality of the library.</p> <p>Basic command line options are:</p> <dl> <dt><code>(file_R1.fastq[.gz] file_R2.fastq[.gz]|file_RN.(fastq[.gz]|fasta|bam|sam))</code></dt> <dd>Paths of input files with sequencing data. File name pattern <a href=../ref-input-file-name-expansion/ >expansion</a> can be used here to merge sequences from multiple sequences or just for convenience.</dd> <dt><code>alignments.vdjca</code></dt> <dd>Path where to write output alignments</dd> <dt><code>-p, --preset &lt;name&gt;</code></dt> <dd>Analysis preset. Sets all important parameters of this and all downstream analysis steps. It is important to carefully select the most appropriate preset for the data you analyse. See <a href=../overview-built-in-presets/ >list of presets</a>. </dd> <dt><code>--trimming-quality-threshold &lt;n&gt;</code></dt> <dd>Read pre-processing: trimming quality threshold. Zero value can be used to skip trimming. Default value determined by the preset.</dd> <dt><code>--trimming-window-size &lt;n&gt;</code></dt> <dd>Read pre-processing: trimming window size. Default value determined by the preset.</dd> <dt><code>--write-all</code></dt> <dd>Write alignment results for all input reads (even if alignment failed). Default value determined by the preset.</dd> <dt><code>--tag-pattern-file &lt;path&gt;</code></dt> <dd>Read tag pattern from a file. Default tag pattern determined by the preset.</dd> <dt><code>--tag-pattern &lt;pattern&gt;</code></dt> <dd>Specify tag pattern for barcoded data.</dd> <dt><code>--tag-parse-unstranded</code></dt> <dd>If paired-end input is used, determines whether to try all combinations of mate-pairs or only match reads to the corresponding pattern sections (i.e. first file to first section, etc...). Default value determined by the preset.</dd> <dt><code>--tag-max-budget &lt;n&gt;</code></dt> <dd>Maximal bit budget, higher values allows more substitutions in small letters. Default value determined by the preset.</dd> <dt><code>--read-id-as-cell-tag</code></dt> <dd>Marks reads, coming from different files, but having the same positions in those files, as reads coming from the same cells. Main use-case is protocols with overlapped alpha-beta, gamma-delta or heavy-light cDNA molecules, where each side was sequenced by separate mate pairs in a paired-end sequencer. Use special expansion group CELLSPLIT instead of R index (i.e. "my_file_R{{CELLSPLIT:n}}.fastq.gz").</dd> </dl> <p>Default value determined by the preset.</p> <dl> <dt><code>-s, --species &lt;species&gt;</code></dt> <dd>Species (organism). Possible values: <code>hsa</code> (or HomoSapiens), <code>mmu</code> (or MusMusculus), <code>rat</code>, <code>spalax</code>, <code>alpaca</code>, <code>lamaGlama</code>, <code>mulatta</code> (<em>Macaca Mulatta</em>), <code>fascicularis</code> (<em>Macaca Fascicularis</em>) or any species from <a href=../../guides/external-libraries/ >IMGT ® library</a>.</dd> <dt><code>-b, --library &lt;library&gt;</code></dt> <dd>V/D/J/C gene library. By default, the <code>default</code> MiXCR reference library is used. One can also use <a href=../../guides/external-libraries/ >external libraries</a></dd> <dt><code>--limit-input &lt;n&gt;</code></dt> <dd>Maximal number of reads to process on <a href=./ >align</a>.</dd> <dt><code>--read-buffer &lt;n&gt;</code></dt> <dd>Size of buffer for FASTQ readers in bytes. Default: 4Mb</dd> <dt><code>--high-compression</code></dt> <dd>Use higher compression for output file, 10~25% slower, minus 30~50% of file size.</dd> <dt><code>--not-aligned-R1 &lt;path.fastq[.gz]&gt;</code></dt> <dd>Pipe not aligned R1 reads into separate file.</dd> <dt><code>--not-aligned-R2 &lt;path.fastq[.gz]&gt;</code></dt> <dd>Pipe not aligned R2 reads into separate file.</dd> <dt><code>--not-parsed-R1 &lt;path.fastq[.gz]&gt;</code></dt> <dd>Pipe not parsed R1 reads into separate file.</dd> <dt><code>--not-parsed-R2 &lt;path.fastq[.gz]&gt;</code></dt> <dd>Pipe not parsed R2 reads into separate file.</dd> <dt><code>-O &lt;key=value&gt;</code></dt> <dd>Overrides aligner parameters from the selected preset (see below)</dd> <dt><code>-M &lt;key=value&gt;</code></dt> <dd>Overrides preset parameters</dd> <dt><code>-r, --report &lt;path&gt;</code></dt> <dd><a href=../report-align/ >Report</a> file (human readable version, see <code>-j / --json-report</code> for machine readable report).</dd> <dt><code>-j, --json-report &lt;path&gt;</code></dt> <dd>JSON formatted <a href=../report-align/ >report</a> file.</dd> <dt><code>-t, --threads &lt;n&gt;</code></dt> <dd>Processing threads</dd> <dt><code>-f, --force-overwrite</code></dt> <dd>Force overwrite of output file(s).</dd> <dt><code>-nw, --no-warnings</code></dt> <dd>Suppress all warning messages.</dd> <dt><code>--verbose</code></dt> <dd>Verbose warning messages.</dd> <dt><code>-h, --help</code></dt> <dd>Show this help message and exit.</dd> </dl> <p>In addition to these parameters, any of the <a href=../overview-mixins-list/ >available mix-in options</a> may ve additionally specify at <code>align</code>. </p> <h2 id=concatenating-across-multiple-files>Concatenating across multiple files</h2> <p>Sometimes it is required to concatenate several fastq files and analyse it as a single sample. This is a common practise when files are separated across sequencing lanes. </p> <p>MiXCR uses <code>{{n}}</code> syntax as a placeholder for any number. Meaning MiXCR will concatenate all files that match the pattern regardless of digits in the place of <code>{{n}}</code></p> <p>Bellow you can see an example of how to pass 8 fastq files (four per each paired read) to <code>mixcr align</code>:</p> <p>Example: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>ls<span class=w> </span>fastq/
<span class=w>    </span>sample1_L001_S25_R1.fastq.gz<span class=w>    </span>sample1_L001_S25_R2.fastq.gz<span class=w> </span>
<span class=w>    </span>sample1_L002_S25_R1.fastq.gz<span class=w>    </span>sample1_L002_S25_R2.fastq.gz
<span class=w>    </span>sample1_L003_S25_R1.fastq.gz<span class=w>    </span>sample1_L003_S25_R2.fastq.gz
<span class=w>    </span>sample1_L004_S25_R1.fastq.gz<span class=w>    </span>sample1_L004_S25_R2.fastq.gz

&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>-s<span class=w> </span>hsa<span class=w> </span><span class=se>\</span>
<span class=w>    </span>fastq/sample1_L<span class=o>{{</span>n<span class=o>}}</span>_S25_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>fastq/sample1_L<span class=o>{{</span>n<span class=o>}}</span>_S25_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/sample1
</code></pre></div></p> <p>MiXCR uses <code>{{a}}</code> syntax which works just like {{n}} but works with any pattern, not just numbers.</p> <p>Example:</p> <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>ls<span class=w> </span>fastq/
<span class=w>    </span>sample1_nameA1_S25_R1.fastq.gz<span class=w>    </span>sample1_nameA1_S25_R2.fastq.gz
<span class=w>    </span>sample1_nameA_S25_R1.fastq.gz<span class=w>     </span>sample1_nameA_S25_R2.fastq.gz
<span class=w>    </span>sample1_nameC3_S25_R1.fastq.gz<span class=w>    </span>sample1_nameC3_S25_R2.fastq.gz
<span class=w>    </span>sample1_fileD_S25_R1.fastq.gz<span class=w>     </span>sample1_fileD_S25_R2.fastq.gz

&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>-s<span class=w> </span>hsa<span class=w> </span><span class=se>\</span>
<span class=w>    </span>fastq/sample1_<span class=o>{{</span>n<span class=o>}}</span>_S25_R1.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>fastq/sample1_<span class=o>{{</span>n<span class=o>}}</span>_S25_R2.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>    </span>results/sample1
</code></pre></div> <p>Finally, MiXCR uses <code>{{R}}</code> syntax that marks the reads' number, so you don't have to submit the second read filename.</p> <p>Bellow is the example of how you can combine multiple MiXCR wildcards and concatenate 16 pairs of files using one pattern. </p> <p>Example: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>ls<span class=w> </span>fastq/
<span class=w>    </span>sampleA_primer1_L001_R1.fastq.gz<span class=w>       </span>sampleA_primer1_L001_R2.fastq.gz<span class=w>    </span>
<span class=w>    </span>sampleA_primer1_L002_R1.fastq.gz<span class=w>       </span>sampleA_primer1_L002_R2.fastq.gz<span class=w>    </span>
<span class=w>    </span>sampleA_primer1_L003_R1.fastq.gz<span class=w>       </span>sampleA_primer1_L003_R2.fastq.gz<span class=w>    </span>
<span class=w>    </span>sampleA_primer1_L004_R1.fastq.gz<span class=w>       </span>sampleA_primer1_L004_R2.fastq.gz<span class=w>    </span>
<span class=w>    </span>sampleA_primer2_L001_R1.fastq.gz<span class=w>       </span>sampleA_primer2_L001_R2.fastq.gz
<span class=w>    </span>sampleA_primer2_L002_R1.fastq.gz<span class=w>       </span>sampleA_primer2_L002_R2.fastq.gz
<span class=w>    </span>sampleA_primer2_L003_R1.fastq.gz<span class=w>       </span>sampleA_primer2_L003_R2.fastq.gz
<span class=w>    </span>sampleA_primer2_L004_R1.fastq.gz<span class=w>       </span>sampleA_primer2_L004_R2.fastq.gz
<span class=w>    </span>sampleA_primer3_L001_R1.fastq.gz<span class=w>       </span>sampleA_primer3_L001_R2.fastq.gz
<span class=w>    </span>sampleA_primer3_L002_R1.fastq.gz<span class=w>       </span>sampleA_primer3_L002_R2.fastq.gz
<span class=w>    </span>sampleA_primer3_L003_R1.fastq.gz<span class=w>       </span>sampleA_primer3_L003_R2.fastq.gz
<span class=w>    </span>sampleA_primer3_L004_R1.fastq.gz<span class=w>       </span>sampleA_primer3_L004_R2.fastq.gz
<span class=w>    </span>sampleA_primer4_L001_R1.fastq.gz<span class=w>       </span>sampleA_primer4_L001_R2.fastq.gz
<span class=w>    </span>sampleA_primer4_L002_R1.fastq.gz<span class=w>       </span>sampleA_primer4_L002_R2.fastq.gz
<span class=w>    </span>sampleA_primer4_L003_R1.fastq.gz<span class=w>       </span>sampleA_primer4_L003_R2.fastq.gz
<span class=w>    </span>sampleA_primer4_L004_R1.fastq.gz<span class=w>       </span>sampleA_primer4_L004_R2.fastq.gz

&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>-s<span class=w> </span>hsa<span class=w> </span><span class=se>\</span>
<span class=w>    </span>fastq/sampleA_<span class=o>{{</span>a<span class=o>}}</span>_L<span class=o>{{</span>n<span class=o>}}</span>_<span class=o>{{</span>R<span class=o>}}</span>.fastq.gz
<span class=w>    </span>results/sampleA
</code></pre></div></p> <p>For details see <a href=/mixcr/reference/ref-input-file-name-expansion>this section</a>.</p> <h2 id=aligner-parameters>Aligner parameters</h2> <p>MiXCR uses a <a href=https://github.com/milaboratory/mixcr/blob/develop/src/main/resources/parameters/vdjcaligner_parameters.json>wide range of parameters</a> that control aligners behaviour. There are global parameters and gene-specific parameters organized in groups: <code>vParameters</code>, <code>dParameters</code>, <code>jParameters</code> and <code>cParameters</code>. Each group of parameters may contain further subgroups of parameters. In order to override a parameter value one can use <code>-O</code> followed by fully qualified parameter name and parameter value (e.g. <code>-Ogroup1.group2.parameter=value</code>).</p> <p>Global aligner parameters include:</p> <dl> <dt><code>-OsaveOriginalReads=false</code></dt> <dd>Save original sequencing reads in <code>.vdjca</code> file (default <code>false</code>).</dd> <dt><code>-OallowPartialAlignments=false</code></dt> <dd>Save incomplete alignments (e.g. only V / only J) in <code>.vdjca</code> file. </dd> <dt><code>-OallowNoCDR3PartAlignments=false</code></dt> <dd>Save alignments which do not fully cover CDR3 region.</dd> <dt><code>-OallowChimeras=false</code></dt> <dd>Accept alignments with different loci of V and J genes (by default such alignments are dropped). </dd> <dt><code>-OminSumScore=120</code></dt> <dd>Minimal total alignment score value of V and J genes. </dd> <dt><code>-OmaxHits=5</code></dt> <dd>Maximal number of hits for each gene type: if input sequence align to more than <code>maxHits</code> targets, then only top <code>maxHits</code> hits will be kept. </dd> <dt><code>-OvjAlignmentOrder=VThenJ</code> (only for single-end analysis)</dt> <dd>Order in which V and J genes aligned in target (possible values <code>JThenV</code> and <code>VThenJ</code>). Parameter affects only <em>single-read</em> alignments and alignments of overlapped paired-end reads. Non-overlapping paired-end reads are always processed in <code>VThenJ</code> mode. <code>JThenV</code> can be used for short reads (~100bp) with full (or nearly full) J gene coverage. </dd> <dt><code>-OrelativeMinVFR3CDR3Score=0.7</code> (only for paired-end analysis)</dt> <dd>Relative minimal alignment score of <code>FR3+VCDR3Part</code> region for V gene. V hit will be kept only if its <code>FR3+VCDR3Part</code> part aligns with score greater than <code>relativeMinVFR3CDR3Score * maxFR3CDR3Score</code>, where <code>maxFR3CDR3Score</code> is the maximal alignment score for <code>FR3+VCDR3Part</code> region among all of V hits for current input reads pair. </dd> <dt><code>-OreadsLayout=Opposite</code> (only for paired-end analysis)</dt> <dd>Relative orientation of paired reads. Available values: <code>Opposite</code>, <code>Collinear</code>, <code>Unknown</code> </dd> <dt><code>-OrelativeMinVScore=0.7</code> (only for paired-end analysis)</dt> <dd>Relative minimum score of V gene. Only those V hits will be considered, which score is greater then <code>relativeMinVScore * maxVScore</code>, where <code>maxVScore</code> is the maximum score throw all obtained V hits.</dd> <dt><code>-OalignmentBoundaryTolerance=5</code></dt> <dd>Force alignment of opposite mate pair if another mate alignment almost reach its boundary (left for right read and right for left read), keeping at most <code>alignmentBoundaryTolerance</code> letters not ailgned.</dd> <dt><code>-OminChimeraDetectionScore=120</code></dt> <dd>Used in V/J chimera detection and elimination mechanism: both alignments in R1 and R2 should be greater than this value.</dd> <dt><code>-OvjOverlapWindow=3</code></dt> <dd>Maximal possible overlap between V and J alignments</dd> <dt><code>includeDScore=false</code></dt> <dd>Add D alignment score to the overall alignment score </dd> <dt><code>includeCScore=false</code></dt> <dd>Add C alignment score to the overall alignment score</dd> </dl> <p>To override these parameters one can e.g. do: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OallowPartialAlignments<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OallowNoCDR3PartAlignments<span class=o>=</span><span class=nb>true</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <h3 id=merging-algorithm-parameters>Merging algorithm parameters</h3> <p>MiXCR overlaps R1 and R2 reads if corresponding alignments overlap. There several parameters controlling merging behaviour: </p> <dl> <dt><code>-OmergerParameters.qualityMergingAlgorithm=MaxSubtraction</code></dt> <dd>Algorithm used to compute Phred quality of overlapping region. Possible values: <code>SumMax</code>, <code>SumSubtraction</code>, <code>MaxSubtraction</code> and <code>MaxMax</code>.</dd> <dt><code>-OmergerParameters.minimalOverlap=17</code></dt> <dd>Minimal length of R1 and R2 to proceed with overlap algorithm.</dd> <dt><code>-OmergerParameters.minimalIdentity=0.9</code></dt> <dd>Minimal allowed percent of matched letters in overlapping region. </dd> </dl> <p>To override these parameters one can e.g. do: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OmergerParameters.minimalIdentity<span class=o>=</span><span class=m>0</span>.8<span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <h3 id=gene-features-to-align>Gene features to align</h3> <p>MiXCR allows to specify particular <a href=../ref-gene-features/ ><code>gene features</code></a> that will be extracted from reference and used as a targets for alignments. Thus, each sequencing read will be aligned to these extracted reference regions. Parameters responsible for target gene regions are:</p> <dl> <dt><code>-OvParameters.geneFeatureToAlign=VRegionWithP</code></dt> <dd>region in V gene which will be used as target</dd> <dt><code>-OdParameters.geneFeatureToAlign=DRegionWithP</code></dt> <dd>region in D gene which will be used as target</dd> <dt><code>-OjParameters.geneFeatureToAlign=JRegionWithP</code></dt> <dd>region in J gene which will be used as target</dd> <dt><code>-OcParameters.geneFeatureToAlign=CExon1</code></dt> <dd>region in C gene which will be used as target</dd> </dl> <p>It is important to specify these gene regions such that they will fully cover target clonal gene region which will be used in <a href=../mixcr-assemble/ ><code>assemble</code></a> (e.g. CDR3).</p> <p>To override these parameters one can e.g. do: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.geneFeatureToAlign<span class=o>=</span>VTranscriptWithP<span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <h3 id=v-j-and-c-aligners-parameters>V, J and C aligners parameters</h3> <p>MiXCR has two types of aligners to align V, J and C genes:</p> <ul> <li><code>kAligner</code> — better suited for linear scoring and better works with T-cell data</li> <li><code>kAligner2</code> — specifically designed for affine scoring, better handles large number of mutations (e.g. hypermutations) and long indels typical for B-cell data</li> </ul> <p>These aligners are based on k-mer seed-and-vote algorithms inspired by <a href=http://nar.oxfordjournals.org/content/41/10/e108>this paper</a>.</p> <p>There are many parameters that may be tuned individually for each aligner for each gene type. MiXCR offers a number of default preset parameters that may be chosen with <a href=#command-line-options><code>--preset</code></a> command line option:</p> <ul> <li><code>default</code> — for T-cell and non-targeted B-cell data</li> <li><code>rna-seq</code> — for non-targeted RNA-Seq (shotgun) data with </li> <li><code>kAligner2</code> — for targeted B-cell data</li> </ul> <p>For <code>default</code> and <code>rna-seq</code> presets MiXCR uses <code>kAligner</code> for all gene types. For <code>kAligner2</code> preset MiXCR uses <code>kAligner2</code> for Variable and Joining segments and <code>kAligner</code> for Constant region.</p> <h4 id=parameters-for-kaligner>Parameters for <code>kAligner</code></h4> <p>The following parameters can be specified for <code>kAligner</code>:</p> <dl> <dt><code>mapperKValue</code></dt> <dd>Length of seeds used in aligner.</dd> <dt><code>floatingLeftBound</code></dt> <dd>Specifies whether left bound of alignment is fixed or float: if <code>floatingLeftBound</code> set to false, the left bound of either target or query will be aligned. Default values are suitable in most cases. If your target molecules have no primer sequences in V Region (e.g. 5'RACE) (or your <code>-OvParameters.geneFeatureToAlign</code> does not cover the region where primer is located) you can change value of this parameter for V gene to <code>false</code> to increase V gene identification accuracy and overall specificity of alignments.</dd> <dt><code>floatingRightBound</code></dt> <dd>Specifies whether right bound of alignment is fixed or float: if <code>floatingRightBound</code> set to false, the right bound of either target or query will be aligned. Default values are suitable in most cases. If your target molecules have no primer sequences in J Region (e.g. library was amplified using primer to the C region) you can change value of this parameter for J gene to <code>false</code> to increase J gene identification accuracy and overall specificity of alignments.</dd> <dt><code>minAlignmentLength</code></dt> <dd>Minimal length of aligned region.</dd> <dt><code>maxAdjacentIndels</code></dt> <dd>Maximum number of indels between two seeds.</dd> <dt><code>absoluteMinScore</code></dt> <dd>Minimal score of alignment: alignments with smaller score will be dropped.</dd> <dt><code>relativeMinScore</code></dt> <dd>Minimal relative score of alignments: if alignment score is smaller than <code>relativeMinScore * maxScore</code>, where <code>maxScore</code> is the best score among all alignments for particular gene type (V, J or C) and input sequence, it will be dropped.</dd> <dt><code>maxHits</code></dt> <dd>Maximal number of hits: if input sequence align with more than <code>maxHits</code> queries, only top <code>maxHits</code> hits will be kept.</dd> <dt><code>mapperAbsoluteMinScore</code></dt> <dd>k-mer mapper absolute min score</dd> <dt><code>mapperRelativeMinScore</code></dt> <dd>k-mer mapper relative min score</dd> <dt><code>mapperMatchScore</code></dt> <dd>Match score for single k-mer match.</dd> <dt><code>mapperMismatchPenalty</code></dt> <dd>Mismatch penalty for single k-mer mismatch.</dd> <dt><code>mapperOffsetShiftPenalty</code></dt> <dd>Penalty for k-mer position shift.</dd> <dt><code>mapperMinSeedsDistance</code></dt> <dd>Min distance between seeds in seed-and-vote strategy.</dd> <dt><code>mapperMaxSeedsDistance</code></dt> <dd>Max distance between neighbor seeds during seeding.</dd> <dt><code>alignmentStopPenalty</code></dt> <dd>Stop penalty.</dd> </dl> <p>Default values for these parameters are:</p> <table> <thead> <tr> <th>Parameter</th> <th>Default V value</th> <th>Default J value</th> <th>Default C value</th> <th>RNA-Seq V value</th> <th>RNA-Seq J value</th> <th>RNA-Seq C value</th> <th>kAligner2 C value</th> </tr> </thead> <tbody> <tr> <td><code>mapperKValue</code></td> <td>5</td> <td>5</td> <td>5</td> <td>5</td> <td>5</td> <td>5</td> <td>5</td> </tr> <tr> <td><code>floatingLeftBound</code></td> <td>true</td> <td>true</td> <td>false</td> <td>false</td> <td>true</td> <td>false</td> <td>false</td> </tr> <tr> <td><code>floatingRightBound</code></td> <td>true</td> <td>false</td> <td>false</td> <td>true</td> <td>false</td> <td>false</td> <td>false</td> </tr> <tr> <td><code>minAlignmentLength</code></td> <td>15</td> <td>15</td> <td>15</td> <td>15</td> <td>15</td> <td>15</td> <td>15</td> </tr> <tr> <td><code>maxAdjacentIndels</code></td> <td>2</td> <td>2</td> <td>2</td> <td>2</td> <td>2</td> <td>2</td> <td>2</td> </tr> <tr> <td><code>absoluteMinScore</code></td> <td>40</td> <td>40.0</td> <td>40.0</td> <td>55.0</td> <td>50.0</td> <td>40.0</td> <td>40.0</td> </tr> <tr> <td><code>relativeMinScore</code></td> <td>0.87</td> <td>0.87</td> <td>0.87</td> <td>0.87</td> <td>0.87</td> <td>0.87</td> <td>0.87</td> </tr> <tr> <td><code>maxHits</code></td> <td>7</td> <td>7</td> <td>7</td> <td>7</td> <td>7</td> <td>7</td> <td>7</td> </tr> <tr> <td><code>mapperAbsoluteMinScore</code></td> <td>1.5</td> <td>1.5</td> <td>1.5</td> <td>1.5</td> <td>1.5</td> <td>1.5</td> <td>1.5</td> </tr> <tr> <td><code>mapperRelativeMinScore</code></td> <td>0.7</td> <td>0.75</td> <td>0.75</td> <td>0.7</td> <td>0.75</td> <td>0.75</td> <td>0.75</td> </tr> <tr> <td><code>mapperMatchScore</code></td> <td>1.0</td> <td>1.0</td> <td>1.0</td> <td>1.0</td> <td>1.0</td> <td>1.0</td> <td>1.0</td> </tr> <tr> <td><code>mapperMismatchPenalty</code></td> <td>-0.1</td> <td>-0.1</td> <td>-0.1</td> <td>-0.1</td> <td>-0.1</td> <td>-0.1</td> <td>-0.1</td> </tr> <tr> <td><code>mapperOffsetShiftPenalty</code></td> <td>-0.3</td> <td>-0.3</td> <td>-0.3</td> <td>-0.3</td> <td>-0.3</td> <td>-0.3</td> <td>-0.3</td> </tr> <tr> <td><code>mapperMinSeedsDistance</code></td> <td>4</td> <td>4</td> <td>4</td> <td>4</td> <td>4</td> <td>4</td> <td>4</td> </tr> <tr> <td><code>mapperMaxSeedsDistance</code></td> <td>10</td> <td>8</td> <td>10</td> <td>10</td> <td>8</td> <td>10</td> <td>10</td> </tr> <tr> <td><code>alignmentStopPenalty</code></td> <td>-1000</td> <td>-1000</td> <td>-1000</td> <td>-1000</td> <td>-1000</td> <td>-1000</td> <td>-1000</td> </tr> </tbody> </table> <p>To override these parameters one can e.g. do: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w>  </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.minAlignmentLength<span class=o>=</span><span class=m>30</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OjParameters.parameters.relativeMinScore<span class=o>=</span><span class=m>0</span>.7<span class=w> </span><span class=se>\ </span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <p>Scoring used in aligners is specified by <code>scoring</code> subgroup of parameters. It contains the following parameters:</p> <dl> <dt><code>subsMatrix=simple(match = &lt;match&gt;, mismatch = &lt;mismatch&gt;&gt;)</code></dt> <dd> <p>Substitution matrix. Available types:</p> <ul> <li><code>simple</code> - a matrix with diagonal elements equal to <code>match</code> and other elements equal to <code>mismatch</code></li> <li><code>raw</code> - a complete set of 16 matrix elements should be specified; for example: <code>raw(5,-9,-9,-9,-9,5,-9,-9,-9,-9,5,-9,-9,-9,-9,5)</code>(<em>equivalent to the default value</em>)</li> </ul> </dd> <dt><code>gapPenalty=&lt;penalty&gt;</code></dt> <dd>Penalty for a gap.</dd> </dl> <p>For <code>default</code> preset MiXCR uses <code>simple(match = 5, mismatch = -9)</code> and <code>gapPenalty=-12</code> for V-, J- and C- aligners. For <code>rna-seq</code> preset MiXCR uses <code>simple(match = 5, mismatch = -11)</code> and <code>gapPenalty=-21</code> values.</p> <p>Scoring parameters can be overridden in the following way:</p> <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.scoring.gapPenalty<span class=o>=</span>-20<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.scoring.subsMatrix<span class=o>=</span>simple<span class=o>(</span><span class=nv>match</span><span class=o>=</span><span class=m>4</span>,mismatch<span class=o>=</span>-11<span class=o>)</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div> <h4 id=parameters-for-kaligner2>Parameters for <code>kAligner2</code></h4> <p>The following parameters can be tuned for <code>kAligner2</code>:</p> <dl> <dt><code>mapperNValue</code></dt> <dd>Length of k-mer seeds used in aligner.</dd> <dt><code>mapperKValue</code></dt> <dd>Allowed number of mutations in k-mer</dd> <dt><code>floatingLeftBound</code></dt> <dd>Specifies whether left bound of alignment is fixed or float: if <code>floatingLeftBound</code> set to false, the left bound of either target or query will be aligned. Default values are suitable in most cases. If your target molecules have no primer sequences in V Region (e.g. 5'RACE) you can change value of this parameter for V gene to <code>false</code> to increase V gene identification accuracy and overall specificity of alignments.</dd> <dt><code>floatingRightBound</code></dt> <dd>Specifies whether right bound of alignment is fixed or float: if <code>floatingRightBound</code> set to <code>false</code>, the right bound of either target or query will be aligned. Default values are suitable in most cases. If your target molecules have no primer sequences in J Region (e.g. library was amplified using primer to the C region) or your <code>-OjParameters.geneFeatureToAlign</code> does not cover the region where primer is located, then you can change value of this parameter for J gene to <code>false</code> to increase J gene identification accuracy and overall specificity of alignments.</dd> <dt><code>absoluteMinScore</code></dt> <dd>Minimal score of alignment: alignments with smaller score will be dropped.</dd> <dt><code>relativeMinScore</code></dt> <dd>Minimal relative score of alignments: if alignment score is smaller than <code>relativeMinScore * maxScore</code>, where <code>maxScore</code> is the best score among all alignments for particular gene type (V, J or C) and input sequence, it will be dropped.</dd> <dt><code>maxHits</code></dt> <dd>Maximal number of hits: if input sequence align with more than <code>maxHits</code> queries, only top <code>maxHits</code> hits will be kept.</dd> <dt><code>mapperAbsoluteMinClusterScore</code></dt> <dd>Minimal allowed absolute hit score obtained by mapper to consider hit as reliable candidate</dd> <dt><code>mapperExtraClusterScore</code></dt> <dd>Extra score for good cluster.</dd> <dt><code>mapperMatchScore</code></dt> <dd>Score for single k-mer match.</dd> <dt><code>mapperMismatchScore</code></dt> <dd>Mismatch penalty for single k-mer mismatch.</dd> <dt><code>mapperOffsetShiftScore</code></dt> <dd>Penalty for k-mer position shift.</dd> <dt><code>mapperSlotCount</code></dt> <dd>Number of simultaneously constructed clusters in one-pass initial cluster detection algorithm.</dd> <dt><code>mapperMaxClusters</code></dt> <dd>Max allowed clusters.</dd> <dt><code>mapperMaxClusterIndels</code></dt> <dd>Max indels inside a cluster (if indel is bigger, alignment will be divided into several clusters).</dd> <dt><code>mapperKMersPerPosition</code></dt> <dd>If <code>mapperKValue &gt; 0</code>, it is possible to map several seeds with holes in different places to the same position in target sequence. This parameter control the maximum.</dd> <dt><code>mapperAbsoluteMinScore</code></dt> <dd>k-mer mapper absolute min score</dd> <dt><code>mapperRelativeMinScore</code></dt> <dd>k-mer mapper relative min score</dd> <dt><code>mapperMinSeedsDistance</code></dt> <dd>min distance between seeds in seed-and-vote strategy</dd> <dt><code>mapperMaxSeedsDistance</code></dt> <dd>max distance between neighbor seeds during seeding</dd> <dt><code>alignmentStopPenalty</code></dt> <dd>stop penalty</dd> </dl> <p>Default values are:</p> <table> <thead> <tr> <th>Parameter</th> <th>Default V value</th> <th>Default J value</th> </tr> </thead> <tbody> <tr> <td>mapperNValue</td> <td>8</td> <td>8</td> </tr> <tr> <td>mapperKValue</td> <td>1</td> <td>1</td> </tr> <tr> <td>floatingLeftBound</td> <td>true</td> <td>true</td> </tr> <tr> <td>floatingRightBound</td> <td>true</td> <td>false</td> </tr> <tr> <td>mapperAbsoluteMinClusterScore</td> <td>102</td> <td>102</td> </tr> <tr> <td>mapperExtraClusterScore</td> <td>-38</td> <td>-38</td> </tr> <tr> <td>mapperMatchScore</td> <td>95</td> <td>95</td> </tr> <tr> <td>mapperMismatchScore</td> <td>-14</td> <td>-14</td> </tr> <tr> <td>mapperOffsetShiftScore</td> <td>-82</td> <td>-82</td> </tr> <tr> <td>mapperSlotCount</td> <td>6</td> <td>6</td> </tr> <tr> <td>mapperMaxClusters</td> <td>4</td> <td>4</td> </tr> <tr> <td>mapperMaxClusterIndels</td> <td>4</td> <td>4</td> </tr> <tr> <td>mapperKMersPerPosition</td> <td>4</td> <td>4</td> </tr> <tr> <td>mapperAbsoluteMinScore</td> <td>100</td> <td>100</td> </tr> <tr> <td>mapperRelativeMinScore</td> <td>0.8</td> <td>0.8</td> </tr> <tr> <td>mapperMinSeedsDistance</td> <td>5</td> <td>5</td> </tr> <tr> <td>mapperMaxSeedsDistance</td> <td>15</td> <td>5</td> </tr> <tr> <td>alignmentStopPenalty</td> <td>0</td> <td>0</td> </tr> <tr> <td>absoluteMinScore</td> <td>150</td> <td>140</td> </tr> <tr> <td>relativeMinScore</td> <td>0.8</td> <td>0.8</td> </tr> <tr> <td>maxHits</td> <td>3</td> <td>3</td> </tr> </tbody> </table> <p>To override these parameters one can e.g. do: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span>--preset<span class=w> </span>kAligner2<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.maxHits<span class=o>=</span><span class=m>5</span><span class=w> </span><span class=se>\ </span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <p>MiXCR uses affine scoring for <code>kAligner2</code>:</p> <dl> <dt><code>subsMatrix=simple(match = &lt;match&gt;, mismatch = &lt;mismatch&gt;&gt;)</code></dt> <dd> <p>Substitution matrix. Available types:</p> <ul> <li><code>simple</code> - a matrix with diagonal elements equal to <code>match</code> and other elements equal to <code>mismatch</code></li> <li><code>raw</code> - a complete set of 16 matrix elements should be specified; for example: <code>raw(5,-9,-9,-9,-9,5,-9,-9,-9,-9,5,-9,-9,-9,-9,5)</code> (equivalent to the default value)</li> </ul> </dd> <dt><code>gapOpenPenalty=&lt;penalty&gt;</code></dt> <dd>Penalty for opening a gap.</dd> <dt><code>gapExtensionPenalty=&lt;penalty&gt;</code></dt> <dd>Penalty gap extension.</dd> </dl> <p>Default values used with <code>kAligner2</code> preset are: <div class=highlight><pre><span></span><code>subsMatrix=simple(match = 10, mismatch = -19)
gapOpenPenalty=-62
gapExtensionPenalty=-11
</code></pre></div></p> <p>Scoring parameters can be overridden in the following way: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span>--species<span class=w> </span>hs<span class=w> </span>--preset<span class=w> </span>kAligner2<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.scoring.gapOpenPenalty<span class=o>=</span>-30<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OvParameters.parameters.scoring.subsMatrix<span class=o>=</span>simple<span class=o>(</span><span class=nv>match</span><span class=o>=</span><span class=m>4</span>,mismatch<span class=o>=</span>-11<span class=o>)</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <h3 id=d-aligner-parameters>D aligner parameters</h3> <p>The following parameters can be overridden for D aligner:</p> <dl> <dt><code>-OdParameters.absoluteMinScore=25.0</code></dt> <dd>Minimal score of alignment: alignments with smaller scores will be dropped. </dd> <dt><code>-OdParameters.relativeMinScore=.85</code></dt> <dd>Minimal relative score of alignment: if alignment score is smaller than <code>relativeMinScore * maxScore</code>, where <code>maxScore</code> is the best score among all alignments for particular sequence, it will be dropped.</dd> <dt><code>-OdParameters.maxHits=3</code></dt> <dd>Maximal number of hits: if input sequence align with more than <code>maxHits</code> queries, only top <code>maxHits</code> hits will be kept. </dd> </dl> <p>One can override these parameters like in the following example:" <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OdParameters.absoluteMinScore<span class=o>=</span><span class=m>10</span><span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <p>Scoring parameters for D aligner are the following:</p> <dl> <dt><code>-OdParameters.type=linear</code></dt> <dd>Type of scoring. Possible values: <code>affine</code>, <code>linear</code>.</dd> <dt><code>-OdParameters.subsMatrix=simple(match = 5,mismatch = -9)</code></dt> <dd> <p>Substitution matrix. Available types:</p> <ul> <li><code>simple</code> - a matrix with diagonal elements equal to <code>match</code> and other elements equal to <code>mismatch</code></li> <li><code>raw</code> - a complete set of 16 matrix elements should be specified; for example: <code>raw(5,-9,-9,-9,-9,5,-9,-9,-9,-9,5,-9,-9,-9,-9,5)</code> (equivalent to the default value) </li> </ul> </dd> <dt><code>-OdParameters.gapPenalty=-12</code></dt> <dd>Penalty for a gap. </dd> </dl> <p>D aligner parameters can be overridden in the following way: <div class=highlight><pre><span></span><code>&gt;<span class=w> </span>mixcr<span class=w> </span>align<span class=w> </span><span class=se>\</span>
<span class=w>    </span>--species<span class=w> </span>hs<span class=w> </span><span class=se>\</span>
<span class=w>    </span>-OdParameters.scoring.gapExtensionPenalty<span class=o>=</span>-5<span class=w> </span><span class=se>\</span>
<span class=w>    </span>input_file1<span class=w> </span><span class=o>[</span>input_file2<span class=o>]</span><span class=w> </span>output_file.vdjca
</code></pre></div></p> <h2 id=hardware-recommendations>Hardware recommendations</h2> <p>Alignment step is CPU-consuming. It utilizes all available CPU kernels (unless <code>--threads</code> option is specified) and scales nearly linearly with the increase of CPU count. The only factor limiting linear scaling is I/O speed (disk read) and decompression of gzipped fastq data. It consumes low amount of RAM and 4Gb should be enough for any size of input files. </p> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../mixcr-analyze/ class="md-footer__link md-footer__link--prev" aria-label="Previous: analyze" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> analyze </div> </div> </a> <a href=../mixcr-refineTagsAndSort/ class="md-footer__link md-footer__link--next" aria-label="Next: refineTagsAndSort" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> refineTagsAndSort </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../../../javascripts/svg.js></script> <script src=../../../javascripts/onload.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>