<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="MiXCR: ultimate software platform for analysis of Next-Generation Sequencing (NGS) data for immune profiling."><meta name=author content="MiLaboratories Inc"><link href=https://docs.milaboratories.com/mixcr/guides/biomed2-bcr/ rel=canonical><link rel=icon href=../../../assets/favicon.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>BioMed2 - Docs @ MiLaboratories</title><link rel=stylesheet href=../../../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link rel=stylesheet href=../../../css/extra.css><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-2N4Z95ZG08"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-2N4Z95ZG08",{page_path:e.pathname})})})</script><script async src="https://www.googletagmanager.com/gtag/js?id=G-2N4Z95ZG08"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#biomed2 class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../.. title="Docs @ MiLaboratories" class="md-header__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Docs @ MiLaboratories </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> BioMed2 </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z"/> </svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z"/> </svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label="Search by name or category" placeholder="Search by name or category" autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../about/ class=md-tabs__link> About </a> </li> <li class=md-tabs__item> <a href=../../getting-started/installation/ class=md-tabs__link> Getting Started </a> </li> <li class=md-tabs__item> <a href=../milaboratories-human-bcr-rna-multi/ class="md-tabs__link md-tabs__link--active"> Guides </a> </li> <li class=md-tabs__item> <a href=../../reference/overview-analysis-overview/ class=md-tabs__link> Docs </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav class="md-nav md-nav--primary md-nav--lifted has-section" aria-label=Navigation data-md-level=0 data-test=1> <!-- Site title --> <label class=md-nav__title for=__drawer> <a href=../../.. title="Docs @ MiLaboratories" class="md-nav__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class=md-nav__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class=md-nav__list data-md-scrollfix> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../about/ class=md-nav__link> About </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_2 type=checkbox id=__nav_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_2 data-level=1> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Getting Started" data-md-level=1> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/installation/ class=md-nav__link> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/docker/ class=md-nav__link> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/milm/ class=md-nav__link> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/first-run/ class=md-nav__link> First run </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/license/ class=md-nav__link> License </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/support/ class=md-nav__link> Support </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3 data-level=1> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Guides data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_1 data-level=2> Kits & Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Kits & Protocols" data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Kits & Protocols </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-bcr-rna-multi/ class=md-nav__link> MiLaboratories Human IG RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-tcr-rna-multi/ class=md-nav__link> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milab-dna-multiplex-tcr/ class=md-nav__link> MiLaboratories Human TCR DNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-hsa-bcr/ class=md-nav__link> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-mmu-tcr/ class=md-nav__link> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../iRepertoire/ class=md-nav__link> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../qiaseq-tcr/ class=md-nav__link> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../cellecta-air-tcr-bcr/ class=md-nav__link> Cellecta AIR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../abhelix-bcr/ class=md-nav__link> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for=__toc> BioMed2 <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> BioMed2 </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-libraries class=md-nav__link> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#under-the-hood-pipeline class=md-nav__link> Under the hood pipeline: </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Under the hood pipeline:"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#align class=md-nav__link> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#assemble class=md-nav__link> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export class=md-nav__link> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#full-length-clonotype-assembly class=md-nav__link> Full-length clonotype assembly </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#reports class=md-nav__link> Reports </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!-- Main navigation item --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../nebnext-bcr/ class=md-nav__link> NEBNext Immune Sequencing </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_2 data-level=2> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Analysis cases" data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-tcr/ class=md-nav__link> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-multiplex-bcr/ class=md-nav__link> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-bcr/ class=md-nav__link> Targeted BCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../rnaseq/ class=md-nav__link> RNA-Seq data </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ig-trees-reconstruction/ class=md-nav__link> Immunoglobulin lineage trees reconstruction </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_3 type=checkbox id=__nav_3_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_3 data-level=2> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Special guides" data-md-level=2> <label class=md-nav__title for=__nav_3_3> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../create-custom-library/ class=md-nav__link> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../external-libraries/ class=md-nav__link> Using external libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../b-cell-lineages-webinar/ class=md-nav__link> B cell lineage tracing </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4 data-level=1> Docs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Docs data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Docs </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_1 data-level=2> Overview <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Overview data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-analysis-overview/ class=md-nav__link> Analysis overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-presets/ class=md-nav__link> Presets </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-mixins-list/ class=md-nav__link> Mix-in options list </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-built-in-presets/ class=md-nav__link> Built-in presets </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_2 data-level=2> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiXCR data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-analyze/ class=md-nav__link> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-align/ class=md-nav__link> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-refineTagsAndSort/ class=md-nav__link> refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemblePartial/ class=md-nav__link> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-extend/ class=md-nav__link> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemble/ class=md-nav__link> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assembleContigs/ class=md-nav__link> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-export/ class=md-nav__link> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPretty/ class=md-nav__link> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportAirr/ class=md-nav__link> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReadsForClones/ class=md-nav__link> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportShmTrees/ class=md-nav__link> exportShmTrees </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReports/ class=md-nav__link> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportQc/ class=md-nav__link> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-postanalysis/ class=md-nav__link> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-downsample/ class=md-nav__link> downsample </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPlots/ class=md-nav__link> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportTables/ class=md-nav__link> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPreprocTables/ class=md-nav__link> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportClonesOverlap/ class=md-nav__link> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-overlapScatterPlot/ class=md-nav__link> overlapScatterPlot </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findAlleles/ class=md-nav__link> findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findShmTrees/ class=md-nav__link> findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_3 data-level=2> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Reports data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-align/ class=md-nav__link> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-refineTagsAndSort/ class=md-nav__link> mixcr refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemblePartial/ class=md-nav__link> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-extend/ class=md-nav__link> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemble/ class=md-nav__link> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assembleContigs/ class=md-nav__link> mixcr assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findAlleles/ class=md-nav__link> mixcr findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findShmTrees/ class=md-nav__link> mixcr findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_4 data-level=2> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=RepSeq.io data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fasta/ class=md-nav__link> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-compile/ class=md-nav__link> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-debug/ class=md-nav__link> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-format/ class=md-nav__link> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromFasta/ class=md-nav__link> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromPaddedFasta/ class=md-nav__link> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-inferPoints/ class=md-nav__link> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-merge/ class=md-nav__link> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-stat/ class=md-nav__link> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-tsv/ class=md-nav__link> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_5 data-level=2> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiTool data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-parse/ class=md-nav__link> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-tag-stat/ class=md-nav__link> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-export-fastq/ class=md-nav__link> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-correct/ class=md-nav__link> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-consensus/ class=md-nav__link> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-process-fastq/ class=md-nav__link> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-format-fastq/ class=md-nav__link> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-compress-sequence-set/ class=md-nav__link> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_6 type=checkbox id=__nav_4_6> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_6 data-level=2> Specifications <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Specifications data-md-level=2> <label class=md-nav__title for=__nav_4_6> <span class="md-nav__icon md-icon"></span> Specifications </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-tag-pattern/ class=md-nav__link> Tag pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-input-file-name-expansion/ class=md-nav__link> Input file name expansion </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-gene-features/ class=md-nav__link> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-translation-rules/ class=md-nav__link> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-mutations-encoding/ class=md-nav__link> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-repseqio-json-format/ class=md-nav__link> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-libraries class=md-nav__link> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#under-the-hood-pipeline class=md-nav__link> Under the hood pipeline: </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Under the hood pipeline:"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#align class=md-nav__link> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#assemble class=md-nav__link> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export class=md-nav__link> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#full-length-clonotype-assembly class=md-nav__link> Full-length clonotype assembly </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#reports class=md-nav__link> Reports </a> <!-- Table of contents list --> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/milaboratory/docs/edit/main/docs/mixcr/guides/biomed2-bcr.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=biomed2>BIOMED2</h1> <p>Here we will discuss how to process BCR cDNA libraries obtained with BIOMED2 kit.</p> <h2 id=data-libraries>Data libraries</h2> <p>This tutorial uses the data from the following publication: <em>High frequency of shared clonotypes in human B cell receptor repertoires.</em> Soto C et al., Nature, 2019 Feb;566(7744):398-402 <a href=https://doi.org/10.1038/s41586-019-0934-8>doi: 10.1038/s41586-019-0934-8</a></p> <p>Peripheral blood samples were collected from three donors: HIP1 (female, 47 y.o.), HIP2 (male, 22 y.o.) and HIP3 (male, 29 y.o.). Samples were collected in multiple replicas. Total RNA was extracted from PBMCs and then RT-PCR was utilized to generate BCR amplicon libraries using BIOMED2 multiplex primers. Subsequent libraries were sequenced using next-generation sequencing.</p> <p>All data is available from SRA (PRJNA511481) using e.g. <a href=https://sra-explorer.info>SRA Explorer</a>.</p> <details class=tip> <summary>Use <a href=https://aria2.github.io>aria2c</a> for efficient download of the full dataset with the proper filenames:</summary> <p><div class=highlight><span class=filename>download.sh</span><pre><span></span><code>mkdir -p raw
aria2c -c -s <span class=m>16</span> -x <span class=m>16</span> -k 1M -j <span class=m>8</span> -i download-list.txt
</code></pre></div> <div class=highlight><span class=filename>download-list.txt</span><pre><span></span><code>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365468/SRR8365468_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365468_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365468/SRR8365468_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365468_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365457/SRR8365457_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365457_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365457/SRR8365457_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365457_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365458/SRR8365458_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365458_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365458/SRR8365458_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365458_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365459/SRR8365459_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365459_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365459/SRR8365459_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365459_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365463/SRR8365463_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365463_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365463/SRR8365463_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365463_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365469/SRR8365469_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365469_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365469/SRR8365469_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365469_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365465/SRR8365465_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365465_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365465/SRR8365465_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365465_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365467/SRR8365467_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365467_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365467/SRR8365467_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365467_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365464/SRR8365464_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365464_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365464/SRR8365464_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365464_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365450/SRR8365450_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365450_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365450/SRR8365450_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365450_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365461/SRR8365461_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365461_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365461/SRR8365461_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365461_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365462/SRR8365462_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365462_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365462/SRR8365462_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365462_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365456/SRR8365456_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365456_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365456/SRR8365456_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365456_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365475/SRR8365475_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365475_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365475/SRR8365475_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365475_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365460/SRR8365460_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365460_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365460/SRR8365460_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365460_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365470/SRR8365470_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365470_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365470/SRR8365470_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365470_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365471/SRR8365471_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365471_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365471/SRR8365471_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365471_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365473/SRR8365473_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365473_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365473/SRR8365473_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365473_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365449/SRR8365449_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365449_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365449/SRR8365449_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365449_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365482/SRR8365482_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365482_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365482/SRR8365482_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365482_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365446/SRR8365446_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365446_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365446/SRR8365446_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365446_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365483/SRR8365483_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365483_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365483/SRR8365483_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365483_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365447/SRR8365447_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365447_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365447/SRR8365447_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365447_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365484/SRR8365484_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365484_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365484/SRR8365484_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365484_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365448/SRR8365448_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365448_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365448/SRR8365448_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365448_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365424/SRR8365424_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365424_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365424/SRR8365424_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365424_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365485/SRR8365485_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365485_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365485/SRR8365485_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365485_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365488/SRR8365488_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365488_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365488/SRR8365488_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365488_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365421/SRR8365421_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365421_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365421/SRR8365421_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365421_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365489/SRR8365489_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365489_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365489/SRR8365489_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365489_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365490/SRR8365490_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365490_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365490/SRR8365490_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365490_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365246/SRR8365246_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365246_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365246/SRR8365246_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365246_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365474/SRR8365474_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365474_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365474/SRR8365474_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365474_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365422/SRR8365422_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365422_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365422/SRR8365422_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365422_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365423/SRR8365423_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365423_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365423/SRR8365423_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365423_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365420/SRR8365420_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365420_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365420/SRR8365420_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365420_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365419/SRR8365419_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365419_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365419/SRR8365419_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365419_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365248/SRR8365248_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365248_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365248/SRR8365248_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365248_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365249/SRR8365249_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365249_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365249/SRR8365249_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365249_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365247/SRR8365247_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365247_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365247/SRR8365247_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365247_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365250/SRR8365250_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365250_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365250/SRR8365250_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365250_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365348/SRR8365348_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365348_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365348/SRR8365348_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365348_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365251/SRR8365251_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365251_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365251/SRR8365251_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365251_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365418/SRR8365418_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365418_HIP3_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365418/SRR8365418_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365418_HIP3_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365310/SRR8365310_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365310_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365310/SRR8365310_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365310_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365252/SRR8365252_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365252_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365252/SRR8365252_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365252_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365308/SRR8365308_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365308_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365308/SRR8365308_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365308_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365309/SRR8365309_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365309_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365309/SRR8365309_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365309_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365253/SRR8365253_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365253_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365253/SRR8365253_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365253_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365307/SRR8365307_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365307_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365307/SRR8365307_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365307_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365259/SRR8365259_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365259_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365259/SRR8365259_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365259_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365258/SRR8365258_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365258_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365258/SRR8365258_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365258_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365306/SRR8365306_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365306_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365306/SRR8365306_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365306_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365305/SRR8365305_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365305_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365305/SRR8365305_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365305_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365304/SRR8365304_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365304_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365304/SRR8365304_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365304_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365260/SRR8365260_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365260_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365260/SRR8365260_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365260_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365303/SRR8365303_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365303_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365303/SRR8365303_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365303_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365261/SRR8365261_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365261_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365261/SRR8365261_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365261_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365262/SRR8365262_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365262_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365262/SRR8365262_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365262_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365302/SRR8365302_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365302_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365302/SRR8365302_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365302_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365301/SRR8365301_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365301_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365301/SRR8365301_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365301_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365263/SRR8365263_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365263_HIP2_male_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365263/SRR8365263_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365263_HIP2_male_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365264/SRR8365264_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365264_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365264/SRR8365264_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365264_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365267/SRR8365267_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365267_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365267/SRR8365267_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365267_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365274/SRR8365274_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365274_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365274/SRR8365274_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365274_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365269/SRR8365269_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365269_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365269/SRR8365269_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365269_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365271/SRR8365271_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365271_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365271/SRR8365271_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365271_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365273/SRR8365273_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365273_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365273/SRR8365273_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365273_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365265/SRR8365265_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365265_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365265/SRR8365265_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365265_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365270/SRR8365270_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365270_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365270/SRR8365270_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365270_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365272/SRR8365272_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365272_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365272/SRR8365272_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365272_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365266/SRR8365266_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365266_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365266/SRR8365266_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365266_HIP1_female_R2.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365268/SRR8365268_1.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365268_HIP1_female_R1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365268/SRR8365268_2.fastq.gz
  <span class=nv>out</span><span class=o>=</span>raw/SRR8365268_HIP1_female_R2.fastq.gz
</code></pre></div></p> </details> <p>The above script downloads all the data to <code>raw/</code> folder. Each file name encodes the data about the donor and its sex. For example for the first file from the above list (<code>SRR8365468_HIP2_male_R1.fastq.gz</code>):</p> <ul> <li><code>SRR8365468</code> - sample id</li> <li><code>HIP2</code> - patient id</li> <li><code>female</code> - patient sex</li> </ul> <p>The project contains 147 FASTQ file pairs. The structure of sequences is shown on the picture bellow. </p> <p>The structure of the library is shown on the picture bellow.</p> <p><img alt src=../../reference/pics/biomed2-human-bcr-kit-light.svg#only-light> <img alt src=../../reference/pics/biomed2-human-bcr-kit-dark.svg#only-dark></p> <p>This data is obtained using multiplex mix of V and J primers that can be found <a href=https://www.nature.com/articles/s41586-019-0934-8/tables/3>here</a>. In the original paper, where <a href=https://www.nature.com/articles/2403202>Biomed2 primer set was published</a> the authors declare that some primers allow upto 2 mismatches. This means that these mismatches will end up in the final sequences and may introduce bias into alignment process.</p> <p>Briefly: 1. There are 16 primers (complimentary to VH, VK and VL segments). These primers' sequences are located in <code>FR1</code>region and the longest primer consists of 32 nucleotides. <code>R1</code> is 250bp long and, depending on a certain V gene, it covers the whole V segment upto to the beginning of<code>CDR3</code> region. 2. There are 5 primers (complimentary to JH, JK and JL segments). These primers are located in <code>FR4</code> and the longest consists of 26 nucleotides. <code>R2</code> starts from one of these primers and ends in <code>FR2</code>.</p> <p>Bellow you can see J primers aligned with IGHJ / IGKL / IGLJ genes sequences.</p> <p><div class=highlight><pre><span></span><code><span class=c1># IGHJ</span>

         &lt;---------FR4-----------------&gt;
NNNNCTTACCTGAGGAGACGGTGACC                                                &lt;- Jh-primer
         CTGAGGAGACGGTGACCAGGGTGCCCTGGCCCCAGTGCTGGAAGTATTCAGC             &lt;- IGHJ1
         CTGAGGAGACAGTGACCAGGGTGCCACGGCCCCAGAGATCGAAGTACCAGTAG            &lt;- IGHJ2
         CTGAAGAGACGGTGACCATTGTCCCTTGGCCCCAGACATCAAAAGCATCA               &lt;- IGHJ3
         CTGAGGAGACGGTGACCAGGGTTCCTTGGCCCCAGTAGTCAAAGTAGT                 &lt;- IGHJ4
         CTGAGGAGACGGTGACCAGGGTTCCTTGGCCCCAGGAGTCGAACCAGTTGT              &lt;- IGHJ5
            AGGAGACGGTGACCGTGGTCCCTTGCCCCCAGACGTCCATACCGTAGTAGTAGTAGTAAT  &lt;- IGHJ6

<span class=c1># IGK</span>
     &lt;------------ FR4----------&gt;
  NNNNTTTGATaTCCAccTTGGTCCC                   &lt;- Jk1-primer 
     GTTTGATTTCCACCTTGGTCCCTTGGCCGAACGTCCAC   &lt;- IGKJ1
     GTTTGATCTCCAGCTTGGTCCCCTGGCCAAAAGTGTACA  &lt;- IGKJ2
     GTTTGATATCCACTTTGGTCCCAGGGCCGAAAGTGAAT   &lt;- IGKJ3
     GTTTGATCTCCACCTTGGTCCCTCCGCCGAAAGTGAGC   &lt;- IGKJ4

  NNNNTTTAATCTCCAGTCGTGTCCC                   &lt;- Jk2-primer
     GTTTAATCTCCAGTCGTGTCCCTTGGCCGAAGGTGATC   &lt;- IGKJ5

<span class=c1># IGL</span>
  &lt;------------ FR4----------&gt;
NNNNAGGACGGTGACCTTGGTCCC                  &lt;- Jl1-primer
NNNNAGGACGGTCAGCTgGGTCCC                  &lt;- Jl2-primer
  CTAGGACGGTGACCTTGGTCCCAGTTCCGAAGACATAA  &lt;- IGLJ1
  CTAGGACGGTCAGCTTGGTCCCTCCGCCGAATACCACA  &lt;- IGLJ2
  CTAGGACGGTCAGCTTGGTCCCTCCGCCGAATACCACA  &lt;- IGLJ3*01
  CTAGGACGGTCAGCTTGGTCCCTCCGCCGAACACCCAA  &lt;- IGLJ3*02
  CTAGGACGGTCAGCTCGGTCCCCTCACCAAACACCCAG  &lt;- IGLJ5
  CGAGGACGGTCACCTTGGTGCCACTGCCGAACACATTA  &lt;- IGLJ6
  CTAAAATGATCAGCTGGGTTCCTCCACCAAATACAAAA  &lt;- IGLJ4
  CGAGGACGGTCAGCTGGGTGCCTCCTCCGAACACAGCA  &lt;- IGLJ7
</code></pre></div> What's important, that in all cases there is enough nucleotides, not covered by the primer, for correct J gene identification. Thus, we can use MiXCR to trim primer sequence.</p> <h2 id=upstream-analysis>Upstream analysis</h2> <p>MiXCR has a dedicated preset for this protocol, thus analysing the data ia as easy as:</p> <div class=highlight><pre><span></span><code>mixcr analyze biomed2-human-bcr-cdr3 <span class=se>\</span>
    raw/SRR8365277_HIP1_female_IgG1_R1.fastq.gz <span class=se>\</span>
    raw/SRR8365277_HIP1_female_IgG1_R2.fastq.gz <span class=se>\</span>
    results/SRR8365277_HIP1_female_IgG1
</code></pre></div> <p>One might also use <a href=https://www.gnu.org/software/parallel/ >GNU Parallel</a> to process all samples at once:</p> <div class=highlight><pre><span></span><code><span class=ch>#!/usr/bin/env bash</span>

mkdir -p results

ls /raw/*R1* <span class=p>|</span>
    parallel -j <span class=m>2</span> --line-buffer <span class=se>\</span>
    <span class=s2>&quot;mixcr analyze biomed2-human-bcr-cdr3 \</span>
<span class=s2>    {} \</span>
<span class=s2>    {=s:R1:R2:=} \</span>
<span class=s2>    {=s:.*/:results/:;s:_R.*::=}&quot;</span>
</code></pre></div> <h3 id=under-the-hood-pipeline>Under the hood pipeline:</h3> <p>Under the hood <code>mixcr analyze biomed2-human-bcr-cdr3</code> executes the following pipeline:</p> <h4 id=align><code>align</code></h4> <p>Alignment of raw sequencing reads against reference database of V-, D-, J- and C- gene segments.</p> <div class=highlight><pre><span></span><code>mixcr align <span class=se>\</span>
    --species hsa <span class=se>\</span>
    -p kAligner2_4.0 <span class=se>\</span>
    --tag-pattern <span class=s2>&quot;^N{32}(R1:*)\^N{26}(R2:*)&quot;</span> <span class=se>\</span>
    -OvParameters.geneFeatureToAlign<span class=o>=</span><span class=s2>&quot;VTranscriptWithout5UTRWithP&quot;</span> <span class=se>\</span>
    -OvParameters.parameters.floatingLeftBound<span class=o>=</span><span class=nb>false</span> <span class=se>\</span>
    -OjParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span> <span class=se>\</span>
    --report results/SRR8365468_HIP2_male.report <span class=se>\</span>
    --json-report results/SRR8365468_HIP2_male.json <span class=se>\</span>
     raw/SRR8365468_HIP2_male_R1.fastq.gz <span class=se>\</span>
     raw/SRR8365468_HIP2_male_R2.fastq.gz <span class=se>\</span>
     results/SRR8365468_HIP2_male.vdjca
</code></pre></div> <p>Option <code>--report</code> is specified here explicitly.</p> <dl> <dt><code>--species hsa</code></dt> <dd>determines the organism species.</dd> <dt><code>-p</code></dt> <dd><code>kAligner2_4.0</code> a preset of MiXCr parameters which includes kAligner2 (a specialized aligner for BCR data).</dd> <dt><code>--tag-pattern "^N{32}(R1:*)\^N{26}(R2:*)"</code></dt> <dd>With this pattern we trim 32 nucleotides (the length of the longest forward primer) from the beginning of R1 file and 26 nucleotide (the length of the longest reverse primer from the beginning of R2 file.</dd> <dt><code>-OvParameters.geneFeatureToAlign="VTranscriptWithout5UTRWithP"</code></dt> <dd>Sets a V gene feature to align. Check <a href=../../reference/ref-gene-features/ >gene features</a> for more info.</dd> <dt><code>-OvParameters.parameters.floatingLeftBound=false</code></dt> <dd>Results in a global alignment algorithm for V gene left bound. We can use global alignment here, because primer sequences were trimmed with <code>--tag-pattern</code>.</dd> <dt><code>-OjParameters.parameters.floatingRightBound=false</code></dt> <dd>Results in a global alignment algorithm for J gene right bound. We can use global alignment here, because primer sequences were trimmed with <code>--tag-pattern</code>.</dd> </dl> <h4 id=assemble><code>assemble</code></h4> <p>Assembles alignments into clonotypes and applies several layers of errors correction(ex. quality-awared correction for sequencing errors, clustering to correct for PCR errors). Check <a href=../../reference/mixcr-assemble/ ><code>mixcr assemble</code></a> for more information. By default clones will be assembled by <code>CDR3</code> gene feature.</p> <div class=highlight><pre><span></span><code>mixcr assemble <span class=se>\</span>
    -OassemblingFeatures<span class=o>=</span><span class=s2>&quot;CDR3&quot;</span> <span class=se>\</span>
    -OseparateByJ<span class=o>=</span><span class=nb>true</span> <span class=se>\</span>
    -OseparateByV<span class=o>=</span><span class=nb>true</span> <span class=se>\</span>
    --report results/SRR8365468_HIP2_male.report <span class=se>\</span>
    --json-report results/SRR8365468_HIP2_male.json <span class=se>\</span>
    results/SRR8365468_HIP2_male.vdjca <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns
</code></pre></div> <dl> <dt><code>-OseparateByV=true</code></dt> <dd>Split clones with the same <code>CDR3</code> sequence and different V-genes</dd> <dt><code>-OseparateByJ=true</code></dt> <dd>Split clones with the same <code>CDR3</code> sequence and different J-genes</dd> </dl> <h4 id=export><code>export</code></h4> <p>Exports clonotypes from .clns file into human-readable tables.</p> <div class=highlight><pre><span></span><code>mixcr exportClones <span class=se>\</span>
    -c IGH <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns <span class=se>\</span>
    results/SRR8365468_HIP2_male.clonotypes.IGH.tsv

mixcr exportClones <span class=se>\</span>
    -c IGL <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns <span class=se>\</span>
    results/SRR8365468_HIP2_male.clonotypes.IGL.tsv

mixcr exportClones <span class=se>\</span>
    -c IGK <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns <span class=se>\</span>
    results/SRR8365468_HIP2_male.clonotypes.IGK.tsv
</code></pre></div> <p>Here <code>-p full</code> defines the full preset of common export columns. Check <a href=../../reference/mixcr-export/ ><code>mixcr export</code></a> for more information.</p> <dl> <dt><code>- &lt;chain&gt;</code></dt> <dd>defines a specific chain to be exported.</dd> </dl> <p>After execution is complete the following list of files is generated for every sample:</p> <div class=highlight><pre><span></span><code><span class=c1># human-readable reports </span>
SRR8365468_HIP2_male.report
<span class=c1># raw alignments (highly compressed binary file)</span>
SRR8365468_HIP2_male.vdjca
<span class=c1># IGH, IGK and IGL CDR3 clonotypes (highly compressed binary file)</span>
SRR8365468_HIP2_male.clns
<span class=c1># IGH, IGK and IGL CDR3 clonotypes exported in tab-delimited txt</span>
SRR8365468_HIP2_male.IGH.tsv
SRR8365468_HIP2_male.IGL.tsv  
SRR8365468_HIP2_male.IGK.tsv  
</code></pre></div> <p>While <code>.clns</code> file holds all data and is used for downstream analysis using <a href=../../reference/mixcr-postanalysis/ ><code>mixcr postanalisis</code></a>, the output <code>.tsv</code> clonotype table will contain exhaustive information about each clonotype as well:</p> <details class=tip> <summary>See first 100 records from clonotype table SRR8365468_HIP2_male:</summary> <table> <thead> <tr> <th align=right>cloneId</th> <th align=right>cloneCount</th> <th align=right>cloneFraction</th> <th align=left>targetSequences</th> <th align=left>targetQualities</th> <th align=left>allVHitsWithScore</th> <th align=left>allDHitsWithScore</th> <th align=left>allJHitsWithScore</th> <th align=right>allCHitsWithScore</th> <th align=left>allVAlignments</th> <th align=left>allDAlignments</th> <th align=left>allJAlignments</th> <th align=right>allCAlignments</th> <th align=right>nSeqFR1</th> <th align=right>minQualFR1</th> <th align=left>nSeqCDR1</th> <th align=right>minQualCDR1</th> <th align=left>nSeqFR2</th> <th align=right>minQualFR2</th> <th align=left>nSeqCDR2</th> <th align=right>minQualCDR2</th> <th align=left>nSeqFR3</th> <th align=right>minQualFR3</th> <th align=left>nSeqCDR3</th> <th align=right>minQualCDR3</th> <th align=right>nSeqFR4</th> <th align=right>minQualFR4</th> <th align=right>aaSeqFR1</th> <th align=left>aaSeqCDR1</th> <th align=left>aaSeqFR2</th> <th align=left>aaSeqCDR2</th> <th align=left>aaSeqFR3</th> <th align=left>aaSeqCDR3</th> <th align=right>aaSeqFR4</th> <th align=left>refPoints</th> </tr> </thead> <tbody> <tr> <td align=right>0</td> <td align=right>2</td> <td align=right>0.0952381</td> <td align=left>AGCAGTGACGTTGATACTTATAACTATGTCTCCTGGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTCATTATTACTGCTGCTCACATTCGACCGGCAGCACTCGTTATGTCTTCGGAACT</td> <td align=left>IIIIIIIIGIGGIGGIGGIGGIIGIIIIIIIIIIIIGIIIIIIIIIIIIIIIIIIIIIIAGIGGGGGGGIGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGGGIIIAG</td> <td align=left>IGLV2-14*00(569)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(257.5)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>112</td> <td align=left>317</td> <td align=right>0</td> <td align=right>37</td> <td align=right>SG88ASG90ASG91C</td> <td align=left>283.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=left>58</td> <td align=right>128</td> <td align=left>144</td> <td align=right></td> <td align=left>160.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGTGACGTTGATACTTATAACTAT</td> <td align=left>38</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>1</td> <td align=right>2</td> <td align=right>0.0952381</td> <td align=left>AGCAATGACGTTGGGGTTTCTCTTCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCAACTCATATACACTCGGCAGCACCCCGGTTTGTGTCTTCGGAACT</td> <td align=left>IIIIIIIIGGIIGIIIIIIIIIIIIIIIIIIIIIGIIIIIIGIIIIIIGIIIIIIIIIIGIGGGIIIIIIIIGGGIGIIIIIGIIIIIIIIIIIIIIIIIIIIIIIIGIIIIIIIIIGGIIIIIIIIIGIIIIIIIGIIGIIIIIIIIIIIAA</td> <td align=left>IGLV2-14*00(737.5)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(206)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>89</td> <td align=left>317</td> <td align=right>0</td> <td align=right>14</td> <td align=right>SG79A</td> <td align=left>111.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=left>58</td> <td align=right>137</td> <td align=left>153</td> <td align=right>SA22G</td> <td align=left>131.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>2</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGGCTCCATCAGTAGTTACTACTGGAGCTGGATCCGGCAGCCCCCAGGGAAGGGACTGGAGTGGATTGGGTATATCTATTACAGTGGGAGCACCAACTACAACCCCTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAAGCTGAGCTCTGTGACCGCTGCGGACACGGCCGTGTATTACTGTGCGAGAGGAGTAATACCGCACCCGCTCGACTACTGGGGCCAGGGA</td> <td align=left>IIIIIIIIIIIGIIIIIGIIIIIGIGGIIIIGIGIIIIGIIEIGIIIGEGGGGIIGIGIGGG&lt;GGGGGG&lt;GIGIIIIGIAGGGIIIIGGGGGGGGGIGGGGGGGGIGIGGGG&lt;GGGIGGIIGGIGIGGGAGGGGGGGGGGGGGGIGIIGIGGGGAAGAAAGGIIIIGIGGIGIGGGGGGGGAGAGIIIGGGGGGGIGIIIIIIIIIGIIGIIIIIGGIIIIIIGGIIIGGGGGGGGIGGGGGGIIGGIGIIIIII</td> <td align=left>IGHV4-59*00(2230)</td> <td align=left>IGHD3-10<em>00(40),IGHD3-22</em>00(40),IGHD1-7*00(36)</td> <td align=left>IGHJ4*00(210)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>292</td> <td align=left>313</td> <td align=right>0</td> <td align=right>217</td> <td align=right></td> <td align=left>2170.0</td> <td align=right>23</td> <td align=left>31</td> <td align=right>93</td> <td align=left>218</td> <td align=right>226</td> <td align=left></td> <td align=right>40.0;23</td> <td align=left>31</td> <td align=right>93</td> <td align=right>218</td> <td align=right>226</td> <td align=right></td> <td align=left>40.0;8</td> <td align=left>18</td> <td align=left>51</td> <td align=left>218</td> <td align=left>228</td> <td align=right>ST11A</td> <td align=left>36.0</td> </tr> <tr> <td align=right>3</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGGCTCCATCAGCAGTGGTGGTTACTACTGGAGCTGGATCCGCCAGCACCCAGGGAAGGGCCTGGAGTGGATTGGGTACATCTATTACAGTGGGAGCACCTACTACAACCCGTCCCTCAAGAGTCGAGTTACCATATCAGTAGACACGTCTAAGAACCAGTTCTCCCTGAAGCTGAGCTCTGTGACTGCCGCGGACACGGCCGTGTATTACTGTGCGAGAGACACCCGTATTACGATTTTTGGAGTGGACTCCAACGGTATGGACGTCTGGGGCCAAGGG</td> <td align=left>GGGGGGGIGGIIIGIIAGGIIIIGGIIGIIIIGIIIIIIIIIIIIIIGGIIIIIIIIIIGAGGEIIIGGGEIIIIGIIIIIIIIGGIIGIIIGGGIIGIIIIGGGIGIIIGGG:GGIIGGGIGIIIGGGIIGIGGGGGGGIIIIIII&lt;GAGGIIIGGIIGGGGGGGGIGIIGIGGIGGGGGGGGIGGIGIGGGGGGGGIIGGAGGIAAAG&lt;&lt;&lt;&lt;&lt;&lt;GGAAAGGAGGIGIIGGGGIIGIIGGGIIGGIIIGIIGIIIGIGGGGIGGGAGGGGGIGGGGIIIGI</td> <td align=left>IGHV4-31*00(2300)</td> <td align=left>IGHD3-3*00(110)</td> <td align=left>IGHJ6*00(312)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>299</td> <td align=left>319</td> <td align=right>0</td> <td align=right>224</td> <td align=right></td> <td align=left>2240.0</td> <td align=right>30</td> <td align=left>52</td> <td align=right>93</td> <td align=left>228</td> <td align=right>250</td> <td align=left></td> <td align=right>110.0</td> <td align=left>29</td> <td align=right>61</td> <td align=right>83</td> <td align=right>250</td> <td align=right>282</td> <td align=left>SA32CST34A</td> <td align=left>262.0</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>GGTGGCTCCATCAGCAGTGGTGGTTACTAC</td> <td align=left>32</td> </tr> <tr> <td align=right>4</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGGCTCCATCAGCAGTAGTAACTGGTGGAGTTGGGTCCGCCAGCCCCCAGGGAAGGGGCTGGAGTGGATTGGGGAAATCTATCATAGTGGGAGCACCAACTACAACCCGTCCCTCAAGAGTCGAGTCACCATATCAGTAGACAAGTCCAAGAACCAGTTCTCCCTGAAGCTGAGCTCTGTGACCGCCGCGGACACGGCCGTGTATTACTGTGCGAGAGTTAGTGGGAGCTACTACTACGGTATGGACGTCTGGGGCCAAGGG</td> <td align=left>IIIIIIIIIIIIIIIGIIIIIIIIIIGGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGEGAAGA&lt;&lt;&lt;&lt;:&lt;&lt;GGIIGIGGGGIGIGIIG&lt;&lt;&lt;GEGGGIIIIIGGGGGGGIIGGGGGGIIIGGGGGIGA&lt;GGGIIIIGGGGIIIIIIIIGGIGGAGGGGGGIIIGIIGGGGGGGIIIIGGGIIGGGGGGGGAAAGG&lt;AGA&lt;&lt;&lt;&lt;&lt;&lt;G&lt;GGIIIIIIIIIIIIIIIIIIIIIIGIIIIIIIIIIIIIIIIIIIIGIIIGGGGGIIII</td> <td align=left>IGHV4-4*00(2260)</td> <td align=left>IGHD1-26*00(45)</td> <td align=left>IGHJ6*00(351)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>295</td> <td align=left>316</td> <td align=right>0</td> <td align=right>220</td> <td align=right></td> <td align=left>2200.0</td> <td align=right>24</td> <td align=left>33</td> <td align=right>60</td> <td align=left>221</td> <td align=right>230</td> <td align=left></td> <td align=right>45.0</td> <td align=left>27</td> <td align=right>61</td> <td align=right>83</td> <td align=right>230</td> <td align=right>264</td> <td align=left></td> <td align=left>340.0</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>GGTGGCTCCATCAGCAGTAGTAACTGG</td> <td align=left>38</td> </tr> <tr> <td align=right>5</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGGGTCCTTCAGTGGTTACTACTGGAGCTGGATCCGCCAGCCCCCAGGGAAGGGGCTGGAGTGGATTGGGGAAATCAATCATAGTGGAAGCACCAACTACAACCCGTCCCTCAAGAGTCGAGTCACCATATCAGTAGACACGTCCAAGAACCAGTTCTCCCTGAAGCTGAGCTCTGTGACCGCCGCGGACACGGCTGTGTATTACTGTGCGAGAGGCCCCTATTACGATTTTTGGAGTGGTTATCGAAACCTCTACTACTACTACGGTATGGACGTCTGGGGCCAAGGG</td> <td align=left>GGIGIIIGGGGGGGGGIIIGIIIIIIIIIIIIIIIIIIIIIGIIIIIIGIIIIIIIIIGGI&lt;&lt;GGGGIIIIIIIGGIIIIIIIIIGGGGGGIIGGGGGGIIIGGGIGIIIIIGGGIGGGGGIGGGGIGGGIGIIIIIIGGIIIIIGIIGGIIIGG:GIIIIGIGIAAAAGGIGIGGGIGGGGGIIIIIGGIIGIGGGIIIIIIIIIIIIIIIIIIIGG&lt;&lt;&lt;&lt;&lt;GGGG&lt;&lt;&lt;GGGGIIIIIIIIIIIIIIIIIGIIIIIIIIGGGGGGIGIIIIIIIIGGGIGIIGGGGIIGI</td> <td align=left>IGHV4-34*00(2261)</td> <td align=left>IGHD3-3*00(115)</td> <td align=left>IGHJ6*00(410)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>298</td> <td align=left>313</td> <td align=right>0</td> <td align=right>223</td> <td align=right>ST295C</td> <td align=left>2201.0</td> <td align=right>33</td> <td align=left>56</td> <td align=right>93</td> <td align=left>223</td> <td align=right>246</td> <td align=left></td> <td align=right>115.0</td> <td align=left>24</td> <td align=right>61</td> <td align=right>83</td> <td align=right>254</td> <td align=right>291</td> <td align=left></td> <td align=left>370.0</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>GGTGGGTCCTTCAGTGGTTACTAC</td> <td align=left>38</td> </tr> <tr> <td align=right>6</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGGGTCCTTCAGTGGTTACATCTGGACCTGGATCCGCCAGACCCCAGGAAAGGGGCTGCAGTAGATTGGACAAATCAATCATAGTGGAAGCGCCAACTACAACCCGTCCCTCAAGAGTCGAGTCACCATATCAGTACACACGTCCAATAGTCAGTTCTCCCTGGAGCTGAGCTCTGTGACCGCCGCGGACACGGCTGTGTATTACTGTGCGAGTCCCAAGATAGCATTCTACAACTGGTTCGACCCCTGGGGCCAGGGA</td> <td align=left>GIIIIIIIIIIIIIIIIIIIIIIIIIIGIIIIIIIIIIIIGGEIIIIIIIIIIGIIIIIIIGGGG&lt;GGGGIGGIIIIGGIIIIIIIIIIIIGAGIIIIIGGGIIIGIIIIIIGIGIGIGGGGIGGGGGGGGGGGIGIIIIIGG:GGIIIIGGIIIGIGIIIIIIGGGIIIIIIIIIIIGGGGIIIIIIIIIG&lt;GGGG&lt;&lt;GAG&lt;GAIIIIGIIGIIIIIIIIIIIIIIIIIGGGIGGIIIIIGGGGGGGIGIIIIGGIIIGI</td> <td align=left>IGHV4-34*00(1775)</td> <td align=left>IGHD7-27<em>00(32),IGHD6-13</em>00(30),IGHD6-25*00(30)</td> <td align=left>IGHJ5*00(351)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>290</td> <td align=left>313</td> <td align=right>0</td> <td align=right>215</td> <td align=right>ST96ASA97TSG103CSC117ASG125ASG135CSG139ASG146ASG147CSA168GSG213CSG224TSA226GSC227TSA240G</td> <td align=left>1715.0</td> <td align=right>0</td> <td align=left>12</td> <td align=right>33</td> <td align=left>215</td> <td align=right>227</td> <td align=left>SC4AST7A</td> <td align=right>32.0;25</td> <td align=left>31</td> <td align=right>63</td> <td align=right>222</td> <td align=right>228</td> <td align=right></td> <td align=left>30.0;22</td> <td align=left>28</td> <td align=left>54</td> <td align=left>222</td> <td align=left>228</td> <td align=right></td> <td align=left>30.0</td> </tr> <tr> <td align=right>7</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGTGACTTCCTCGACGATAGTAGGTGGTGGAGTTGGGTCCGCCAGCCCCCAGGGAAGGGGCTGGAGTGGATTGGAGAGATCCATCATACGGGGATCACCAACTACATCCCGTCCCTCAAGAGTCGGGTCACCATGTCAGTGGACAAGAACAAGAACCAGGTCTCCCTAGAGGTGTATTCTGTGACCGCCGCGGACACGGCCGTGTATTACTGTGCGAGTGGGGCCCACTATATTTGGAATGGGTGGGGCCAGGGA</td> <td align=left>GGIIIIIIIGIIGIIGGIGGIGGGIIIIIIIIIIIGIGIIIIIIIIIIIIIIIIIIIIIGIIIIGGGGI&lt;&lt;&lt;IIGIIIIIIIIIIIIIGIIIIIIIIIIIIIIIIIIIIIIIGAGGGGIGIIGGGIGGIGIIIIIGGGIGIIGGIIGGIIIIIGGIIGGGGGIIIIIIIIGGIIIIIGIIIIIIGGIIGIIIIGIGIGGGIIGIIGGGGGGIIIIIGGIGIIIIIIIIIIIIIGIIIGIIIIIIIIIIGGIIIGG</td> <td align=left>IGHV4-4*00(1399)</td> <td align=left>IGHD5-18<em>00(50),IGHD1-26</em>00(45),IGHD5-12*00(45)</td> <td align=left>IGHJ4<em>00(170),IGHJ5</em>00(170)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>293</td> <td align=left>316</td> <td align=right>0</td> <td align=right>218</td> <td align=right>SG79ASC82TSA84CSA87GSG88ASA90GSG91ASA97GSC98GSG149ASA152GST156CSG163CST164GSG169TSA181TSA200GSA209GSA215GST222ASC223AST234GSG242ASA243GSC246GSA249TSG250ASC251T</td> <td align=left>1368.0</td> <td align=right>10</td> <td align=left>20</td> <td align=right>69</td> <td align=left>224</td> <td align=right>234</td> <td align=left></td> <td align=right>50.0;9</td> <td align=left>18</td> <td align=right>60</td> <td align=right>223</td> <td align=right>232</td> <td align=right></td> <td align=left>45.0;10</td> <td align=left>19</td> <td align=left>69</td> <td align=left>224</td> <td align=left>233</td> <td align=right></td> <td align=left>45.0</td> </tr> <tr> <td align=right>8</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>GGATTCACCTTCAGTAGCTACTGGATGCACTGGGTCCGCCAAGCTCCAGGGAAGGGGCTGGTGTGGGTCTCACGTATTAATAGTGATGGGAGTATCACAAGCTACGCGGACTCCGTGAAGGGCCGATTCACCATCTCCAGAGACAACGCCAAGAGCACGCTGTATCTGCAAATGAACAGTCTGAGAGCCGAGGACACGGCTGTGTATTACTGTGCCCCATACGGTAGTAACCACATTTTCGACTACTGGGGCCAGGGA</td> <td align=left>IIGIIIIIIIIIGIIIIIIIIIIIIIGIIIIIIIIIIIIIGIIIIIIIIIGIIIIII&lt;:GIGGGGIGIIIIIIIIIIGGIIIIGGIIIIIIGGGGGGGIIGGIGGGIGIIIGGGGGGGGGIIIIIIIIIGIIIIGIIIIIIGIIIIGIIIGIIIGGGIGIGGGGGGIGGIGGGIIIIIGGGGIIGIGIIGGGGGGGGGAGGIIIIIIIIIIIIIGIIIGGIIIIIGIIIIIIIGGGIGIGGIIGIGIIIIGGAIIIII</td> <td align=left>IGHV3-74*00(2152)</td> <td align=left>IGHD4-23<em>00(46),IGHD1-14</em>00(41),IGHD3-22*00(41)</td> <td align=left>IGHJ4*00(221)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>290</td> <td align=left>316</td> <td align=right>0</td> <td align=right>215</td> <td align=right>SG169TSA229G</td> <td align=left>2092.0</td> <td align=right>23</td> <td align=left>35</td> <td align=right>57</td> <td align=left>219</td> <td align=right>231</td> <td align=left>SG29A</td> <td align=right>46.0;12</td> <td align=left>26</td> <td align=right>51</td> <td align=right>218</td> <td align=right>232</td> <td align=right>DC15I21G</td> <td align=left>41.0;3</td> <td align=left>14</td> <td align=left>93</td> <td align=left>223</td> <td align=left>234</td> <td align=right>SA6G</td> <td align=left>41.0</td> </tr> <tr> <td align=right>9</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGTTCCAACATCGGAAGCAATACTGTAAACTGGTCTGGCACCTCAGCCTCCCTGGCCATCAGTGGTCTCCAGTCTGAGGATGAAGCTGATTATTACTGTGCAGTCTGGGATGACAAGCTGCGTGGTCGGATAATCGGCGGA</td> <td align=left>IIIIIIGIGG&lt;GAG&lt;G&lt;&lt;G&lt;GGGIIGIIIIIIIIIIIGGGIIIIGGGIGIIIGIIIIIGG&lt;&lt;&lt;&lt;&lt;GGIIIGGGGGGGGIIIIIIIIIIGIGIIIIIIIIIIIIGIIIIIGGIGGIGGGGGIIIIIIIIIIIGIIIGIIGGG</td> <td align=left>IGLV1-44*00(627)</td> <td align=left>nan</td> <td align=left>IGLJ2<em>00(223),IGLJ3</em>00(222)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>109</td> <td align=left>316</td> <td align=right>0</td> <td align=right>34</td> <td align=right>SC77TST92C</td> <td align=left>282.0</td> <td align=right>nan</td> <td align=left>25</td> <td align=right>36</td> <td align=left>58</td> <td align=right>130</td> <td align=left>141</td> <td align=right>ST27A</td> <td align=left>81.0;28</td> <td align=right>36</td> <td align=right>58</td> <td align=right>133</td> <td align=right>141</td> <td align=left></td> <td align=left>80.0</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>AGTTCCAACATCGGAAGCAATACT</td> <td align=left>27</td> </tr> <tr> <td align=right>10</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGATATTGGGGGTTATAACCATGTCTCCTGGTACCAACAACACCCAGGCAAAGCCCCCAAACTCATCATTTATGAGCACCCGTTATGTCGTCGGAACA</td> <td align=left>GIIIIGGIG&lt;&lt;GAGGGIIIGGIIGGGGGIIIIIIIIIIGIIIIIIGIIIIIIIIIIIIGGIIIIIIGIIIGIIGIIIIIGGGGGGGGGGGIGIIIIIIIIIGG</td> <td align=left>IGLV2-23<em>00(1357),IGLV2-14</em>00(1318)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(264)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>156</td> <td align=left>317</td> <td align=right>0</td> <td align=right>81</td> <td align=right>SG84ASA90GST100ASG119ASG146C</td> <td align=left>665.0;75</td> <td align=right>155</td> <td align=left>317</td> <td align=right>0</td> <td align=left>80</td> <td align=right>SC83TSG84AST89GST99CSG119ASG146C</td> <td align=left>626.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=right>58</td> <td align=right>87</td> <td align=right>103</td> <td align=left>ST27GST35A</td> <td align=left>102.0</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>AGCAGTGATATTGGGGGTTATAACCAT</td> <td align=left>27</td> </tr> <tr> <td align=right>11</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGATATTGGGAGGTCTGGCAACACGTCCTCCCTGACAATGTTTGGGCTCCAGGGTGAGGACGAGGCTGATAATTACTGCTACTCATATGCGTTTAATTGGACGTCTTCTAAACT</td> <td align=left>IGGIIIGGIGIGIGGGIGIIGIIIIIGIIIIIIIIGGIGGGGGIIIIIIIIIIGIIIIIGGIGIIIIIIIIIIIGGGGIIIIIIGIIIIIIGIIGIIIIIIIIIIIIII7AAGGAG&lt;&lt;7</td> <td align=left>IGLV2-23*00(167)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(224)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>92</td> <td align=left>317</td> <td align=right>0</td> <td align=right>17</td> <td align=right>SG84A</td> <td align=left>141.0</td> <td align=right>nan</td> <td align=left>24</td> <td align=right>36</td> <td align=left>58</td> <td align=right>107</td> <td align=left>119</td> <td align=right>SG30TSG31A</td> <td align=left>62.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>12</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACGTTGGTGGTTATAACTATGTCTCCTGGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCACCTCATATACAAGCAGTGCGAATTATGTCTTCGGAACT</td> <td align=left>GIIIIIIGGGGGIIIIIGIGIGIIIIIIIIIIIIIIIIIGIIIIIIIIIIIIIIIIIIIG&lt;&lt;GGGIIIIIGIIIGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGGGGGIIIIIIIIIIIIIIGIIIIIGIAG</td> <td align=left>IGLV2-14*00(808)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(370)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>112</td> <td align=left>317</td> <td align=right>0</td> <td align=right>37</td> <td align=right></td> <td align=left>370.0</td> <td align=right>nan</td> <td align=left>18</td> <td align=right>36</td> <td align=left>58</td> <td align=right>123</td> <td align=left>141</td> <td align=right></td> <td align=left>180.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGTGACGTTGGTGGTTATAACTAT</td> <td align=left>38</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>13</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACGTTGGTGGTTATAACTATGTCTCCTGGTACCAACAGCACCCAGGCAAAGCCCCCAAACTCATGATTTATGATGTCAGTAATCGGCCCTCAGGGGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCAGCTCATATACAAGCAGCAGCACTCTCTATGTCTTCGGAACT</td> <td align=left>&lt;&lt;G&lt;&lt;&lt;GGGAGGGGIGIIAGAGIEIIIIIIIIIIGGIIGGIIIIIIIGGIGGIIIGIGIGGGGGGGAAGGGGGIIGIGGGGGAAG:GIIGGIIIGIIIGIIGGGGGGGIIIIGGGIGIIGGGGGIIIIIGGGIIGGIIIIIIIIIIIIIIIIIIIIIIIIIIIGGGG&lt;&lt;&lt;&lt;GIIGIIIIGIIIGIIIIIIIIIGIIIGIIIIIIIIIIGIIIIIIIGGIGIIIIIIIIIIIIIIIIG</td> <td align=left>IGLV2-14*00(2850)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(190)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>297</td> <td align=left>317</td> <td align=right>0</td> <td align=right>222</td> <td align=right></td> <td align=left>2220.0</td> <td align=right>nan</td> <td align=left>21</td> <td align=right>36</td> <td align=left>58</td> <td align=right>222</td> <td align=left>237</td> <td align=right></td> <td align=left>150.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGTGACGTTGGTGGTTATAACTAT</td> <td align=left>27</td> <td align=left>GTCTCCTGGTACCAACAGCACCCAGGCAAAGCCCCCAAACTCATGATTTAT</td> <td align=left>32</td> <td align=left>GATGTCAGT</td> <td align=left>25</td> <td align=right>AATCGGCCCTCAGGGGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTAC</td> <td align=left>27</td> </tr> <tr> <td align=right>14</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACGTTGGGGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCGACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCAACTCATATACAAGCAGCAGCACTTATGTCTTCGGAACT</td> <td align=left>GGIIIIIIIIGGGGGIIGIIIIIIIIIIIIIIIIIIIIIGGGGGIIIIIIIIGG:IIIIIGGIGGGGGGGGIGGGGAIGGGGGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGGIIGGGGGIGIIIIIIGAA</td> <td align=left>IGLV2-14*00(845)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(281)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>294</td> <td align=left>317</td> <td align=right>0</td> <td align=right>132</td> <td align=right>DT89DG90DG91DT92DT93DA94DT95DA96DA97DC98DT99DA100DT101DG102DT103DC104DT105DC106DC107DT108DG109DG110DT111DA112DC113DC114DA115DA116DC117DA118DG119DC120DA121DC122DC123DC124DA125DG126DG127DC128DA129DA130DA131DG132DC133DC134DC135DC136DC137DA138DA139DA140DC141DT142DC143DA144DT145DG146DA147DT148DT149DT150DA151DT152DG153DA154DT155DG156DT157DC158DA159DG160DT161DA162DA163DT164DC165DG166DG167DC168DC169DC170DT171DC172DA173DG174DG175SA210GSG271A</td> <td align=left>254.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=left>58</td> <td align=right>131</td> <td align=left>147</td> <td align=right></td> <td align=left>160.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGTGACGTTGG</td> <td align=left>38</td> <td align=left>nan</td> <td align=left>0</td> <td align=left>nan</td> <td align=left>0</td> <td align=right>GGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCGACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTAC</td> <td align=left>25</td> </tr> <tr> <td align=right>15</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACGTTGGGATCTCGGATCGCTTCTCTGGTTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCAGCTCGTTTACAACCAGCGACACTTATGTCTTCGGAAGT</td> <td align=left>IGIIGGGIGIGIGGGGIGGGIIIIIIGGIIIIIIIIGGGGIGGGIIIIIIIIIIIIIIIIG&lt;GGGGGGIIGGGGGIGIIGIIGGGIIGIIIGIIIIIGIGIIIIIIIIIIIIIIIGIIGIIIIIIIIIGIGIIIGG:IIIIIIIGIG</td> <td align=left>IGLV2-14*00(721)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(271)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>89</td> <td align=left>317</td> <td align=right>0</td> <td align=right>14</td> <td align=right></td> <td align=left>140.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=left>58</td> <td align=right>131</td> <td align=left>147</td> <td align=right>SC34G</td> <td align=left>131.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>16</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACATTGGTCTGGCAACACGGCCTCCCTGACCATCTCTGGACTCCAGGCTGAGGACGAGGCTGATTATTACTGCAGCTCATATACAAGCAGCAGCACTGATGTCTTCGGAACT</td> <td align=left>IIIIIIIIGGGGGGGGIIIIGIIIIIIIIIIIIIIGIIIIIIIGGGIIIIGGGGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGGIIIIIIIGG&lt;G</td> <td align=left>IGLV2-14*00(416)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(302)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>90</td> <td align=left>317</td> <td align=right>0</td> <td align=right>15</td> <td align=right>SG84A</td> <td align=left>121.0</td> <td align=right>nan</td> <td align=left>22</td> <td align=right>36</td> <td align=left>58</td> <td align=right>105</td> <td align=left>119</td> <td align=right></td> <td align=left>140.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> <tr> <td align=right>17</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGTGACATTGGGGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTACTGCACCTCATGGACAACCAGCACCACTATGATATTCGGCGGA</td> <td align=left>II&lt;IIGIIGG&lt;&lt;&lt;GGGGIIIIIIIGIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIGIIIGIGGGGGIGIGIGGGGGIIIIIIGIIIIIIIIIIIIIIIIIIIIGIIIIIIIIIIIIIGGGGIIGIIIIIIIIIIGGGG&lt;IIGGGIA</td> <td align=left>IGLV2-14*00(859)</td> <td align=left>nan</td> <td align=left>IGLJ2*00(164)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>294</td> <td align=left>317</td> <td align=right>0</td> <td align=right>132</td> <td align=right>SG84ADT89DG90DG91DT92DT93DA94DT95DA96DA97DC98DT99DA100DT101DG102DT103DC104DT105DC106DC107DT108DG109DG110DT111DA112DC113DC114DA115DA116DC117DA118DG119DC120DA121DC122DC123DC124DA125DG126DG127DC128DA129DA130DA131DG132DC133DC134DC135DC136DC137DA138DA139DA140DC141DT142DC143DA144DT145DG146DA147DT148DT149DT150DA151DT152DG153DA154DT155DG156DT157DC158DA159DG160DT161DA162DA163DT164DC165DG166DG167DC168DC169DC170DT171DC172DA173DG174DG175SG271CSA277GST278GSG283CSG289C</td> <td align=left>138.0</td> <td align=right>nan</td> <td align=left>22</td> <td align=right>36</td> <td align=left>58</td> <td align=right>133</td> <td align=left>147</td> <td align=right>SG24A</td> <td align=left>111.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGTGACATTGG</td> <td align=left>27</td> <td align=left>nan</td> <td align=left>0</td> <td align=left>nan</td> <td align=left>0</td> <td align=right>GGTTTCTAATCGCTTCTCTGGCTCCAAGTCTGGCAACACGGCCTCCCTGACCATCTCTGGGCTCCAGGCTGAGGACGAGGCTGATTATTAC</td> <td align=left>38</td> </tr> <tr> <td align=right>18</td> <td align=right>1</td> <td align=right>0.047619</td> <td align=left>AGCAGAGATATTGGGACTTCTAACCTTGTCTCCTGGTACCAACAGTATCCAGGTATTTCTACTTCCCTTTTTGTCTTCGGAACA</td> <td align=left>IIIIGIIIG&lt;GGGGG&lt;GGGIIIIGIIIIIIIIIIIIIGIIIGIIIIIIIIIIIIIIIIIIIGIIIIGIGGIIIIIIIIIIIGG&lt;</td> <td align=left>IGLV2-23*00(1048)</td> <td align=left>nan</td> <td align=left>IGLJ1*00(264)</td> <td align=right>nan</td> <td align=left>75</td> <td align=left>128</td> <td align=left>317</td> <td align=right>0</td> <td align=right>53</td> <td align=right>ST80ASG84ASG91CSA94CSC120TSC122T</td> <td align=left>356.0</td> <td align=right>nan</td> <td align=left>20</td> <td align=right>36</td> <td align=left>58</td> <td align=right>68</td> <td align=left>84</td> <td align=right>SA22TST35A</td> <td align=left>102.0</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>nan</td> <td align=right>AGCAGAGATATTGGGACTTCTAACCTT</td> <td align=left>27</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=left>nan</td> <td align=right>nan</td> <td align=left>nan</td> </tr> </tbody> </table> </details> <h2 id=quality-control>Quality control</h2> <p>Now when we have all files processed lets perform Quality Control. The first thing to check is the alignment rate. That can be easily done using <a href=../../reference/mixcr-exportQc/#alignment-reports><code>mixcr exportQc align</code></a> function.</p> <div class=highlight><pre><span></span><code>mixcr exportQc align results/*.clns figs/alignQc.pdf
</code></pre></div> <p><img alt=alignQc.svg src=figs/alignQc.svg></p> <p>Here we can see a percentage of successfully aligned reads for every sample as well as the percentage of reads that failed to align for various reasons. Some samples have a good alignment rate (more than 90%). But a lot of samples had some issues that lead to an alignment as low as 10-20%. Two major reasons for that outcome according to the plot are:</p> <ul> <li>the lack of TCR/IG sequence</li> <li>absence of J hits</li> </ul> <p>Let's take one of the bad quality samples (ex. SRR8365459_HIP1_female) and examine it. To look at the reads' alignments for that sample we first will run <a href=../../reference/mixcr-align/ ><code>mixcr align</code></a> command for that sample once again, but this time we will specify additional options (<code>-OallowPartialAlignments=true -OallowNoCDR3PartAlignments=true</code>) that will preserve partially aligned reads (ex. reads that may lack J gene) and reads that lack <code>CDR3</code> sequence.</p> <div class=highlight><pre><span></span><code>mkdir -p debug
mixcr align -f <span class=se>\</span>
    --species hsa <span class=se>\</span>
    -p kAligner2_4.0 <span class=se>\</span>
    --tag-pattern <span class=s2>&quot;^N{32}(R1:*)\^N{26}(R2:*)&quot;</span> <span class=se>\</span>
    -OvParameters.geneFeatureToAlign<span class=o>=</span><span class=s2>&quot;VTranscriptWithout5UTRWithP&quot;</span> <span class=se>\</span>
    -OvParameters.parameters.floatingLeftBound<span class=o>=</span><span class=nb>false</span> <span class=se>\</span>
    -OjParameters.parameters.floatingRightBound<span class=o>=</span><span class=nb>false</span> <span class=se>\</span>
    -OallowNoCDR3PartAlignments<span class=o>=</span><span class=nb>true</span> <span class=se>\</span>
    -OallowPartialAlignments<span class=o>=</span><span class=nb>true</span> <span class=se>\</span>
    --not-aligned-R1 debug/SRR8365459_HIP1_female_notAligned_R1.fastq <span class=se>\</span>
    --not-aligned-R2 debug/SRR8365459_HIP1_female_notAligned_R2.fastq <span class=se>\</span>
    --report debug/SRR8365459_HIP1_female_debug.report <span class=se>\</span>
     raw/SRR8365459_HIP1_female_R1.fastq.gz raw/SRR8365459_HIP1_female_R2.fastq.gz <span class=se>\</span>
     debug/SRR8365459_HIP1_female_debug.vdjca
</code></pre></div> <p>Now, when we have a new <code>.vdjca</code> file let's visualize how reads cover FRs and CDRs regions for that sample.</p> <div class=highlight><pre><span></span><code>mixcr exportQc coverage <span class=se>\</span>
      debug/SRR8365459_HIP1_female_debug.vdjca <span class=se>\</span>
      figs/SRR8365459_HIP1_female_debug.vdjca.coverage.pdf
</code></pre></div> <p>This will generate three <code>.pdf</code> formatted plots: <code>R1</code> alignment, <code>R2</code> alignment and alignment of overlapped reads. These plots can tell us the percentage of reads that cover each region at a certain position. Briefly, for this sample, only those reads that overlap show a good coverage pattern.</p> <div class="tabbed-set tabbed-alternate" data-tabs=1:3><input checked=checked id=__tabbed_1_1 name=__tabbed_1 type=radio><input id=__tabbed_1_2 name=__tabbed_1 type=radio><input id=__tabbed_1_3 name=__tabbed_1 type=radio><div class=tabbed-labels><label for=__tabbed_1_1>R1</label><label for=__tabbed_1_2>R2</label><label for=__tabbed_1_3>Overlap</label></div> <div class=tabbed-content> <div class=tabbed-block> <p><img alt=SRR8365459_HIP_female_debug_coverageQc_R1.svg src=figs/SRR8365459_HIP_female_debug_coverageQc_R1.svg></p> </div> <div class=tabbed-block> <p><img alt=SRR8365459_HIP_female_debug_coverageQc_R2.svg src=figs/SRR8365459_HIP_female_debug_coverageQc_R2.svg></p> </div> <div class=tabbed-block> <p><img alt=SRR8365459_HIP_female_debug_coverageQc_Overlap.svg src=figs/SRR8365459_HIP_female_debug_coverageQc_Overlap.svg></p> </div> </div> </div> <p>Finally, we can look at raw alignments using <a href=../../reference/mixcr-exportPretty/#raw-alignments><code>mixcr exportAlignmentsPretty</code></a>.</p> <p>The function bellow will generate a <code>.txt</code> human-readable file with alignments. We use parameter <code>--skip 1000</code> to skip first 1000 reads, as first reads usually have bad quality, and <code>--limit 100</code> will export only 100 alignments as we usually don't need to examine every alignment to see the issue. </p> <div class=highlight><pre><span></span><code>mixcr exportAlignmentsPretty
    --skip <span class=m>1000</span> <span class=se>\</span>
    --limit <span class=m>100</span> <span class=se>\</span>
    debug/SRR8365459_HIP1_female_debug.vdjca <span class=se>\</span>
    debug/SRR8365459_HIP1_female_debug.alignments.txt
</code></pre></div> <p>Bellow you can see a few alignments from the generated file. The first one is an example of well aligned read.</p> <div class=highlight><pre><span></span><code>&gt;&gt;&gt; Read ids: <span class=m>1840</span>


                                     FR1&gt;&lt;CDR1              CDR1&gt;&lt;FR2                               
                _ L  R  L  S  C  A  A  S  G  F  T  L  S  D  Y  Y  M  S  W  I  R  Q  A  P  G  K      
    Quality     <span class=m>77767826778888888888888887788888888888878888888888888887888778888888888888887878</span>    
    Target0   <span class=m>0</span> CCTGAGACTCTCCTGTGCAGCCTCTGGATTCACCTTGAGTGACTACTACATGAGCTGGATCCGCCAGGCTCCAGGGAAGG <span class=m>79</span>   Score
IGHV3-11*00 <span class=m>107</span> cctgagactctcctgtgcagcctctggattcaccttCagtgactactacatgagctggatccgccaggctccagggaagg <span class=m>186</span>  <span class=m>2102</span>

                                FR2&gt;&lt;CDR2              CDR2&gt;&lt;FR3                                    
                G  L  E  W  V  S  H  I  S  G  S  G  N  T  I  D  Y  A  D  S  V  K  G  R  F  T  I     
    Quality     <span class=m>88788887778888888888887788888888888888888888888888888888887888888888888888888888</span>    
    Target0  <span class=m>80</span> GGCTGGAGTGGGTTTCACACATTAGTGGCAGTGGTAATACCATAGACTACGCAGACTCTGTGAAGGGCCGATTCACCATC <span class=m>159</span>  Score
IGHV3-11*00 <span class=m>187</span> ggctggagtgggtttcaTacattagtAgTagtggtaGtaccataTactacgcagactctgtgaagggccgattcaccatc <span class=m>266</span>  <span class=m>2102</span>

                                                                                       FR3&gt;&lt;CDR3    
                 S  R  D  N  A  K  N  S  L  Y  L  Q  M  N  S  L  R  D  D  D  T  A  V  F  Y  C  A    
    Quality     <span class=m>88888888888888888888888888888888888888888888888888888888888888888888888887888788</span>    
    Target0 <span class=m>160</span> TCCAGGGACAACGCCAAGAACTCGCTCTATCTGCAAATGAACAGCCTGAGAGACGACGACACGGCCGTGTTTTATTGTGC <span class=m>239</span>  Score
IGHV3-11*00 <span class=m>267</span> tccagggacaacgccaagaactcActGtatctgcaaatgaacagcctgagagCcgaGgacacggccgtgtAttaCtgtgc <span class=m>346</span>  <span class=m>2102</span>

                   V&gt;  &lt;D     D&gt;   &lt;J  CDR3&gt;&lt;FR4                       FR4&gt;             
                  R  G  R  Y  A  L  D  Y  W  G  Q  G  T  R  V  T  V  S  S _             
    Quality     <span class=m>88878788888888788888888888888888888888888888888888888888888888877776</span>    
    Target0 <span class=m>240</span> GAGAGGCCGTTATGCCCTAGATTATTGGGGCCAGGGAACCCGGGTCACCGTCTCCTCAGGTAAGCCCC <span class=m>307</span>  Score
IGHV3-11*00 <span class=m>347</span> gagag                                                                <span class=m>351</span>  <span class=m>2102</span>
IGHD3-16*00  <span class=m>65</span>        cgttatAcc                                                     <span class=m>73</span>   <span class=m>31</span>
 IGHD2-2*00  <span class=m>56</span>           tatgcc                                                     <span class=m>61</span>   <span class=m>30</span>
IGHD3-10*00  <span class=m>61</span>        cgttat                                                        <span class=m>66</span>   <span class=m>30</span>
   IGHJ4*00  <span class=m>28</span>                    gaCtaCtggggccagggaacccTggtcaccgtctcctcag          <span class=m>67</span>   <span class=m>313</span>
   IGHJ5*00  <span class=m>37</span>                          tggggccagggaacccTggtcaccgtctcctcag          <span class=m>70</span>   <span class=m>311</span>
</code></pre></div> <p>Now, the following alignment is a troubled one. And for this particular sample the majority of alignments look similar. We can see that one read does not align to any reference sequences and has a lot of low-quality nucleotides.</p> <div class=highlight><pre><span></span><code>&gt;&gt;&gt; Read ids: <span class=m>1853</span>



Quality   <span class=m>25762677888767572572527252276555757625555572625275777255225225525252725222252552</span>   
Target0 <span class=m>0</span> CGGCATTCCTGCTGAAACGAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTCACATTAAA <span class=m>79</span>  Score


Quality    <span class=m>57222255256566257772557225555565256777672262255775556255567762277262262677727762</span>    
Target0 <span class=m>80</span> AAAAAAAACCAGCAGTGATGTTGGCAGTTATGACTATGTCTCCTGGTACCAACAGCACCCAGGCACAGTCTCCAAACCCA <span class=m>159</span>  Score


Quality     <span class=m>42222547622525426677662772660000000000000000000000000000000000000000000000000000</span>    
Target0 <span class=m>160</span> TGACGTACAATGACAATACTCAGCCCTCAGGGGTCCCTGATCGATTCTCTGGCTTCAAGTCTGGCAATACGCCCTCCATG <span class=m>239</span>  Score


Quality     <span class=m>00000000000</span>    
Target0 <span class=m>240</span> ACCATCCTTAG <span class=m>250</span>  Score

                                                                                        CDR2&gt;&lt;F    
                            CDR1&gt;&lt;FR2                                           FR2&gt;&lt;CDR2          
                F  G  S  Y  V  Y  V  S  W  Y  Q  Q  H  S  S  T  V  P  K  P  M  I  D  N  V  N  T    
   Quality     <span class=m>00000000000000000000000000000000000000000006247675667452527762544226662675276772</span>    
   Target1   <span class=m>0</span> TTTGGGAGTTATGTCTATGTCTCCTGGTACCAACAGCACTCAAGCACAGTCCCCAAACCCATGATCGACAATGTCAATAC <span class=m>79</span>   Score
IGLV2-5*00 <span class=m>141</span>  ttgggagttatgActatgtctcctggtaccaacagcacCcaGgcacagtccccaaacccatgatcTacaatgtcaatac <span class=m>219</span>  <span class=m>1445</span>

               R3                                                                                  

                 Q  P  S  G  V  P  D  R  F  S  G  S  K  S  G  N  T  A  S  M  T  I  S  G  L  *      
   Quality     <span class=m>22262275272725267265252267676726267252555266222725267626677675277777757552652222</span>    
   Target1  <span class=m>80</span> TCAGCCCTCAGGGGTCCCTGATCGTTTCTCTGGCTCCAAGTCTGGCAATACGGCCTCCATGACCATCTCTGGACTCTAGG <span class=m>159</span>  Score
IGLV2-5*00 <span class=m>220</span> tcagccctcaggggtccctgatcgtttctctggctccaagtctggcaatacggcctccatgaccatctctggactcCagg <span class=m>299</span>  <span class=m>1445</span>



            V  Y  K  Q  K  T  A  Y  E  C  Q  S  R  S  R  H  S  C  *  T  A  L  P  I  S  A  F     
Quality     <span class=m>55222252675222525222752252222225622525266672252275226775565552222577665575255625</span>    
Target1 <span class=m>160</span> TTTACAAGCAGAAGACGGCATACGAGTGCCAGTCCCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCTCGGCATTC <span class=m>239</span>  Score



             L  L  T _     
Quality     <span class=m>72552252777</span>    
Target1 <span class=m>240</span> CTGCTGACCCG <span class=m>250</span>  Score
</code></pre></div> <p>Another quality report we should investigate is chain abundance plot.</p> <div class=highlight><pre><span></span><code>mixcr exportQc chainUsage <span class=se>\</span>
    results/*.clns <span class=se>\</span>
    figs/chainUsage.pdf
</code></pre></div> <p><img alt=chainUsage.svg src=figs/chainUsage.svg></p> <p>From that plot we can see another issue. According to the publication, the data was generated using a V and J primers multiplex protocol in such a way that every sample should have sequences for both heavy and light IG chains. But we see, that most samples have only one of the chains present in the sample, and those samples that have both still have a unexpected distribution, which has to be about 50\50, as every cell has both chains.</p> <h2 id=full-length-clonotype-assembly>Full-length clonotype assembly</h2> <p>Biomed2 BCR protocol allows to recover a broader BCR receptor sequence then just <code>CDR3</code> region. According to the protocol, forward primers are located in <code>FR1</code> region, thus we can safely use an assembling feature that starts from <code>CDR1</code> and be sure that no primers will affect the original sequence. The reverse primers are located in <code>FR4</code> region very close to <code>CDR3</code>, thus there is not much left from to include in clone assembly.</p> <p>Taking into account what is mentioned above, the longest possible assembling feature for this protocol is <code>"{CDR1Begin:CDR3End}"</code>.</p> <p>MiXCR has a specific preset to obtain full-length BCR clones with Biomed2 protocol:</p> <div class=highlight><pre><span></span><code>mixcr analyze biomed2-human-bcr-full-length<span class=se>\</span>
    raw/SRR8365468_HIP2_male_R1.fastq.gz <span class=se>\</span>
    raw/SRR8365468_HIP2_male_R2.fastq.gz <span class=se>\</span>
    results/SRR8365468_HIP2_male
</code></pre></div> <p>The <code>mixcr assemble</code> step in this preset differs from the one above in the following manner:</p> <div class=highlight><pre><span></span><code>mixcr assemble <span class=se>\</span>
    -OassemblingFeatures<span class=o>=</span><span class=s2>&quot;{CDR1Begin:CDR3End}&quot;</span> <span class=se>\</span>
    <span class=sb>`</span>-OseparateByJ<span class=o>=</span><span class=nb>true</span><span class=sb>`</span> <span class=se>\</span>
    --report results/SRR8365468_HIP2_male.report <span class=se>\</span>
    --json-report results/SRR8365468_HIP2_male.json <span class=se>\</span>
    results/SRR8365468_HIP2_male.vdjca <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns
</code></pre></div> <dl> <dt><code>-OassemblingFeatures="{CDR1Begin:CDR3End}"</code></dt> <dd>sets the assembling feature to the region which starts from <code>CDR1Begin</code> and ends at the end of <code>CDR3</code>.</dd> </dl> <p>Notice that we do not use <code>-OseparateByV=true</code> in this case because assembling feature already covers most of the V regions, thus in case if clones have identical <code>CDR3</code> they will still be separated.</p> <h2 id=reports>Reports</h2> <p>Finally, MiXCR provides a very convenient way to look at the reports generated at ech step. Every <code>.vdjca</code>, <code>.clns</code> and <code>.clna</code> file holds all the reports for every MiXCR function that has been applied to this sample. E.g. in our case <code>.clns</code> file contains reports for <code>mixcr align</code> and <code>mixcr assemble</code>. To output this report use <a href=../../reference/mixcr-exportReports/ ><code>mixcr exportReports</code></a> as shown bellow. Note <code>--json</code> parameter will output a JSON-formatted report.</p> <div class=highlight><pre><span></span><code>mixcr exportReports <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns <span class=se>\</span>
    figs/SRR8365468_HIP2_male.report.txt
</code></pre></div> <div class=highlight><pre><span></span><code>mixcr exportReports <span class=se>\</span>
    --json <span class=se>\</span>
    results/SRR8365468_HIP2_male.clns <span class=se>\</span>
    figs/SRR8365468_HIP2_male.report.txt
</code></pre></div> <details> <summary>Show report file</summary> <div class="tabbed-set tabbed-alternate" data-tabs=2:2><input checked=checked id=__tabbed_2_1 name=__tabbed_2 type=radio><input id=__tabbed_2_2 name=__tabbed_2 type=radio><div class=tabbed-labels><label for=__tabbed_2_1><code>.txt</code></label><label for=__tabbed_2_2><code>.json</code></label></div> <div class=tabbed-content> <div class=tabbed-block> <div class=highlight><pre><span></span><code><span class=o>==============</span> Align <span class=nv>Report</span> <span class=o>==============</span>
Input file<span class=o>(</span>s<span class=o>)</span>: /raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R1.fastq.gz,/raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R2.fastq.gz
Output file<span class=o>(</span>s<span class=o>)</span>: results/SRR8365468_HIP2_male.vdjca
Version: <span class=p>;</span> <span class=nv>built</span><span class=o>=</span>Mon Sep <span class=m>26</span> <span class=m>10</span>:55:18 CEST <span class=m>2022</span><span class=p>;</span> <span class=nv>rev</span><span class=o>=</span>8c998df1ab<span class=p>;</span> <span class=nv>lib</span><span class=o>=</span>repseqio.v2.0
Command line arguments: --report results/SRR8365468_HIP2_male.align.report.txt --json-report results/SRR8365468_HIP2_male.align.report.json --preset local:biomed2-human-bcr-full-length +limitInput <span class=m>100000</span> /raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R1.fastq.gz /raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R2.fastq.gz results/SRR8365468_HIP2_male.vdjca
Analysis time: 0ns
Total sequencing reads: <span class=m>100000</span>
Successfully aligned reads: <span class=m>760</span> <span class=o>(</span><span class=m>0</span>.76%<span class=o>)</span>
Paired-end alignment conflicts eliminated: <span class=m>98</span> <span class=o>(</span><span class=m>0</span>.1%<span class=o>)</span>
Alignment failed, no hits <span class=o>(</span>not TCR/IG?<span class=o>)</span>: <span class=m>82357</span> <span class=o>(</span><span class=m>82</span>.36%<span class=o>)</span>
Alignment failed because of absence of V hits: <span class=m>51</span> <span class=o>(</span><span class=m>0</span>.05%<span class=o>)</span>
Alignment failed because of absence of J hits: <span class=m>16810</span> <span class=o>(</span><span class=m>16</span>.81%<span class=o>)</span>
No target with both V and J alignments: <span class=m>22</span> <span class=o>(</span><span class=m>0</span>.02%<span class=o>)</span>
Overlapped: <span class=m>25788</span> <span class=o>(</span><span class=m>25</span>.79%<span class=o>)</span>
Overlapped and aligned: <span class=m>547</span> <span class=o>(</span><span class=m>0</span>.55%<span class=o>)</span>
Alignment-aided overlaps: <span class=m>20</span> <span class=o>(</span><span class=m>3</span>.66%<span class=o>)</span>
Overlapped and not aligned: <span class=m>25241</span> <span class=o>(</span><span class=m>25</span>.24%<span class=o>)</span>
No CDR3 parts alignments, percent of successfully aligned: <span class=m>2</span> <span class=o>(</span><span class=m>0</span>.26%<span class=o>)</span>
Partial aligned reads, percent of successfully aligned: <span class=m>9</span> <span class=o>(</span><span class=m>1</span>.18%<span class=o>)</span>
V gene chimeras: <span class=m>9</span> <span class=o>(</span><span class=m>0</span>.01%<span class=o>)</span>
IGH chains: <span class=m>44</span> <span class=o>(</span><span class=m>5</span>.79%<span class=o>)</span>
IGH non-functional: <span class=m>1</span> <span class=o>(</span><span class=m>2</span>.27%<span class=o>)</span>
IGK chains: <span class=m>496</span> <span class=o>(</span><span class=m>65</span>.26%<span class=o>)</span>
IGK non-functional: <span class=m>15</span> <span class=o>(</span><span class=m>3</span>.02%<span class=o>)</span>
IGL chains: <span class=m>220</span> <span class=o>(</span><span class=m>28</span>.95%<span class=o>)</span>
IGL non-functional: <span class=m>11</span> <span class=o>(</span><span class=m>5</span>%<span class=o>)</span>
Realigned with forced non-floating bound: <span class=m>148464</span> <span class=o>(</span><span class=m>148</span>.46%<span class=o>)</span>
Realigned with forced non-floating right bound <span class=k>in</span> left read: <span class=m>997</span> <span class=o>(</span><span class=m>1</span>%<span class=o>)</span>
Realigned with forced non-floating left bound <span class=k>in</span> right read: <span class=m>997</span> <span class=o>(</span><span class=m>1</span>%<span class=o>)</span>
<span class=o>==============</span> Assemble <span class=nv>Report</span> <span class=o>==============</span>
Input file<span class=o>(</span>s<span class=o>)</span>: results/SRR8365468_HIP2_male.vdjca
Output file<span class=o>(</span>s<span class=o>)</span>: results/SRR8365468_HIP2_male.clns
Version: <span class=p>;</span> <span class=nv>built</span><span class=o>=</span>Mon Sep <span class=m>26</span> <span class=m>10</span>:55:18 CEST <span class=m>2022</span><span class=p>;</span> <span class=nv>rev</span><span class=o>=</span>8c998df1ab<span class=p>;</span> <span class=nv>lib</span><span class=o>=</span>repseqio.v2.0
Command line arguments: --report results/SRR8365468_HIP2_male.assemble.report.txt --json-report results/SRR8365468_HIP2_male.assemble.report.json results/SRR8365468_HIP2_male.vdjca results/SRR8365468_HIP2_male.clns
Analysis time: 0ns
Final clonotype count: <span class=m>19</span>
Average number of reads per clonotype: <span class=m>1</span>.11
Reads used <span class=k>in</span> clonotypes, percent of total: <span class=m>21</span> <span class=o>(</span><span class=m>0</span>.02%<span class=o>)</span>
Reads used <span class=k>in</span> clonotypes before clustering, percent of total: <span class=m>21</span> <span class=o>(</span><span class=m>0</span>.02%<span class=o>)</span>
Number of reads used as a core, percent of used: <span class=m>21</span> <span class=o>(</span><span class=m>100</span>%<span class=o>)</span>
Mapped low quality reads, percent of used: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads clustered <span class=k>in</span> PCR error correction, percent of used: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads pre-clustered due to the similar VJC-lists, percent of used: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads dropped due to the lack of a clone sequence, percent of total: <span class=m>610</span> <span class=o>(</span><span class=m>0</span>.61%<span class=o>)</span>
Reads dropped due to a too short clonal sequence, percent of total: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads dropped due to low quality, percent of total: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Reads dropped due to failed mapping, percent of total: <span class=m>129</span> <span class=o>(</span><span class=m>0</span>.13%<span class=o>)</span>
Reads dropped with low quality clones, percent of total: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
Clonotypes eliminated by PCR error correction: <span class=m>0</span>
Clonotypes dropped as low quality: <span class=m>0</span>
Clonotypes pre-clustered due to the similar VJC-lists: <span class=m>0</span>
IGH chains: <span class=m>7</span> <span class=o>(</span><span class=m>36</span>.84%<span class=o>)</span>
IGH non-functional: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
IGL chains: <span class=m>12</span> <span class=o>(</span><span class=m>63</span>.16%<span class=o>)</span>
IGL non-functional: <span class=m>0</span> <span class=o>(</span><span class=m>0</span>%<span class=o>)</span>
</code></pre></div> </div> <div class=tabbed-block> <div class=highlight><pre><span></span><code><span class=p>{</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;type&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;alignerReport&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;commandLine&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;--report results/SRR8365468_HIP2_male.align.report.txt --json-report results/SRR8365468_HIP2_male.align.report.json --preset local:biomed2-human-bcr-full-length +limitInput 100000 /raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R1.fastq.gz /raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R2.fastq.gz results/SRR8365468_HIP2_male.vdjca&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;inputFiles&quot;</span><span class=p>:</span><span class=w> </span><span class=p>[</span><span class=w></span>
<span class=w>    </span><span class=s2>&quot;/raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R1.fastq.gz&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=s2>&quot;/raw/PRJNA511481-HIP/BIOMED2/SRR8365468_HIP2_male_R2.fastq.gz&quot;</span><span class=w></span>
<span class=w>  </span><span class=p>],</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;outputFiles&quot;</span><span class=p>:</span><span class=w> </span><span class=p>[</span><span class=w></span>
<span class=w>    </span><span class=s2>&quot;results/SRR8365468_HIP2_male.vdjca&quot;</span><span class=w></span>
<span class=w>  </span><span class=p>],</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;version&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;; built=Mon Sep 26 10:55:18 CEST 2022; rev=8c998df1ab; lib=repseqio.v2.0&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;trimmingReport&quot;</span><span class=p>:</span><span class=w> </span><span class=kc>null</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;totalReadsProcessed&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>100000</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;aligned&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>760</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;notAligned&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>99240</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;notAlignedReasons&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;VAndJOnDifferentTargets&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>22</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;NoVHits&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>51</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;NoJHits&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>16810</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;LowTotalScore&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;NoBarcode&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;NoHits&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>82357</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;NoCDR3Parts&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=w></span>
<span class=w>  </span><span class=p>},</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;chimeras&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;overlapped&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>25788</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;alignmentAidedOverlaps&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>20</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;overlappedAligned&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>547</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;overlappedNotAligned&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>25241</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;pairedEndAlignmentConflicts&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>98</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;vChimeras&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>9</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;jChimeras&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;chainUsage&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;type&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;chainUsage&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;chimeras&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>760</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;chains&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>      </span><span class=nt>&quot;IGH&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>44</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;nonFunctional&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>1</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;isOOF&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;hasStops&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>1</span><span class=w></span>
<span class=w>      </span><span class=p>},</span><span class=w></span>
<span class=w>      </span><span class=nt>&quot;IGK&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>496</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;nonFunctional&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>15</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;isOOF&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>10</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;hasStops&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>5</span><span class=w></span>
<span class=w>      </span><span class=p>},</span><span class=w></span>
<span class=w>      </span><span class=nt>&quot;IGL&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>220</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;nonFunctional&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>11</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;isOOF&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>7</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;hasStops&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>4</span><span class=w></span>
<span class=w>      </span><span class=p>}</span><span class=w></span>
<span class=w>    </span><span class=p>}</span><span class=w></span>
<span class=w>  </span><span class=p>},</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;realignedWithForcedNonFloatingBound&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>148464</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;realignedWithForcedNonFloatingRightBoundInLeftRead&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>997</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;realignedWithForcedNonFloatingLeftBoundInRightRead&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>997</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;noCDR3PartsAlignments&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>2</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;partialAlignments&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>9</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;tagReport&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;type&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;tagReport&quot;</span><span class=w></span>
<span class=w>  </span><span class=p>}</span><span class=w></span>
<span class=p>}</span><span class=w></span>
<span class=p>{</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;type&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;assemblerReport&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;commandLine&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;--report results/SRR8365468_HIP2_male.assemble.report.txt --json-report results/SRR8365468_HIP2_male.assemble.report.json results/SRR8365468_HIP2_male.vdjca results/SRR8365468_HIP2_male.clns&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;inputFiles&quot;</span><span class=p>:</span><span class=w> </span><span class=p>[</span><span class=w></span>
<span class=w>    </span><span class=s2>&quot;results/SRR8365468_HIP2_male.vdjca&quot;</span><span class=w></span>
<span class=w>  </span><span class=p>],</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;outputFiles&quot;</span><span class=p>:</span><span class=w> </span><span class=p>[</span><span class=w></span>
<span class=w>    </span><span class=s2>&quot;results/SRR8365468_HIP2_male.clns&quot;</span><span class=w></span>
<span class=w>  </span><span class=p>],</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;version&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;; built=Mon Sep 26 10:55:18 CEST 2022; rev=8c998df1ab; lib=repseqio.v2.0&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;preCloneAssemblerReport&quot;</span><span class=p>:</span><span class=w> </span><span class=kc>null</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;totalReadsProcessed&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>100000</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;initialClonesCreated&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>19</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsDroppedNoTargetSequence&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>610</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsDroppedTooShortClonalSequence&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsDroppedLowQuality&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;coreReads&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>21</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsDroppedFailedMapping&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>129</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;lowQualityRescued&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;clonesClustered&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsClustered&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;clones&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>19</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;clonesDroppedAsLowQuality&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;clonesPreClustered&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsPreClustered&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsInClones&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>21</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsInClonesBeforeClustering&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>21</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;readsDroppedWithLowQualityClones&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>  </span><span class=nt>&quot;clonalChainUsage&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;type&quot;</span><span class=p>:</span><span class=w> </span><span class=s2>&quot;chainUsage&quot;</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;chimeras&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>19</span><span class=p>,</span><span class=w></span>
<span class=w>    </span><span class=nt>&quot;chains&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>      </span><span class=nt>&quot;IGH&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>7</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;nonFunctional&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;isOOF&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;hasStops&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=w></span>
<span class=w>      </span><span class=p>},</span><span class=w></span>
<span class=w>      </span><span class=nt>&quot;IGL&quot;</span><span class=p>:</span><span class=w> </span><span class=p>{</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;total&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>12</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;nonFunctional&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;isOOF&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=p>,</span><span class=w></span>
<span class=w>        </span><span class=nt>&quot;hasStops&quot;</span><span class=p>:</span><span class=w> </span><span class=mi>0</span><span class=w></span>
<span class=w>      </span><span class=p>}</span><span class=w></span>
<span class=w>    </span><span class=p>}</span><span class=w></span>
<span class=w>  </span><span class=p>}</span><span class=w></span>
<span class=p>}</span><span class=w></span>
</code></pre></div> </div> </div> </div> </details> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../abhelix-bcr/ class="md-footer__link md-footer__link--prev" aria-label="Previous: AbHelix BCR" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> AbHelix BCR </div> </div> </a> <a href=../nebnext-bcr/ class="md-footer__link md-footer__link--next" aria-label="Next: NEBNext Immune Sequencing" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> NEBNext Immune Sequencing </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../../../javascripts/svg.js></script> <script src=../../../javascripts/onload.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>