<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="MiXCR: ultimate software platform for analysis of Next-Generation Sequencing (NGS) data for immune profiling."><meta name=author content="MiLaboratories Inc"><link href=https://docs.milaboratories.com/mixcr/guides/b-cell-lineages-webinar/ rel=canonical><link rel=icon href=../../../assets/favicon.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>B cell lineage tracing - Docs @ MiLaboratories</title><link rel=stylesheet href=../../../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link rel=stylesheet href=../../../css/extra.css><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-2N4Z95ZG08"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-2N4Z95ZG08",{page_path:e.pathname})})})</script><script async src="https://www.googletagmanager.com/gtag/js?id=G-2N4Z95ZG08"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#b-cell-lineage-tracing class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../.. title="Docs @ MiLaboratories" class="md-header__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Docs @ MiLaboratories </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> B cell lineage tracing </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z"/> </svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <path fill-rule=evenodd clip-rule=evenodd fill=white d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z"/> </svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label="Search by name or category" placeholder="Search by name or category" autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg class=magnify width=24 height=24 viewbox="0 0 24 24" fill=none xmlns=http://www.w3.org/2000/svg> <circle cx=11 cy=11 r=7.25 stroke=white stroke-width=1.5 /> <path stroke=white stroke-width=1.5 d="M16 16L21.5 21.5"/> </svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class="md-tabs__inner md-grid"> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../about/ class=md-tabs__link> About </a> </li> <li class=md-tabs__item> <a href=../../getting-started/installation/ class=md-tabs__link> Getting Started </a> </li> <li class=md-tabs__item> <a href=../milaboratories-human-bcr-rna-multi/ class="md-tabs__link md-tabs__link--active"> Guides </a> </li> <li class=md-tabs__item> <a href=../../reference/overview-analysis-overview/ class=md-tabs__link> Docs </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav class="md-nav md-nav--primary md-nav--lifted has-section" aria-label=Navigation data-md-level=0 data-test=1> <!-- Site title --> <label class=md-nav__title for=__drawer> <a href=../../.. title="Docs @ MiLaboratories" class="md-nav__button md-logo" aria-label="Docs @ MiLaboratories" data-md-component=logo> <img src=../../../assets/logo.svg alt=logo> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class=md-nav__source> <a href=https://github.com/milaboratory/docs title="Go to repository" class=md-source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class=md-nav__list data-md-scrollfix> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../about/ class=md-nav__link> About </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_2 type=checkbox id=__nav_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_2 data-level=1> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Getting Started" data-md-level=1> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/installation/ class=md-nav__link> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/docker/ class=md-nav__link> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/milm/ class=md-nav__link> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/first-run/ class=md-nav__link> First run </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/license/ class=md-nav__link> License </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../getting-started/support/ class=md-nav__link> Support </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3 data-level=1> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Guides data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_1 data-level=2> Kits & Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Kits & Protocols" data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Kits & Protocols </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-bcr-rna-multi/ class=md-nav__link> MiLaboratories Human IG RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milaboratories-human-tcr-rna-multi/ class=md-nav__link> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../milab-dna-multiplex-tcr/ class=md-nav__link> MiLaboratories Human TCR DNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-hsa-bcr/ class=md-nav__link> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../takara-mmu-tcr/ class=md-nav__link> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../iRepertoire/ class=md-nav__link> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../qiaseq-tcr/ class=md-nav__link> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../cellecta-air-tcr-bcr/ class=md-nav__link> Cellecta AIR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../abhelix-bcr/ class=md-nav__link> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../biomed2-bcr/ class=md-nav__link> BioMed2 </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../nebnext-bcr/ class=md-nav__link> NEBNext Immune Sequencing </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_2 data-level=2> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Analysis cases" data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-tcr/ class=md-nav__link> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-multiplex-bcr/ class=md-nav__link> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../generic-umi-bcr/ class=md-nav__link> Targeted BCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../rnaseq/ class=md-nav__link> RNA-Seq data </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../ig-trees-reconstruction/ class=md-nav__link> Immunoglobulin lineage trees reconstruction </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_3 type=checkbox id=__nav_3_3 checked> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_3_3 data-level=2> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label="Special guides" data-md-level=2> <label class=md-nav__title for=__nav_3_3> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../create-custom-library/ class=md-nav__link> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../external-libraries/ class=md-nav__link> Using external libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for=__toc> B cell lineage tracing <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> B cell lineage tracing </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-and-experimental-design class=md-nav__link> Data and experimental design </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#allele-inference class=md-nav__link> Allele inference </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Allele inference"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-clones class=md-nav__link> Export clones </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#generate-lineage-trees class=md-nav__link> Generate lineage trees </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#lineage-analysis class=md-nav__link> Lineage analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Lineage analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#load-clonesets-lineages-and-calculate-statistics class=md-nav__link> Load clonesets, lineages and calculate statistics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#visualizations class=md-nav__link> Visualizations </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#analysis-for-individual-trees class=md-nav__link> Analysis for individual trees </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Analysis for individual trees"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-13628 class=md-nav__link> Draw all plots for lineage 13628 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-10511 class=md-nav__link> Draw all plots for lineage 10511 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-12796 class=md-nav__link> Draw all plots for lineage 12796 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alternative-representations-for-lineage-trees class=md-nav__link> Alternative representations for lineage trees </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </li> <!-- Main navigation item --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4 data-level=1> Docs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Docs data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Docs </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_1 data-level=2> Overview <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Overview data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Overview </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-analysis-overview/ class=md-nav__link> Analysis overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-presets/ class=md-nav__link> Presets </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-mixins-list/ class=md-nav__link> Mix-in options list </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/overview-built-in-presets/ class=md-nav__link> Built-in presets </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_2 data-level=2> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiXCR data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-analyze/ class=md-nav__link> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-align/ class=md-nav__link> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-refineTagsAndSort/ class=md-nav__link> refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemblePartial/ class=md-nav__link> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-extend/ class=md-nav__link> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assemble/ class=md-nav__link> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-assembleContigs/ class=md-nav__link> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-export/ class=md-nav__link> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPretty/ class=md-nav__link> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportAirr/ class=md-nav__link> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReadsForClones/ class=md-nav__link> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportShmTrees/ class=md-nav__link> exportShmTrees </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportReports/ class=md-nav__link> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportQc/ class=md-nav__link> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-postanalysis/ class=md-nav__link> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-downsample/ class=md-nav__link> downsample </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPlots/ class=md-nav__link> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportTables/ class=md-nav__link> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportPreprocTables/ class=md-nav__link> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-exportClonesOverlap/ class=md-nav__link> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-overlapScatterPlot/ class=md-nav__link> overlapScatterPlot </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findAlleles/ class=md-nav__link> findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mixcr-findShmTrees/ class=md-nav__link> findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_3 data-level=2> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Reports data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-align/ class=md-nav__link> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-refineTagsAndSort/ class=md-nav__link> mixcr refineTagsAndSort </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemblePartial/ class=md-nav__link> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-extend/ class=md-nav__link> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assemble/ class=md-nav__link> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-assembleContigs/ class=md-nav__link> mixcr assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findAlleles/ class=md-nav__link> mixcr findAlleles </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/report-findShmTrees/ class=md-nav__link> mixcr findShmTrees </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_4 data-level=2> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=RepSeq.io data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fasta/ class=md-nav__link> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-compile/ class=md-nav__link> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-debug/ class=md-nav__link> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-format/ class=md-nav__link> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromFasta/ class=md-nav__link> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-fromPaddedFasta/ class=md-nav__link> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-inferPoints/ class=md-nav__link> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-merge/ class=md-nav__link> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-stat/ class=md-nav__link> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/repseqio-tsv/ class=md-nav__link> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_5 data-level=2> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=MiTool data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-parse/ class=md-nav__link> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-filter/ class=md-nav__link> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-tag-stat/ class=md-nav__link> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-export-fastq/ class=md-nav__link> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-correct/ class=md-nav__link> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-consensus/ class=md-nav__link> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-process-fastq/ class=md-nav__link> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-format-fastq/ class=md-nav__link> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/mitool-compress-sequence-set/ class=md-nav__link> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_6 type=checkbox id=__nav_4_6> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class=md-nav__link for=__nav_4_6 data-level=2> Specifications <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav class=md-nav aria-label=Specifications data-md-level=2> <label class=md-nav__title for=__nav_4_6> <span class="md-nav__icon md-icon"></span> Specifications </label> <ul class=md-nav__list data-md-scrollfix> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-tag-pattern/ class=md-nav__link> Tag pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-input-file-name-expansion/ class=md-nav__link> Input file name expansion </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-gene-features/ class=md-nav__link> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-translation-rules/ class=md-nav__link> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-mutations-encoding/ class=md-nav__link> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a href=../../reference/ref-repseqio-json-format/ class=md-nav__link> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav class="md-nav md-nav--secondary hello" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#data-and-experimental-design class=md-nav__link> Data and experimental design </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#upstream-analysis class=md-nav__link> Upstream analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Upstream analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#quality-control class=md-nav__link> Quality control </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#allele-inference class=md-nav__link> Allele inference </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Allele inference"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#export-clones class=md-nav__link> Export clones </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#generate-lineage-trees class=md-nav__link> Generate lineage trees </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#lineage-analysis class=md-nav__link> Lineage analysis </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Lineage analysis"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#load-clonesets-lineages-and-calculate-statistics class=md-nav__link> Load clonesets, lineages and calculate statistics </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#visualizations class=md-nav__link> Visualizations </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#analysis-for-individual-trees class=md-nav__link> Analysis for individual trees </a> <!-- Table of contents list --> <nav class=md-nav aria-label="Analysis for individual trees"> <ul class=md-nav__list> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-13628 class=md-nav__link> Draw all plots for lineage 13628 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-10511 class=md-nav__link> Draw all plots for lineage 10511 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#draw-all-plots-for-lineage-12796 class=md-nav__link> Draw all plots for lineage 12796 </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item  item"> <a href=#alternative-representations-for-lineage-trees class=md-nav__link> Alternative representations for lineage trees </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/milaboratory/docs/edit/main/docs/mixcr/guides/b-cell-lineages-webinar.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=b-cell-lineage-tracing>B cell lineage tracing</h1> <p>MiXCR allows V- and J-gene allele inference and somatic hypermutation lineage trees reconstruction. This guide covers these two major advances in BCR repertoire analysis and provides examples.</p> <p>See full webinar video:</p> <iframe width=560 height=315 style="display: block; margin: 0 auto;" src=https://www.youtube.com/embed/HblPe8NhZL4 title="YouTube video player" frameborder=0 allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe> <h2 id=data-and-experimental-design>Data and experimental design</h2> <p>This tutorial uses the longitudinal data on IGH repertoires obtained for one healthy middle aged donor who experienced two COVID-19 infections during the course of the last two years. Peripheral blood samples were obtained at 8 timepoints in replicates.</p> <p><img alt src=../b-cell-lineage-tracing/experiment-design.svg></p> <p>PBMC was isolated using ficoll density gradient centrifugation and RNA was isolated for cDNA library preparation using <a href=../../reference/overview-built-in-presets/#milaboratories>MiLaboratories Human IG RNA Multiplex kit</a>.</p> <p><img alt src=../b-cell-lineage-tracing/pipeline.svg></p> <p>Let's download the full dataset for the donor:</p> <h2 id=upstream-analysis>Upstream analysis</h2> <p><img alt src=../b-cell-lineage-tracing/upstream.svg></p> <p>Running upstream analysis is easy with the <a href="../reference/mixcr-analyze.md`"><code>mixcr analyze</code></a> command, which requires only a few arguments: <div class=highlight><pre><span></span><code>mixcr analyze milab-human-bcr-multiplex-full-length <span class=se>\</span>
      file_R1.fastq.gz <span class=se>\</span>
      file_R2.fastq.gz <span class=se>\</span>
      output_id
</code></pre></div></p> <dl> <dt><code>milab-human-bcr-multiplex-full-length</code></dt> <dd>is the preset name for <a href=../../reference/overview-built-in-presets/#milaboratories>MiLaboratories Human IG RNA Multiplex kit</a>. The presets for <a href=../../reference/overview-built-in-presets/ >most of the commercially available platforms</a> are supported along with generic presets allowing running the pipeline for custom protocols.</dd> <dt><code>file_R1.fastq.gz</code> and <code>file_R2.fastq.gz</code></dt> <dd>the filenames for raw files with sequencing reads, no preprocessing is required (e.g. primers/adapters trimming or UMI-processing).</dd> <dt><code>output_id</code></dt> <dd>output prefix which will be used in the names of the output files.</dd> </dl> <p>The command above will generate the following set of output files: <div class=highlight><pre><span></span><code>&gt; ls mixcr/DCh_T1_pbmc_H_2*

<span class=c1># human-readable reports for every step in txt and json format</span>
DCh_T1_pbmc_H_2.align.report.json
DCh_T1_pbmc_H_2.align.report.txt
DCh_T1_pbmc_H_2.assemble.report.json
DCh_T1_pbmc_H_2.assemble.report.txt
DCh_T1_pbmc_H_2.refine.report.json
DCh_T1_pbmc_H_2.refine.report.txt

<span class=c1># alignments (highly compressed binary file)</span>
DCh_T1_pbmc_H_2.vdjca

<span class=c1># alignments with corrected UMI barcode sequences</span>
DCh_T1_pbmc_H_2.refined.vdjca

<span class=c1># IG clonotypes (highly compressed binary file)</span>
DCh_T1_pbmc_H_2.clns
</code></pre></div></p> <p>We can utilize GNU parallel to run the pipeline for multiple files simultaneously: <div class=highlight><pre><span></span><code>ls raw/*R1*.fastq.gz <span class=p>|</span> parallel --line-buffer <span class=s1>&#39;mixcr analyze \</span>
<span class=s1>   milab-human-bcr-multiplex-full-length \</span>
<span class=s1>   --remove-step exportClones \</span>
<span class=s1>   {}   {=s:R1:R2:=}  \</span>
<span class=s1>   mixcr/{=s:.*/:: ; s:_R1.fastq.gz:: =}&#39;</span>
</code></pre></div></p> <p>Note that we also have added <code>--remove-step exportClones</code> <a href=../../reference/overview-presets/ >'mix-in' option</a>. Initially this preset includes <code>mixcr exportClones</code> step, but we will skip it for now, as we need to infer the allelic variants first and reassign clonotypes to a newly generated individualized gene segment reference library.</p> <h3 id=quality-control>Quality control</h3> <p>Before we move on to lineage trees reconstruction, we will look at the basic QC plots.</p> <p>First, lets generate and investigate alignment rates QC plot. <div class=highlight><pre><span></span><code>mixcr exportQc align <span class=se>\</span>
      mixcr/*.clns <span class=se>\</span>
      figures/alignQc.pdf
</code></pre></div> <img alt src=../b-cell-lineage-tracing/qc-align.svg></p> <p>We observe that all the libraries aligned very well, more than 90% of the reads were successfully aligned to the reference gene segment library.</p> <p>Next we would like to inspect the number of sequencing reads for each of the libraries, we add <code>--absolute-values</code> flag for the same command as above: <div class=highlight><pre><span></span><code>mixcr exportQc align <span class=se>\</span>
      --absolute-values <span class=se>\</span>
      mixcr/*.clns <span class=se>\</span>
      figures/alignQc.pdf
</code></pre></div> <img alt src=../b-cell-lineage-tracing/qc-align-abs.svg></p> <p>We see that depth by reads is a little uneven between our samples, however for each of them we have at least one million sequencing reads, which is generally enough for this analysis.</p> <p>We can investigate this further exporting UMI coverage by reads plot: <div class=highlight><pre><span></span><code>mixcr exportQc tags <span class=se>\</span>
      mixcr/*.clns   <span class=se>\</span>
      figures/readsPerUmiQc.pdf
</code></pre></div> <img alt src=../b-cell-lineage-tracing/qc-coverage.svg></p> <p>This plot demonstrates the distribution of how many times unique cDNA molecules (each represented by a UMI) were covered by sequencing reads. It is common to observe a bimodal distribution, with the lower peak attributing to the dubious sequences, generated by PCR- and sequencing errors. MiXCR automatically calculates the optimal threshold and filters out UMI groups with low coverage by reads to account for this.</p> <h2 id=allele-inference>Allele inference</h2> <p>The next step, after obtaining clonotypes is to perform allele inference to separate true allelic variants from hypermutations alterations. This is done using <a href=../../reference/mixcr-findAlleles/ ><code>mixcr findAlleles</code></a> commands which works in two steps:</p> <ul> <li>a new personalized reference gene library is generated based on the provided samples;</li> <li>the newly generated reference is used to realign each clone. New .clns files are generated.</li> </ul> <p>Only samples obtained from a single donor (or genetic mice strain) should be provided for allele inference.</p> <p><div class=highlight><pre><span></span><code>mixcr findAlleles \
      --report mixcr/DCh.findAlleles.report.txt \
      --export-alleles-mutations mixcr/DCh_alleles.tsv \
      --export-library mixcr/DCh_alleles.json \
      --output-template {file_dir_path}/{file_name}.reassigned.clns \
      mixcr/*.clns
</code></pre></div> The following parameters and options are used:</p> <dl> <dt><code>--export-alleles-mutations</code></dt> <dd>specifies a path to write the descriptions and stats for all original and new alleles</dd> <dt><code>--export-library</code></dt> <dd>specifies a path where to write the new library with found alleles. Supports exporting in <code>.fasta</code> or MiXCR-compatible json formats, which is defined by output filename extension</dd> <dt><code>--output-template {file_dir_path}/{file_name}_.reassigned.clns</code></dt> <dd>specifies the output file path. <code>{file_dir_path}/{file_name}_.reassigned.clns</code> means that each realigned <code>.clns</code> file will be placed in the same folder and will have the same base name as the original <code>.clns</code> file.</dd> </dl> <p>For more information on how to read the output tables see <a href=../../reference/mixcr-findAlleles/ ><code>mixcr findAlleles</code></a>.</p> <h3 id=export-clones>Export clones</h3> <p>Now, when the alleles have been reassigned we can export clonotype tables in human-readable format: <div class=highlight><pre><span></span><code>mixcr exportClones \ 
      mixcr/DCh_T1_pbmc_H_2.reassigned.clns \
      mixcr/DCh_T1_pbmc_H_2.txt
</code></pre></div></p> <h2 id=generate-lineage-trees>Generate lineage trees</h2> <p>Now we can reconstruct clonal lineage trees with <a href=../../reference/mixcr-findShmTrees/ ><code>mixcr findShmTrees</code></a> command: <div class=highlight><pre><span></span><code>mixcr findShmTrees \
      --report mixcr/DCh_trees.log \
      mixcr/*.reassigned.clns \
      mixcr/DCh.shmt
</code></pre></div></p> <p>To export information on lineage trees in human-readable format use <a href=../../reference/mixcr-exportShmTrees/ ><code>mixcr exportShmTreesWithNodes</code></a>: <div class=highlight><pre><span></span><code>mixcr exportShmTreesWithNodes \
      mixcr/DCh.shmt \
      mixcr/DCh_trees.txt
</code></pre></div></p> <h2 id=lineage-analysis>Lineage analysis</h2> <p>Load necessary packages, define several handy helper functions, and color pallets:</p> <details class=quote> <summary>Expand</summary> <div class=highlight><pre><span></span><code><span class=c1>#| code-fold: true</span>
<span class=nf>library</span><span class=p>(</span><span class=n>tidyr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>readr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>stringr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>stringdist</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggplot2</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>msa</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>magrittr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggbeeswarm</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>tibble</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>dplyr</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>plotly</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggtree</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>ggrepel</span><span class=p>)</span>
<span class=nf>library</span><span class=p>(</span><span class=n>data.table</span><span class=p>)</span>

<span class=n>`%+%`</span> <span class=o>&lt;-</span> <span class=nf>function</span><span class=p>(</span><span class=n>a</span><span class=p>,</span> <span class=n>b</span><span class=p>)</span> <span class=nf>paste0</span><span class=p>(</span><span class=n>a</span><span class=p>,</span> <span class=n>b</span><span class=p>)</span>

<span class=n>`+.uneval`</span> <span class=o>&lt;-</span> <span class=nf>function</span><span class=p>(</span><span class=n>a</span><span class=p>,</span><span class=n>b</span><span class=p>)</span> <span class=p>{</span>
    <span class=nf>`class&lt;-`</span><span class=p>(</span><span class=nf>modifyList</span><span class=p>(</span><span class=n>a</span><span class=p>,</span><span class=n>b</span><span class=p>),</span> <span class=s>&quot;uneval&quot;</span><span class=p>)</span>
<span class=p>}</span>

<span class=n>alignment2Fasta</span> <span class=o>&lt;-</span> <span class=nf>function</span><span class=p>(</span><span class=n>alignment</span><span class=p>,</span> <span class=n>filename</span><span class=p>)</span> <span class=p>{</span>
  <span class=nf>sink</span><span class=p>(</span><span class=n>filename</span><span class=p>)</span>

  <span class=n>n</span> <span class=o>&lt;-</span> <span class=nf>length</span><span class=p>(</span><span class=nf>rownames</span><span class=p>(</span><span class=n>alignment</span><span class=p>))</span>
  <span class=nf>for</span><span class=p>(</span><span class=n>i</span> <span class=n>in</span> <span class=nf>seq</span><span class=p>(</span><span class=m>1</span><span class=p>,</span> <span class=n>n</span><span class=p>))</span> <span class=p>{</span>
    <span class=nf>cat</span><span class=p>(</span><span class=nf>paste0</span><span class=p>(</span><span class=s>&#39;&gt;&#39;</span><span class=p>,</span> <span class=nf>rownames</span><span class=p>(</span><span class=n>alignment</span><span class=p>)[</span><span class=n>i</span><span class=p>]))</span>
    <span class=nf>cat</span><span class=p>(</span><span class=s>&#39;\n&#39;</span><span class=p>)</span>
    <span class=n>the.sequence</span> <span class=o>&lt;-</span> <span class=nf>toString</span><span class=p>(</span><span class=nf>unmasked</span><span class=p>(</span><span class=n>alignment</span><span class=p>)[[</span><span class=n>i</span><span class=p>]])</span>
    <span class=nf>cat</span><span class=p>(</span><span class=n>the.sequence</span><span class=p>)</span>
    <span class=nf>cat</span><span class=p>(</span><span class=s>&#39;\n&#39;</span><span class=p>)</span>  
  <span class=p>}</span>

  <span class=nf>sink</span><span class=p>(</span><span class=kc>NULL</span><span class=p>)</span>
<span class=p>}</span>

<span class=n>mi_27</span><span class=o>&lt;-</span><span class=nf>c</span><span class=p>(</span>
  <span class=s>&quot;#99E099&quot;</span><span class=p>,</span><span class=s>&quot;#198020&quot;</span><span class=p>,</span><span class=s>&quot;#42B842&quot;</span><span class=p>,</span><span class=s>&quot;#C1ADFF&quot;</span><span class=p>,</span><span class=s>&quot;#845CFF&quot;</span><span class=p>,</span><span class=s>&quot;#5F31CC&quot;</span><span class=p>,</span><span class=s>&quot;#FFCB8F&quot;</span><span class=p>,</span>
  <span class=s>&quot;#FF9429&quot;</span><span class=p>,</span><span class=s>&quot;#C26A27&quot;</span><span class=p>,</span><span class=s>&quot;#90E0E0&quot;</span><span class=p>,</span><span class=s>&quot;#27C2C2&quot;</span><span class=p>,</span><span class=s>&quot;#068A94&quot;</span><span class=p>,</span><span class=s>&quot;#FAAAFA&quot;</span><span class=p>,</span><span class=s>&quot;#E553E5&quot;</span><span class=p>,</span>
  <span class=s>&quot;#A324B2&quot;</span><span class=p>,</span><span class=s>&quot;#CBEB67&quot;</span><span class=p>,</span><span class=s>&quot;#95C700&quot;</span><span class=p>,</span><span class=s>&quot;#659406&quot;</span><span class=p>,</span><span class=s>&quot;#99CCFF&quot;</span><span class=p>,</span><span class=s>&quot;#2D93FA&quot;</span><span class=p>,</span><span class=s>&quot;#105BCC&quot;</span><span class=p>,</span>
  <span class=s>&quot;#FFADBA&quot;</span><span class=p>,</span><span class=s>&quot;#F05670&quot;</span><span class=p>,</span><span class=s>&quot;#AD3757&quot;</span><span class=p>,</span><span class=s>&quot;#D3D7E0&quot;</span><span class=p>,</span><span class=s>&quot;#929BAD&quot;</span><span class=p>,</span><span class=s>&quot;#5E5E70&quot;</span><span class=p>)</span>

<span class=n>mi_dark</span><span class=o>&lt;-</span><span class=nf>c</span><span class=p>(</span>
  <span class=s>&quot;#198020&quot;</span><span class=p>,</span>
  <span class=s>&quot;#5F31CC&quot;</span><span class=p>,</span>
  <span class=s>&quot;#C26A27&quot;</span><span class=p>,</span>
  <span class=s>&quot;#068A94&quot;</span><span class=p>,</span>
  <span class=s>&quot;#A324B2&quot;</span><span class=p>,</span>
  <span class=s>&quot;#659406&quot;</span><span class=p>,</span>
  <span class=s>&quot;#105BCC&quot;</span><span class=p>,</span>
  <span class=s>&quot;#AD3757&quot;</span><span class=p>,</span>
  <span class=s>&quot;#5E5E70&quot;</span><span class=p>)</span>

<span class=n>aminoacids_pal</span><span class=o>&lt;-</span><span class=nf>c</span><span class=p>(</span>
  <span class=s>&quot;#99E099&quot;</span><span class=p>,</span><span class=s>&quot;#198020&quot;</span><span class=p>,</span><span class=s>&quot;#42B842&quot;</span><span class=p>,</span><span class=s>&quot;#C1ADFF&quot;</span><span class=p>,</span><span class=s>&quot;#845CFF&quot;</span><span class=p>,</span><span class=s>&quot;#5F31CC&quot;</span><span class=p>,</span><span class=s>&quot;#FFCB8F&quot;</span><span class=p>,</span>
  <span class=s>&quot;#FF9429&quot;</span><span class=p>,</span><span class=s>&quot;#C26A27&quot;</span><span class=p>,</span><span class=s>&quot;#90E0E0&quot;</span><span class=p>,</span><span class=s>&quot;#27C2C2&quot;</span><span class=p>,</span><span class=s>&quot;#068A94&quot;</span><span class=p>,</span><span class=s>&quot;#FAAAFA&quot;</span><span class=p>,</span><span class=s>&quot;#E553E5&quot;</span><span class=p>,</span>
  <span class=s>&quot;#A324B2&quot;</span><span class=p>,</span><span class=s>&quot;#CBEB67&quot;</span><span class=p>,</span><span class=s>&quot;#95C700&quot;</span><span class=p>,</span><span class=s>&quot;#659406&quot;</span><span class=p>,</span><span class=s>&quot;#99CCFF&quot;</span><span class=p>,</span><span class=s>&quot;#2D93FA&quot;</span><span class=p>,</span><span class=s>&quot;#105BCC&quot;</span><span class=p>,</span>
  <span class=s>&quot;#FFADBA&quot;</span><span class=p>,</span><span class=s>&quot;#F05670&quot;</span><span class=p>,</span><span class=s>&quot;#AD3757&quot;</span><span class=p>,</span><span class=s>&quot;#D3D7E0&quot;</span><span class=p>,</span><span class=s>&quot;#929BAD&quot;</span><span class=p>,</span><span class=s>&quot;#5E5E70&quot;</span><span class=p>)</span>

<span class=n>isotype_pal</span><span class=o>&lt;-</span><span class=nf>c</span><span class=p>(</span>
  <span class=n>IGHA1</span> <span class=o>=</span> <span class=s>&quot;#F05670&quot;</span><span class=p>,</span>
  <span class=n>IGHA2</span> <span class=o>=</span> <span class=s>&quot;#FFADBA&quot;</span><span class=p>,</span>
  <span class=n>IGHM</span> <span class=o>=</span> <span class=s>&quot;#99CCFF&quot;</span><span class=p>,</span>
  <span class=n>IGHD</span> <span class=o>=</span> <span class=s>&quot;#105BCC&quot;</span><span class=p>,</span>
  <span class=n>IGHG1</span> <span class=o>=</span><span class=s>&quot;#198020&quot;</span><span class=p>,</span>
  <span class=n>IGHG2</span> <span class=o>=</span><span class=s>&quot;#42B842&quot;</span><span class=p>,</span>
  <span class=n>IGHG3</span> <span class=o>=</span><span class=s>&quot;#99E099&quot;</span><span class=p>,</span>
  <span class=n>IGHG4</span> <span class=o>=</span><span class=s>&quot;#95C700&quot;</span><span class=p>)</span>
</code></pre></div> </details> <h3 id=load-clonesets-lineages-and-calculate-statistics>Load clonesets, lineages and calculate statistics</h3> <p>Loading <code>mixcr exportClones</code> command output and calculating number of clonotypes and unique cDNA molecules</p> <div class=highlight><pre><span></span><code><span class=n>folder</span><span class=o>&lt;-</span><span class=s>&quot;~/webinar/mixcr/&quot;</span>
<span class=n>cloneset_files</span><span class=o>&lt;-</span><span class=nf>dir</span><span class=p>(</span><span class=n>folder</span><span class=p>,</span><span class=s>&quot;.reassigned_IGH.tsv&quot;</span><span class=p>,</span><span class=n>full.names</span> <span class=o>=</span> <span class=bp>T</span><span class=p>)</span>

<span class=n>clonesets</span><span class=o>&lt;-</span><span class=nf>read_tsv</span><span class=p>(</span><span class=n>cloneset_files</span><span class=p>,</span><span class=n>id</span><span class=o>=</span><span class=s>&quot;fileName&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>id</span><span class=o>=</span><span class=nf>str_remove_all</span><span class=p>(</span><span class=n>fileName</span><span class=p>,</span><span class=s>&quot;.*\\/|.reassigned_IGH.tsv&quot;</span><span class=p>),</span>
         <span class=n>isotype</span><span class=o>=</span><span class=nf>str_remove</span><span class=p>(</span><span class=n>allCHitsWithScore</span><span class=p>,</span> <span class=s>&quot;\\*.*&quot;</span><span class=p>))</span>

<span class=n>stats_clonesets</span><span class=o>&lt;-</span><span class=n>clonesets</span> <span class=o>%&gt;%</span>
  <span class=nf>group_by</span><span class=p>(</span><span class=n>id</span><span class=p>)</span> <span class=o>%&gt;%</span>
  <span class=nf>summarise</span><span class=p>(</span><span class=n>totalCloneInSample</span><span class=o>=</span><span class=nf>n</span><span class=p>(),</span>
            <span class=n>totalUMIsInSample</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>uniqueUMICount</span><span class=p>))</span>
</code></pre></div> <p>Load <code>mixcr exportShmTreesWithNodes</code> command output, parse metadata from filenames, and join with statistics table generated on previous step</p> <div class=highlight><pre><span></span><code><span class=n>lng_db</span><span class=o>&lt;-</span><span class=nf>read_tsv</span><span class=p>(</span><span class=s>&quot;~/ch_covid/mixcr/shmTrees/DCh_trees.tsv&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>id</span><span class=o>=</span><span class=nf>str_remove_all</span><span class=p>(</span><span class=n>fileName</span><span class=p>,</span><span class=s>&quot;.*\\/|.reassigned.clns&quot;</span><span class=p>),</span>
         <span class=n>isotype</span><span class=o>=</span><span class=nf>str_remove</span><span class=p>(</span><span class=n>bestCHit</span><span class=p>,</span><span class=s>&quot;\\*00&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=nf>separate</span><span class=p>(</span><span class=s>&quot;id&quot;</span><span class=p>,</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;donor&quot;</span><span class=p>,</span><span class=s>&quot;timepoint&quot;</span><span class=p>,</span><span class=s>&quot;tissue&quot;</span><span class=p>,</span><span class=s>&quot;status&quot;</span><span class=p>,</span><span class=s>&quot;repnum&quot;</span><span class=p>),</span>
           <span class=n>sep</span><span class=o>=</span><span class=s>&quot;_&quot;</span><span class=p>,</span>
           <span class=n>remove</span><span class=o>=</span><span class=bp>F</span><span class=p>)</span> <span class=o>%&gt;%</span>
  <span class=nf>left_join</span><span class=p>(</span><span class=n>stats_clonesets</span><span class=p>,</span><span class=n>by</span> <span class=o>=</span><span class=s>&quot;id&quot;</span><span class=p>)</span>
<span class=c1># Note rows with NA in cloneId and fileName fields - these are the inferred nodes</span>
</code></pre></div> <p>For further processing it is useful to pivot to R's wide format:</p> <details class=quote> <summary>Pivot to wide format</summary> <div class=highlight><pre><span></span><code><span class=n>lngs_trees_wide</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span>
  <span class=nf>filter</span><span class=p>(</span><span class=o>!</span><span class=n>fileName</span><span class=o>==</span><span class=s>&quot;&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>group_by</span><span class=p>(</span><span class=n>treeId</span><span class=p>,</span><span class=n>timepoint</span><span class=p>,</span><span class=n>repnum</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>summarise</span><span class=p>(</span><span class=n>nClones</span><span class=o>=</span><span class=nf>n</span><span class=p>(),</span>
            <span class=n>nClonesNormalized</span><span class=o>=</span><span class=nf>n</span><span class=p>()</span><span class=o>/</span><span class=n>dplyr</span><span class=o>::</span><span class=nf>first</span><span class=p>(</span><span class=n>totalCloneInSample</span><span class=p>),</span>
            <span class=n>nUMIsNormalized</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>uniqueUMICount</span><span class=p>)</span><span class=o>/</span><span class=n>dplyr</span><span class=o>::</span><span class=nf>first</span><span class=p>(</span><span class=n>totalUMIsInSample</span><span class=p>),</span>
            <span class=n>nUMIs</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>uniqueUMICount</span><span class=p>),</span>
            <span class=n>isotypes</span><span class=o>=</span><span class=nf>list</span><span class=p>(</span><span class=nf>unique</span><span class=p>(</span><span class=n>isotype</span><span class=p>)),</span>
            <span class=n>nConditions</span><span class=o>=</span><span class=nf>n_distinct</span><span class=p>(</span><span class=n>status</span><span class=p>),</span>
            <span class=n>conditions</span><span class=o>=</span><span class=nf>paste</span><span class=p>(</span><span class=nf>unique</span><span class=p>(</span><span class=n>status</span><span class=p>),</span><span class=n>collapse</span><span class=o>=</span><span class=s>&quot;,&quot;</span><span class=p>),</span>
            <span class=n>nTissues</span><span class=o>=</span><span class=nf>n_distinct</span><span class=p>(</span><span class=n>tissue</span><span class=p>),</span>
            <span class=n>tissues</span><span class=o>=</span><span class=nf>paste</span><span class=p>(</span><span class=nf>unique</span><span class=p>(</span><span class=n>tissue</span><span class=p>),</span><span class=n>collapse</span><span class=o>=</span><span class=s>&quot;,&quot;</span><span class=p>),</span>
            <span class=n>totalCloneInSample</span><span class=o>=</span><span class=n>dplyr</span><span class=o>::</span><span class=nf>first</span><span class=p>(</span><span class=n>totalCloneInSample</span><span class=p>),</span>
            <span class=n>totalUMIsInSample</span><span class=o>=</span><span class=n>dplyr</span><span class=o>::</span><span class=nf>first</span><span class=p>(</span><span class=n>totalUMIsInSample</span><span class=p>)</span>
  <span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>ungroup</span><span class=p>()</span> <span class=o>%&gt;%</span>
  <span class=nf>group_by</span><span class=p>(</span><span class=n>treeId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>timepoint</span><span class=o>=</span><span class=nf>factor</span><span class=p>(</span><span class=n>timepoint</span><span class=p>,</span><span class=n>levels</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;T1&quot;</span><span class=p>,</span><span class=s>&quot;T2&quot;</span><span class=p>,</span><span class=s>&quot;T3&quot;</span><span class=p>,</span><span class=s>&quot;T4&quot;</span><span class=p>,</span><span class=s>&quot;T5&quot;</span><span class=p>,</span><span class=s>&quot;T6&quot;</span><span class=p>,</span><span class=s>&quot;T7&quot;</span><span class=p>,</span><span class=s>&quot;T8&quot;</span><span class=p>)),</span>
         <span class=n>timepoints</span><span class=o>=</span><span class=n>timepoint</span> <span class=o>%&gt;%</span><span class=nf>unique</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>sort</span><span class=p>()</span> <span class=o>%&gt;%</span>  <span class=nf>paste</span><span class=p>(</span><span class=n>collapse</span><span class=o>=</span><span class=s>&quot;,&quot;</span><span class=p>),</span>
         <span class=n>isotypes</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=n>isotypes</span><span class=p>)</span> <span class=o>%&gt;%</span> <span class=nf>unlist</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>unique</span><span class=p>()</span> <span class=o>%&gt;%</span><span class=nf>sort</span><span class=p>()</span> <span class=o>%&gt;%</span>  <span class=nf>paste</span><span class=p>(</span><span class=n>collapse</span><span class=o>=</span><span class=s>&quot;,&quot;</span><span class=p>),</span>
         <span class=n>totalNClones</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>nClones</span><span class=p>),</span>
         <span class=n>totalCount</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>nUMIs</span><span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=nf>ungroup</span><span class=p>()</span> <span class=o>%&gt;%</span> 
  <span class=nf>arrange</span><span class=p>(</span><span class=n>timepoint</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>pivot_wider</span><span class=p>(</span><span class=n>id_cols</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>treeId</span><span class=p>,</span><span class=n>isotypes</span><span class=p>,</span><span class=n>timepoints</span><span class=p>,</span><span class=n>totalNClones</span><span class=p>,</span><span class=n>totalCount</span><span class=p>),</span> 
              <span class=n>names_from</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>timepoint</span><span class=p>,</span><span class=n>repnum</span><span class=p>),</span>
              <span class=n>values_from</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>nClones</span><span class=p>,</span><span class=n>nUMIs</span><span class=p>,</span><span class=n>nClonesNormalized</span><span class=p>,</span><span class=n>nUMIsNormalized</span><span class=p>),</span>
              <span class=n>values_fill</span> <span class=o>=</span> <span class=m>0.00</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=nf>across</span><span class=p>(</span><span class=nf>starts_with</span><span class=p>(</span><span class=s>&quot;nClonesN&quot;</span><span class=p>),</span><span class=n>.fns</span> <span class=o>=</span> <span class=o>~</span> <span class=nf>ifelse</span><span class=p>(</span><span class=n>.x</span><span class=o>==</span><span class=m>0</span><span class=p>,</span> <span class=nf>min</span><span class=p>(</span><span class=n>.x</span><span class=p>[</span><span class=n>.x</span><span class=o>&gt;</span><span class=m>0</span><span class=p>])</span><span class=o>/</span><span class=m>2</span><span class=p>,</span> <span class=n>.x</span> <span class=p>)),</span>
         <span class=nf>across</span><span class=p>(</span><span class=nf>starts_with</span><span class=p>(</span><span class=s>&quot;nUmisN&quot;</span><span class=p>),</span><span class=n>.fns</span> <span class=o>=</span> <span class=o>~</span> <span class=nf>ifelse</span><span class=p>(</span><span class=n>.x</span><span class=o>==</span><span class=m>0</span><span class=p>,</span> <span class=nf>min</span><span class=p>(</span><span class=n>.x</span><span class=p>[</span><span class=n>.x</span><span class=o>&gt;</span><span class=m>0</span><span class=p>])</span><span class=o>/</span><span class=m>2</span><span class=p>,</span> <span class=n>.x</span> <span class=p>)))</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>switchedIsotypePresent</span><span class=o>=</span><span class=nf>ifelse</span><span class=p>(</span><span class=nf>str_detect</span><span class=p>(</span><span class=n>isotypes</span><span class=p>,</span> <span class=s>&quot;IGHA|IGHG|IGHE&quot;</span><span class=p>),</span> <span class=kc>TRUE</span><span class=p>,</span> <span class=kc>FALSE</span><span class=p>))</span>
</code></pre></div> </details> <h3 id=visualizations>Visualizations</h3> <details class=quote> <summary>Define function to draw lineage abundance scatter plots</summary> <div class=highlight><pre><span></span><code><span class=n>drawScatter</span><span class=o>&lt;-</span><span class=nf>function</span><span class=p>(</span><span class=n>sample1</span><span class=p>,</span><span class=n>sample2</span><span class=p>,</span><span class=n>metric</span><span class=p>,</span><span class=n>coloring</span><span class=p>){</span>
  <span class=c1>#Two possible options for metric - nClonesNormalized and nUMIsNormalized</span>
  <span class=c1>#Two possible options for coloring - changedTrees and switchedIsotypePresent</span>
  <span class=c1>#Set limits for plots depending on metric</span>
  <span class=nf>if </span><span class=p>(</span><span class=n>metric</span><span class=o>==</span><span class=s>&quot;nClonesNormalized&quot;</span><span class=p>){</span>
    <span class=n>upperLimit</span><span class=o>=</span><span class=m>0.2</span><span class=o>*</span><span class=m>1e-2</span>
    <span class=n>lowerLimit</span><span class=o>=</span><span class=m>0.8</span><span class=o>*</span><span class=m>1e-05</span>
  <span class=p>}</span> <span class=n>else</span> <span class=nf>if </span><span class=p>(</span><span class=n>metric</span><span class=o>==</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>){</span>
    <span class=n>upperLimit</span><span class=o>=</span><span class=m>1</span><span class=o>*</span><span class=m>1e-1</span>
    <span class=n>lowerLimit</span><span class=o>=</span><span class=m>0.5</span><span class=o>*</span><span class=m>1e-05</span>
  <span class=p>}</span>

  <span class=n>g</span><span class=o>&lt;-</span><span class=nf>ggplot</span><span class=p>(</span><span class=n>lngs_trees_wide</span><span class=p>,</span>
         <span class=nf>aes_string</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>metric</span> <span class=o>%+%</span> <span class=s>&quot;_&quot;</span> <span class=o>%+%</span> <span class=n>sample1</span><span class=p>,</span>
                    <span class=n>y</span><span class=o>=</span><span class=n>metric</span> <span class=o>%+%</span> <span class=s>&quot;_&quot;</span> <span class=o>%+%</span> <span class=n>sample2</span><span class=p>)</span> <span class=o>+</span>
          <span class=nf>aes</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;timepoints: &quot;</span> <span class=o>%+%</span> <span class=n>timepoints</span> <span class=o>%+%</span> <span class=s>&quot;\n&quot;</span>  <span class=o>%+%</span> 
                <span class=s>&quot;isotypes: &quot;</span>  <span class=o>%+%</span> <span class=n>isotypes</span> <span class=o>%+%</span> <span class=s>&quot;\n&quot;</span>  <span class=o>%+%</span>
                <span class=s>&quot;N Clones in &quot;</span> <span class=o>%+%</span> <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;sample1&quot;</span><span class=p>))</span> <span class=o>%+%</span> <span class=s>&quot;: &quot;</span> <span class=o>%+%</span>
                <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;nClones&quot;</span> <span class=o>%+%</span> <span class=s>&quot;_&quot;</span> <span class=o>%+%</span>
                             <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;sample1&quot;</span><span class=p>))))</span> <span class=o>%+%</span> <span class=s>&quot;\n&quot;</span> <span class=o>%+%</span>
                <span class=s>&quot;N Clones in &quot;</span> <span class=o>%+%</span> <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;sample2&quot;</span><span class=p>))</span> <span class=o>%+%</span> <span class=s>&quot;: &quot;</span> <span class=o>%+%</span>
                <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;nClones&quot;</span> <span class=o>%+%</span> <span class=s>&quot;_&quot;</span> <span class=o>%+%</span>
                             <span class=nf>eval</span><span class=p>(</span><span class=nf>parse</span><span class=p>(</span><span class=n>text</span> <span class=o>=</span> <span class=s>&quot;sample2&quot;</span><span class=p>))))</span> <span class=o>%+%</span> <span class=s>&quot;\n&quot;</span> <span class=o>%+%</span>
                <span class=s>&quot;total N Clonotypes: &quot;</span> <span class=o>%+%</span> <span class=n>totalNClones</span> <span class=o>%+%</span> <span class=s>&quot;\n&quot;</span> <span class=o>%+%</span>
                <span class=s>&quot;total N UMIs: &quot;</span> <span class=o>%+%</span> <span class=n>totalCount</span>
                <span class=p>))</span><span class=o>+</span>   
    <span class=nf>geom_point</span><span class=p>(</span><span class=nf>aes_string</span><span class=p>(</span><span class=n>color</span><span class=o>=</span><span class=n>coloring</span><span class=p>),</span>
               <span class=n>alpha</span><span class=o>=</span><span class=m>0.5</span><span class=p>,</span><span class=n>size</span><span class=o>=</span><span class=m>4</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>theme_bw</span><span class=p>()</span><span class=o>+</span>
    <span class=nf>scale_x_log10</span><span class=p>()</span><span class=o>+</span>
    <span class=nf>scale_y_log10</span><span class=p>()</span><span class=o>+</span>
    <span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&quot;none&quot;</span><span class=p>)</span> <span class=o>+</span>
    <span class=nf>geom_abline</span><span class=p>(</span><span class=n>slope</span><span class=o>=</span><span class=m>1</span><span class=p>,</span> <span class=n>intercept</span><span class=o>=</span><span class=m>0</span><span class=p>,</span> <span class=n>linetype</span><span class=o>=</span><span class=m>2</span><span class=p>,</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;grey28&quot;</span><span class=p>)</span>

   <span class=nf>if</span><span class=p>(</span><span class=nf>exists</span><span class=p>(</span><span class=s>&quot;upperLimit&quot;</span><span class=p>))</span> <span class=p>{</span>
    <span class=n>g</span><span class=o>&lt;-</span><span class=n>g</span><span class=o>+</span> <span class=nf>expand_limits</span><span class=p>(</span><span class=n>y</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=n>lowerLimit</span><span class=p>,</span><span class=n>upperLimit</span><span class=p>),</span><span class=n>x</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=n>lowerLimit</span><span class=p>,</span><span class=n>upperLimit</span><span class=p>))</span>
  <span class=p>}</span>

  <span class=nf>if</span><span class=p>(</span><span class=n>coloring</span><span class=o>==</span><span class=s>&quot;changedTrees&quot;</span><span class=p>)</span> <span class=p>{</span>
  <span class=n>g</span><span class=o>&lt;-</span><span class=n>g</span><span class=o>+</span><span class=nf>geom_point</span><span class=p>(</span><span class=n>data</span><span class=o>=</span><span class=n>lngs_trees_wide</span> <span class=o>%&gt;%</span> <span class=nf>filter</span><span class=p>(</span><span class=n>changedTrees</span><span class=p>),</span>
                  <span class=n>color</span><span class=o>=</span><span class=s>&quot;#F05670&quot;</span><span class=p>,</span>
                  <span class=n>alpha</span><span class=o>=</span><span class=m>0.4</span><span class=p>,</span>
                  <span class=n>size</span><span class=o>=</span><span class=m>4</span><span class=p>)</span> <span class=o>+</span>
    <span class=nf>scale_color_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=s>&quot;grey76&quot;</span><span class=p>,</span><span class=s>&quot;#F05670&quot;</span><span class=p>))</span>
  <span class=p>}</span> <span class=n>else</span> <span class=nf>if </span><span class=p>(</span><span class=n>coloring</span><span class=o>==</span><span class=s>&quot;switchedIsotypePresent&quot;</span><span class=p>)</span> <span class=p>{</span>
   <span class=n>g</span><span class=o>&lt;-</span><span class=n>g</span><span class=o>+</span> <span class=nf>scale_color_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>mi_dark</span><span class=p>[</span><span class=m>2</span><span class=p>],</span><span class=n>mi_dark</span><span class=p>[</span><span class=m>1</span><span class=p>]))</span>
  <span class=p>}</span>
<span class=p>}</span>
</code></pre></div> </details> <p>Each point represents a lineage, values represent number of cDNA molecules (each represented by one UMI) normalized by total number of cDNA molecules in the corresponding sample. In the steady state time point (T6) many of the most abundant (by cDNA) lineages are not replicated. Presumably these lineages are derived from antibody-secreting cells (ASCs), with high expression of IGH mRNA, but with low number of cells in them. At peak of response timepoint (T7) ASCs subset is represented by lineages undergoing active clonal expansions, with many cells present in peripheral blood, that's why it is easier to detect them in replicates.</p> <p><div class=highlight><pre><span></span><code><span class=n>g1</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T6_1&quot;</span><span class=p>,</span><span class=s>&quot;T6_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;switchedIsotypePresent&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T6, pre-COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T6 pre-COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=n>g2</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T7_1&quot;</span><span class=p>,</span><span class=s>&quot;T7_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;switchedIsotypePresent&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=n>g3</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T6_1&quot;</span><span class=p>,</span><span class=s>&quot;T7_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;switchedIsotypePresent&quot;</span><span class=p>)</span><span class=o>+</span><span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T6, pre-COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g1</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g2</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g3</span><span class=p>)</span>
</code></pre></div> <img alt src=../b-cell-lineage-tracing/lineage-1.svg></p> <p>Now we can label the responding lineages, looking at the deviation of abundance between timepoints T6 and T7 and comparing it to deviation between replicates at peak of response (T7). We also filter lineages which have more than 5 different clonotypes at T7.</p> <details class=quote> <summary>Define function to draw changed lineages</summary> <div class=highlight><pre><span></span><code><span class=n>lngs_trees_wide</span> <span class=o>%&lt;&gt;%</span> 
   <span class=nf>mutate</span><span class=p>(</span><span class=n>logFCp61_71</span><span class=o>=</span><span class=nf>log10</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>/</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>),</span>
          <span class=n>logFCp61_72</span><span class=o>=</span><span class=nf>log10</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>/</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>),</span>
          <span class=n>logFCp62_71</span><span class=o>=</span><span class=nf>log10</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_2</span><span class=o>/</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>),</span>
          <span class=n>logFCp62_72</span><span class=o>=</span><span class=nf>log10</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>/</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>))</span> <span class=o>%&gt;%</span>
  <span class=c1>#Calculate standard deviation </span>
  <span class=nf>mutate</span><span class=p>(</span>
    <span class=n>sd71_72</span><span class=o>=</span><span class=p>(</span><span class=n>nUMIsNormalized_T7_1</span><span class=o>-</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>^</span><span class=m>2</span> <span class=o>/</span> <span class=p>((</span><span class=n>nUMIsNormalized_T7_1</span><span class=o>+</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>/</span><span class=m>2</span><span class=p>),</span>
    <span class=n>sd61_71</span><span class=o>=</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>-</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>)</span><span class=o>^</span><span class=m>2</span> <span class=o>/</span> <span class=p>((</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>+</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>)</span><span class=o>/</span><span class=m>2</span><span class=p>),</span>
    <span class=n>sd61_72</span><span class=o>=</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>-</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>^</span><span class=m>2</span> <span class=o>/</span> <span class=p>((</span><span class=n>nUMIsNormalized_T6_1</span><span class=o>+</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>/</span><span class=m>2</span><span class=p>),</span>
    <span class=n>sd62_71</span><span class=o>=</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_2</span><span class=o>-</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>)</span><span class=o>^</span><span class=m>2</span> <span class=o>/</span> <span class=p>((</span><span class=n>nUMIsNormalized_T6_2</span><span class=o>+</span><span class=n>nUMIsNormalized_T7_1</span><span class=p>)</span><span class=o>/</span><span class=m>2</span><span class=p>),</span>
    <span class=n>sd62_72</span><span class=o>=</span><span class=p>(</span><span class=n>nUMIsNormalized_T6_2</span><span class=o>-</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>^</span><span class=m>2</span> <span class=o>/</span> <span class=p>((</span><span class=n>nUMIsNormalized_T6_2</span><span class=o>+</span><span class=n>nUMIsNormalized_T7_2</span><span class=p>)</span><span class=o>/</span><span class=m>2</span><span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=c1>#Label lineages that have sd between T6 and T7 higher than between replicates in T7</span>
  <span class=c1>#and havr more than 5 unique clonotypes at T7</span>
  <span class=c1>#and also increased abundance in T7 compared to T6 by comparison of all replicates</span>
  <span class=nf>mutate</span><span class=p>(</span>
    <span class=n>changedTrees</span><span class=o>=</span><span class=n>sd71_72</span><span class=o>&lt;</span><span class=n>sd61_71</span> <span class=o>&amp;</span> <span class=n>sd71_72</span><span class=o>&lt;</span><span class=n>sd61_72</span> <span class=o>&amp;</span> <span class=n>sd71_72</span><span class=o>&lt;</span><span class=n>sd62_71</span> <span class=o>&amp;</span> <span class=n>sd71_72</span><span class=o>&lt;</span><span class=n>sd62_72</span> <span class=o>&amp;</span><span class=p>(</span><span class=n>nClones_T7_1</span><span class=o>+</span><span class=n>nClones_T7_2</span><span class=o>&gt;</span><span class=m>5</span><span class=p>)</span> <span class=o>&amp;</span> <span class=n>logFCp61_71</span><span class=o>&lt;</span><span class=m>0</span>  <span class=o>&amp;</span> <span class=n>logFCp61_72</span><span class=o>&lt;</span><span class=m>0</span> <span class=o>&amp;</span> <span class=n>logFCp62_71</span><span class=o>&lt;</span><span class=m>0</span> <span class=o>&amp;</span> <span class=n>logFCp62_72</span><span class=o>&lt;</span><span class=m>0</span><span class=p>)</span> 
</code></pre></div> </details> <p>Draw responding lineages:</p> <p><div class=highlight><pre><span></span><code><span class=n>g1</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T6_1&quot;</span><span class=p>,</span><span class=s>&quot;T6_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;changedTrees&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T6, pre-COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T6 pre-COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=n>g2</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T7_1&quot;</span><span class=p>,</span><span class=s>&quot;T7_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;changedTrees&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=n>g3</span><span class=o>&lt;-</span><span class=nf>drawScatter</span><span class=p>(</span><span class=s>&quot;T6_1&quot;</span><span class=p>,</span><span class=s>&quot;T7_2&quot;</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>,</span><span class=n>coloring</span><span class=o>=</span><span class=s>&quot;changedTrees&quot;</span><span class=p>)</span><span class=o>+</span><span class=nf>labs</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=s>&quot;T6, pre-COVID19, replicate 1&quot;</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;T7, d7 COVID19, replicate 2&quot;</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g1</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g2</span><span class=p>)</span>
<span class=nf>ggplotly</span><span class=p>(</span><span class=n>g3</span><span class=p>)</span>
</code></pre></div> <img alt src=b-cell-lineage-tracing/lineage-2.svg></p> <p>In pink we show lineages presumably responding to second COVID19 infection, defined at the previous step. We observe that some of these lineages also increased their abundance at first COVID19 response time point (T4).</p> <details class=quote> <summary>Draw lineage tracing plot</summary> <div class=highlight><pre><span></span><code><span class=c1>#Reshape a little bit to account to the replicates present at some of the timepoints</span>
<span class=n>lngs_trees_tracking</span><span class=o>&lt;-</span><span class=n>lngs_trees_wide</span> <span class=o>%&gt;%</span> 
  <span class=nf>select</span><span class=p>(</span><span class=n>treeId</span><span class=p>,</span><span class=n>changedTrees</span><span class=p>,</span> <span class=n>nUMIsNormalized_T1_1</span><span class=o>:</span><span class=n>nUMIsNormalized_T8_1</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>pivot_longer</span><span class=p>(</span><span class=n>cols</span><span class=o>=</span><span class=n>nUMIsNormalized_T1_1</span><span class=o>:</span><span class=n>nUMIsNormalized_T8_1</span><span class=p>,</span><span class=n>names_to</span><span class=o>=</span> <span class=s>&quot;timepoint&quot;</span><span class=p>,</span><span class=n>values_to</span> <span class=o>=</span> <span class=s>&quot;nUMIsNormalized&quot;</span><span class=p>)</span>  <span class=o>%&gt;%</span>
  <span class=nf>mutate</span><span class=p>(</span><span class=n>timepoint</span><span class=o>=</span><span class=nf>str_remove</span><span class=p>(</span><span class=n>timepoint</span><span class=p>,</span><span class=s>&quot;nUMIsNormalized_&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=nf>separate</span><span class=p>(</span><span class=s>&quot;timepoint&quot;</span><span class=p>,</span><span class=n>into</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;timepoint&quot;</span><span class=p>,</span><span class=s>&quot;repnum&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span>
  <span class=nf>group_by</span><span class=p>(</span><span class=n>timepoint</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>nReplicates</span><span class=o>=</span><span class=nf>n_distinct</span><span class=p>(</span><span class=n>repnum</span><span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=nf>ungroup</span><span class=p>()</span> <span class=o>%&gt;%</span> 
  <span class=nf>pivot_wider</span><span class=p>(</span><span class=n>id_cols</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>treeId</span><span class=p>,</span><span class=n>changedTrees</span><span class=p>,</span><span class=n>timepoint</span><span class=p>,</span><span class=n>nReplicates</span><span class=p>),</span>
              <span class=n>names_from</span> <span class=o>=</span> <span class=n>repnum</span><span class=p>,</span>
              <span class=n>names_prefix</span> <span class=o>=</span> <span class=s>&quot;rep_&quot;</span><span class=p>,</span>
              <span class=n>values_from</span> <span class=o>=</span> <span class=n>nUMIsNormalized</span><span class=p>,</span>
              <span class=n>values_fill</span> <span class=o>=</span> <span class=m>0.0</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>rowwise</span><span class=p>()</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>meanNUMIsNormalized</span><span class=o>=</span><span class=nf>ifelse</span><span class=p>(</span><span class=n>nReplicates</span><span class=o>==</span><span class=m>1</span><span class=p>,</span> <span class=n>rep_1</span> <span class=p>,</span> <span class=nf>mean</span><span class=p>(</span><span class=nf>c</span><span class=p>(</span><span class=n>rep_1</span><span class=p>,</span><span class=n>rep_2</span><span class=p>))),</span>
         <span class=n>minNUMIsNormalized</span><span class=o>=</span><span class=nf>ifelse</span><span class=p>(</span><span class=n>nReplicates</span><span class=o>==</span><span class=m>1</span><span class=p>,</span> <span class=n>rep_1</span> <span class=p>,</span> <span class=nf>min</span><span class=p>(</span><span class=n>rep_1</span><span class=p>,</span><span class=n>rep_2</span><span class=p>)),</span>
         <span class=n>maxNUMIsNormalized</span><span class=o>=</span><span class=nf>ifelse</span><span class=p>(</span><span class=n>nReplicates</span><span class=o>==</span><span class=m>1</span><span class=p>,</span> <span class=n>rep_1</span> <span class=p>,</span> <span class=nf>max</span><span class=p>(</span><span class=n>rep_1</span><span class=p>,</span><span class=n>rep_2</span><span class=p>)))</span>

<span class=c1>#Draw lineage tracing plot</span>
<span class=n>lngs_trees_tracking</span> <span class=o>%&gt;%</span> 
  <span class=nf>ggplot</span><span class=p>()</span><span class=o>+</span>
  <span class=nf>geom_line</span><span class=p>(</span><span class=n>data</span><span class=o>=</span> <span class=n>lngs_trees_tracking</span> <span class=o>%&gt;%</span> <span class=nf>filter</span><span class=p>(</span><span class=o>!</span><span class=n>changedTrees</span><span class=p>),</span>
            <span class=nf>aes</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>timepoint</span><span class=p>,</span> <span class=n>y</span><span class=o>=</span><span class=n>meanNUMIsNormalized</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>treeId</span><span class=p>),</span>
            <span class=n>color</span><span class=o>=</span><span class=s>&quot;#929BAD&quot;</span><span class=p>,</span>
            <span class=n>alpha</span><span class=o>=</span><span class=m>0.2</span><span class=p>,</span><span class=n>size</span><span class=o>=</span><span class=m>1</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>geom_line</span><span class=p>(</span><span class=n>data</span><span class=o>=</span> <span class=n>lngs_trees_tracking</span> <span class=o>%&gt;%</span> <span class=nf>filter</span><span class=p>(</span><span class=n>changedTrees</span><span class=p>),</span>
            <span class=nf>aes</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>timepoint</span><span class=p>,</span> <span class=n>y</span><span class=o>=</span><span class=n>meanNUMIsNormalized</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>treeId</span><span class=p>),</span>
            <span class=n>color</span><span class=o>=</span><span class=s>&quot;#F05670&quot;</span><span class=p>,</span>
            <span class=n>alpha</span><span class=o>=</span><span class=m>0.8</span><span class=p>,</span><span class=n>size</span><span class=o>=</span><span class=m>1</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>theme_bw</span><span class=p>()</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;Fraction of the lineage by cDNA molecules&quot;</span><span class=p>,</span>
       <span class=n>x</span><span class=o>=</span><span class=s>&quot;Time point&quot;</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>geom_segment</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=m>5</span><span class=p>,</span> <span class=n>y</span> <span class=o>=</span> <span class=m>0.085</span><span class=p>,</span> <span class=n>xend</span> <span class=o>=</span> <span class=m>4</span><span class=p>,</span> <span class=n>yend</span> <span class=o>=</span> <span class=m>0.08</span><span class=p>),</span>
                                      <span class=n>arrow</span> <span class=o>=</span> <span class=nf>arrow</span><span class=p>(</span><span class=n>length</span> <span class=o>=</span> <span class=nf>unit</span><span class=p>(</span><span class=m>0.2</span><span class=p>,</span> <span class=s>&quot;cm&quot;</span><span class=p>)))</span><span class=o>+</span>
    <span class=nf>geom_segment</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span> <span class=o>=</span> <span class=m>6</span><span class=p>,</span> <span class=n>y</span> <span class=o>=</span> <span class=m>0.085</span><span class=p>,</span> <span class=n>xend</span> <span class=o>=</span><span class=m>7</span><span class=p>,</span> <span class=n>yend</span> <span class=o>=</span> <span class=m>0.08</span><span class=p>),</span>
                 <span class=n>arrow</span> <span class=o>=</span> <span class=nf>arrow</span><span class=p>(</span><span class=n>length</span> <span class=o>=</span> <span class=nf>unit</span><span class=p>(</span><span class=m>0.2</span><span class=p>,</span> <span class=s>&quot;cm&quot;</span><span class=p>)))</span><span class=o>+</span>
    <span class=nf>annotate</span><span class=p>(</span><span class=s>&quot;text&quot;</span><span class=p>,</span> <span class=n>x</span><span class=o>=</span><span class=m>5.5</span><span class=p>,</span> <span class=n>y</span><span class=o>=</span><span class=m>0.086</span><span class=p>,</span> <span class=n>label</span><span class=o>=</span> <span class=s>&quot;COVID19&quot;</span><span class=p>)</span> <span class=o>+</span>
    <span class=nf>geom_vline</span><span class=p>(</span><span class=n>xintercept</span> <span class=o>=</span> <span class=m>4</span><span class=p>,</span><span class=n>linetype</span><span class=o>=</span><span class=m>2</span><span class=p>,</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;grey76&quot;</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>geom_vline</span><span class=p>(</span><span class=n>xintercept</span> <span class=o>=</span> <span class=m>7</span><span class=p>,</span><span class=n>linetype</span><span class=o>=</span><span class=m>2</span><span class=p>,</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;grey76&quot;</span><span class=p>)</span>
</code></pre></div> </details> <p><img alt src=../b-cell-lineage-tracing/lineage-3.svg></p> <h3 id=analysis-for-individual-trees>Analysis for individual trees</h3> <details class=quote> <summary>Define function for drawing phylogenetic trees with amino acid sequence alignments</summary> <div class=highlight><pre><span></span><code><span class=n>drawTreeWithMSA</span><span class=o>&lt;-</span> <span class=nf>function</span><span class=p>(</span><span class=n>trId</span><span class=p>){</span>
  <span class=nf>library</span><span class=p>(</span><span class=n>ggtree</span><span class=p>)</span>
  <span class=n>tree</span><span class=o>&lt;-</span><span class=nf>read.tree</span><span class=p>(</span><span class=s>&quot;~/ch_covid/mixcr/shmTrees/newick/&quot;</span> <span class=o>%+%</span> <span class=n>trId</span> <span class=o>%+%</span> <span class=s>&quot;.tree&quot;</span><span class=p>)</span>


  <span class=n>tree_meta</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span> 
    <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=nf>select</span><span class=p>(</span><span class=n>nodeId</span><span class=p>,</span>
           <span class=n>isotype</span><span class=p>,</span>
           <span class=n>timepoint</span><span class=p>,</span>
           <span class=n>tissue</span><span class=p>,</span>
           <span class=n>aaSeqCDR3</span><span class=p>,</span>
           <span class=n>uniqueUMICount</span><span class=p>,</span>
           <span class=n>cloneId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=n>dplyr</span><span class=o>::</span><span class=nf>rename</span><span class=p>(</span><span class=n>ID</span><span class=o>=</span><span class=n>nodeId</span><span class=p>)</span>

  <span class=c1>## draw initial tree </span>
  <span class=n>g</span><span class=o>&lt;-</span><span class=nf>ggtree</span><span class=p>(</span><span class=n>tree</span><span class=p>)</span> <span class=o>%&lt;+%</span> <span class=n>tree_meta</span> <span class=o>+</span>
    <span class=nf>geom_text</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>label</span><span class=o>=</span><span class=n>timepoint</span><span class=p>),</span> <span class=n>hjust</span><span class=o>=</span><span class=m>-.5</span><span class=p>,</span><span class=n>size</span><span class=o>=</span><span class=m>4</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>geom_tippoint</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>color</span> <span class=o>=</span><span class=n>isotype</span> <span class=p>,</span> <span class=n>size</span><span class=o>=</span><span class=n>uniqueUMICount</span><span class=p>))</span><span class=o>+</span>
    <span class=nf>scale_color_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span><span class=n>isotype_pal</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&#39;bottom&#39;</span><span class=p>)</span><span class=o>+</span>
      <span class=nf>scale_size_continuous</span><span class=p>(</span><span class=n>range</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=m>2</span><span class=p>,</span> <span class=m>7</span><span class=p>))</span>

  <span class=n>MiXCRtreeVDJ</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span>
    <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>`aaSeq{CDR1Begin:FR4End}`</span><span class=p>)</span> <span class=o>%&gt;%</span> <span class=nf>AAStringSet</span><span class=p>()</span>

  <span class=nf>names</span><span class=p>(</span><span class=n>MiXCRtreeVDJ</span><span class=p>)</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span>
    <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>nodeId</span><span class=p>)</span>

  <span class=n>msaMiXCRtreeVDJ</span><span class=o>&lt;-</span><span class=nf>msa</span><span class=p>(</span><span class=n>MiXCRtreeVDJ</span><span class=p>)</span>

  <span class=nf>alignment2Fasta</span><span class=p>(</span><span class=n>msaMiXCRtreeVDJ</span><span class=p>,</span><span class=s>&quot;~/ch_covid/mixcr/shmTrees/export/&quot;</span> <span class=o>%+%</span> <span class=n>trId</span> <span class=o>%+%</span> <span class=s>&quot;_aln.fasta&quot;</span><span class=p>)</span>

  <span class=nf>pdf</span><span class=p>(</span><span class=n>file</span> <span class=o>=</span> <span class=s>&quot;~/ch_covid/mixcr/shmTrees/export/&quot;</span> <span class=o>%+%</span> <span class=n>trId</span> <span class=o>%+%</span> <span class=s>&quot;.pdf&quot;</span><span class=p>,</span>  
      <span class=n>width</span> <span class=o>=</span> <span class=m>16</span><span class=p>,</span> <span class=c1># The width of the plot in inches</span>
      <span class=n>height</span> <span class=o>=</span> <span class=m>12</span><span class=p>)</span>
  <span class=nf>print</span><span class=p>(</span><span class=nf>msaplot</span><span class=p>(</span><span class=n>g</span><span class=p>,</span><span class=s>&quot;~/ch_covid/mixcr/shmTrees/export/&quot;</span> <span class=o>%+%</span> <span class=n>trId</span> <span class=o>%+%</span> <span class=s>&quot;_aln.fasta&quot;</span><span class=p>,</span>
            <span class=n>offset</span><span class=o>=</span><span class=m>8</span><span class=p>,</span><span class=n>width</span> <span class=o>=</span> <span class=m>2</span><span class=p>,</span><span class=n>color</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;grey88&quot;</span><span class=p>,</span><span class=n>mi_27</span><span class=p>))</span>
  <span class=p>)</span>
<span class=p>}</span>
</code></pre></div> </details> <details class=quote> <summary>Define function for drawing SHM frequencies by position plot</summary> <div class=highlight><pre><span></span><code><span class=n>mutationPositionPlot</span><span class=o>&lt;-</span><span class=nf>function</span><span class=p>(</span><span class=n>trId</span><span class=p>){</span>
  <span class=n>treeDb</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span> <span class=nf>filter</span><span class=p>(</span><span class=o>!</span><span class=nf>is.na</span><span class=p>(</span><span class=n>cloneId</span><span class=p>),</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=nf>mutate</span><span class=p>(</span><span class=n>CDR1BeginPosition</span><span class=o>=</span><span class=m>0</span><span class=p>,</span>
           <span class=n>FR2BeginPosition</span><span class=o>=</span><span class=n>positionInReferenceOfFR2Begin</span><span class=o>-</span><span class=n>positionInReferenceOfCDR1Begin</span><span class=p>,</span>
           <span class=n>CDR2BeginPosition</span><span class=o>=</span><span class=n>positionInReferenceOfCDR2Begin</span><span class=o>-</span><span class=n>positionInReferenceOfCDR1Begin</span><span class=p>,</span>
           <span class=n>FR3BeginPosition</span><span class=o>=</span><span class=n>positionInReferenceOfFR3Begin</span><span class=o>-</span><span class=n>positionInReferenceOfCDR1Begin</span><span class=p>,</span>
           <span class=n>CDR3BeginPosition</span><span class=o>=</span><span class=n>positionInReferenceOfCDR3Begin</span><span class=o>-</span><span class=n>positionInReferenceOfCDR1Begin</span><span class=p>,</span>
           <span class=n>FR4BeginPosition</span><span class=o>=</span><span class=n>positionInReferenceOfCDR3Begin</span><span class=o>-</span><span class=n>positionInReferenceOfCDR1Begin</span><span class=o>+</span><span class=nf>nchar</span><span class=p>(</span><span class=n>nSeqCDR3</span><span class=p>))</span>


  <span class=n>FR2Pos</span><span class=o>&lt;-</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>FR2BeginPosition</span><span class=p>)</span> <span class=o>%&gt;%</span>
    <span class=nf>unique</span><span class=p>()</span><span class=o>%&gt;%</span> <span class=nf>min</span><span class=p>()</span>

  <span class=n>CDR2Pos</span><span class=o>&lt;-</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>CDR2BeginPosition</span><span class=p>)</span> <span class=o>%&gt;%</span>
    <span class=nf>unique</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>min</span><span class=p>()</span>

  <span class=n>FR3Pos</span><span class=o>&lt;-</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>FR3BeginPosition</span><span class=p>)</span> <span class=o>%&gt;%</span>
    <span class=nf>unique</span><span class=p>()</span><span class=o>%&gt;%</span> <span class=nf>min</span><span class=p>()</span>

 <span class=n>CDR3Pos</span><span class=o>&lt;-</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>CDR3BeginPosition</span><span class=p>)</span> <span class=o>%&gt;%</span>
    <span class=nf>unique</span><span class=p>()</span><span class=o>%&gt;%</span> <span class=nf>min</span><span class=p>()</span>

 <span class=n>FR4Pos</span><span class=o>&lt;-</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
    <span class=nf>pull</span><span class=p>(</span><span class=n>FR4BeginPosition</span><span class=p>)</span> <span class=o>%&gt;%</span>
    <span class=nf>unique</span><span class=p>()</span><span class=o>%&gt;%</span> <span class=nf>min</span><span class=p>()</span>

<span class=n>mutByRegion</span><span class=o>&lt;-</span><span class=nf>str_extract_all</span><span class=p>(</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> 
                        <span class=nf>pull</span><span class=p>(</span><span class=n>`nMutations{CDR1Begin:FR3End}BasedOnGermline`</span><span class=p>),</span>
                <span class=s>&quot;[0-9]*[0-9][0-9]*&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> <span class=nf>unlist</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>table</span><span class=p>()</span> <span class=o>%&gt;%</span> 
  <span class=nf>enframe</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;position&quot;</span><span class=p>,</span><span class=n>value</span><span class=o>=</span><span class=s>&quot;nMut&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>position</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>position</span><span class=p>),</span><span class=n>nMut</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>nMut</span><span class=p>),</span>
         <span class=n>region</span><span class=o>=</span><span class=nf>case_when</span><span class=p>(</span>
           <span class=n>position</span><span class=o>&lt;</span><span class=n>FR2Pos</span> <span class=o>~</span> <span class=s>&quot;CDR1&quot;</span><span class=p>,</span>
           <span class=n>position</span><span class=o>&lt;</span><span class=n>CDR2Pos</span> <span class=o>~</span> <span class=s>&quot;FR2&quot;</span><span class=p>,</span>
           <span class=n>position</span><span class=o>&lt;</span><span class=n>FR3Pos</span> <span class=o>~</span> <span class=s>&quot;CDR2&quot;</span><span class=p>,</span>
           <span class=n>position</span><span class=o>&lt;</span><span class=n>CDR3Pos</span> <span class=o>~</span> <span class=s>&quot;FR3&quot;</span>
         <span class=p>))</span> <span class=o>%&gt;%</span> 
  <span class=nf>bind_rows</span><span class=p>(</span>
    <span class=nf>str_extract_all</span><span class=p>(</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> <span class=nf>pull</span><span class=p>(</span><span class=n>nMutationsCDR3BasedOnGermline</span><span class=p>),</span>
                    <span class=s>&quot;[0-9]*[0-9][0-9]*&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> <span class=nf>unlist</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>table</span><span class=p>()</span> <span class=o>%&gt;%</span> 
      <span class=nf>enframe</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;position&quot;</span><span class=p>,</span><span class=n>value</span><span class=o>=</span><span class=s>&quot;nMut&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
      <span class=nf>mutate</span><span class=p>(</span><span class=n>position</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>position</span><span class=p>)</span><span class=o>+</span><span class=n>CDR3Pos</span><span class=p>,</span><span class=n>nMut</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>nMut</span><span class=p>),</span>
             <span class=n>region</span><span class=o>=</span><span class=s>&quot;CDR3&quot;</span><span class=p>)</span>
  <span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>bind_rows</span><span class=p>(</span>
    <span class=nf>str_extract_all</span><span class=p>(</span><span class=n>treeDb</span> <span class=o>%&gt;%</span> <span class=nf>pull</span><span class=p>(</span><span class=n>nMutationsFR4BasedOnGermline</span><span class=p>),</span>
                    <span class=s>&quot;[0-9]*[0-9][0-9]*&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> <span class=nf>unlist</span><span class=p>()</span> <span class=o>%&gt;%</span> <span class=nf>table</span><span class=p>()</span> <span class=o>%&gt;%</span> 
      <span class=nf>enframe</span><span class=p>(</span><span class=n>name</span> <span class=o>=</span> <span class=s>&quot;position&quot;</span><span class=p>,</span><span class=n>value</span><span class=o>=</span><span class=s>&quot;nMut&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span> 
      <span class=nf>mutate</span><span class=p>(</span><span class=n>position</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>position</span><span class=p>)</span><span class=o>+</span><span class=n>FR4Pos</span><span class=p>,</span><span class=n>nMut</span><span class=o>=</span><span class=nf>as.integer</span><span class=p>(</span><span class=n>nMut</span><span class=p>),</span>
             <span class=n>region</span><span class=o>=</span><span class=s>&quot;FR4&quot;</span><span class=p>)</span>
  <span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>freqMut</span><span class=o>=</span><span class=n>nMut</span><span class=o>/</span><span class=nf>nrow</span><span class=p>(</span><span class=n>treeDb</span><span class=p>))</span>


<span class=n>g</span><span class=o>&lt;-</span><span class=n>mutByRegion</span><span class=o>%&gt;%</span> 
  <span class=nf>ggplot</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>position</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=n>freqMut</span><span class=p>,</span><span class=n>fill</span><span class=o>=</span><span class=n>region</span><span class=p>))</span> <span class=o>+</span>
  <span class=nf>theme_bw</span><span class=p>()</span><span class=o>+</span>
  <span class=nf>geom_bar</span><span class=p>(</span><span class=n>stat</span><span class=o>=</span><span class=s>&quot;identity&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>scale_fill_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span><span class=n>mi_dark</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>geom_vline</span><span class=p>(</span><span class=n>xintercept</span> <span class=o>=</span> <span class=nf>c</span><span class=p>(</span><span class=n>FR2Pos</span><span class=p>,</span><span class=n>CDR2Pos</span><span class=p>,</span><span class=n>FR3Pos</span><span class=p>,</span><span class=n>CDR3Pos</span><span class=p>,</span><span class=n>FR4Pos</span><span class=p>),</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;grey28&quot;</span><span class=p>,</span><span class=n>linetype</span><span class=o>=</span><span class=m>2</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&quot;none&quot;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>scale_x_continuous</span><span class=p>(</span><span class=n>expand</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>0</span><span class=p>))</span> <span class=o>+</span>
  <span class=nf>scale_y_continuous</span><span class=p>(</span><span class=n>expand</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=m>0</span><span class=p>,</span><span class=m>0</span><span class=p>))</span> <span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;Mutation freq.&quot;</span><span class=p>,</span> <span class=n>x</span><span class=o>=</span><span class=s>&quot;Position&quot;</span><span class=p>)</span>

<span class=nf>print</span><span class=p>(</span><span class=n>g</span><span class=p>)</span>

<span class=p>}</span>
</code></pre></div> </details> <details class=quote> <summary>Define function for drawing lineage longitudinal trajectory</summary> <div class=highlight><pre><span></span><code><span class=n>lineageTrajectoryPlot</span><span class=o>&lt;-</span><span class=nf>function</span><span class=p>(</span><span class=n>trId</span><span class=p>){</span>
  <span class=n>g</span><span class=o>&lt;-</span><span class=n>lngs_trees_tracking</span> <span class=o>%&gt;%</span> 
    <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
    <span class=nf>ggplot</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>timepoint</span><span class=p>,</span> <span class=n>y</span><span class=o>=</span><span class=n>meanNUMIsNormalized</span><span class=p>,</span> <span class=n>group</span><span class=o>=</span><span class=n>treeId</span><span class=p>))</span><span class=o>+</span>
    <span class=nf>geom_line</span><span class=p>(</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;#F05670&quot;</span><span class=p>,</span><span class=n>size</span><span class=o>=</span><span class=m>1</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>geom_point</span><span class=p>(</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;grey20&quot;</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>theme_bw</span><span class=p>()</span><span class=o>+</span>
    <span class=nf>labs</span><span class=p>(</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;Lineage fraction in\nrepertoire by cDNA count&quot;</span><span class=p>,</span>
         <span class=n>x</span><span class=o>=</span><span class=s>&quot;Time point&quot;</span><span class=p>)</span><span class=o>+</span>
    <span class=nf>geom_errorbar</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>ymin</span><span class=o>=</span><span class=n>minNUMIsNormalized</span><span class=p>,</span>
                      <span class=n>ymax</span><span class=o>=</span><span class=n>maxNUMIsNormalized</span><span class=p>),</span>
                  <span class=n>width</span> <span class=o>=</span> <span class=m>0.15</span><span class=p>,</span>
                  <span class=n>color</span><span class=o>=</span><span class=s>&quot;grey26&quot;</span><span class=p>)</span>
  <span class=nf>print</span><span class=p>(</span><span class=n>g</span><span class=p>)</span>
<span class=p>}</span>
</code></pre></div> </details> <details class=quote> <summary>Define function for drawing lineage isotype abundance by timepoints</summary> <div class=highlight><pre><span></span><code><span class=n>lineageIsotypesPlot</span><span class=o>&lt;-</span><span class=nf>function</span><span class=p>(</span><span class=n>trId</span><span class=p>){</span>
<span class=n>g</span><span class=o>&lt;-</span>  <span class=n>lngs_trees_tracking</span> <span class=o>%&gt;%</span> 
      <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span>
      <span class=c1>#calculate isotype proportions in each timepoint and join to tracking data.frame</span>
      <span class=nf>left_join</span><span class=p>(</span>
        <span class=n>lng_db</span> <span class=o>%&gt;%</span>
          <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>,</span><span class=o>!</span><span class=nf>is.na</span><span class=p>(</span><span class=n>cloneId</span><span class=p>))</span> <span class=o>%&gt;%</span> 
          <span class=nf>group_by</span><span class=p>(</span><span class=n>isotype</span><span class=p>,</span><span class=n>timepoint</span><span class=p>)</span> <span class=o>%&gt;%</span> 
          <span class=nf>summarise</span><span class=p>(</span><span class=n>UMICount</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>uniqueUMICount</span><span class=p>))</span> <span class=o>%&gt;%</span> 
          <span class=nf>ungroup</span><span class=p>()</span> <span class=o>%&gt;%</span> 
          <span class=nf>group_by</span><span class=p>(</span><span class=n>timepoint</span><span class=p>)</span> <span class=o>%&gt;%</span> 
          <span class=nf>mutate</span><span class=p>(</span><span class=n>UMIFraction</span><span class=o>=</span><span class=n>UMICount</span><span class=o>/</span><span class=nf>sum</span><span class=p>(</span><span class=n>UMICount</span><span class=p>)),</span>
      <span class=n>by</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;timepoint&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span> 
      <span class=nf>mutate</span><span class=p>(</span><span class=n>isotypeUMIFraction</span><span class=o>=</span><span class=n>UMIFraction</span><span class=o>*</span><span class=n>meanNUMIsNormalized</span><span class=p>)</span> <span class=o>%&gt;%</span> 
      <span class=nf>ggplot</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>x</span><span class=o>=</span><span class=n>timepoint</span><span class=p>,</span><span class=n>y</span><span class=o>=</span><span class=n>isotypeUMIFraction</span><span class=p>,</span><span class=n>fill</span><span class=o>=</span><span class=n>isotype</span><span class=p>))</span><span class=o>+</span>
        <span class=nf>geom_histogram</span><span class=p>(</span><span class=n>stat</span><span class=o>=</span><span class=s>&quot;identity&quot;</span><span class=p>)</span><span class=o>+</span>
        <span class=nf>theme_bw</span><span class=p>()</span><span class=o>+</span>
        <span class=nf>scale_fill_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span><span class=n>isotype_pal</span><span class=p>)</span><span class=o>+</span>
        <span class=nf>labs</span><span class=p>(</span><span class=n>y</span><span class=o>=</span><span class=s>&quot;Isotype fraction in\nrepertoire by cDNA count&quot;</span><span class=p>,</span>
             <span class=n>x</span><span class=o>=</span><span class=s>&quot;Time point&quot;</span><span class=p>)</span><span class=o>+</span>
        <span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&quot;none&quot;</span><span class=p>)</span>
<span class=nf>print</span><span class=p>(</span><span class=n>g</span><span class=p>)</span>
<span class=p>}</span>
</code></pre></div> </details> <h4 id=draw-all-plots-for-lineage-13628>Draw all plots for lineage 13628</h4> <div class=highlight><pre><span></span><code><span class=c1>#| warning: false</span>
<span class=nf>drawTreeWithMSA</span><span class=p>(</span><span class=m>13628</span><span class=p>)</span>
<span class=nf>mutationPositionPlot</span><span class=p>(</span><span class=m>13628</span><span class=p>)</span>
<span class=nf>lineageTrajectoryPlot</span><span class=p>(</span><span class=m>13628</span><span class=p>)</span>
<span class=nf>lineageIsotypesPlot</span><span class=p>(</span><span class=m>13628</span><span class=p>)</span>
</code></pre></div> <p><img alt src=../b-cell-lineage-tracing/lineage-4.svg></p> <h4 id=draw-all-plots-for-lineage-10511>Draw all plots for lineage 10511</h4> <div class=highlight><pre><span></span><code><span class=c1>#| warning: false</span>
<span class=nf>drawTreeWithMSA</span><span class=p>(</span><span class=m>10511</span><span class=p>)</span>
<span class=nf>mutationPositionPlot</span><span class=p>(</span><span class=m>10511</span><span class=p>)</span>
<span class=nf>lineageTrajectoryPlot</span><span class=p>(</span><span class=m>10511</span><span class=p>)</span>
<span class=nf>lineageIsotypesPlot</span><span class=p>(</span><span class=m>10511</span><span class=p>)</span>
</code></pre></div> <p><img alt src=../b-cell-lineage-tracing/lineage-5.svg></p> <h4 id=draw-all-plots-for-lineage-12796>Draw all plots for lineage 12796</h4> <div class=highlight><pre><span></span><code><span class=c1>#| warning: false</span>
<span class=nf>drawTreeWithMSA</span><span class=p>(</span><span class=m>12796</span><span class=p>)</span>
<span class=nf>mutationPositionPlot</span><span class=p>(</span><span class=m>12796</span><span class=p>)</span>
<span class=nf>lineageTrajectoryPlot</span><span class=p>(</span><span class=m>12796</span><span class=p>)</span>
<span class=nf>lineageIsotypesPlot</span><span class=p>(</span><span class=m>12796</span><span class=p>)</span>
</code></pre></div> <p><img alt src=../b-cell-lineage-tracing/lineage-6.svg></p> <h4 id=alternative-representations-for-lineage-trees>Alternative representations for lineage trees</h4> <p>Lets calculate the number of synonymous and non-synonymous mutations in clonotypes within lineages by parsing <code>mutationsDetailed</code> columns from <code>'mixcr exportShmTreesWithNodes</code>' command output:</p> <details class=quote> <summary>Code</summary> <div class=highlight><pre><span></span><code><span class=nf>suppressWarnings</span><span class=p>(</span>
<span class=n>lng_db</span> <span class=o>%&lt;&gt;%</span> 
 <span class=nf>mutate</span><span class=p>(</span>
    <span class=nf>across</span><span class=p>(</span><span class=nf>starts_with</span><span class=p>(</span><span class=s>&quot;mutationsDetailed&quot;</span><span class=p>),</span>
           <span class=n>.fns</span> <span class=o>=</span> <span class=o>~</span> <span class=nf>str_split</span><span class=p>(</span><span class=n>.x</span><span class=p>,</span><span class=n>pattern</span> <span class=o>=</span> <span class=s>&quot;,&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
                             <span class=nf>lapply</span><span class=p>(</span><span class=nf>function</span><span class=p>(</span><span class=n>mut</span><span class=p>){</span>
                               <span class=nf>if </span><span class=p>(</span><span class=nf>is.na</span><span class=p>(</span><span class=n>mut</span><span class=p>))</span> <span class=p>{</span> 
                                 <span class=nf>return</span><span class=p>(</span><span class=m>0</span><span class=p>)</span>
                               <span class=p>}</span> <span class=n>else</span> <span class=p>{</span>
                               <span class=nf>str_split</span><span class=p>(</span><span class=n>mut</span><span class=p>,</span><span class=n>pattern</span><span class=o>=</span><span class=s>&quot;:&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
                                 <span class=nf>transpose</span><span class=p>()</span> <span class=o>%&gt;%</span>
                                 <span class=nf>as.data.table</span><span class=p>()</span> <span class=o>%&gt;%</span>
                                 <span class=nf>summarise</span><span class=p>(</span><span class=n>nMut</span><span class=o>=</span><span class=nf>n</span><span class=p>())</span> <span class=o>%&gt;%</span> 
                                 <span class=nf>pull</span><span class=p>(</span><span class=n>nMut</span><span class=p>)}})</span> <span class=o>%&gt;%</span> 
                             <span class=nf>unlist</span><span class=p>(),</span>
           <span class=n>.names</span> <span class=o>=</span> <span class=s>&quot;mutNum_{.col}&quot;</span> <span class=p>),</span>
    <span class=nf>across</span><span class=p>(</span><span class=nf>starts_with</span><span class=p>(</span><span class=s>&quot;mutationsDetailed&quot;</span><span class=p>),</span>
           <span class=n>.fns</span> <span class=o>=</span> <span class=o>~</span> <span class=nf>str_split</span><span class=p>(</span><span class=n>.x</span><span class=p>,</span><span class=n>pattern</span> <span class=o>=</span> <span class=s>&quot;,&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
             <span class=nf>lapply</span><span class=p>(</span><span class=nf>function</span><span class=p>(</span><span class=n>mut</span><span class=p>){</span>
               <span class=nf>if </span><span class=p>(</span><span class=nf>is.na</span><span class=p>(</span><span class=n>mut</span><span class=p>))</span> <span class=p>{</span> 
                 <span class=nf>return</span><span class=p>(</span><span class=m>0</span><span class=p>)</span>
               <span class=p>}</span> <span class=n>else</span> <span class=p>{</span>
                 <span class=nf>str_split</span><span class=p>(</span><span class=n>mut</span><span class=p>,</span><span class=n>pattern</span><span class=o>=</span><span class=s>&quot;:&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
                   <span class=nf>transpose</span><span class=p>()</span> <span class=o>%&gt;%</span>
                   <span class=nf>as.data.table</span><span class=p>()</span> <span class=o>%&gt;%</span>
                   <span class=nf>summarise</span><span class=p>(</span><span class=n>nMutSyn</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>V2</span><span class=o>==</span><span class=s>&quot;&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span> 
                   <span class=nf>pull</span><span class=p>(</span><span class=n>nMutSyn</span><span class=p>)}})</span> <span class=o>%&gt;%</span> 
             <span class=nf>unlist</span><span class=p>(),</span>
           <span class=n>.names</span> <span class=o>=</span> <span class=s>&quot;mutNumSyn_{.col}&quot;</span> <span class=p>),</span>
    <span class=nf>across</span><span class=p>(</span><span class=nf>starts_with</span><span class=p>(</span><span class=s>&quot;mutationsDetailed&quot;</span><span class=p>),</span>
           <span class=n>.fns</span> <span class=o>=</span> <span class=o>~</span> <span class=nf>str_split</span><span class=p>(</span><span class=n>.x</span><span class=p>,</span><span class=n>pattern</span> <span class=o>=</span> <span class=s>&quot;,&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
             <span class=nf>lapply</span><span class=p>(</span><span class=nf>function</span><span class=p>(</span><span class=n>mut</span><span class=p>){</span>
               <span class=nf>if </span><span class=p>(</span><span class=nf>is.na</span><span class=p>(</span><span class=n>mut</span><span class=p>))</span> <span class=p>{</span> 
                 <span class=nf>return</span><span class=p>(</span><span class=m>0</span><span class=p>)</span>
               <span class=p>}</span> <span class=n>else</span> <span class=p>{</span>
                 <span class=nf>str_split</span><span class=p>(</span><span class=n>mut</span><span class=p>,</span><span class=n>pattern</span><span class=o>=</span><span class=s>&quot;:&quot;</span><span class=p>)</span> <span class=o>%&gt;%</span>
                   <span class=nf>transpose</span><span class=p>()</span> <span class=o>%&gt;%</span>
                   <span class=nf>as.data.table</span><span class=p>()</span> <span class=o>%&gt;%</span>
                   <span class=nf>summarise</span><span class=p>(</span><span class=n>nMutNonSyn</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=n>V2</span><span class=o>!=</span><span class=s>&quot;&quot;</span><span class=p>))</span> <span class=o>%&gt;%</span> 
                   <span class=nf>pull</span><span class=p>(</span><span class=n>nMutNonSyn</span><span class=p>)}})</span> <span class=o>%&gt;%</span> 
             <span class=nf>unlist</span><span class=p>(),</span>
           <span class=n>.names</span> <span class=o>=</span> <span class=s>&quot;mutNumNonSyn_{.col}&quot;</span> <span class=p>)</span>

    <span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>rename_at</span><span class=p>(</span><span class=nf>vars</span><span class=p>(</span><span class=nf>matches</span><span class=p>(</span><span class=s>&quot;^mutNum_|^mutNumSyn_|^mutNumNonSyn_&quot;</span><span class=p>)),</span> <span class=o>~</span> <span class=nf>str_remove</span><span class=p>(</span><span class=n>.</span><span class=p>,</span> <span class=s>&quot;_mutationsDetailed&quot;</span><span class=p>))</span> <span class=p>)</span>

<span class=n>lng_db</span><span class=o>%&lt;&gt;%</span> 
  <span class=nf>rowwise</span><span class=p>()</span> <span class=o>%&gt;%</span> 
  <span class=nf>mutate</span><span class=p>(</span><span class=n>totalSynMut</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=nf>across</span><span class=p>(</span><span class=n>mutNumSynInCDR1BasedOnGermline</span><span class=o>:</span><span class=n>mutNumSynInFR4BasedOnGermline</span><span class=p>)),</span>
         <span class=n>totalNonSynMut</span><span class=o>=</span><span class=nf>sum</span><span class=p>(</span><span class=nf>across</span><span class=p>(</span><span class=n>mutNumNonSynInCDR1BasedOnGermline</span><span class=o>:</span><span class=n>mutNumNonSynInFR4BasedOnGermline</span><span class=p>)))</span> <span class=o>%&gt;%</span> 
 <span class=nf>ungroup</span><span class=p>()</span>
</code></pre></div> </details> <p>Let's plot the phylogenetic tree for one of the lineages in three alternative colorings: by isotype, by timepoint and by number of non-synonymous mutations</p> <div class=highlight><pre><span></span><code><span class=n>trId</span><span class=o>=</span><span class=m>12796</span>

<span class=n>tree</span><span class=o>&lt;-</span><span class=nf>read.tree</span><span class=p>(</span><span class=s>&quot;~/ch_covid/mixcr/shmTrees/newick/&quot;</span> <span class=o>%+%</span> <span class=n>trId</span> <span class=o>%+%</span> <span class=s>&quot;.tree&quot;</span><span class=p>)</span>

<span class=n>tree_meta</span><span class=o>&lt;-</span><span class=n>lng_db</span> <span class=o>%&gt;%</span> 
  <span class=nf>filter</span><span class=p>(</span><span class=n>treeId</span><span class=o>==</span><span class=n>trId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=nf>select</span><span class=p>(</span><span class=n>nodeId</span><span class=p>,</span>
         <span class=n>isotype</span><span class=p>,</span>
         <span class=n>timepoint</span><span class=p>,</span>
         <span class=n>totalNonSynMut</span><span class=p>,</span>
         <span class=n>uniqueUMICount</span><span class=p>,</span>
         <span class=n>cloneId</span><span class=p>)</span> <span class=o>%&gt;%</span> 
  <span class=n>dplyr</span><span class=o>::</span><span class=nf>rename</span><span class=p>(</span><span class=n>ID</span><span class=o>=</span><span class=n>nodeId</span><span class=p>)</span>

<span class=c1>## draw initial tree </span>
<span class=n>g1</span><span class=o>&lt;-</span><span class=nf>ggtree</span><span class=p>(</span><span class=n>tree</span><span class=p>)</span> <span class=o>%&lt;+%</span> <span class=n>tree_meta</span> <span class=o>+</span>
  <span class=nf>geom_text_repel</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>label</span><span class=o>=</span><span class=n>isotype</span><span class=p>))</span><span class=o>+</span>
  <span class=nf>geom_tippoint</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>color</span> <span class=o>=</span><span class=n>isotype</span><span class=p>,</span> <span class=n>size</span><span class=o>=</span><span class=n>uniqueUMICount</span><span class=p>))</span><span class=o>+</span>
  <span class=nf>scale_color_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span><span class=n>isotype_pal</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&#39;bottom&#39;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;Isotype&quot;</span><span class=p>)</span>

<span class=n>g2</span><span class=o>&lt;-</span><span class=nf>ggtree</span><span class=p>(</span><span class=n>tree</span><span class=p>)</span> <span class=o>%&lt;+%</span> <span class=n>tree_meta</span> <span class=o>+</span>
  <span class=nf>geom_text_repel</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>label</span><span class=o>=</span><span class=n>timepoint</span><span class=p>))</span><span class=o>+</span>
  <span class=nf>geom_tippoint</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>color</span> <span class=o>=</span><span class=n>timepoint</span><span class=p>,</span> <span class=n>size</span><span class=o>=</span><span class=n>uniqueUMICount</span> <span class=p>))</span><span class=o>+</span>
  <span class=nf>scale_colour_manual</span><span class=p>(</span><span class=n>values</span><span class=o>=</span><span class=n>mi_dark</span><span class=p>)</span><span class=o>+</span>
<span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&#39;bottom&#39;</span><span class=p>)</span> 


<span class=n>g3</span><span class=o>&lt;-</span><span class=nf>ggtree</span><span class=p>(</span><span class=n>tree</span><span class=p>)</span> <span class=o>%&lt;+%</span> <span class=n>tree_meta</span> <span class=o>+</span>
  <span class=nf>geom_text_repel</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>label</span><span class=o>=</span><span class=n>totalNonSynMut</span><span class=p>))</span><span class=o>+</span>
  <span class=nf>geom_tippoint</span><span class=p>(</span><span class=nf>aes</span><span class=p>(</span><span class=n>color</span> <span class=o>=</span><span class=n>totalNonSynMut</span><span class=p>,</span> <span class=n>size</span><span class=o>=</span><span class=n>uniqueUMICount</span> <span class=p>))</span><span class=o>+</span>
  <span class=nf>scale_colour_gradientn</span><span class=p>(</span><span class=n>colours</span><span class=o>=</span><span class=nf>c</span><span class=p>(</span><span class=s>&quot;#DFFADC&quot;</span><span class=p>,</span> <span class=s>&quot;#611347&quot;</span><span class=p>))</span><span class=o>+</span>
<span class=nf>theme</span><span class=p>(</span><span class=n>legend.position</span> <span class=o>=</span> <span class=s>&#39;bottom&#39;</span><span class=p>)</span><span class=o>+</span>
  <span class=nf>labs</span><span class=p>(</span><span class=n>color</span><span class=o>=</span><span class=s>&quot;# of non-syn. SHM&quot;</span><span class=p>)</span>

<span class=n>cowplot</span><span class=o>::</span><span class=nf>plot_grid</span><span class=p>(</span><span class=n>g1</span><span class=p>,</span><span class=n>g2</span><span class=p>,</span><span class=n>g3</span><span class=p>,</span><span class=n>nrow</span><span class=o>=</span><span class=m>1</span><span class=p>)</span>
</code></pre></div> <p><img alt src=../b-cell-lineage-tracing/lineage-7.svg></p> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../external-libraries/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Using external libraries" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Using external libraries </div> </div> </a> <a href=../../reference/overview-analysis-overview/ class="md-footer__link md-footer__link--next" aria-label="Next: Analysis overview" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Analysis overview </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../../../javascripts/svg.js></script> <script src=../../../javascripts/onload.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>