# When the analysis results look poor

<div align="right">
<i>"Garbage in, garbage out"</i> - <br>
<a href="https://en.wikipedia.org/wiki/Garbage_in,_garbage_out">GIGO principle</a>
</div>


MiXCR takes raw sequencing data as input and extracts repertoire information based on the provided parameters and library architecture. Millions of sequencing datasets prepared with hundreds of different wet lab protocols and technologies have been analyzed by MiXCR through the years.

When results of the analysis look poor (high rate of failed alignments, low number of clones etc), there might be only two global reasons:

 - wet lab issues (in our experience this covers 90% of cases) or 
 - wrong analysis settings (this is another 10%).

Below we elaborate on how to assess the basic quality of the libraries, read MiXCR reports, discover wet lab issues and troubleshoot to get maximum information from the data you have. Check out:

 - [Checking sequencing quality](qc-sequencing-quality.md)
 - [Assessing quality of Rep-Seq libraries](qc-repseq.md)
 - [Non-targeted shotgun libraries](qc-rnaseq.md)
